{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "6cec1869",
   "metadata": {},
   "outputs": [],
   "source": [
    "%run ../scripts/notebook_settings.py\n",
    "import glob"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "b3ff37b7",
   "metadata": {},
   "source": [
    "Performing some checks to understand Vasilis metadata."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "09fd7694",
   "metadata": {},
   "outputs": [],
   "source": [
    "table_desc = \"~/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/plots/SupTable_Sample_Stats_wGT_QC.tsv\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "49434a18",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index(['ID', 'ID_PGDP', 'group', 'family', 'genus', 'species', 'subspecies',\n",
       "       'species_full', 'species_short', 'species_genotyping', 'reference',\n",
       "       'SEX', 'len_chrA', 'len_chrA_contigs_shorter_than_1Mb', 'len_chrX',\n",
       "       'len_chrX_contigs_shorter_than_1Mb', 'len_chrY',\n",
       "       'len_chrY_contigs_shorter_than_1Mb', 'total_seq_chrA', 'total_seq_chrX',\n",
       "       'total_seq_chrY', 'len_cov_chrA', 'len_cov_chrX', 'len_cov_chrY',\n",
       "       'frac_cov_chrA', 'frac_cov_chrX', 'frac_cov_chrY', 'frac_cov_gw',\n",
       "       'rel_frac_cov_YX', 'nonN_frac', 'frac_cov_gw_cor', 'threshold',\n",
       "       'threshold_chrX', 'good_frac_cov', 'cov_chrA', 'cov_chrX', 'cov_chrY',\n",
       "       'cov_gw', 'rel_cov_chrX', 'gSEX', 'gSEX_source', 'callable_chrA',\n",
       "       'callable_chrX', 'GT00_chrA', 'GT01_chrA', 'GT11_chrA', 'GTmis_chrA',\n",
       "       'GT00_chrX', 'GT0_chrX', 'GT01_chrX', 'GT11_chrX', 'GT1_chrX',\n",
       "       'GTmis_chrX', 'het_chrA', 'het_chrX', 'rel_chrX_het', 'hom_alt_chrA',\n",
       "       'hom_alt_chrX', 'hem_alt_chrX', 'rel_chrX_hom_alt', 'mis_chrA',\n",
       "       'mis_chrX', 'div_chrA', 'clade', 'subclade_split', 'sample_used_for_q',\n",
       "       'q25_div_chrA', 'q50_div_chrA', 'q75_div_chrA', 'q25_frac_cov_chrA',\n",
       "       'q50_frac_cov_chrA', 'q75_frac_cov_chrA', 'zscore_div_chrA',\n",
       "       'zscore_frac_cov_chrA', 'dupl_group', 'remove_as_duplicate', 'low_mis',\n",
       "       'low_frac_cov_qc', 'high_div_chrA_qc', 'high_mis_qc', 'low_div_chrA_qc',\n",
       "       'masterQC', 'manual_fix', 'reason', 'finalQC', 'Kuderna_AB_QC',\n",
       "       'Kuderna_COV_QC', 'relatives', 'remove_as_relative', 'generation',\n",
       "       'founder', 'remove_manual', 'pedigree_source'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vasili_table = pd.read_csv(table_desc, sep=\"\\t\")\n",
    "vasili_table.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "638874b8",
   "metadata": {},
   "outputs": [],
   "source": [
    "#vasili_table.loc[(vasili_table.genus == \"Papio\") & (~vasili_table.relatives.isna())\n",
    "#                & (vasili_table.ID_PGDP.str.startswith(\"PD\"))][[\"ID_PGDP\", \"ID\", \"species\",\n",
    "#                                                                \"relatives\", \"remove_as_relative\",\n",
    "#                                                                \"remove_as_duplicate\"]].to_csv(\"papio_all.csv\", index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "1da1332a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0       Allenopithecus_nigroviridis_ssp\n",
       "1       Allenopithecus_nigroviridis_ssp\n",
       "2       Allenopithecus_nigroviridis_ssp\n",
       "3                      Aotus_azarai_ssp\n",
       "4                      Aotus_azarai_ssp\n",
       "                     ...               \n",
       "3235              Varecia_variegata_ssp\n",
       "3236              Varecia_variegata_ssp\n",
       "3237              Varecia_variegata_ssp\n",
       "3238              Varecia_variegata_ssp\n",
       "3239              Varecia_variegata_ssp\n",
       "Name: species_full, Length: 3240, dtype: object"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vasili_table.species_full"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "81a4450a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0       Allenopithecus_nigroviridis\n",
       "1       Allenopithecus_nigroviridis\n",
       "2       Allenopithecus_nigroviridis\n",
       "3                   Aotus_nancymaae\n",
       "4                   Aotus_nancymaae\n",
       "                   ...             \n",
       "3235              Varecia_variegata\n",
       "3236              Varecia_variegata\n",
       "3237              Varecia_variegata\n",
       "3238              Varecia_variegata\n",
       "3239              Varecia_variegata\n",
       "Name: reference, Length: 3240, dtype: object"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vasili_table.reference"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "32fcf0e7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(3240, array(['chrX', 'metadata', 'default', 'chrY'], dtype=object))"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(vasili_table), vasili_table.gSEX_source.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "a6dc1d8e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "256"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(vasili_table.species.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "2e808998",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(240, 1390)"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sub_vasili = vasili_table.loc[(vasili_table.finalQC != \"fail\")\n",
    "                              & (vasili_table.cov_chrA >= 20)\n",
    "                              & (vasili_table.remove_as_relative != True)\n",
    "                             # & (vasili_table.mis_chrA < 0.01)\n",
    "                              & (vasili_table.remove_manual != True)\n",
    "                              & (~vasili_table.ID.str.startswith(\"SAMEA11633\"))\n",
    "                             ]\n",
    "# Can get six more species if you use cov >= 10, but probably not worth the effort\n",
    "len(sub_vasili.species.unique()), len(sub_vasili)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "e46831aa",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "species_genotyping\n",
       "Macaca_fascicularis_ssp        132\n",
       "Macaca_mulatta_ssp             103\n",
       "Papio_anubis_ssp                96\n",
       "Pan_troglodytes_ssp             94\n",
       "Papio_cynocephalus_ssp          70\n",
       "                              ... \n",
       "Microcebus_griseorufus_ssp       1\n",
       "Microcebus_jonahi_ssp            1\n",
       "Microcebus_mittermeieri_ssp      1\n",
       "Microcebus_tavaratra_ssp         1\n",
       "Cercocebus_lunulatus_ssp         1\n",
       "Name: count, Length: 251, dtype: int64"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sub_vasili.species_genotyping.value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "1535f95a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>ID_PGDP</th>\n",
       "      <th>group</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "      <th>subspecies</th>\n",
       "      <th>species_full</th>\n",
       "      <th>species_short</th>\n",
       "      <th>species_genotyping</th>\n",
       "      <th>...</th>\n",
       "      <th>reason</th>\n",
       "      <th>finalQC</th>\n",
       "      <th>Kuderna_AB_QC</th>\n",
       "      <th>Kuderna_COV_QC</th>\n",
       "      <th>relatives</th>\n",
       "      <th>remove_as_relative</th>\n",
       "      <th>generation</th>\n",
       "      <th>founder</th>\n",
       "      <th>remove_manual</th>\n",
       "      <th>pedigree_source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>199</th>\n",
       "      <td>SAMN11334319</td>\n",
       "      <td>SAMN11334319</td>\n",
       "      <td>STR</td>\n",
       "      <td>Cheirogaleidae</td>\n",
       "      <td>Cheirogaleus</td>\n",
       "      <td>crossleyi</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Cheirogaleus_crossleyi_ssp</td>\n",
       "      <td>C.crossleyi_ssp</td>\n",
       "      <td>Cheirogaleus_crossleyi_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>wrong label; correct: C.major</td>\n",
       "      <td>fail</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>395</th>\n",
       "      <td>PD_0055</td>\n",
       "      <td>PD_0055</td>\n",
       "      <td>OWM</td>\n",
       "      <td>Cercopithecidae</td>\n",
       "      <td>Colobus</td>\n",
       "      <td>polykomos</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Colobus_polykomos_ssp</td>\n",
       "      <td>C.polykomos_ssp</td>\n",
       "      <td>Colobus_polykomos_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fail</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>473</th>\n",
       "      <td>SAMEA116330139</td>\n",
       "      <td>SAMEA116330139</td>\n",
       "      <td>STR</td>\n",
       "      <td>Lemuridae</td>\n",
       "      <td>Eulemur</td>\n",
       "      <td>cinereiceps</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Eulemur_cinereiceps_ssp</td>\n",
       "      <td>E.cinereiceps_ssp</td>\n",
       "      <td>Eulemur_cinereiceps_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>731</th>\n",
       "      <td>SAMEA116330175</td>\n",
       "      <td>SAMEA116330175</td>\n",
       "      <td>STR</td>\n",
       "      <td>Lemuridae</td>\n",
       "      <td>Hapalemur</td>\n",
       "      <td>aureus</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Hapalemur_aureus_ssp</td>\n",
       "      <td>H.aureus_ssp</td>\n",
       "      <td>Hapalemur_aureus_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fail</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>742</th>\n",
       "      <td>PD_0160</td>\n",
       "      <td>PD_0160</td>\n",
       "      <td>STR</td>\n",
       "      <td>Lemuridae</td>\n",
       "      <td>Hapalemur</td>\n",
       "      <td>occidentalis</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Hapalemur_occidentalis_ssp</td>\n",
       "      <td>H.occidentalis_ssp</td>\n",
       "      <td>Hapalemur_occidentalis_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fail</td>\n",
       "      <td>PASS</td>\n",
       "      <td>WARN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1593</th>\n",
       "      <td>SAMEA116330194</td>\n",
       "      <td>SAMEA116330194</td>\n",
       "      <td>STR</td>\n",
       "      <td>Lepilemuridae</td>\n",
       "      <td>Lepilemur</td>\n",
       "      <td>sahamalazensis</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Lepilemur_sahamalazensis_ssp</td>\n",
       "      <td>L.sahamalazensis_ssp</td>\n",
       "      <td>Lepilemur_sahamalazensis_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2737</th>\n",
       "      <td>PD_0083</td>\n",
       "      <td>PD_0083</td>\n",
       "      <td>PLA</td>\n",
       "      <td>Pitheciidae</td>\n",
       "      <td>Pithecia</td>\n",
       "      <td>pissinatti</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Pithecia_pissinatti_ssp</td>\n",
       "      <td>P.pissinatti_ssp</td>\n",
       "      <td>Pithecia_pissinatti_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fail</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2738</th>\n",
       "      <td>PD_0144</td>\n",
       "      <td>PD_0144</td>\n",
       "      <td>PLA</td>\n",
       "      <td>Pitheciidae</td>\n",
       "      <td>Pithecia</td>\n",
       "      <td>pissinatti</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Pithecia_pissinatti_ssp</td>\n",
       "      <td>P.pissinatti_ssp</td>\n",
       "      <td>Pithecia_pissinatti_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fail</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2739</th>\n",
       "      <td>PD_0331</td>\n",
       "      <td>PD_0331</td>\n",
       "      <td>PLA</td>\n",
       "      <td>Pitheciidae</td>\n",
       "      <td>Pithecia</td>\n",
       "      <td>pissinatti</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Pithecia_pissinatti_ssp</td>\n",
       "      <td>P.pissinatti_ssp</td>\n",
       "      <td>Pithecia_pissinatti_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>fail</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2846</th>\n",
       "      <td>SAMEA116330132</td>\n",
       "      <td>SAMEA116330132</td>\n",
       "      <td>STR</td>\n",
       "      <td>Indriidae</td>\n",
       "      <td>Avahi</td>\n",
       "      <td>occidentalis</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Avahi_occidentalis_ssp</td>\n",
       "      <td>A.occidentalis_ssp</td>\n",
       "      <td>Avahi_occidentalis_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>10 rows × 93 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                  ID         ID_PGDP group           family         genus  \\\n",
       "199     SAMN11334319    SAMN11334319   STR   Cheirogaleidae  Cheirogaleus   \n",
       "395          PD_0055         PD_0055   OWM  Cercopithecidae       Colobus   \n",
       "473   SAMEA116330139  SAMEA116330139   STR        Lemuridae       Eulemur   \n",
       "731   SAMEA116330175  SAMEA116330175   STR        Lemuridae     Hapalemur   \n",
       "742          PD_0160         PD_0160   STR        Lemuridae     Hapalemur   \n",
       "1593  SAMEA116330194  SAMEA116330194   STR    Lepilemuridae     Lepilemur   \n",
       "2737         PD_0083         PD_0083   PLA      Pitheciidae      Pithecia   \n",
       "2738         PD_0144         PD_0144   PLA      Pitheciidae      Pithecia   \n",
       "2739         PD_0331         PD_0331   PLA      Pitheciidae      Pithecia   \n",
       "2846  SAMEA116330132  SAMEA116330132   STR        Indriidae         Avahi   \n",
       "\n",
       "             species subspecies                  species_full  \\\n",
       "199        crossleyi        ssp    Cheirogaleus_crossleyi_ssp   \n",
       "395        polykomos        ssp         Colobus_polykomos_ssp   \n",
       "473      cinereiceps        ssp       Eulemur_cinereiceps_ssp   \n",
       "731           aureus        ssp          Hapalemur_aureus_ssp   \n",
       "742     occidentalis        ssp    Hapalemur_occidentalis_ssp   \n",
       "1593  sahamalazensis        ssp  Lepilemur_sahamalazensis_ssp   \n",
       "2737      pissinatti        ssp       Pithecia_pissinatti_ssp   \n",
       "2738      pissinatti        ssp       Pithecia_pissinatti_ssp   \n",
       "2739      pissinatti        ssp       Pithecia_pissinatti_ssp   \n",
       "2846    occidentalis        ssp        Avahi_occidentalis_ssp   \n",
       "\n",
       "             species_short            species_genotyping  ...  \\\n",
       "199        C.crossleyi_ssp    Cheirogaleus_crossleyi_ssp  ...   \n",
       "395        C.polykomos_ssp         Colobus_polykomos_ssp  ...   \n",
       "473      E.cinereiceps_ssp       Eulemur_cinereiceps_ssp  ...   \n",
       "731           H.aureus_ssp          Hapalemur_aureus_ssp  ...   \n",
       "742     H.occidentalis_ssp    Hapalemur_occidentalis_ssp  ...   \n",
       "1593  L.sahamalazensis_ssp  Lepilemur_sahamalazensis_ssp  ...   \n",
       "2737      P.pissinatti_ssp       Pithecia_pissinatti_ssp  ...   \n",
       "2738      P.pissinatti_ssp       Pithecia_pissinatti_ssp  ...   \n",
       "2739      P.pissinatti_ssp       Pithecia_pissinatti_ssp  ...   \n",
       "2846    A.occidentalis_ssp        Avahi_occidentalis_ssp  ...   \n",
       "\n",
       "                             reason finalQC  Kuderna_AB_QC  Kuderna_COV_QC  \\\n",
       "199   wrong label; correct: C.major    fail            NaN             NaN   \n",
       "395                             NaN    fail            NaN             NaN   \n",
       "473                             NaN      ok            NaN             NaN   \n",
       "731                             NaN    fail            NaN             NaN   \n",
       "742                             NaN    fail           PASS            WARN   \n",
       "1593                            NaN      ok            NaN             NaN   \n",
       "2737                            NaN    fail            NaN             NaN   \n",
       "2738                            NaN    fail            NaN             NaN   \n",
       "2739                            NaN    fail            NaN             NaN   \n",
       "2846                            NaN      ok            NaN             NaN   \n",
       "\n",
       "      relatives  remove_as_relative  generation  founder  remove_manual  \\\n",
       "199         NaN                 NaN         NaN      NaN            NaN   \n",
       "395         NaN                 NaN         NaN      NaN            NaN   \n",
       "473         NaN                 NaN         NaN      NaN            NaN   \n",
       "731         NaN                 NaN         NaN      NaN            NaN   \n",
       "742         NaN                 NaN         NaN      NaN            NaN   \n",
       "1593        NaN                 NaN         NaN      NaN            NaN   \n",
       "2737        NaN                 NaN         NaN      NaN            NaN   \n",
       "2738        NaN                 NaN         NaN      NaN            NaN   \n",
       "2739        NaN                 NaN         NaN      NaN            NaN   \n",
       "2846        NaN                 NaN         NaN      NaN            NaN   \n",
       "\n",
       "      pedigree_source  \n",
       "199               NaN  \n",
       "395               NaN  \n",
       "473               NaN  \n",
       "731               NaN  \n",
       "742               NaN  \n",
       "1593              NaN  \n",
       "2737              NaN  \n",
       "2738              NaN  \n",
       "2739              NaN  \n",
       "2846              NaN  \n",
       "\n",
       "[10 rows x 93 columns]"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vasili_table.loc[(~vasili_table.species.isin(sub_vasili.species.unique())) & (vasili_table.cov_chrA >= 20)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "4ff522b4",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>ID_PGDP</th>\n",
       "      <th>group</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "      <th>subspecies</th>\n",
       "      <th>species_full</th>\n",
       "      <th>species_short</th>\n",
       "      <th>species_genotyping</th>\n",
       "      <th>...</th>\n",
       "      <th>reason</th>\n",
       "      <th>finalQC</th>\n",
       "      <th>Kuderna_AB_QC</th>\n",
       "      <th>Kuderna_COV_QC</th>\n",
       "      <th>relatives</th>\n",
       "      <th>remove_as_relative</th>\n",
       "      <th>generation</th>\n",
       "      <th>founder</th>\n",
       "      <th>remove_manual</th>\n",
       "      <th>pedigree_source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>146</th>\n",
       "      <td>PD_0629</td>\n",
       "      <td>PD_0629</td>\n",
       "      <td>TAR</td>\n",
       "      <td>Tarsiidae</td>\n",
       "      <td>Carlito</td>\n",
       "      <td>syrichta</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Carlito_syrichta_ssp</td>\n",
       "      <td>C.syrichta_ssp</td>\n",
       "      <td>Carlito_syrichta_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>WARN</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 93 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "          ID  ID_PGDP group     family    genus   species subspecies  \\\n",
       "146  PD_0629  PD_0629   TAR  Tarsiidae  Carlito  syrichta        ssp   \n",
       "\n",
       "             species_full   species_short    species_genotyping  ... reason  \\\n",
       "146  Carlito_syrichta_ssp  C.syrichta_ssp  Carlito_syrichta_ssp  ...    NaN   \n",
       "\n",
       "    finalQC  Kuderna_AB_QC  Kuderna_COV_QC  relatives  remove_as_relative  \\\n",
       "146      ok           WARN            PASS        NaN                 NaN   \n",
       "\n",
       "     generation  founder  remove_manual  pedigree_source  \n",
       "146         NaN      NaN            NaN              NaN  \n",
       "\n",
       "[1 rows x 93 columns]"
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "vasili_table.loc[vasili_table.species == \"syrichta\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "f6ddf9ca",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(240, 92)"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_sub = sub_vasili[[\"gSEX\", \"species\"]].value_counts().reset_index()\n",
    "len(count_sub.species.unique()), len(count_sub.loc[(count_sub.gSEX == \"M\") & (count_sub[\"count\"] == 1)])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "e39796dd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "240"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(count_sub.species.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "d1acd458",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "193"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(count_sub.loc[(count_sub.gSEX == \"F\") | ((count_sub.gSEX == \"M\") & (count_sub[\"count\"] >= 2))].species.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "1e5eee3a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>gSEX</th>\n",
       "      <th>species</th>\n",
       "      <th>count</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>F</td>\n",
       "      <td>fascicularis</td>\n",
       "      <td>104</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>F</td>\n",
       "      <td>troglodytes</td>\n",
       "      <td>55</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>F</td>\n",
       "      <td>anubis</td>\n",
       "      <td>54</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>F</td>\n",
       "      <td>mulatta</td>\n",
       "      <td>53</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>F</td>\n",
       "      <td>fuscata</td>\n",
       "      <td>42</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>339</th>\n",
       "      <td>F</td>\n",
       "      <td>bicolor</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>340</th>\n",
       "      <td>F</td>\n",
       "      <td>hamlyni</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>341</th>\n",
       "      <td>F</td>\n",
       "      <td>brunnescens</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>342</th>\n",
       "      <td>F</td>\n",
       "      <td>avunculus</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>345</th>\n",
       "      <td>F</td>\n",
       "      <td>brunneus</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>166 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    gSEX       species  count\n",
       "0      F  fascicularis    104\n",
       "1      F   troglodytes     55\n",
       "2      F        anubis     54\n",
       "3      F       mulatta     53\n",
       "6      F       fuscata     42\n",
       "..   ...           ...    ...\n",
       "339    F       bicolor      1\n",
       "340    F       hamlyni      1\n",
       "341    F   brunnescens      1\n",
       "342    F     avunculus      1\n",
       "345    F      brunneus      1\n",
       "\n",
       "[166 rows x 3 columns]"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count_sub.loc[(count_sub.gSEX == \"F\")]"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "531a9931",
   "metadata": {},
   "source": [
    "So after filtering, there are 166 species with at least one female in which you can determine chromosome X.\n",
    "Adding samples with more than 1 male, we reach 195 (and assuming a present chrX). The last species only have a single male (or no identified females)."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "becfb96e",
   "metadata": {},
   "outputs": [],
   "source": [
    "sub_vasili = vasili_table.loc[(vasili_table.finalQC != \"fail\")\n",
    "                              & (vasili_table.cov_chrA >= 20)\n",
    "                              & (vasili_table.remove_as_relative != True)\n",
    "                              & (vasili_table.remove_manual != True)\n",
    "                              & (~vasili_table.ID.str.startswith(\"SAMEA11633\"))\n",
    "                             ]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "id": "17715451",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "starting_dir = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024/{}/filteredVCF/all_samples/bcf_step1/\"\n",
    "\n",
    "for species in sub_vasili.species_full.unique()[:2]:\n",
    "    species_inds = sub_vasili.loc[sub_vasili.species_full == species]\n",
    "    short_form = species.split(\"_\")[0]\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    reference = species_inds[\"reference\"].unique()\n",
    "    sample_names = \",\".join(species_inds[\"ID\"])\n",
    "    all_chr_file = starting_dir.format(species)+species+\"_all_chr.sorted.bcf\"\n",
    "    if not os.path.exists(all_chr_file):\n",
    "        print(\"Filtered VCF does not exist \", species)\n",
    "        continue\n",
    "    if len(reference) > 1:\n",
    "        print(\"Too many references \", species)\n",
    "        continue\n",
    "    zarr_input = []\n",
    "    # Chromosome X\n",
    "    zarr_input.extend(regions_df.loc[(regions_df.END >= 1000000) &\n",
    "                                  (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                                  (regions_df.MALE_PLOIDY == 1) &\n",
    "          (regions_df.REFERENCE_FOLDER == reference[0]+\"_ssp\")].CONTIG_ID.unique())"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "2f4899e9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['NW_026988415.1',\n",
       " 'NW_026988472.1',\n",
       " 'NW_026988488.1',\n",
       " 'NW_026988496.1',\n",
       " 'NW_026988553.1']"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "zarr_input"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "id": "91eb5b06",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "(array(['Aotus_nancymaae'], dtype=object), 'Aotus_azarai_ssp')"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "reference, species"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "dc6e387a",
   "metadata": {},
   "source": [
    "Some files fail dus to lack of information."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "0a336faa",
   "metadata": {},
   "outputs": [],
   "source": [
    "starting_dir = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024/{}/filteredVCF/all_samples/bcf_step1/\"\n",
    "# starting_dir_gvcf = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024/{}/gVCF/\"\n",
    "metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\" # Performing all selection with Vasili's stats\n",
    "zarr_dir = \"zarr_data_20x/\"\n",
    "vasili_stats = \"~/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/plots/SupTable_Sample_Stats_wGT_QC.tsv\"\n",
    "\n",
    "\n",
    "vasili_table = pd.read_csv(vasili_stats, sep=\"\\t\")\n",
    "sub_vasili = vasili_table.loc[(vasili_table.finalQC != \"fail\")\n",
    "                              & (vasili_table.cov_chrA >= 20)\n",
    "                              & (vasili_table.remove_as_relative != True)\n",
    "                              & (vasili_table.remove_manual != True)\n",
    "                              & (~vasili_table.ID.str.startswith(\"SAMEA11633\"))\n",
    "                             ]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "7d437567",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Allenopithecus_nigroviridis_ssp Allenopithecus\n",
      "Aotus_azarai_ssp Aotus\n"
     ]
    }
   ],
   "source": [
    "\n",
    "for species in sub_vasili.species_full.unique()[:2]:\n",
    "    species_inds = sub_vasili.loc[sub_vasili.species_full == species]\n",
    "    short_form = species.split(\"_\")[0]\n",
    "    print(species, short_form)\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    reference = species_inds[\"reference\"].unique()\n",
    "    sample_names = \",\".join(species_inds[\"ID\"])\n",
    "    all_chr_file = starting_dir.format(species)+species+\"_all_chr.sorted.bcf\"\n",
    "    if not os.path.exists(all_chr_file):\n",
    "        print(\"Filtered VCF does not exist \", species)\n",
    "        continue\n",
    "    if len(reference) > 1:\n",
    "        print(\"Too many references \", species)\n",
    "        continue\n",
    "    zarr_input = []"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "1125b54f",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ID</th>\n",
       "      <th>ID_PGDP</th>\n",
       "      <th>group</th>\n",
       "      <th>family</th>\n",
       "      <th>genus</th>\n",
       "      <th>species</th>\n",
       "      <th>subspecies</th>\n",
       "      <th>species_full</th>\n",
       "      <th>species_short</th>\n",
       "      <th>species_genotyping</th>\n",
       "      <th>...</th>\n",
       "      <th>reason</th>\n",
       "      <th>finalQC</th>\n",
       "      <th>Kuderna_AB_QC</th>\n",
       "      <th>Kuderna_COV_QC</th>\n",
       "      <th>relatives</th>\n",
       "      <th>remove_as_relative</th>\n",
       "      <th>generation</th>\n",
       "      <th>founder</th>\n",
       "      <th>remove_manual</th>\n",
       "      <th>pedigree_source</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>495</th>\n",
       "      <td>SAMEA2697037</td>\n",
       "      <td>SAMEA2697037</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>beringei</td>\n",
       "      <td>Gorilla_beringei_beringei</td>\n",
       "      <td>G.beringei_beringei</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>497</th>\n",
       "      <td>SAMEA2697039</td>\n",
       "      <td>SAMEA2697039</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>beringei</td>\n",
       "      <td>Gorilla_beringei_beringei</td>\n",
       "      <td>G.beringei_beringei</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>498</th>\n",
       "      <td>SAMEA3939556</td>\n",
       "      <td>SAMEA3939556</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_beringei_beringei</td>\n",
       "      <td>G.beringei_beringei</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>499</th>\n",
       "      <td>SAMEA3939557</td>\n",
       "      <td>SAMEA3939557</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_beringei_beringei</td>\n",
       "      <td>G.beringei_beringei</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>500</th>\n",
       "      <td>SAMEA3939558</td>\n",
       "      <td>SAMEA3939558</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_beringei_beringei</td>\n",
       "      <td>G.beringei_beringei</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>502</th>\n",
       "      <td>SAMEA3939560</td>\n",
       "      <td>SAMEA3939560</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_beringei_beringei</td>\n",
       "      <td>G.beringei_beringei</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>503</th>\n",
       "      <td>SAMEA3939561</td>\n",
       "      <td>SAMEA3939561</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_beringei_beringei</td>\n",
       "      <td>G.beringei_beringei</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>504</th>\n",
       "      <td>SAMEA3991715</td>\n",
       "      <td>SAMEA3991715</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_beringei_beringei</td>\n",
       "      <td>G.beringei_beringei</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>509</th>\n",
       "      <td>SAMEA2697040</td>\n",
       "      <td>SAMEA2697040</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>graueri</td>\n",
       "      <td>Gorilla_beringei_graueri</td>\n",
       "      <td>G.beringei_graueri</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>510</th>\n",
       "      <td>SAMEA2697041</td>\n",
       "      <td>SAMEA2697041</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>graueri</td>\n",
       "      <td>Gorilla_beringei_graueri</td>\n",
       "      <td>G.beringei_graueri</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>511</th>\n",
       "      <td>SAMEA2697042</td>\n",
       "      <td>SAMEA2697042</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>graueri</td>\n",
       "      <td>Gorilla_beringei_graueri</td>\n",
       "      <td>G.beringei_graueri</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>514</th>\n",
       "      <td>SAMN01920474</td>\n",
       "      <td>SAMN01920474</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>graueri</td>\n",
       "      <td>Gorilla_beringei_graueri</td>\n",
       "      <td>G.beringei_graueri</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>515</th>\n",
       "      <td>SAMN01920475</td>\n",
       "      <td>SAMN01920475</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>graueri</td>\n",
       "      <td>Gorilla_beringei_graueri</td>\n",
       "      <td>G.beringei_graueri</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>WARN</td>\n",
       "      <td>PASS</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>516</th>\n",
       "      <td>PD_0179</td>\n",
       "      <td>PD_0179</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>G.beringei_ssp</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>warn</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>517</th>\n",
       "      <td>SAMEA1692350</td>\n",
       "      <td>SAMEA1692350</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>G.beringei_ssp</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>518</th>\n",
       "      <td>SAMEA1692351</td>\n",
       "      <td>SAMEA1692351</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>G.beringei_ssp</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMEA2697038</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>519</th>\n",
       "      <td>SAMEA1692352</td>\n",
       "      <td>SAMEA1692352</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>G.beringei_ssp</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>520</th>\n",
       "      <td>SAMEA1692353</td>\n",
       "      <td>SAMEA1692353</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>G.beringei_ssp</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>521</th>\n",
       "      <td>SAMEA1692354</td>\n",
       "      <td>SAMEA1692354</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>G.beringei_ssp</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>522</th>\n",
       "      <td>SAMEA1692355</td>\n",
       "      <td>SAMEA1692355</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>beringei</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>G.beringei_ssp</td>\n",
       "      <td>Gorilla_beringei_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>630</th>\n",
       "      <td>SAMEA5204233</td>\n",
       "      <td>SAMEA5204233</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>Gorilla_gorilla_gorilla</td>\n",
       "      <td>G.gorilla_gorilla</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMEA5204235; SAMEA5204237</td>\n",
       "      <td>False</td>\n",
       "      <td>P</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ncbi</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>631</th>\n",
       "      <td>SAMEA5204234</td>\n",
       "      <td>SAMEA5204234</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>Gorilla_gorilla_gorilla</td>\n",
       "      <td>G.gorilla_gorilla</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMEA5204235; SAMN01920489</td>\n",
       "      <td>False</td>\n",
       "      <td>P</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ncbi</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>633</th>\n",
       "      <td>SAMEA5204236</td>\n",
       "      <td>SAMEA5204236</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>Gorilla_gorilla_gorilla</td>\n",
       "      <td>G.gorilla_gorilla</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMEA5204237</td>\n",
       "      <td>False</td>\n",
       "      <td>P</td>\n",
       "      <td>True</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ncbi</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>635</th>\n",
       "      <td>SAMEA5204238</td>\n",
       "      <td>SAMEA5204238</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>Gorilla_gorilla_gorilla</td>\n",
       "      <td>G.gorilla_gorilla</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMEA5204237</td>\n",
       "      <td>False</td>\n",
       "      <td>F2</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ncbi</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>648</th>\n",
       "      <td>SAMN01920484</td>\n",
       "      <td>SAMN01920484</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>Gorilla_gorilla_gorilla</td>\n",
       "      <td>G.gorilla_gorilla</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>649</th>\n",
       "      <td>SAMN01920485</td>\n",
       "      <td>SAMN01920485</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>Gorilla_gorilla_gorilla</td>\n",
       "      <td>G.gorilla_gorilla</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>650</th>\n",
       "      <td>SAMN01920486</td>\n",
       "      <td>SAMN01920486</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>Gorilla_gorilla_gorilla</td>\n",
       "      <td>G.gorilla_gorilla</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>651</th>\n",
       "      <td>SAMN01920487</td>\n",
       "      <td>SAMN01920487</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>Gorilla_gorilla_gorilla</td>\n",
       "      <td>G.gorilla_gorilla</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>652</th>\n",
       "      <td>SAMN01920488</td>\n",
       "      <td>SAMN01920488</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>Gorilla_gorilla_gorilla</td>\n",
       "      <td>G.gorilla_gorilla</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>675</th>\n",
       "      <td>SAMN12714420</td>\n",
       "      <td>SAMN12714420</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_gorilla_gorilla</td>\n",
       "      <td>G.gorilla_gorilla</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>676</th>\n",
       "      <td>SAMN12714421</td>\n",
       "      <td>SAMN12714421</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_gorilla_gorilla</td>\n",
       "      <td>G.gorilla_gorilla</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>677</th>\n",
       "      <td>SAMN35877944</td>\n",
       "      <td>SAMN35877944</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>Gorilla_gorilla_gorilla</td>\n",
       "      <td>G.gorilla_gorilla</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMN04003007</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>679</th>\n",
       "      <td>PD_0180</td>\n",
       "      <td>PD_0180</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>G.gorilla_ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>682</th>\n",
       "      <td>PD_0183</td>\n",
       "      <td>PD_0183</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>G.gorilla_ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>683</th>\n",
       "      <td>PD_0184</td>\n",
       "      <td>PD_0184</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>G.gorilla_ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>685</th>\n",
       "      <td>PD_0186</td>\n",
       "      <td>PD_0186</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>G.gorilla_ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>686</th>\n",
       "      <td>PD_0187</td>\n",
       "      <td>PD_0187</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>G.gorilla_ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>PASS</td>\n",
       "      <td>FAIL</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>687</th>\n",
       "      <td>PD_0188</td>\n",
       "      <td>PD_0188</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>G.gorilla_ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>689</th>\n",
       "      <td>PD_0190</td>\n",
       "      <td>PD_0190</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>G.gorilla_ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>690</th>\n",
       "      <td>PD_0191</td>\n",
       "      <td>PD_0191</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>G.gorilla_ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>691</th>\n",
       "      <td>PD_0192</td>\n",
       "      <td>PD_0192</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>G.gorilla_ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>692</th>\n",
       "      <td>PD_0193</td>\n",
       "      <td>PD_0193</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>G.gorilla_ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693</th>\n",
       "      <td>PD_0194</td>\n",
       "      <td>PD_0194</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>G.gorilla_ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>694</th>\n",
       "      <td>PD_0195</td>\n",
       "      <td>PD_0195</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>G.gorilla_ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>695</th>\n",
       "      <td>PD_0196</td>\n",
       "      <td>PD_0196</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>G.gorilla_ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>696</th>\n",
       "      <td>PD_0197</td>\n",
       "      <td>PD_0197</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>G.gorilla_ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>SAMEA5204235</td>\n",
       "      <td>False</td>\n",
       "      <td>F2</td>\n",
       "      <td>False</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ncbi</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>697</th>\n",
       "      <td>PD_0198</td>\n",
       "      <td>PD_0198</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>G.gorilla_ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>698</th>\n",
       "      <td>PD_0260</td>\n",
       "      <td>PD_0260</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>G.gorilla_ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>PASS</td>\n",
       "      <td>WARN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>701</th>\n",
       "      <td>SAMEA2446072</td>\n",
       "      <td>SAMEA2446072</td>\n",
       "      <td>APE</td>\n",
       "      <td>Hominidae</td>\n",
       "      <td>Gorilla</td>\n",
       "      <td>gorilla</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>G.gorilla_ssp</td>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ok</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>49 rows × 93 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               ID       ID_PGDP group     family    genus   species  \\\n",
       "495  SAMEA2697037  SAMEA2697037   APE  Hominidae  Gorilla  beringei   \n",
       "497  SAMEA2697039  SAMEA2697039   APE  Hominidae  Gorilla  beringei   \n",
       "498  SAMEA3939556  SAMEA3939556   APE  Hominidae  Gorilla  beringei   \n",
       "499  SAMEA3939557  SAMEA3939557   APE  Hominidae  Gorilla  beringei   \n",
       "500  SAMEA3939558  SAMEA3939558   APE  Hominidae  Gorilla  beringei   \n",
       "502  SAMEA3939560  SAMEA3939560   APE  Hominidae  Gorilla  beringei   \n",
       "503  SAMEA3939561  SAMEA3939561   APE  Hominidae  Gorilla  beringei   \n",
       "504  SAMEA3991715  SAMEA3991715   APE  Hominidae  Gorilla  beringei   \n",
       "509  SAMEA2697040  SAMEA2697040   APE  Hominidae  Gorilla  beringei   \n",
       "510  SAMEA2697041  SAMEA2697041   APE  Hominidae  Gorilla  beringei   \n",
       "511  SAMEA2697042  SAMEA2697042   APE  Hominidae  Gorilla  beringei   \n",
       "514  SAMN01920474  SAMN01920474   APE  Hominidae  Gorilla  beringei   \n",
       "515  SAMN01920475  SAMN01920475   APE  Hominidae  Gorilla  beringei   \n",
       "516       PD_0179       PD_0179   APE  Hominidae  Gorilla  beringei   \n",
       "517  SAMEA1692350  SAMEA1692350   APE  Hominidae  Gorilla  beringei   \n",
       "518  SAMEA1692351  SAMEA1692351   APE  Hominidae  Gorilla  beringei   \n",
       "519  SAMEA1692352  SAMEA1692352   APE  Hominidae  Gorilla  beringei   \n",
       "520  SAMEA1692353  SAMEA1692353   APE  Hominidae  Gorilla  beringei   \n",
       "521  SAMEA1692354  SAMEA1692354   APE  Hominidae  Gorilla  beringei   \n",
       "522  SAMEA1692355  SAMEA1692355   APE  Hominidae  Gorilla  beringei   \n",
       "630  SAMEA5204233  SAMEA5204233   APE  Hominidae  Gorilla   gorilla   \n",
       "631  SAMEA5204234  SAMEA5204234   APE  Hominidae  Gorilla   gorilla   \n",
       "633  SAMEA5204236  SAMEA5204236   APE  Hominidae  Gorilla   gorilla   \n",
       "635  SAMEA5204238  SAMEA5204238   APE  Hominidae  Gorilla   gorilla   \n",
       "648  SAMN01920484  SAMN01920484   APE  Hominidae  Gorilla   gorilla   \n",
       "649  SAMN01920485  SAMN01920485   APE  Hominidae  Gorilla   gorilla   \n",
       "650  SAMN01920486  SAMN01920486   APE  Hominidae  Gorilla   gorilla   \n",
       "651  SAMN01920487  SAMN01920487   APE  Hominidae  Gorilla   gorilla   \n",
       "652  SAMN01920488  SAMN01920488   APE  Hominidae  Gorilla   gorilla   \n",
       "675  SAMN12714420  SAMN12714420   APE  Hominidae  Gorilla   gorilla   \n",
       "676  SAMN12714421  SAMN12714421   APE  Hominidae  Gorilla   gorilla   \n",
       "677  SAMN35877944  SAMN35877944   APE  Hominidae  Gorilla   gorilla   \n",
       "679       PD_0180       PD_0180   APE  Hominidae  Gorilla   gorilla   \n",
       "682       PD_0183       PD_0183   APE  Hominidae  Gorilla   gorilla   \n",
       "683       PD_0184       PD_0184   APE  Hominidae  Gorilla   gorilla   \n",
       "685       PD_0186       PD_0186   APE  Hominidae  Gorilla   gorilla   \n",
       "686       PD_0187       PD_0187   APE  Hominidae  Gorilla   gorilla   \n",
       "687       PD_0188       PD_0188   APE  Hominidae  Gorilla   gorilla   \n",
       "689       PD_0190       PD_0190   APE  Hominidae  Gorilla   gorilla   \n",
       "690       PD_0191       PD_0191   APE  Hominidae  Gorilla   gorilla   \n",
       "691       PD_0192       PD_0192   APE  Hominidae  Gorilla   gorilla   \n",
       "692       PD_0193       PD_0193   APE  Hominidae  Gorilla   gorilla   \n",
       "693       PD_0194       PD_0194   APE  Hominidae  Gorilla   gorilla   \n",
       "694       PD_0195       PD_0195   APE  Hominidae  Gorilla   gorilla   \n",
       "695       PD_0196       PD_0196   APE  Hominidae  Gorilla   gorilla   \n",
       "696       PD_0197       PD_0197   APE  Hominidae  Gorilla   gorilla   \n",
       "697       PD_0198       PD_0198   APE  Hominidae  Gorilla   gorilla   \n",
       "698       PD_0260       PD_0260   APE  Hominidae  Gorilla   gorilla   \n",
       "701  SAMEA2446072  SAMEA2446072   APE  Hominidae  Gorilla   gorilla   \n",
       "\n",
       "    subspecies               species_full        species_short  \\\n",
       "495   beringei  Gorilla_beringei_beringei  G.beringei_beringei   \n",
       "497   beringei  Gorilla_beringei_beringei  G.beringei_beringei   \n",
       "498        ssp  Gorilla_beringei_beringei  G.beringei_beringei   \n",
       "499        ssp  Gorilla_beringei_beringei  G.beringei_beringei   \n",
       "500        ssp  Gorilla_beringei_beringei  G.beringei_beringei   \n",
       "502        ssp  Gorilla_beringei_beringei  G.beringei_beringei   \n",
       "503        ssp  Gorilla_beringei_beringei  G.beringei_beringei   \n",
       "504        ssp  Gorilla_beringei_beringei  G.beringei_beringei   \n",
       "509    graueri   Gorilla_beringei_graueri   G.beringei_graueri   \n",
       "510    graueri   Gorilla_beringei_graueri   G.beringei_graueri   \n",
       "511    graueri   Gorilla_beringei_graueri   G.beringei_graueri   \n",
       "514    graueri   Gorilla_beringei_graueri   G.beringei_graueri   \n",
       "515    graueri   Gorilla_beringei_graueri   G.beringei_graueri   \n",
       "516        ssp       Gorilla_beringei_ssp       G.beringei_ssp   \n",
       "517        ssp       Gorilla_beringei_ssp       G.beringei_ssp   \n",
       "518        ssp       Gorilla_beringei_ssp       G.beringei_ssp   \n",
       "519        ssp       Gorilla_beringei_ssp       G.beringei_ssp   \n",
       "520        ssp       Gorilla_beringei_ssp       G.beringei_ssp   \n",
       "521        ssp       Gorilla_beringei_ssp       G.beringei_ssp   \n",
       "522        ssp       Gorilla_beringei_ssp       G.beringei_ssp   \n",
       "630    gorilla    Gorilla_gorilla_gorilla    G.gorilla_gorilla   \n",
       "631    gorilla    Gorilla_gorilla_gorilla    G.gorilla_gorilla   \n",
       "633    gorilla    Gorilla_gorilla_gorilla    G.gorilla_gorilla   \n",
       "635    gorilla    Gorilla_gorilla_gorilla    G.gorilla_gorilla   \n",
       "648    gorilla    Gorilla_gorilla_gorilla    G.gorilla_gorilla   \n",
       "649    gorilla    Gorilla_gorilla_gorilla    G.gorilla_gorilla   \n",
       "650    gorilla    Gorilla_gorilla_gorilla    G.gorilla_gorilla   \n",
       "651    gorilla    Gorilla_gorilla_gorilla    G.gorilla_gorilla   \n",
       "652    gorilla    Gorilla_gorilla_gorilla    G.gorilla_gorilla   \n",
       "675        ssp    Gorilla_gorilla_gorilla    G.gorilla_gorilla   \n",
       "676        ssp    Gorilla_gorilla_gorilla    G.gorilla_gorilla   \n",
       "677    gorilla    Gorilla_gorilla_gorilla    G.gorilla_gorilla   \n",
       "679        ssp        Gorilla_gorilla_ssp        G.gorilla_ssp   \n",
       "682        ssp        Gorilla_gorilla_ssp        G.gorilla_ssp   \n",
       "683        ssp        Gorilla_gorilla_ssp        G.gorilla_ssp   \n",
       "685        ssp        Gorilla_gorilla_ssp        G.gorilla_ssp   \n",
       "686        ssp        Gorilla_gorilla_ssp        G.gorilla_ssp   \n",
       "687        ssp        Gorilla_gorilla_ssp        G.gorilla_ssp   \n",
       "689        ssp        Gorilla_gorilla_ssp        G.gorilla_ssp   \n",
       "690        ssp        Gorilla_gorilla_ssp        G.gorilla_ssp   \n",
       "691        ssp        Gorilla_gorilla_ssp        G.gorilla_ssp   \n",
       "692        ssp        Gorilla_gorilla_ssp        G.gorilla_ssp   \n",
       "693        ssp        Gorilla_gorilla_ssp        G.gorilla_ssp   \n",
       "694        ssp        Gorilla_gorilla_ssp        G.gorilla_ssp   \n",
       "695        ssp        Gorilla_gorilla_ssp        G.gorilla_ssp   \n",
       "696        ssp        Gorilla_gorilla_ssp        G.gorilla_ssp   \n",
       "697        ssp        Gorilla_gorilla_ssp        G.gorilla_ssp   \n",
       "698        ssp        Gorilla_gorilla_ssp        G.gorilla_ssp   \n",
       "701        ssp        Gorilla_gorilla_ssp        G.gorilla_ssp   \n",
       "\n",
       "       species_genotyping  ... reason finalQC  Kuderna_AB_QC  Kuderna_COV_QC  \\\n",
       "495  Gorilla_beringei_ssp  ...    NaN      ok            NaN             NaN   \n",
       "497  Gorilla_beringei_ssp  ...    NaN      ok            NaN             NaN   \n",
       "498  Gorilla_beringei_ssp  ...    NaN      ok            NaN             NaN   \n",
       "499  Gorilla_beringei_ssp  ...    NaN      ok            NaN             NaN   \n",
       "500  Gorilla_beringei_ssp  ...    NaN      ok            NaN             NaN   \n",
       "502  Gorilla_beringei_ssp  ...    NaN      ok            NaN             NaN   \n",
       "503  Gorilla_beringei_ssp  ...    NaN      ok            NaN             NaN   \n",
       "504  Gorilla_beringei_ssp  ...    NaN      ok            NaN             NaN   \n",
       "509  Gorilla_beringei_ssp  ...    NaN      ok            NaN             NaN   \n",
       "510  Gorilla_beringei_ssp  ...    NaN      ok            NaN             NaN   \n",
       "511  Gorilla_beringei_ssp  ...    NaN      ok            NaN             NaN   \n",
       "514  Gorilla_beringei_ssp  ...    NaN      ok            NaN             NaN   \n",
       "515  Gorilla_beringei_ssp  ...    NaN      ok           WARN            PASS   \n",
       "516  Gorilla_beringei_ssp  ...    NaN    warn            NaN             NaN   \n",
       "517  Gorilla_beringei_ssp  ...    NaN      ok            NaN             NaN   \n",
       "518  Gorilla_beringei_ssp  ...    NaN      ok            NaN             NaN   \n",
       "519  Gorilla_beringei_ssp  ...    NaN      ok            NaN             NaN   \n",
       "520  Gorilla_beringei_ssp  ...    NaN      ok            NaN             NaN   \n",
       "521  Gorilla_beringei_ssp  ...    NaN      ok            NaN             NaN   \n",
       "522  Gorilla_beringei_ssp  ...    NaN      ok            NaN             NaN   \n",
       "630   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "631   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "633   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "635   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "648   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "649   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "650   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "651   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "652   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "675   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "676   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "677   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "679   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "682   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "683   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "685   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "686   Gorilla_gorilla_ssp  ...    NaN      ok           PASS            FAIL   \n",
       "687   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "689   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "690   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "691   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "692   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "693   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "694   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "695   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "696   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "697   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "698   Gorilla_gorilla_ssp  ...    NaN      ok           PASS            WARN   \n",
       "701   Gorilla_gorilla_ssp  ...    NaN      ok            NaN             NaN   \n",
       "\n",
       "                      relatives  remove_as_relative  generation  founder  \\\n",
       "495                         NaN                 NaN         NaN      NaN   \n",
       "497                         NaN                 NaN         NaN      NaN   \n",
       "498                         NaN                 NaN         NaN      NaN   \n",
       "499                         NaN                 NaN         NaN      NaN   \n",
       "500                         NaN                 NaN         NaN      NaN   \n",
       "502                         NaN                 NaN         NaN      NaN   \n",
       "503                         NaN                 NaN         NaN      NaN   \n",
       "504                         NaN                 NaN         NaN      NaN   \n",
       "509                         NaN                 NaN         NaN      NaN   \n",
       "510                         NaN                 NaN         NaN      NaN   \n",
       "511                         NaN                 NaN         NaN      NaN   \n",
       "514                         NaN                 NaN         NaN      NaN   \n",
       "515                         NaN                 NaN         NaN      NaN   \n",
       "516                         NaN                 NaN         NaN      NaN   \n",
       "517                         NaN                 NaN         NaN      NaN   \n",
       "518                SAMEA2697038               False         NaN      NaN   \n",
       "519                         NaN                 NaN         NaN      NaN   \n",
       "520                         NaN                 NaN         NaN      NaN   \n",
       "521                         NaN                 NaN         NaN      NaN   \n",
       "522                         NaN                 NaN         NaN      NaN   \n",
       "630  SAMEA5204235; SAMEA5204237               False           P     True   \n",
       "631  SAMEA5204235; SAMN01920489               False           P     True   \n",
       "633                SAMEA5204237               False           P     True   \n",
       "635                SAMEA5204237               False          F2    False   \n",
       "648                         NaN                 NaN         NaN      NaN   \n",
       "649                         NaN                 NaN         NaN      NaN   \n",
       "650                         NaN                 NaN         NaN      NaN   \n",
       "651                         NaN                 NaN         NaN      NaN   \n",
       "652                         NaN                 NaN         NaN      NaN   \n",
       "675                         NaN                 NaN         NaN      NaN   \n",
       "676                         NaN                 NaN         NaN      NaN   \n",
       "677                SAMN04003007               False         NaN      NaN   \n",
       "679                         NaN                 NaN         NaN      NaN   \n",
       "682                         NaN                 NaN         NaN      NaN   \n",
       "683                         NaN                 NaN         NaN      NaN   \n",
       "685                         NaN                 NaN         NaN      NaN   \n",
       "686                         NaN                 NaN         NaN      NaN   \n",
       "687                         NaN                 NaN         NaN      NaN   \n",
       "689                         NaN                 NaN         NaN      NaN   \n",
       "690                         NaN                 NaN         NaN      NaN   \n",
       "691                         NaN                 NaN         NaN      NaN   \n",
       "692                         NaN                 NaN         NaN      NaN   \n",
       "693                         NaN                 NaN         NaN      NaN   \n",
       "694                         NaN                 NaN         NaN      NaN   \n",
       "695                         NaN                 NaN         NaN      NaN   \n",
       "696                SAMEA5204235               False          F2    False   \n",
       "697                         NaN                 NaN         NaN      NaN   \n",
       "698                         NaN                 NaN         NaN      NaN   \n",
       "701                         NaN                 NaN         NaN      NaN   \n",
       "\n",
       "     remove_manual  pedigree_source  \n",
       "495            NaN              NaN  \n",
       "497            NaN              NaN  \n",
       "498            NaN              NaN  \n",
       "499            NaN              NaN  \n",
       "500            NaN              NaN  \n",
       "502            NaN              NaN  \n",
       "503            NaN              NaN  \n",
       "504            NaN              NaN  \n",
       "509            NaN              NaN  \n",
       "510            NaN              NaN  \n",
       "511            NaN              NaN  \n",
       "514            NaN              NaN  \n",
       "515            NaN              NaN  \n",
       "516            NaN              NaN  \n",
       "517            NaN              NaN  \n",
       "518            NaN              NaN  \n",
       "519            NaN              NaN  \n",
       "520            NaN              NaN  \n",
       "521            NaN              NaN  \n",
       "522            NaN              NaN  \n",
       "630            NaN             ncbi  \n",
       "631            NaN             ncbi  \n",
       "633            NaN             ncbi  \n",
       "635            NaN             ncbi  \n",
       "648            NaN              NaN  \n",
       "649            NaN              NaN  \n",
       "650            NaN              NaN  \n",
       "651            NaN              NaN  \n",
       "652            NaN              NaN  \n",
       "675            NaN              NaN  \n",
       "676            NaN              NaN  \n",
       "677            NaN              NaN  \n",
       "679            NaN              NaN  \n",
       "682            NaN              NaN  \n",
       "683            NaN              NaN  \n",
       "685            NaN              NaN  \n",
       "686            NaN              NaN  \n",
       "687            NaN              NaN  \n",
       "689            NaN              NaN  \n",
       "690            NaN              NaN  \n",
       "691            NaN              NaN  \n",
       "692            NaN              NaN  \n",
       "693            NaN              NaN  \n",
       "694            NaN              NaN  \n",
       "695            NaN              NaN  \n",
       "696            NaN             ncbi  \n",
       "697            NaN              NaN  \n",
       "698            NaN              NaN  \n",
       "701            NaN              NaN  \n",
       "\n",
       "[49 rows x 93 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sub_vasili.loc[sub_vasili.genus == \"Gorilla\"]#.species_full.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "f6a1392b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>REFERENCE_FOLDER</th>\n",
       "      <th>REGION</th>\n",
       "      <th>CONTIG_ID</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "      <th>BATCH</th>\n",
       "      <th>FEMALE_PLOIDY</th>\n",
       "      <th>MALE_PLOIDY</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>81</th>\n",
       "      <td>Aotus_nancymaae_ssp</td>\n",
       "      <td>NW_026988493.1</td>\n",
       "      <td>NW_026988493.1</td>\n",
       "      <td>0</td>\n",
       "      <td>2145027</td>\n",
       "      <td>39</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       REFERENCE_FOLDER          REGION       CONTIG_ID  START      END  \\\n",
       "81  Aotus_nancymaae_ssp  NW_026988493.1  NW_026988493.1      0  2145027   \n",
       "\n",
       "    BATCH  FEMALE_PLOIDY  MALE_PLOIDY  \n",
       "81     39              2            2  "
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "regions_df.loc[regions_df.CONTIG_ID == \"NW_026988493.1\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "d74d9cc9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['Allenopithecus_nigroviridis_ssp', 'Aotus_azarai_ssp',\n",
       "       'Aotus_griseimembra_ssp', 'Aotus_nancymaae_ssp',\n",
       "       'Aotus_trivirgatus_ssp', 'Aotus_vociferans_ssp',\n",
       "       'Alouatta_belzebul_ssp', 'Alouatta_caraya_ssp',\n",
       "       'Alouatta_discolor_ssp', 'Alouatta_juara_ssp',\n",
       "       'Alouatta_macconnelli_ssp', 'Alouatta_palliata_ssp',\n",
       "       'Alouatta_seniculus_puruensis', 'Alouatta_seniculus_ssp',\n",
       "       'Ateles_belzebuth_ssp', 'Ateles_chamek_ssp',\n",
       "       'Ateles_geoffroyi_ssp', 'Ateles_marginatus_ssp',\n",
       "       'Ateles_paniscus_ssp', 'Brachyteles_hypoxanthus_ssp',\n",
       "       'Lagothrix_lagotricha_ssp', 'Callimico_goeldii_ssp',\n",
       "       'Callithrix_geoffroyi_ssp', 'Callithrix_jacchus_ssp',\n",
       "       'Callithrix_kuhlii_ssp', 'Cebuella_niveiventris_ssp',\n",
       "       'Cebuella_pygmaea_ssp', 'Mico_argentatus_ssp',\n",
       "       'Mico_humeralifer_ssp', 'Mico_humilis_ssp', 'Carlito_syrichta_ssp',\n",
       "       'Cephalopachus_bancanus_ssp', 'Tarsius_dentatus_ssp',\n",
       "       'Tarsius_lariang_ssp', 'Tarsius_wallacei_ssp',\n",
       "       'Cercopithecus_ascanius_ssp', 'Cercopithecus_campbelli_lowei',\n",
       "       'Cercopithecus_campbelli_ssp', 'Cercopithecus_cephus_ssp',\n",
       "       'Cercopithecus_denti_ssp', 'Cercopithecus_diana_ssp',\n",
       "       'Cercopithecus_hamlyni_ssp', 'Cercopithecus_mitis_ssp',\n",
       "       'Cercopithecus_mitis_stuhlmanni', 'Cercopithecus_mona_ssp',\n",
       "       'Cercopithecus_neglectus_ssp', 'Cercopithecus_nictitans_ssp',\n",
       "       'Cercopithecus_petaurista_ssp', 'Cercopithecus_pogonias_ssp',\n",
       "       'Cercopithecus_roloway_ssp', 'Cercopithecus_wolfi_elegans',\n",
       "       'Cercopithecus_wolfi_ssp', 'Cheirogaleus_major_ssp',\n",
       "       'Cheirogaleus_medius_ssp', 'Cheirogaleus_sibreei_ssp',\n",
       "       'Allochrocebus_lhoesti_ssp', 'Allochrocebus_preussi_ssp',\n",
       "       'Allochrocebus_solatus_ssp', 'Chlorocebus_dryas_ssp',\n",
       "       'Chlorocebus_pygerythrus_ssp', 'Erythrocebus_patas_ssp',\n",
       "       'Colobus_angolensis_palliatus', 'Colobus_guereza_kikuyuensis',\n",
       "       'Colobus_guereza_ssp', 'Daubentonia_madagascariensis_ssp',\n",
       "       'Eulemur_coronatus_ssp', 'Eulemur_flavifrons_ssp',\n",
       "       'Eulemur_macaco_ssp', 'Eulemur_mongoz_ssp',\n",
       "       'Eulemur_rubriventer_ssp', 'Eulemur_albifrons_ssp',\n",
       "       'Eulemur_collaris_ssp', 'Eulemur_fulvus_ssp',\n",
       "       'Eulemur_rufifrons_ssp', 'Eulemur_rufus_ssp',\n",
       "       'Eulemur_sanfordi_ssp', 'Gorilla_beringei_beringei',\n",
       "       'Gorilla_beringei_graueri', 'Gorilla_beringei_ssp',\n",
       "       'Gorilla_gorilla_gorilla', 'Gorilla_gorilla_ssp',\n",
       "       'Hoolock_hoolock_ssp', 'Hoolock_leuconedys_ssp',\n",
       "       'Hylobates_abbotti_ssp', 'Hylobates_agilis_ssp',\n",
       "       'Hylobates_klossii_ssp', 'Hylobates_lar_ssp',\n",
       "       'Hylobates_moloch_ssp', 'Hylobates_muelleri_ssp',\n",
       "       'Hylobates_pileatus_ssp', 'Hapalemur_griseus_alaotrensis',\n",
       "       'Hapalemur_griseus_ssp', 'Hapalemur_meridionalis_ssp',\n",
       "       'Lemur_catta_ssp', 'Prolemur_simus_ssp',\n",
       "       'Leontopithecus_chrysomelas_ssp', 'Leontopithecus_rosalia_ssp',\n",
       "       'Macaca_fascicularis_aureus', 'Macaca_fascicularis_ssp',\n",
       "       'Macaca_cyclopis_ssp'], dtype=object)"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sub_vasili.species_full.unique()[:100]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "8bb45881",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>REFERENCE_FOLDER</th>\n",
       "      <th>REGION</th>\n",
       "      <th>CONTIG_ID</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "      <th>BATCH</th>\n",
       "      <th>FEMALE_PLOIDY</th>\n",
       "      <th>MALE_PLOIDY</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073224.2</td>\n",
       "      <td>NC_073224.2</td>\n",
       "      <td>0</td>\n",
       "      <td>243847345</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_086017.1</td>\n",
       "      <td>NC_086017.1</td>\n",
       "      <td>0</td>\n",
       "      <td>215253457</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073227.2</td>\n",
       "      <td>NC_073227.2</td>\n",
       "      <td>0</td>\n",
       "      <td>215373162</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073228.2</td>\n",
       "      <td>NC_073228.2</td>\n",
       "      <td>0</td>\n",
       "      <td>195332687</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073229.2</td>\n",
       "      <td>NC_073229.2</td>\n",
       "      <td>0</td>\n",
       "      <td>199452338</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073230.2</td>\n",
       "      <td>NC_073230.2</td>\n",
       "      <td>0</td>\n",
       "      <td>170803427</td>\n",
       "      <td>5</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073231.2</td>\n",
       "      <td>NC_073231.2</td>\n",
       "      <td>0</td>\n",
       "      <td>158533252</td>\n",
       "      <td>6</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073232.2</td>\n",
       "      <td>NC_073232.2</td>\n",
       "      <td>0</td>\n",
       "      <td>150804130</td>\n",
       "      <td>7</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073233.2</td>\n",
       "      <td>NC_073233.2</td>\n",
       "      <td>0</td>\n",
       "      <td>147876701</td>\n",
       "      <td>8</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073234.2</td>\n",
       "      <td>NC_073234.2</td>\n",
       "      <td>0</td>\n",
       "      <td>154228409</td>\n",
       "      <td>9</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073235.2</td>\n",
       "      <td>NC_073235.2</td>\n",
       "      <td>0</td>\n",
       "      <td>150248535</td>\n",
       "      <td>10</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073236.2</td>\n",
       "      <td>NC_073236.2</td>\n",
       "      <td>0</td>\n",
       "      <td>145906006</td>\n",
       "      <td>11</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073237.2</td>\n",
       "      <td>NC_073237.2</td>\n",
       "      <td>0</td>\n",
       "      <td>134695396</td>\n",
       "      <td>12</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073238.2</td>\n",
       "      <td>NC_073238.2</td>\n",
       "      <td>0</td>\n",
       "      <td>132018348</td>\n",
       "      <td>13</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073239.2</td>\n",
       "      <td>NC_073239.2</td>\n",
       "      <td>0</td>\n",
       "      <td>131692671</td>\n",
       "      <td>14</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073240.2</td>\n",
       "      <td>NC_073240.2</td>\n",
       "      <td>0</td>\n",
       "      <td>107915467</td>\n",
       "      <td>15</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073241.2</td>\n",
       "      <td>NC_073241.2</td>\n",
       "      <td>0</td>\n",
       "      <td>116054869</td>\n",
       "      <td>16</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073242.2</td>\n",
       "      <td>NC_073242.2</td>\n",
       "      <td>0</td>\n",
       "      <td>131347499</td>\n",
       "      <td>17</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073243.2</td>\n",
       "      <td>NC_073243.2</td>\n",
       "      <td>0</td>\n",
       "      <td>120452965</td>\n",
       "      <td>18</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073244.2</td>\n",
       "      <td>NC_073244.2</td>\n",
       "      <td>0</td>\n",
       "      <td>80312928</td>\n",
       "      <td>19</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073245.2</td>\n",
       "      <td>NC_073245.2</td>\n",
       "      <td>0</td>\n",
       "      <td>84443219</td>\n",
       "      <td>20</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073246.2</td>\n",
       "      <td>NC_073246.2</td>\n",
       "      <td>0</td>\n",
       "      <td>59507314</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_086018.1</td>\n",
       "      <td>NC_086018.1</td>\n",
       "      <td>0</td>\n",
       "      <td>54769797</td>\n",
       "      <td>22</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>prePAR1</td>\n",
       "      <td>NC_073247.2</td>\n",
       "      <td>0</td>\n",
       "      <td>9544188</td>\n",
       "      <td>23</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>PAR1</td>\n",
       "      <td>NC_073247.2</td>\n",
       "      <td>9544188</td>\n",
       "      <td>12039748</td>\n",
       "      <td>23</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>postPAR1</td>\n",
       "      <td>NC_073247.2</td>\n",
       "      <td>12039748</td>\n",
       "      <td>177558554</td>\n",
       "      <td>23</td>\n",
       "      <td>2</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_073248.2</td>\n",
       "      <td>NC_073248.2</td>\n",
       "      <td>0</td>\n",
       "      <td>67405748</td>\n",
       "      <td>24</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>Gorilla_gorilla_ssp</td>\n",
       "      <td>NC_011120.1</td>\n",
       "      <td>NC_011120.1</td>\n",
       "      <td>0</td>\n",
       "      <td>16412</td>\n",
       "      <td>25</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       REFERENCE_FOLDER       REGION    CONTIG_ID     START        END  BATCH  \\\n",
       "0   Gorilla_gorilla_ssp  NC_073224.2  NC_073224.2         0  243847345      0   \n",
       "1   Gorilla_gorilla_ssp  NC_086017.1  NC_086017.1         0  215253457      1   \n",
       "2   Gorilla_gorilla_ssp  NC_073227.2  NC_073227.2         0  215373162      2   \n",
       "3   Gorilla_gorilla_ssp  NC_073228.2  NC_073228.2         0  195332687      3   \n",
       "4   Gorilla_gorilla_ssp  NC_073229.2  NC_073229.2         0  199452338      4   \n",
       "5   Gorilla_gorilla_ssp  NC_073230.2  NC_073230.2         0  170803427      5   \n",
       "6   Gorilla_gorilla_ssp  NC_073231.2  NC_073231.2         0  158533252      6   \n",
       "7   Gorilla_gorilla_ssp  NC_073232.2  NC_073232.2         0  150804130      7   \n",
       "8   Gorilla_gorilla_ssp  NC_073233.2  NC_073233.2         0  147876701      8   \n",
       "9   Gorilla_gorilla_ssp  NC_073234.2  NC_073234.2         0  154228409      9   \n",
       "10  Gorilla_gorilla_ssp  NC_073235.2  NC_073235.2         0  150248535     10   \n",
       "11  Gorilla_gorilla_ssp  NC_073236.2  NC_073236.2         0  145906006     11   \n",
       "12  Gorilla_gorilla_ssp  NC_073237.2  NC_073237.2         0  134695396     12   \n",
       "13  Gorilla_gorilla_ssp  NC_073238.2  NC_073238.2         0  132018348     13   \n",
       "14  Gorilla_gorilla_ssp  NC_073239.2  NC_073239.2         0  131692671     14   \n",
       "15  Gorilla_gorilla_ssp  NC_073240.2  NC_073240.2         0  107915467     15   \n",
       "16  Gorilla_gorilla_ssp  NC_073241.2  NC_073241.2         0  116054869     16   \n",
       "17  Gorilla_gorilla_ssp  NC_073242.2  NC_073242.2         0  131347499     17   \n",
       "18  Gorilla_gorilla_ssp  NC_073243.2  NC_073243.2         0  120452965     18   \n",
       "19  Gorilla_gorilla_ssp  NC_073244.2  NC_073244.2         0   80312928     19   \n",
       "20  Gorilla_gorilla_ssp  NC_073245.2  NC_073245.2         0   84443219     20   \n",
       "21  Gorilla_gorilla_ssp  NC_073246.2  NC_073246.2         0   59507314     21   \n",
       "22  Gorilla_gorilla_ssp  NC_086018.1  NC_086018.1         0   54769797     22   \n",
       "23  Gorilla_gorilla_ssp      prePAR1  NC_073247.2         0    9544188     23   \n",
       "24  Gorilla_gorilla_ssp         PAR1  NC_073247.2   9544188   12039748     23   \n",
       "25  Gorilla_gorilla_ssp     postPAR1  NC_073247.2  12039748  177558554     23   \n",
       "26  Gorilla_gorilla_ssp  NC_073248.2  NC_073248.2         0   67405748     24   \n",
       "27  Gorilla_gorilla_ssp  NC_011120.1  NC_011120.1         0      16412     25   \n",
       "\n",
       "    FEMALE_PLOIDY  MALE_PLOIDY  \n",
       "0               2            2  \n",
       "1               2            2  \n",
       "2               2            2  \n",
       "3               2            2  \n",
       "4               2            2  \n",
       "5               2            2  \n",
       "6               2            2  \n",
       "7               2            2  \n",
       "8               2            2  \n",
       "9               2            2  \n",
       "10              2            2  \n",
       "11              2            2  \n",
       "12              2            2  \n",
       "13              2            2  \n",
       "14              2            2  \n",
       "15              2            2  \n",
       "16              2            2  \n",
       "17              2            2  \n",
       "18              2            2  \n",
       "19              2            2  \n",
       "20              2            2  \n",
       "21              2            2  \n",
       "22              2            2  \n",
       "23              2            1  \n",
       "24              2            2  \n",
       "25              2            1  \n",
       "26              0            1  \n",
       "27              1            1  "
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(\"Gorilla\"), sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "id": "1163e34c",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['../zarr_20x_inds/Aotus_azarai_ssp/NW_026988425.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988517.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988622.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988490.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988467.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988555.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988446.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988480.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988477.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988507.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988414.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988435.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988526.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988588.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988415.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988608.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988565.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988476.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988481.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988457.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988544.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988447.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988466.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988491.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988516.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988537.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988489.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988426.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988514.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988464.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988556.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988445.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988455.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988546.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988483.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988504.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988417.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988499.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988436.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988437.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988498.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988505.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988416.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988482.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988454.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988444.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988465.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988576.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988492.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988534.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988427.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988488.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988530.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988423.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988549.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988496.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988418.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988553.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988440.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988450.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988562.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988429.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988471.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988595.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988501.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988412.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988615.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988433.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988520.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988432.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988521.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988413.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988487.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988428.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988470.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988552.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988419.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988441.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988584.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988460.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988573.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988438.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988625.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988510.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988422.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988478.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988533.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988420.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988512.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988462.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988827.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988571.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988586.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988495.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988443.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988665.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988675.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988453.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988518.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988682.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988472.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988539.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988485.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988449.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988430.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988523.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988431.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988522.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988448.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988503.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988484.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988473.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988519.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988452.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988442.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988509.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988587.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988494.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988463.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988570.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988513.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988458.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988532.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988479.1/',\n",
       " '../zarr_20x_inds/Aotus_azarai_ssp/NW_026988421.1/']"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "glob.glob(\"../zarr_20x_inds/Aotus_azarai_ssp/*/\")"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ce27f925",
   "metadata": {},
   "source": [
    "Check of failed species for the first run.\n",
    "They failed as the regions_df called them Macaca_thibetana_thibetana instead of Macaca_thibetana_ssp, for example. Fixed by using the correct metadata column."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "id": "f041230a",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Saimiri_macrodon_ssp            ID  ID_PGDP group   family    genus   species subspecies  \\\n",
      "3045  PD_0361  PD_0361   PLA  Cebidae  Saimiri  macrodon        ssp   \n",
      "3046  PD_0362  PD_0362   PLA  Cebidae  Saimiri  macrodon        ssp   \n",
      "3047  PD_0372  PD_0372   PLA  Cebidae  Saimiri  macrodon        ssp   \n",
      "\n",
      "              species_full   species_short    species_genotyping  ... reason  \\\n",
      "3045  Saimiri_macrodon_ssp  S.macrodon_ssp  Saimiri_macrodon_ssp  ...    NaN   \n",
      "3046  Saimiri_macrodon_ssp  S.macrodon_ssp  Saimiri_macrodon_ssp  ...    NaN   \n",
      "3047  Saimiri_macrodon_ssp  S.macrodon_ssp  Saimiri_macrodon_ssp  ...    NaN   \n",
      "\n",
      "     finalQC  Kuderna_AB_QC  Kuderna_COV_QC  relatives  remove_as_relative  \\\n",
      "3045      ok            NaN             NaN        NaN                 NaN   \n",
      "3046      ok            NaN             NaN        NaN                 NaN   \n",
      "3047      ok            NaN             NaN        NaN                 NaN   \n",
      "\n",
      "      generation  founder  remove_manual  pedigree_source  \n",
      "3045         NaN      NaN            NaN              NaN  \n",
      "3046         NaN      NaN            NaN              NaN  \n",
      "3047         NaN      NaN            NaN              NaN  \n",
      "\n",
      "[3 rows x 93 columns]\n",
      "Macaca_thibetana_ssp                 ID       ID_PGDP group           family   genus    species  \\\n",
      "1550  SAMN02390221  SAMN02390221   OWM  Cercopithecidae  Macaca  thibetana   \n",
      "1552  SAMN15194904  SAMN15194904   OWM  Cercopithecidae  Macaca  thibetana   \n",
      "\n",
      "     subspecies          species_full    species_short    species_genotyping  \\\n",
      "1550        ssp  Macaca_thibetana_ssp  M.thibetana_ssp  Macaca_thibetana_ssp   \n",
      "1552        ssp  Macaca_thibetana_ssp  M.thibetana_ssp  Macaca_thibetana_ssp   \n",
      "\n",
      "      ... reason finalQC  Kuderna_AB_QC  Kuderna_COV_QC  relatives  \\\n",
      "1550  ...    NaN      ok           PASS            WARN        NaN   \n",
      "1552  ...    NaN      ok            NaN             NaN        NaN   \n",
      "\n",
      "      remove_as_relative  generation  founder  remove_manual  pedigree_source  \n",
      "1550                 NaN         NaN      NaN            NaN              NaN  \n",
      "1552                 NaN         NaN      NaN            NaN              NaN  \n",
      "\n",
      "[2 rows x 93 columns]\n"
     ]
    }
   ],
   "source": [
    "vasili_table = pd.read_csv(vasili_stats, sep=\"\\t\")\n",
    "sub_vasili = vasili_table.loc[(vasili_table.finalQC != \"fail\")\n",
    "                              & (vasili_table.cov_chrA >= 20)\n",
    "                              & (vasili_table.remove_as_relative != True)\n",
    "                              & (vasili_table.remove_manual != True)\n",
    "                              & (~vasili_table.ID.str.startswith(\"SAMEA11633\"))\n",
    "                             ]\n",
    "\n",
    "failed_species = ['Saimiri_macrodon_ssp',\n",
    " 'Macaca_thibetana_ssp',\n",
    " 'Saimiri_sciureus_ssp',\n",
    " 'Saimiri_oerstedii_ssp',\n",
    " 'Saimiri_ustus_ssp',\n",
    " 'Macaca_arctoides_ssp',\n",
    " 'Macaca_leucogenys_ssp',\n",
    " 'Macaca_assamensis_ssp',\n",
    " 'Macaca_radiata_ssp',\n",
    " 'Saimiri_cassiquiarensis_ssp',\n",
    " 'Saimiri_boliviensis_ssp']\n",
    "\n",
    "\n",
    "for species in failed_species[:2]:\n",
    "    species_inds = sub_vasili.loc[sub_vasili.species_genotyping == species]\n",
    "    print(species, species_inds)\n",
    "    short_form = species.split(\"_\")[0]\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    reference = species_inds[\"reference\"].unique()\n",
    "    sample_names = \",\".join(species_inds[\"ID\"])\n",
    "    all_chr_file = starting_dir.format(species)+species+\"_all_chr.sorted.bcf\"\n",
    "    if not os.path.exists(all_chr_file):\n",
    "        print(\"Filtered VCF does not exist \", species)\n",
    "        continue\n",
    "    if len(reference) > 1:\n",
    "        print(\"Too many references \", species)\n",
    "        continue\n",
    "    zarr_input = []\n",
    "    # Chromosome X\n",
    "    zarr_input.extend(regions_df.loc[(regions_df.END >= 1000000) &\n",
    "                                  (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                                  (regions_df.MALE_PLOIDY == 1) &\n",
    "          (regions_df.REFERENCE_FOLDER == reference[0]+\"_ssp\")].CONTIG_ID.unique()[:])\n",
    "    # Autosomes\n",
    "    zarr_input.extend(regions_df.loc[(regions_df.END >= 1000000) &\n",
    "                                  (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                                  (regions_df.MALE_PLOIDY == 2) &\n",
    "                                  (~regions_df.CONTIG_ID.isin((zarr_input))) &\n",
    "          (regions_df.REFERENCE_FOLDER == reference[0]+\"_ssp\")].CONTIG_ID.unique()[:])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "f727bed5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['NC_065578.1', 'NC_065579.1', 'NC_065580.1', 'NC_065581.1',\n",
       "       'NC_065582.1', 'NC_065583.1', 'NC_065584.1', 'NC_065585.1',\n",
       "       'NC_065586.1', 'NC_065587.1', 'NC_065588.1', 'NC_065589.1',\n",
       "       'NC_065590.1', 'NC_065591.1', 'NC_065592.1', 'NC_065593.1',\n",
       "       'NC_065594.1', 'NC_065595.1', 'NC_065596.1', 'NC_065597.1'],\n",
       "      dtype=object)"
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "regions_df.loc[(regions_df.END >= 1000000) &\n",
    "                                  (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                                  (regions_df.MALE_PLOIDY == 2) &\n",
    "          (regions_df.REFERENCE_FOLDER == reference[0]+\"_thibetana\")].CONTIG_ID.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "id": "6fda8ec8",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>REFERENCE_FOLDER</th>\n",
       "      <th>REGION</th>\n",
       "      <th>CONTIG_ID</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "      <th>BATCH</th>\n",
       "      <th>FEMALE_PLOIDY</th>\n",
       "      <th>MALE_PLOIDY</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3308</th>\n",
       "      <td>Macaca_thibetana_thibetana</td>\n",
       "      <td>NC_065578.1</td>\n",
       "      <td>NC_065578.1</td>\n",
       "      <td>0</td>\n",
       "      <td>220358725</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3309</th>\n",
       "      <td>Macaca_thibetana_thibetana</td>\n",
       "      <td>NC_065579.1</td>\n",
       "      <td>NC_065579.1</td>\n",
       "      <td>0</td>\n",
       "      <td>194429148</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3310</th>\n",
       "      <td>Macaca_thibetana_thibetana</td>\n",
       "      <td>NC_065580.1</td>\n",
       "      <td>NC_065580.1</td>\n",
       "      <td>0</td>\n",
       "      <td>185122172</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3311</th>\n",
       "      <td>Macaca_thibetana_thibetana</td>\n",
       "      <td>NC_065581.1</td>\n",
       "      <td>NC_065581.1</td>\n",
       "      <td>0</td>\n",
       "      <td>168803367</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3312</th>\n",
       "      <td>Macaca_thibetana_thibetana</td>\n",
       "      <td>NC_065582.1</td>\n",
       "      <td>NC_065582.1</td>\n",
       "      <td>0</td>\n",
       "      <td>185943000</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3904</th>\n",
       "      <td>Macaca_thibetana_thibetana</td>\n",
       "      <td>NW_026089386.1</td>\n",
       "      <td>NW_026089386.1</td>\n",
       "      <td>0</td>\n",
       "      <td>5000</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3905</th>\n",
       "      <td>Macaca_thibetana_thibetana</td>\n",
       "      <td>NW_026089387.1</td>\n",
       "      <td>NW_026089387.1</td>\n",
       "      <td>0</td>\n",
       "      <td>5000</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3906</th>\n",
       "      <td>Macaca_thibetana_thibetana</td>\n",
       "      <td>NW_026089388.1</td>\n",
       "      <td>NW_026089388.1</td>\n",
       "      <td>0</td>\n",
       "      <td>161000</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3907</th>\n",
       "      <td>Macaca_thibetana_thibetana</td>\n",
       "      <td>NW_026089389.1</td>\n",
       "      <td>NW_026089389.1</td>\n",
       "      <td>0</td>\n",
       "      <td>150144</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3908</th>\n",
       "      <td>Macaca_thibetana_thibetana</td>\n",
       "      <td>NW_026089390.1</td>\n",
       "      <td>NW_026089390.1</td>\n",
       "      <td>0</td>\n",
       "      <td>10000</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>601 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                REFERENCE_FOLDER          REGION       CONTIG_ID  START  \\\n",
       "3308  Macaca_thibetana_thibetana     NC_065578.1     NC_065578.1      0   \n",
       "3309  Macaca_thibetana_thibetana     NC_065579.1     NC_065579.1      0   \n",
       "3310  Macaca_thibetana_thibetana     NC_065580.1     NC_065580.1      0   \n",
       "3311  Macaca_thibetana_thibetana     NC_065581.1     NC_065581.1      0   \n",
       "3312  Macaca_thibetana_thibetana     NC_065582.1     NC_065582.1      0   \n",
       "...                          ...             ...             ...    ...   \n",
       "3904  Macaca_thibetana_thibetana  NW_026089386.1  NW_026089386.1      0   \n",
       "3905  Macaca_thibetana_thibetana  NW_026089387.1  NW_026089387.1      0   \n",
       "3906  Macaca_thibetana_thibetana  NW_026089388.1  NW_026089388.1      0   \n",
       "3907  Macaca_thibetana_thibetana  NW_026089389.1  NW_026089389.1      0   \n",
       "3908  Macaca_thibetana_thibetana  NW_026089390.1  NW_026089390.1      0   \n",
       "\n",
       "            END  BATCH  FEMALE_PLOIDY  MALE_PLOIDY  \n",
       "3308  220358725      0              2            2  \n",
       "3309  194429148      1              2            2  \n",
       "3310  185122172      2              2            2  \n",
       "3311  168803367      3              2            2  \n",
       "3312  185943000      4              2            2  \n",
       "...         ...    ...            ...          ...  \n",
       "3904       5000     21              2            2  \n",
       "3905       5000     21              2            2  \n",
       "3906     161000     21              2            2  \n",
       "3907     150144     21              2            2  \n",
       "3908      10000     21              2            2  \n",
       "\n",
       "[601 rows x 8 columns]"
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "regions_df.loc[regions_df.REFERENCE_FOLDER.str.startswith(reference[0])]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "id": "96711c39",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>REFERENCE_FOLDER</th>\n",
       "      <th>REGION</th>\n",
       "      <th>CONTIG_ID</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "      <th>BATCH</th>\n",
       "      <th>FEMALE_PLOIDY</th>\n",
       "      <th>MALE_PLOIDY</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>Macaca_fascicularis_ssp</td>\n",
       "      <td>CP141361.1</td>\n",
       "      <td>CP141361.1</td>\n",
       "      <td>0</td>\n",
       "      <td>234122563</td>\n",
       "      <td>0</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>Macaca_fascicularis_ssp</td>\n",
       "      <td>CP141360.1</td>\n",
       "      <td>CP141360.1</td>\n",
       "      <td>0</td>\n",
       "      <td>203129947</td>\n",
       "      <td>1</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>Macaca_fascicularis_ssp</td>\n",
       "      <td>CP141359.1</td>\n",
       "      <td>CP141359.1</td>\n",
       "      <td>0</td>\n",
       "      <td>200656507</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>Macaca_fascicularis_ssp</td>\n",
       "      <td>CP141358.1</td>\n",
       "      <td>CP141358.1</td>\n",
       "      <td>0</td>\n",
       "      <td>173030664</td>\n",
       "      <td>3</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>Macaca_fascicularis_ssp</td>\n",
       "      <td>CP141357.1</td>\n",
       "      <td>CP141357.1</td>\n",
       "      <td>0</td>\n",
       "      <td>194799049</td>\n",
       "      <td>4</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3904</th>\n",
       "      <td>Macaca_thibetana_thibetana</td>\n",
       "      <td>NW_026089386.1</td>\n",
       "      <td>NW_026089386.1</td>\n",
       "      <td>0</td>\n",
       "      <td>5000</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3905</th>\n",
       "      <td>Macaca_thibetana_thibetana</td>\n",
       "      <td>NW_026089387.1</td>\n",
       "      <td>NW_026089387.1</td>\n",
       "      <td>0</td>\n",
       "      <td>5000</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3906</th>\n",
       "      <td>Macaca_thibetana_thibetana</td>\n",
       "      <td>NW_026089388.1</td>\n",
       "      <td>NW_026089388.1</td>\n",
       "      <td>0</td>\n",
       "      <td>161000</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3907</th>\n",
       "      <td>Macaca_thibetana_thibetana</td>\n",
       "      <td>NW_026089389.1</td>\n",
       "      <td>NW_026089389.1</td>\n",
       "      <td>0</td>\n",
       "      <td>150144</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3908</th>\n",
       "      <td>Macaca_thibetana_thibetana</td>\n",
       "      <td>NW_026089390.1</td>\n",
       "      <td>NW_026089390.1</td>\n",
       "      <td>0</td>\n",
       "      <td>10000</td>\n",
       "      <td>21</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3909 rows × 8 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                REFERENCE_FOLDER          REGION       CONTIG_ID  START  \\\n",
       "0        Macaca_fascicularis_ssp      CP141361.1      CP141361.1      0   \n",
       "1        Macaca_fascicularis_ssp      CP141360.1      CP141360.1      0   \n",
       "2        Macaca_fascicularis_ssp      CP141359.1      CP141359.1      0   \n",
       "3        Macaca_fascicularis_ssp      CP141358.1      CP141358.1      0   \n",
       "4        Macaca_fascicularis_ssp      CP141357.1      CP141357.1      0   \n",
       "...                          ...             ...             ...    ...   \n",
       "3904  Macaca_thibetana_thibetana  NW_026089386.1  NW_026089386.1      0   \n",
       "3905  Macaca_thibetana_thibetana  NW_026089387.1  NW_026089387.1      0   \n",
       "3906  Macaca_thibetana_thibetana  NW_026089388.1  NW_026089388.1      0   \n",
       "3907  Macaca_thibetana_thibetana  NW_026089389.1  NW_026089389.1      0   \n",
       "3908  Macaca_thibetana_thibetana  NW_026089390.1  NW_026089390.1      0   \n",
       "\n",
       "            END  BATCH  FEMALE_PLOIDY  MALE_PLOIDY  \n",
       "0     234122563      0              2            2  \n",
       "1     203129947      1              2            2  \n",
       "2     200656507      2              2            2  \n",
       "3     173030664      3              2            2  \n",
       "4     194799049      4              2            2  \n",
       "...         ...    ...            ...          ...  \n",
       "3904       5000     21              2            2  \n",
       "3905       5000     21              2            2  \n",
       "3906     161000     21              2            2  \n",
       "3907     150144     21              2            2  \n",
       "3908      10000     21              2            2  \n",
       "\n",
       "[3909 rows x 8 columns]"
      ]
     },
     "execution_count": 29,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "regions_df"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
