{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "551183f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "%run ../scripts/notebook_settings.py\n",
    "import sgkit as sg\n",
    "import xarray as xr\n",
    "import glob"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "75c2c2ce",
   "metadata": {},
   "source": [
    "Workflow to load in the respective individuals and generate base statistics on them."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d7fa0b04",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Functions\n",
    "\n",
    "def read_beds(long_form):\n",
    "    bed_path_x = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024/{}/filteredVCF/pos_bed_cov_based/{}_batch*_fploidy2_mploidy1.bed\".format(long_form, long_form)\n",
    "    bed_path_all = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024/{}/filteredVCF/pos_bed_cov_based/{}_batch*_fploidy2_mploidy2.bed\".format(long_form, long_form)\n",
    "    bed_l = []\n",
    "    for b in glob.glob(bed_path_all):\n",
    "        bed_file = pd.read_csv(b, sep=\"\\t\", names=[\"chrom\", \"start\", \"end\"])\n",
    "        bed_l.append(bed_file)\n",
    "    bed_files = pd.concat(bed_l)\n",
    "    bed_l = []\n",
    "    for b in glob.glob(bed_path_x):\n",
    "        #print(b)\n",
    "        bed_file = pd.read_csv(b, sep=\"\\t\", names=[\"chrom\", \"start\", \"end\"])\n",
    "        bed_l.append(bed_file)\n",
    "    if len(bed_l) > 0:\n",
    "        bed_x = pd.concat(bed_l)\n",
    "        bed_files = bed_files.loc[~(bed_files.chrom.isin(bed_x.chrom.unique()))]\n",
    "        bed_files = pd.concat([bed_files, bed_x]).sort_values(by=[\"chrom\", \"start\", \"end\"])\n",
    "    return bed_files\n",
    "\n",
    "def pos_windows(bed_l, window_size, chrom_order):\n",
    "    # Input a bed file and the window size of intervals desired. Multiple chromosomes accepted.\n",
    "    # It has to be sorted.\n",
    "    df_l = []\n",
    "    for c in chrom_order:\n",
    "        #print(c)\n",
    "        frac_l = []\n",
    "        b = bed_l.loc[bed_l[\"chrom\"] == c].copy()\n",
    "        b[\"w_s\"] = b.end-b.start\n",
    "        w_start = b.start.iloc[0]\n",
    "        current_pos, callable_bases = 0, 0\n",
    "        for i, j, k in zip(b.start, b.end, b.w_s):\n",
    "            # Nothing called in the current window under investigation.\n",
    "            while i-window_size >= current_pos:\n",
    "                frac_l.append(callable_bases/window_size)\n",
    "                callable_bases = 0\n",
    "                current_pos += window_size\n",
    "            # Window starts in current. We know this is true because of the previous while loop.\n",
    "            callable_bases += min(k, current_pos+window_size-i)\n",
    "            # Everything called in current.\n",
    "            while j-window_size >= current_pos:\n",
    "                frac_l.append(callable_bases/window_size)\n",
    "                callable_bases = 0\n",
    "                current_pos += window_size\n",
    "                if j-window_size >= current_pos:\n",
    "                    callable_bases += window_size\n",
    "                else:\n",
    "                # Window stops in current. Again, know this is true.\n",
    "                    callable_bases += j-current_pos\n",
    "        # Last window.\n",
    "        frac_l.append(callable_bases/(window_size))\n",
    "        df_l.append(pd.DataFrame({\"chrom\": c, \"window_start\": list(range(0, len(frac_l)*window_size, window_size)),\n",
    "                                  \"window_end\": list(range(window_size, (len(frac_l)+1)*window_size, window_size)),\n",
    "                                  \"callable_frac\": frac_l}))\n",
    "    return pd.concat(df_l)\n",
    "\n",
    "def haploid_double(ds, variable, dim):\n",
    "    unmasked = ~ds[f\"{variable}_mask\"]\n",
    "    overwrite = ds.call_genotype[:,:,0]\n",
    "    overwrite_2 = ds.call_genotype_mask[:,:,0]\n",
    "    return ds.assign(**{\n",
    "        f\"{variable}\": ds[variable].where(\n",
    "        unmasked, \n",
    "        overwrite),\n",
    "        f\"{variable}_mask\": ds[f\"{variable}_mask\"].where(\n",
    "        unmasked,\n",
    "        overwrite_2)})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "daf7b95d",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "zarr_path = \"../zarr_data/\"\n",
    "metadata_folders = glob.glob(metadata_path+\"*_individuals.txt\")\n",
    "\n",
    "size_cutoff = 1000000\n",
    "window_size = 100000\n",
    "missing_filter = 0.5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "f5a97f3f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['../zarr_data/Macaca_maura_ssp/NC_092145.1',\n",
       " '../zarr_data/Macaca_maura_ssp/NW_027257673.1',\n",
       " '../zarr_data/Macaca_maura_ssp/NC_092146.1',\n",
       " '../zarr_data/Macaca_maura_ssp/NC_092125.1']"
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "glob.glob(\"../zarr_data/Macaca_maura_ssp/*\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "id": "7b80c137",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading metadata\n",
      "Loading bed files\n",
      "Loading genetic data\n",
      "../zarr_data/Macaca_maura_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_maura_ssp/NW_027257673.1\n",
      "../zarr_data/Macaca_maura_ssp/NC_092146.1\n",
      "../zarr_data/Macaca_maura_ssp/NC_092125.1\n"
     ]
    }
   ],
   "source": [
    "\n",
    "print(\"Loading metadata\")\n",
    "zarr_path = \"../zarr_data/Macaca_maura_ssp\"\n",
    "metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "\n",
    "short_form = zarr_path.split(\"/\")[-1].split(\"_\")[0]\n",
    "long_form = zarr_path.split(\"/\")[-1]\n",
    "# Loading the various metadata files. Metadata, contig information, callability bed.\n",
    "metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "regions_df[\"chr_type\"] = [\"chrX\" if x == 2 and y == 1 else \"aut\" for x, y in zip(regions_df.FEMALE_PLOIDY, regions_df.MALE_PLOIDY)]\n",
    "large_contigs = regions_df.loc[(regions_df.LENGTH >= 1000000) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "large_x = regions_df.loc[(regions_df.LENGTH >= 1000000) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID\n",
    "print(\"Loading bed files\")\n",
    "bed_files = read_beds(long_form)\n",
    "# Loading the genetic data.\n",
    "print(\"Loading genetic data\")\n",
    "df_l = []\n",
    "for c in glob.glob(zarr_path+\"/*\"):\n",
    "    ds = sg.load_dataset(c)\n",
    "    print(c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "id": "45afc487",
   "metadata": {},
   "outputs": [],
   "source": [
    "ds_var_stats = sg.variant_stats(ds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "47c5efb9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><svg style=\"position: absolute; width: 0; height: 0; overflow: hidden\">\n",
       "<defs>\n",
       "<symbol id=\"icon-database\" viewBox=\"0 0 32 32\">\n",
       "<path d=\"M16 0c-8.837 0-16 2.239-16 5v4c0 2.761 7.163 5 16 5s16-2.239 16-5v-4c0-2.761-7.163-5-16-5z\"></path>\n",
       "<path d=\"M16 17c-8.837 0-16-2.239-16-5v6c0 2.761 7.163 5 16 5s16-2.239 16-5v-6c0 2.761-7.163 5-16 5z\"></path>\n",
       "<path d=\"M16 26c-8.837 0-16-2.239-16-5v6c0 2.761 7.163 5 16 5s16-2.239 16-5v-6c0 2.761-7.163 5-16 5z\"></path>\n",
       "</symbol>\n",
       "<symbol id=\"icon-file-text2\" viewBox=\"0 0 32 32\">\n",
       "<path d=\"M28.681 7.159c-0.694-0.947-1.662-2.053-2.724-3.116s-2.169-2.030-3.116-2.724c-1.612-1.182-2.393-1.319-2.841-1.319h-15.5c-1.378 0-2.5 1.121-2.5 2.5v27c0 1.378 1.122 2.5 2.5 2.5h23c1.378 0 2.5-1.122 2.5-2.5v-19.5c0-0.448-0.137-1.23-1.319-2.841zM24.543 5.457c0.959 0.959 1.712 1.825 2.268 2.543h-4.811v-4.811c0.718 0.556 1.584 1.309 2.543 2.268zM28 29.5c0 0.271-0.229 0.5-0.5 0.5h-23c-0.271 0-0.5-0.229-0.5-0.5v-27c0-0.271 0.229-0.5 0.5-0.5 0 0 15.499-0 15.5 0v7c0 0.552 0.448 1 1 1h7v19.5z\"></path>\n",
       "<path d=\"M23 26h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z\"></path>\n",
       "<path d=\"M23 22h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z\"></path>\n",
       "<path d=\"M23 18h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z\"></path>\n",
       "</symbol>\n",
       "</defs>\n",
       "</svg>\n",
       "<style>/* CSS stylesheet for displaying xarray objects in jupyterlab.\n",
       " *\n",
       " */\n",
       "\n",
       ":root {\n",
       "  --xr-font-color0: var(--jp-content-font-color0, rgba(0, 0, 0, 1));\n",
       "  --xr-font-color2: var(--jp-content-font-color2, rgba(0, 0, 0, 0.54));\n",
       "  --xr-font-color3: var(--jp-content-font-color3, rgba(0, 0, 0, 0.38));\n",
       "  --xr-border-color: var(--jp-border-color2, #e0e0e0);\n",
       "  --xr-disabled-color: var(--jp-layout-color3, #bdbdbd);\n",
       "  --xr-background-color: var(--jp-layout-color0, white);\n",
       "  --xr-background-color-row-even: var(--jp-layout-color1, white);\n",
       "  --xr-background-color-row-odd: var(--jp-layout-color2, #eeeeee);\n",
       "}\n",
       "\n",
       "html[theme=\"dark\"],\n",
       "html[data-theme=\"dark\"],\n",
       "body[data-theme=\"dark\"],\n",
       "body.vscode-dark {\n",
       "  --xr-font-color0: rgba(255, 255, 255, 1);\n",
       "  --xr-font-color2: rgba(255, 255, 255, 0.54);\n",
       "  --xr-font-color3: rgba(255, 255, 255, 0.38);\n",
       "  --xr-border-color: #1f1f1f;\n",
       "  --xr-disabled-color: #515151;\n",
       "  --xr-background-color: #111111;\n",
       "  --xr-background-color-row-even: #111111;\n",
       "  --xr-background-color-row-odd: #313131;\n",
       "}\n",
       "\n",
       ".xr-wrap {\n",
       "  display: block !important;\n",
       "  min-width: 300px;\n",
       "  max-width: 700px;\n",
       "}\n",
       "\n",
       ".xr-text-repr-fallback {\n",
       "  /* fallback to plain text repr when CSS is not injected (untrusted notebook) */\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-header {\n",
       "  padding-top: 6px;\n",
       "  padding-bottom: 6px;\n",
       "  margin-bottom: 4px;\n",
       "  border-bottom: solid 1px var(--xr-border-color);\n",
       "}\n",
       "\n",
       ".xr-header > div,\n",
       ".xr-header > ul {\n",
       "  display: inline;\n",
       "  margin-top: 0;\n",
       "  margin-bottom: 0;\n",
       "}\n",
       "\n",
       ".xr-obj-type,\n",
       ".xr-array-name {\n",
       "  margin-left: 2px;\n",
       "  margin-right: 10px;\n",
       "}\n",
       "\n",
       ".xr-obj-type {\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-sections {\n",
       "  padding-left: 0 !important;\n",
       "  display: grid;\n",
       "  grid-template-columns: 150px auto auto 1fr 0 20px 0 20px;\n",
       "}\n",
       "\n",
       ".xr-section-item {\n",
       "  display: contents;\n",
       "}\n",
       "\n",
       ".xr-section-item input {\n",
       "  display: inline-block;\n",
       "  opacity: 0;\n",
       "  height: 0;\n",
       "}\n",
       "\n",
       ".xr-section-item input + label {\n",
       "  color: var(--xr-disabled-color);\n",
       "}\n",
       "\n",
       ".xr-section-item input:enabled + label {\n",
       "  cursor: pointer;\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-section-item input:focus + label {\n",
       "  border: 2px solid var(--xr-font-color0);\n",
       "}\n",
       "\n",
       ".xr-section-item input:enabled + label:hover {\n",
       "  color: var(--xr-font-color0);\n",
       "}\n",
       "\n",
       ".xr-section-summary {\n",
       "  grid-column: 1;\n",
       "  color: var(--xr-font-color2);\n",
       "  font-weight: 500;\n",
       "}\n",
       "\n",
       ".xr-section-summary > span {\n",
       "  display: inline-block;\n",
       "  padding-left: 0.5em;\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:disabled + label {\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-section-summary-in + label:before {\n",
       "  display: inline-block;\n",
       "  content: \"►\";\n",
       "  font-size: 11px;\n",
       "  width: 15px;\n",
       "  text-align: center;\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:disabled + label:before {\n",
       "  color: var(--xr-disabled-color);\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:checked + label:before {\n",
       "  content: \"▼\";\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:checked + label > span {\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-section-summary,\n",
       ".xr-section-inline-details {\n",
       "  padding-top: 4px;\n",
       "  padding-bottom: 4px;\n",
       "}\n",
       "\n",
       ".xr-section-inline-details {\n",
       "  grid-column: 2 / -1;\n",
       "}\n",
       "\n",
       ".xr-section-details {\n",
       "  display: none;\n",
       "  grid-column: 1 / -1;\n",
       "  margin-bottom: 5px;\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:checked ~ .xr-section-details {\n",
       "  display: contents;\n",
       "}\n",
       "\n",
       ".xr-array-wrap {\n",
       "  grid-column: 1 / -1;\n",
       "  display: grid;\n",
       "  grid-template-columns: 20px auto;\n",
       "}\n",
       "\n",
       ".xr-array-wrap > label {\n",
       "  grid-column: 1;\n",
       "  vertical-align: top;\n",
       "}\n",
       "\n",
       ".xr-preview {\n",
       "  color: var(--xr-font-color3);\n",
       "}\n",
       "\n",
       ".xr-array-preview,\n",
       ".xr-array-data {\n",
       "  padding: 0 5px !important;\n",
       "  grid-column: 2;\n",
       "}\n",
       "\n",
       ".xr-array-data,\n",
       ".xr-array-in:checked ~ .xr-array-preview {\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-array-in:checked ~ .xr-array-data,\n",
       ".xr-array-preview {\n",
       "  display: inline-block;\n",
       "}\n",
       "\n",
       ".xr-dim-list {\n",
       "  display: inline-block !important;\n",
       "  list-style: none;\n",
       "  padding: 0 !important;\n",
       "  margin: 0;\n",
       "}\n",
       "\n",
       ".xr-dim-list li {\n",
       "  display: inline-block;\n",
       "  padding: 0;\n",
       "  margin: 0;\n",
       "}\n",
       "\n",
       ".xr-dim-list:before {\n",
       "  content: \"(\";\n",
       "}\n",
       "\n",
       ".xr-dim-list:after {\n",
       "  content: \")\";\n",
       "}\n",
       "\n",
       ".xr-dim-list li:not(:last-child):after {\n",
       "  content: \",\";\n",
       "  padding-right: 5px;\n",
       "}\n",
       "\n",
       ".xr-has-index {\n",
       "  font-weight: bold;\n",
       "}\n",
       "\n",
       ".xr-var-list,\n",
       ".xr-var-item {\n",
       "  display: contents;\n",
       "}\n",
       "\n",
       ".xr-var-item > div,\n",
       ".xr-var-item label,\n",
       ".xr-var-item > .xr-var-name span {\n",
       "  background-color: var(--xr-background-color-row-even);\n",
       "  margin-bottom: 0;\n",
       "}\n",
       "\n",
       ".xr-var-item > .xr-var-name:hover span {\n",
       "  padding-right: 5px;\n",
       "}\n",
       "\n",
       ".xr-var-list > li:nth-child(odd) > div,\n",
       ".xr-var-list > li:nth-child(odd) > label,\n",
       ".xr-var-list > li:nth-child(odd) > .xr-var-name span {\n",
       "  background-color: var(--xr-background-color-row-odd);\n",
       "}\n",
       "\n",
       ".xr-var-name {\n",
       "  grid-column: 1;\n",
       "}\n",
       "\n",
       ".xr-var-dims {\n",
       "  grid-column: 2;\n",
       "}\n",
       "\n",
       ".xr-var-dtype {\n",
       "  grid-column: 3;\n",
       "  text-align: right;\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-var-preview {\n",
       "  grid-column: 4;\n",
       "}\n",
       "\n",
       ".xr-index-preview {\n",
       "  grid-column: 2 / 5;\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-var-name,\n",
       ".xr-var-dims,\n",
       ".xr-var-dtype,\n",
       ".xr-preview,\n",
       ".xr-attrs dt {\n",
       "  white-space: nowrap;\n",
       "  overflow: hidden;\n",
       "  text-overflow: ellipsis;\n",
       "  padding-right: 10px;\n",
       "}\n",
       "\n",
       ".xr-var-name:hover,\n",
       ".xr-var-dims:hover,\n",
       ".xr-var-dtype:hover,\n",
       ".xr-attrs dt:hover {\n",
       "  overflow: visible;\n",
       "  width: auto;\n",
       "  z-index: 1;\n",
       "}\n",
       "\n",
       ".xr-var-attrs,\n",
       ".xr-var-data,\n",
       ".xr-index-data {\n",
       "  display: none;\n",
       "  background-color: var(--xr-background-color) !important;\n",
       "  padding-bottom: 5px !important;\n",
       "}\n",
       "\n",
       ".xr-var-attrs-in:checked ~ .xr-var-attrs,\n",
       ".xr-var-data-in:checked ~ .xr-var-data,\n",
       ".xr-index-data-in:checked ~ .xr-index-data {\n",
       "  display: block;\n",
       "}\n",
       "\n",
       ".xr-var-data > table {\n",
       "  float: right;\n",
       "}\n",
       "\n",
       ".xr-var-name span,\n",
       ".xr-var-data,\n",
       ".xr-index-name div,\n",
       ".xr-index-data,\n",
       ".xr-attrs {\n",
       "  padding-left: 25px !important;\n",
       "}\n",
       "\n",
       ".xr-attrs,\n",
       ".xr-var-attrs,\n",
       ".xr-var-data,\n",
       ".xr-index-data {\n",
       "  grid-column: 1 / -1;\n",
       "}\n",
       "\n",
       "dl.xr-attrs {\n",
       "  padding: 0;\n",
       "  margin: 0;\n",
       "  display: grid;\n",
       "  grid-template-columns: 125px auto;\n",
       "}\n",
       "\n",
       ".xr-attrs dt,\n",
       ".xr-attrs dd {\n",
       "  padding: 0;\n",
       "  margin: 0;\n",
       "  float: left;\n",
       "  padding-right: 10px;\n",
       "  width: auto;\n",
       "}\n",
       "\n",
       ".xr-attrs dt {\n",
       "  font-weight: normal;\n",
       "  grid-column: 1;\n",
       "}\n",
       "\n",
       ".xr-attrs dt:hover span {\n",
       "  display: inline-block;\n",
       "  background: var(--xr-background-color);\n",
       "  padding-right: 10px;\n",
       "}\n",
       "\n",
       ".xr-attrs dd {\n",
       "  grid-column: 2;\n",
       "  white-space: pre-wrap;\n",
       "  word-break: break-all;\n",
       "}\n",
       "\n",
       ".xr-icon-database,\n",
       ".xr-icon-file-text2,\n",
       ".xr-no-icon {\n",
       "  display: inline-block;\n",
       "  vertical-align: middle;\n",
       "  width: 1em;\n",
       "  height: 1.5em !important;\n",
       "  stroke-width: 0;\n",
       "  stroke: currentColor;\n",
       "  fill: currentColor;\n",
       "}\n",
       "</style><pre class='xr-text-repr-fallback'>&lt;xarray.DataArray &#x27;variant_n_hom_alt&#x27; (variants: 20)&gt; Size: 160B\n",
       "array([0, 0, 0, 2, 5, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1])\n",
       "Dimensions without coordinates: variants</pre><div class='xr-wrap' style='display:none'><div class='xr-header'><div class='xr-obj-type'>xarray.DataArray</div><div class='xr-array-name'>'variant_n_hom_alt'</div><ul class='xr-dim-list'><li><span>variants</span>: 20</li></ul></div><ul class='xr-sections'><li class='xr-section-item'><div class='xr-array-wrap'><input id='section-442a0a91-6236-4408-9afd-9891da30dbae' class='xr-array-in' type='checkbox' checked><label for='section-442a0a91-6236-4408-9afd-9891da30dbae' title='Show/hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-array-preview xr-preview'><span>0 0 0 2 5 2 0 0 0 0 0 0 0 0 0 1 0 0 0 1</span></div><div class='xr-array-data'><pre>array([0, 0, 0, 2, 5, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1])</pre></div></div></li><li class='xr-section-item'><input id='section-e8366235-f3ab-4b34-b75f-00f612ee7a8d' class='xr-section-summary-in' type='checkbox' disabled ><label for='section-e8366235-f3ab-4b34-b75f-00f612ee7a8d' class='xr-section-summary'  title='Expand/collapse section'>Coordinates: <span>(0)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><ul class='xr-var-list'></ul></div></li><li class='xr-section-item'><input id='section-3da53ecc-accb-4907-af81-d7b03d8ecf92' class='xr-section-summary-in' type='checkbox' disabled ><label for='section-3da53ecc-accb-4907-af81-d7b03d8ecf92' class='xr-section-summary'  title='Expand/collapse section'>Indexes: <span>(0)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><ul class='xr-var-list'></ul></div></li><li class='xr-section-item'><input id='section-a07c986e-0c73-41a6-bce0-995820e3b3a0' class='xr-section-summary-in' type='checkbox' disabled ><label for='section-a07c986e-0c73-41a6-bce0-995820e3b3a0' class='xr-section-summary'  title='Expand/collapse section'>Attributes: <span>(0)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><dl class='xr-attrs'></dl></div></li></ul></div></div>"
      ],
      "text/plain": [
       "<xarray.DataArray 'variant_n_hom_alt' (variants: 20)> Size: 160B\n",
       "array([0, 0, 0, 2, 5, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1])\n",
       "Dimensions without coordinates: variants"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ds_var_stats.variant_n_hom_alt[:20].compute()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "id": "6c7bda37",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sgkit.cohorts import _cohorts_to_array\n",
    "from sgkit.stats.utils import assert_array_shape\n",
    "from sgkit.utils import (\n",
    "    conditional_merge_datasets,\n",
    "    create_dataset,\n",
    "    define_variable_if_absent,\n",
    ")\n",
    "from sgkit.window import has_windows, window_statistic\n",
    "\n",
    "from sgkit import variables\n",
    "from sgkit.stats.aggregation import (\n",
    "    count_cohort_alleles,\n",
    "    count_variant_alleles,\n",
    "    individual_heterozygosity,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "18f7aeaf",
   "metadata": {},
   "outputs": [],
   "source": [
    "def alt_hom(ds):\n",
    "    an = ac.sum(axis=2)\n",
    "    n_pairs = an * (an - 1) / 2\n",
    "    n_same = (ac * (ac - 1) / 2).sum(axis=2)\n",
    "    n_diff = n_pairs - n_same\n",
    "    # replace zeros to avoid divide by zero error\n",
    "    n_pairs_na = n_pairs.where(n_pairs != 0)\n",
    "    pi = n_diff / n_pairs_na\n",
    "\n",
    "    if has_windows(ds):\n",
    "        div = window_statistic(\n",
    "            pi,\n",
    "            np.sum,\n",
    "            ds.window_start.values,\n",
    "            ds.window_stop.values,\n",
    "            dtype=pi.dtype,\n",
    "            axis=0,\n",
    "        )\n",
    "        new_ds = create_dataset(\n",
    "            {\n",
    "                variables.stat_diversity: (\n",
    "                    (\"windows\", \"cohorts\"),\n",
    "                    div,\n",
    "                )\n",
    "            }\n",
    "        )\n",
    "    else:\n",
    "        new_ds = create_dataset(\n",
    "            {\n",
    "                variables.stat_diversity: (\n",
    "                    (\"variants\", \"cohorts\"),\n",
    "                    pi.data,\n",
    "                )\n",
    "            }\n",
    "        )\n",
    "    return conditional_merge_datasets(ds, new_ds, merge)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "id": "73edcdda",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[False, False, False, ..., False, False, False],\n",
       "       [False, False, False, ..., False, False, False],\n",
       "       [False, False, False, ..., False, False, False],\n",
       "       ...,\n",
       "       [ True, False,  True, ...,  True,  True,  True],\n",
       "       [False, False, False, ..., False, False, False],\n",
       "       [False, False, False, ..., False, False, False]])"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "((ds.call_genotype[:,:,0] == ds.call_genotype[:,:,1]) & (ds.call_genotype[:,:,0] >= 1)).values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "id": "f8a626c7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[[ 1,  1],\n",
       "        [-1, -1],\n",
       "        [ 1,  1],\n",
       "        [ 1,  1],\n",
       "        [ 1,  1],\n",
       "        [ 1,  1],\n",
       "        [ 1,  1]],\n",
       "\n",
       "       [[ 0,  0],\n",
       "        [-1, -1],\n",
       "        [ 0,  0],\n",
       "        [ 0,  0],\n",
       "        [ 0,  0],\n",
       "        [ 0,  0],\n",
       "        [ 0,  1]],\n",
       "\n",
       "       [[ 0,  0],\n",
       "        [-1, -1],\n",
       "        [-1, -1],\n",
       "        [ 0,  1],\n",
       "        [ 0,  0],\n",
       "        [-1, -1],\n",
       "        [ 0,  0]]], dtype=int8)"
      ]
     },
     "execution_count": 54,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ds.call_genotype[-3:].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "f09c4bcd",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'Hashable' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[42], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m cohort_allele_count: \u001b[43mHashable\u001b[49m \u001b[38;5;241m=\u001b[39m variables\u001b[38;5;241m.\u001b[39mcohort_allele_count,\n\u001b[1;32m      2\u001b[0m ds \u001b[38;5;241m=\u001b[39m define_variable_if_absent(\n\u001b[1;32m      3\u001b[0m         ds, variables\u001b[38;5;241m.\u001b[39mcohort_allele_count, cohort_allele_count, count_cohort_alleles\n\u001b[1;32m      4\u001b[0m     )\n",
      "\u001b[0;31mNameError\u001b[0m: name 'Hashable' is not defined"
     ]
    }
   ],
   "source": [
    "\n",
    "ds = define_variable_if_absent(\n",
    "        ds, variables.cohort_allele_count, cohort_allele_count, count_cohort_alleles\n",
    "    )\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "b0260bbe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<function sgkit.utils.define_variable_if_absent(ds: xarray.core.dataset.Dataset, default_variable_name: Hashable, variable_name: Optional[Hashable], func: Callable[[xarray.core.dataset.Dataset], xarray.core.dataset.Dataset], **kwargs) -> xarray.core.dataset.Dataset>"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "define_variable_if_absent"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "858db6ba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Avahi_laniger_ssp\n",
      "../zarr_data/Avahi_laniger_ssp/HiC_scaffold_2\n",
      "../zarr_data/Avahi_laniger_ssp/HiC_scaffold_1\n",
      "1 Lepilemur_sahamalazensis_ssp\n",
      "Skipping, sample size 1\n",
      "3 Allenopithecus_nigroviridis_ssp\n",
      "../zarr_data/Allenopithecus_nigroviridis_ssp/HiC_scaffold_1\n",
      "../zarr_data/Allenopithecus_nigroviridis_ssp/HiC_scaffold_2\n",
      "3 Alouatta_belzebul_ssp\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000060.1\n",
      "2 Cercopithecus_denti_ssp\n",
      "../zarr_data/Cercopithecus_denti_ssp/CM053398.1\n",
      "../zarr_data/Cercopithecus_denti_ssp/CM053363.1\n",
      "10 Saimiri_ustus_ssp\n",
      "../zarr_data/Saimiri_ustus_ssp/NW_024100917.1\n",
      "../zarr_data/Saimiri_ustus_ssp/NW_024100929.1\n",
      "../zarr_data/Saimiri_ustus_ssp/NW_024100959.1\n",
      "0 Trachypithecus_vetulus_ssp\n",
      "Skipping, sample size 0\n",
      "9 Trachypithecus_francoisi_ssp\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681432.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022680469.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681471.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022680461.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022680471.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022680465.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681437.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681445.1\n",
      "1 Nomascus_gabriellae_ssp\n",
      "Skipping, sample size 1\n",
      "2 Cercopithecus_hamlyni_ssp\n",
      "../zarr_data/Cercopithecus_hamlyni_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_hamlyni_ssp/CM053398.1\n",
      "1 Leontocebus_nigricollis_ssp\n",
      "Skipping, sample size 1\n",
      "4 Chiropotes_albinasus_ssp\n",
      "../zarr_data/Chiropotes_albinasus_ssp/CM052647.1\n",
      "../zarr_data/Chiropotes_albinasus_ssp/CM052624.1\n",
      "8 Macaca_tonkeana_ssp\n",
      "../zarr_data/Macaca_tonkeana_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_tonkeana_ssp/NC_092146.1\n",
      "../zarr_data/Macaca_tonkeana_ssp/NC_092125.1\n",
      "../zarr_data/Macaca_tonkeana_ssp/NW_027257673.1\n",
      "22 Gorilla_beringei_ssp\n",
      "Skipping, sample size 22\n",
      "5 Cebus_albifrons_ssp\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437031.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437119.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437042.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437094.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437043.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437085.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022436941.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437013.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437038.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437131.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437033.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437082.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437044.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437065.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437089.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022436957.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437055.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437104.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437015.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437081.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437116.1\n",
      "1 Mico_humeralifer_ssp\n",
      "Skipping, sample size 1\n",
      "19 Cacajao_calvus_ssp\n",
      "../zarr_data/Cacajao_calvus_ssp/CM052624.1\n",
      "../zarr_data/Cacajao_calvus_ssp/CM052647.1\n",
      "26 Macaca_nemestrina_ssp\n",
      "Skipping, sample size 26\n",
      "1 Macaca_sinica_ssp\n",
      "Skipping, sample size 1\n",
      "4 Aotus_griseimembra_ssp\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988472.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988412.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988496.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988488.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988415.1\n",
      "3 Cercopithecus_neglectus_ssp\n",
      "../zarr_data/Cercopithecus_neglectus_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_neglectus_ssp/CM053398.1\n",
      "1 Pithecia_vanzolinii_ssp\n",
      "Skipping, sample size 1\n",
      "2 Cebus_olivaceus_ssp\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437081.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437015.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437116.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022436957.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437089.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437104.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437055.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437082.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437065.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437044.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437033.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437013.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437131.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437038.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437043.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022436941.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437085.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437119.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437031.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437094.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437042.1\n",
      "1 Semnopithecus_priam_ssp\n",
      "Skipping, sample size 1\n",
      "9 Semnopithecus_entellus_ssp\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022680465.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681445.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681437.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022680469.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681471.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681432.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022680461.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022680471.1\n",
      "3 Hylobates_pileatus_ssp\n",
      "../zarr_data/Hylobates_pileatus_ssp/CM038521.1\n",
      "../zarr_data/Hylobates_pileatus_ssp/CM038516.1\n",
      "2 Galago_moholi_ssp\n",
      "../zarr_data/Galago_moholi_ssp/HiC_scaffold_31\n",
      "../zarr_data/Galago_moholi_ssp/HiC_scaffold_1\n",
      "3 Rhinopithecus_strykeri_ssp\n",
      "../zarr_data/Rhinopithecus_strykeri_ssp/NC_044549.1\n",
      "../zarr_data/Rhinopithecus_strykeri_ssp/NC_044555.1\n",
      "5 Otolemur_garnettii_ssp\n",
      "../zarr_data/Otolemur_garnettii_ssp/HiC_scaffold_1\n",
      "1 Colobus_polykomos_ssp\n",
      "Skipping, sample size 1\n",
      "2 Alouatta_discolor_ssp\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000017.1\n",
      "4 Macaca_leucogenys_ssp\n",
      "../zarr_data/Macaca_leucogenys_ssp/NC_065578.1\n",
      "../zarr_data/Macaca_leucogenys_ssp/NC_065597.1\n",
      "../zarr_data/Macaca_leucogenys_ssp/NC_065598.1\n",
      "../zarr_data/Macaca_leucogenys_ssp/NC_065599.1\n",
      "15 Cercopithecus_mona_ssp\n",
      "../zarr_data/Cercopithecus_mona_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_mona_ssp/CM053398.1\n",
      "1 Ateles_marginatus_ssp\n",
      "Skipping, sample size 1\n",
      "1 Callithrix_kuhlii_ssp\n",
      "Skipping, sample size 1\n",
      "7 Propithecus_diadema_ssp\n",
      "../zarr_data/Propithecus_diadema_ssp/HiC_scaffold_2\n",
      "../zarr_data/Propithecus_diadema_ssp/HiC_scaffold_1\n",
      "4 Rhinopithecus_brelichi_ssp\n",
      "../zarr_data/Rhinopithecus_brelichi_ssp/NC_044549.1\n",
      "../zarr_data/Rhinopithecus_brelichi_ssp/NC_044555.1\n",
      "6 Eulemur_coronatus_ssp\n",
      "../zarr_data/Eulemur_coronatus_ssp/HiC_scaffold_1\n",
      "1 Macaca_brunnescens_ssp\n",
      "Skipping, sample size 1\n",
      "5 Cercopithecus_ascanius_ssp\n",
      "../zarr_data/Cercopithecus_ascanius_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_ascanius_ssp/CM053398.1\n",
      "12 Callithrix_jacchus_ssp\n",
      "../zarr_data/Callithrix_jacchus_ssp/NC_071442.1\n",
      "../zarr_data/Callithrix_jacchus_ssp/NC_071464.1\n",
      "2 Avahi_peyrierasi_ssp\n",
      "../zarr_data/Avahi_peyrierasi_ssp/HiC_scaffold_2\n",
      "../zarr_data/Avahi_peyrierasi_ssp/HiC_scaffold_1\n",
      "1 Tarsius_dentatus_ssp\n",
      "Skipping, sample size 1\n",
      "2 Indri_indri_ssp\n",
      "../zarr_data/Indri_indri_ssp/HiC_scaffold_2\n",
      "../zarr_data/Indri_indri_ssp/HiC_scaffold_1\n",
      "17 Macaca_silenus_ssp\n",
      "../zarr_data/Macaca_silenus_ssp/NW_027257673.1\n",
      "../zarr_data/Macaca_silenus_ssp/NC_092146.1\n",
      "../zarr_data/Macaca_silenus_ssp/NC_092125.1\n",
      "../zarr_data/Macaca_silenus_ssp/NC_092145.1\n",
      "1 Allochrocebus_preussi_ssp\n",
      "Skipping, sample size 1\n",
      "1 Cercopithecus_petaurista_ssp\n",
      "Skipping, sample size 1\n",
      "4 Mico_argentatus_ssp\n",
      "../zarr_data/Mico_argentatus_ssp/NC_071464.1\n",
      "../zarr_data/Mico_argentatus_ssp/NC_071442.1\n",
      "1 Plecturocebus_hoffmannsi_ssp\n",
      "Skipping, sample size 1\n",
      "2 Alouatta_juara_ssp\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000001.1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000034.1\n",
      "1 Hylobates_klossii_ssp\n",
      "Skipping, sample size 1\n",
      "4 Cheracebus_lucifer_ssp\n",
      "../zarr_data/Cheracebus_lucifer_ssp/CM080837.1\n",
      "../zarr_data/Cheracebus_lucifer_ssp/CM080815.1\n",
      "1 Propithecus_deckenii_ssp\n",
      "Skipping, sample size 1\n",
      "2 Cheracebus_regulus_ssp\n",
      "../zarr_data/Cheracebus_regulus_ssp/CM080815.1\n",
      "../zarr_data/Cheracebus_regulus_ssp/CM080837.1\n",
      "2 Alouatta_palliata_ssp\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000040.1\n",
      "1 Colobus_angolensis_ssp\n",
      "Skipping, sample size 1\n",
      "0 Chlorocebus_cynosuros_ssp\n",
      "Skipping, sample size 0\n",
      "4 Hylobates_moloch_ssp\n",
      "../zarr_data/Hylobates_moloch_ssp/CM038516.1\n",
      "../zarr_data/Hylobates_moloch_ssp/CM038521.1\n",
      "18 Lemur_catta_ssp\n",
      "../zarr_data/Lemur_catta_ssp/NC_059155.1\n",
      "../zarr_data/Lemur_catta_ssp/NW_025423732.1\n",
      "../zarr_data/Lemur_catta_ssp/NC_059128.1\n",
      "23 Pongo_abelii_ssp\n",
      "Skipping, sample size 23\n",
      "1 Hapalemur_occidentalis_ssp\n",
      "Skipping, sample size 1\n",
      "1 Allochrocebus_solatus_ssp\n",
      "Skipping, sample size 1\n",
      "2 Cebuella_niveiventris_ssp\n",
      "../zarr_data/Cebuella_niveiventris_ssp/NC_071464.1\n",
      "../zarr_data/Cebuella_niveiventris_ssp/NC_071442.1\n",
      "11 Hoolock_hoolock_ssp\n",
      "../zarr_data/Hoolock_hoolock_ssp/CM104451.1\n",
      "../zarr_data/Hoolock_hoolock_ssp/CM104452.1\n",
      "../zarr_data/Hoolock_hoolock_ssp/CM104434.1\n",
      "2 Chlorocebus_pygerythrus_ssp\n",
      "../zarr_data/Chlorocebus_pygerythrus_ssp/NC_132904.1\n",
      "../zarr_data/Chlorocebus_pygerythrus_ssp/NC_132933.1\n",
      "1 Cercopithecus_roloway_ssp\n",
      "Skipping, sample size 1\n",
      "9 Colobus_guereza_ssp\n",
      "../zarr_data/Colobus_guereza_ssp/CM058074.1\n",
      "../zarr_data/Colobus_guereza_ssp/CM058096.1\n",
      "1 Callithrix_geoffroyi_ssp\n",
      "Skipping, sample size 1\n",
      "3 Trachypithecus_obscurus_ssp\n",
      "../zarr_data/Trachypithecus_obscurus_ssp/NW_022680469.1\n",
      "../zarr_data/Trachypithecus_obscurus_ssp/NW_022681471.1\n",
      "../zarr_data/Trachypithecus_obscurus_ssp/NW_022681432.1\n",
      "../zarr_data/Trachypithecus_obscurus_ssp/NW_022680461.1\n",
      "../zarr_data/Trachypithecus_obscurus_ssp/NW_022680471.1\n",
      "../zarr_data/Trachypithecus_obscurus_ssp/NW_022680465.1\n",
      "../zarr_data/Trachypithecus_obscurus_ssp/NW_022681445.1\n",
      "../zarr_data/Trachypithecus_obscurus_ssp/NW_022681437.1\n",
      "2 Saguinus_imperator_ssp\n",
      "../zarr_data/Saguinus_imperator_ssp/CM063169.1\n",
      "../zarr_data/Saguinus_imperator_ssp/JASSZA010000432.1\n",
      "../zarr_data/Saguinus_imperator_ssp/CM063191.1\n",
      "63 Gorilla_gorilla_ssp\n",
      "Skipping, sample size 63\n",
      "1 Microcebus_tavaratra_ssp\n",
      "Skipping, sample size 1\n",
      "9 Nycticebus_bengalensis_ssp\n",
      "../zarr_data/Nycticebus_bengalensis_ssp/NC_069804.1\n",
      "../zarr_data/Nycticebus_bengalensis_ssp/NC_069805.1\n",
      "../zarr_data/Nycticebus_bengalensis_ssp/NC_069780.1\n",
      "1 Trachypithecus_cristatus_ssp\n",
      "Skipping, sample size 1\n",
      "6 Eulemur_rufifrons_ssp\n",
      "../zarr_data/Eulemur_rufifrons_ssp/NC_090983.1\n",
      "../zarr_data/Eulemur_rufifrons_ssp/NC_091012.1\n",
      "1 Pithecia_albicans_ssp\n",
      "Skipping, sample size 1\n",
      "171 Macaca_mulatta_ssp\n",
      "Skipping, sample size 171\n",
      "2 Cacajao_melanocephalus_ssp\n",
      "../zarr_data/Cacajao_melanocephalus_ssp/CM052624.1\n",
      "../zarr_data/Cacajao_melanocephalus_ssp/CM052647.1\n",
      "1 Trachypithecus_hatinhensis_ssp\n",
      "Skipping, sample size 1\n",
      "1 Galagoides_demidoff_ssp\n",
      "Skipping, sample size 1\n",
      "1 Aotus_trivirgatus_ssp\n",
      "Skipping, sample size 1\n",
      "1 Pithecia_mittermeieri_ssp\n",
      "Skipping, sample size 1\n",
      "2 Hoolock_leuconedys_ssp\n",
      "../zarr_data/Hoolock_leuconedys_ssp/CM104452.1\n",
      "../zarr_data/Hoolock_leuconedys_ssp/CM104451.1\n",
      "../zarr_data/Hoolock_leuconedys_ssp/CM104434.1\n",
      "1 Rhinopithecus_avunculus_ssp\n",
      "Skipping, sample size 1\n",
      "1 Pongo_tapanuliensis_ssp\n",
      "Skipping, sample size 1\n",
      "1 Nomascus_annamensis_ssp\n",
      "Skipping, sample size 1\n",
      "4 Macaca_nigra_ssp\n",
      "../zarr_data/Macaca_nigra_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_nigra_ssp/NC_092146.1\n",
      "../zarr_data/Macaca_nigra_ssp/NC_092125.1\n",
      "../zarr_data/Macaca_nigra_ssp/NW_027257673.1\n",
      "1 Cercopithecus_diana_ssp\n",
      "Skipping, sample size 1\n",
      "4 Nomascus_leucogenys_ssp\n",
      "../zarr_data/Nomascus_leucogenys_ssp/NC_044381.1\n",
      "../zarr_data/Nomascus_leucogenys_ssp/NC_044406.1\n",
      "1 Plecturocebus_grovesi_ssp\n",
      "Skipping, sample size 1\n",
      "4 Hylobates_lar_ssp\n",
      "../zarr_data/Hylobates_lar_ssp/CM038516.1\n",
      "../zarr_data/Hylobates_lar_ssp/CM038521.1\n",
      "106 Papio_anubis_ssp\n",
      "Skipping, sample size 106\n",
      "4 Leontocebus_fuscicollis_ssp\n",
      "../zarr_data/Leontocebus_fuscicollis_ssp/CM063191.1\n",
      "../zarr_data/Leontocebus_fuscicollis_ssp/CM063169.1\n",
      "2 Macaca_nigrescens_ssp\n",
      "../zarr_data/Macaca_nigrescens_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_nigrescens_ssp/NC_092146.1\n",
      "../zarr_data/Macaca_nigrescens_ssp/NC_092125.1\n",
      "../zarr_data/Macaca_nigrescens_ssp/NW_027257673.1\n",
      "1 Trachypithecus_germaini_ssp\n",
      "Skipping, sample size 1\n",
      "3 Ateles_geoffroyi_ssp\n",
      "../zarr_data/Ateles_geoffroyi_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Ateles_geoffroyi_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Ateles_geoffroyi_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Ateles_geoffroyi_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Ateles_geoffroyi_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Ateles_geoffroyi_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Ateles_geoffroyi_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Ateles_geoffroyi_ssp/CAJZLT010000017.1\n",
      "1 Cephalopachus_bancanus_ssp\n",
      "Skipping, sample size 1\n",
      "8 Cheracebus_lugens_ssp\n",
      "../zarr_data/Cheracebus_lugens_ssp/CM080837.1\n",
      "../zarr_data/Cheracebus_lugens_ssp/CM080815.1\n",
      "5 Cercopithecus_mitis_ssp\n",
      "../zarr_data/Cercopithecus_mitis_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_mitis_ssp/CM053398.1\n",
      "1 Nomascus_concolor_ssp\n",
      "Skipping, sample size 1\n",
      "2 Plecturocebus_cinerascens_ssp\n",
      "../zarr_data/Plecturocebus_cinerascens_ssp/CM080815.1\n",
      "../zarr_data/Plecturocebus_cinerascens_ssp/CM080837.1\n",
      "2 Mandrillus_sphinx_ssp\n",
      "../zarr_data/Mandrillus_sphinx_ssp/HiC_scaffold_111\n",
      "../zarr_data/Mandrillus_sphinx_ssp/HiC_scaffold_1\n",
      "../zarr_data/Mandrillus_sphinx_ssp/HiC_scaffold_21\n",
      "0 Mico_humilis_ssp\n",
      "Skipping, sample size 0\n",
      "4 Cebus_unicolor_ssp\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437015.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437081.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437116.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437082.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437065.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437044.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022436957.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437089.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437055.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437104.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437013.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437038.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437131.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437033.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437031.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437119.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437094.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437042.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437043.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437085.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022436941.1\n",
      "7 Eulemur_mongoz_ssp\n",
      "../zarr_data/Eulemur_mongoz_ssp/CM052853.1\n",
      "7 Mandrillus_leucophaeus_ssp\n",
      "../zarr_data/Mandrillus_leucophaeus_ssp/HiC_scaffold_1\n",
      "../zarr_data/Mandrillus_leucophaeus_ssp/HiC_scaffold_21\n",
      "../zarr_data/Mandrillus_leucophaeus_ssp/HiC_scaffold_111\n",
      "10 Pithecia_pithecia_ssp\n",
      "../zarr_data/Pithecia_pithecia_ssp/CM052624.1\n",
      "../zarr_data/Pithecia_pithecia_ssp/CM052647.1\n",
      "28 Papio_kindae_ssp\n",
      "Skipping, sample size 28\n",
      "1 Cheracebus_torquatus_ssp\n",
      "Skipping, sample size 1\n",
      "2 Lepilemur_septentrionalis_ssp\n",
      "../zarr_data/Lepilemur_septentrionalis_ssp/NC_033660.1\n",
      "../zarr_data/Lepilemur_septentrionalis_ssp/NC_033692.1\n",
      "1 Saguinus_bicolor_ssp\n",
      "Skipping, sample size 1\n",
      "6 Mirza_zaza_ssp\n",
      "../zarr_data/Mirza_zaza_ssp/NC_033660.1\n",
      "../zarr_data/Mirza_zaza_ssp/NC_033692.1\n",
      "67 Macaca_fuscata_ssp\n",
      "Skipping, sample size 67\n",
      "143 Pan_troglodytes_ssp\n",
      "Skipping, sample size 143\n",
      "14 Papio_papio_ssp\n",
      "../zarr_data/Papio_papio_ssp/NC_044997.1\n",
      "../zarr_data/Papio_papio_ssp/NC_044996.1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Papio_papio_ssp/NC_044976.1\n",
      "3 Galago_senegalensis_ssp\n",
      "../zarr_data/Galago_senegalensis_ssp/HiC_scaffold_1\n",
      "../zarr_data/Galago_senegalensis_ssp/HiC_scaffold_31\n",
      "2 Ateles_paniscus_ssp\n",
      "../zarr_data/Ateles_paniscus_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Ateles_paniscus_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Ateles_paniscus_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Ateles_paniscus_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Ateles_paniscus_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Ateles_paniscus_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Ateles_paniscus_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Ateles_paniscus_ssp/CAJZLT010000017.1\n",
      "5 Varecia_variegata_ssp\n",
      "../zarr_data/Varecia_variegata_ssp/CM052457.1\n",
      "../zarr_data/Varecia_variegata_ssp/CM052436.1\n",
      "1 Cercocebus_lunulatus_ssp\n",
      "Skipping, sample size 1\n",
      "1 Macaca_siberu_ssp\n",
      "Skipping, sample size 1\n",
      "2 Xanthonycticebus_pygmaeus_ssp\n",
      "../zarr_data/Xanthonycticebus_pygmaeus_ssp/NC_069780.1\n",
      "../zarr_data/Xanthonycticebus_pygmaeus_ssp/NC_069804.1\n",
      "../zarr_data/Xanthonycticebus_pygmaeus_ssp/NC_069805.1\n",
      "1 Avahi_occidentalis_ssp\n",
      "Skipping, sample size 1\n",
      "1 Microcebus_mittermeieri_ssp\n",
      "Skipping, sample size 1\n",
      "1 Propithecus_edwardsi_ssp\n",
      "Skipping, sample size 1\n",
      "7 Hapalemur_griseus_ssp\n",
      "../zarr_data/Hapalemur_griseus_ssp/NC_059128.1\n",
      "../zarr_data/Hapalemur_griseus_ssp/NC_059155.1\n",
      "2 Macaca_sylvanus_ssp\n",
      "../zarr_data/Macaca_sylvanus_ssp/NC_041774.1\n",
      "../zarr_data/Macaca_sylvanus_ssp/NC_041754.1\n",
      "../zarr_data/Macaca_sylvanus_ssp/NW_021161850.1\n",
      "1 Pithecia_chrysocephala_ssp\n",
      "Skipping, sample size 1\n",
      "31 Aotus_nancymaae_ssp\n",
      "Skipping, sample size 31\n",
      "5 Macaca_assamensis_ssp\n",
      "../zarr_data/Macaca_assamensis_ssp/NC_065599.1\n",
      "../zarr_data/Macaca_assamensis_ssp/NC_065598.1\n",
      "../zarr_data/Macaca_assamensis_ssp/NC_065597.1\n",
      "../zarr_data/Macaca_assamensis_ssp/NC_065578.1\n",
      "1 Plecturocebus_donacophilus_ssp\n",
      "Skipping, sample size 1\n",
      "3 Plecturocebus_bernhardi_ssp\n",
      "../zarr_data/Plecturocebus_bernhardi_ssp/CM080837.1\n",
      "../zarr_data/Plecturocebus_bernhardi_ssp/CM080815.1\n",
      "1 Trachypithecus_ebenus_ssp\n",
      "Skipping, sample size 1\n",
      "7 Macaca_maura_ssp\n",
      "../zarr_data/Macaca_maura_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_maura_ssp/NW_027257673.1\n",
      "../zarr_data/Macaca_maura_ssp/NC_092146.1\n",
      "../zarr_data/Macaca_maura_ssp/NC_092125.1\n",
      "5 Alouatta_macconnelli_ssp\n",
      "../zarr_data/Alouatta_macconnelli_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_macconnelli_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_macconnelli_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_macconnelli_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_macconnelli_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_macconnelli_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_macconnelli_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Alouatta_macconnelli_ssp/CAJZLT010000017.1\n",
      "1 Eulemur_cinereiceps_ssp\n",
      "Skipping, sample size 1\n",
      "11 Ateles_chamek_ssp\n",
      "../zarr_data/Ateles_chamek_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Ateles_chamek_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Ateles_chamek_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Ateles_chamek_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Ateles_chamek_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Ateles_chamek_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Ateles_chamek_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Ateles_chamek_ssp/CAJZLT010000017.1\n",
      "1 Plecturocebus_moloch_ssp\n",
      "Skipping, sample size 1\n",
      "6 Eulemur_macaco_ssp\n",
      "../zarr_data/Eulemur_macaco_ssp/HiC_scaffold_1\n",
      "2 Pithecia_hirsuta_ssp\n",
      "../zarr_data/Pithecia_hirsuta_ssp/CM052647.1\n",
      "../zarr_data/Pithecia_hirsuta_ssp/CM052624.1\n",
      "1 Hylobates_abbotti_ssp\n",
      "Skipping, sample size 1\n",
      "2 Cacajao_hosomi_ssp\n",
      "../zarr_data/Cacajao_hosomi_ssp/CM052624.1\n",
      "../zarr_data/Cacajao_hosomi_ssp/CM052647.1\n",
      "1 Saguinus_geoffroyi_ssp\n",
      "Skipping, sample size 1\n",
      "3 Pithecia_pissinatti_ssp\n",
      "Skipping, no completed zarrs\n",
      "3 Nasalis_larvatus_ssp\n",
      "../zarr_data/Nasalis_larvatus_ssp/CM106345.1\n",
      "../zarr_data/Nasalis_larvatus_ssp/CM106344.1\n",
      "2 Propithecus_perrieri_ssp\n",
      "../zarr_data/Propithecus_perrieri_ssp/HiC_scaffold_1\n",
      "../zarr_data/Propithecus_perrieri_ssp/HiC_scaffold_2\n",
      "13 Macaca_radiata_ssp\n",
      "../zarr_data/Macaca_radiata_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_radiata_ssp/NW_027257673.1\n",
      "../zarr_data/Macaca_radiata_ssp/NC_092146.1\n",
      "../zarr_data/Macaca_radiata_ssp/NC_092125.1\n",
      "1 Lepilemur_mustelinus_ssp\n",
      "Skipping, sample size 1\n",
      "8 Prolemur_simus_ssp\n",
      "../zarr_data/Prolemur_simus_ssp/NC_059128.1\n",
      "../zarr_data/Prolemur_simus_ssp/NC_059155.1\n",
      "1 Saguinus_oedipus_ssp\n",
      "Skipping, sample size 1\n",
      "1 Piliocolobus_gordonorum_ssp\n",
      "Skipping, sample size 1\n",
      "2 Alouatta_caraya_ssp\n",
      "../zarr_data/Alouatta_caraya_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_caraya_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_caraya_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_caraya_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_caraya_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Alouatta_caraya_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_caraya_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_caraya_ssp/CAJZLT010000034.1\n",
      "1 Lagothrix_lagotricha_ssp\n",
      "Skipping, sample size 1\n",
      "3 Leontopithecus_rosalia_ssp\n",
      "../zarr_data/Leontopithecus_rosalia_ssp/HiC_scaffold_23\n",
      "../zarr_data/Leontopithecus_rosalia_ssp/HiC_scaffold_1\n",
      "22 Daubentonia_madagascariensis_ssp\n",
      "Skipping, sample size 22\n",
      "8 Propithecus_tattersalli_ssp\n",
      "../zarr_data/Propithecus_tattersalli_ssp/HiC_scaffold_1\n",
      "../zarr_data/Propithecus_tattersalli_ssp/HiC_scaffold_2\n",
      "2 Hylobates_agilis_ssp\n",
      "../zarr_data/Hylobates_agilis_ssp/CM038521.1\n",
      "../zarr_data/Hylobates_agilis_ssp/CM038516.1\n",
      "3 Cacajao_ayresi_ssp\n",
      "../zarr_data/Cacajao_ayresi_ssp/CM052647.1\n",
      "../zarr_data/Cacajao_ayresi_ssp/CM052624.1\n",
      "2 Lepilemur_ruficaudatus_ssp\n",
      "../zarr_data/Lepilemur_ruficaudatus_ssp/NC_033660.1\n",
      "../zarr_data/Lepilemur_ruficaudatus_ssp/NC_033692.1\n",
      "10 Saimiri_boliviensis_ssp\n",
      "../zarr_data/Saimiri_boliviensis_ssp/NW_024100917.1\n",
      "../zarr_data/Saimiri_boliviensis_ssp/NW_024100929.1\n",
      "../zarr_data/Saimiri_boliviensis_ssp/NW_024100959.1\n",
      "5 Macaca_hecki_ssp\n",
      "../zarr_data/Macaca_hecki_ssp/NW_027257673.1\n",
      "../zarr_data/Macaca_hecki_ssp/NC_092125.1\n",
      "../zarr_data/Macaca_hecki_ssp/NC_092145.1\n",
      "3 Macaca_thibetana_ssp\n",
      "../zarr_data/Macaca_thibetana_ssp/NC_065598.1\n",
      "../zarr_data/Macaca_thibetana_ssp/NC_065578.1\n",
      "../zarr_data/Macaca_thibetana_ssp/NC_065597.1\n",
      "4 Varecia_rubra_ssp\n",
      "../zarr_data/Varecia_rubra_ssp/CM052436.1\n",
      "../zarr_data/Varecia_rubra_ssp/CM052457.1\n",
      "3 Cercopithecus_nictitans_ssp\n",
      "../zarr_data/Cercopithecus_nictitans_ssp/CM053398.1\n",
      "../zarr_data/Cercopithecus_nictitans_ssp/CM053363.1\n",
      "45 Theropithecus_gelada_ssp\n",
      "Skipping, sample size 45\n",
      "1 Microcebus_jonahi_ssp\n",
      "Skipping, sample size 1\n",
      "4 Semnopithecus_hypoleucos_ssp\n",
      "../zarr_data/Semnopithecus_hypoleucos_ssp/NW_022680471.1\n",
      "../zarr_data/Semnopithecus_hypoleucos_ssp/NW_022680461.1\n",
      "../zarr_data/Semnopithecus_hypoleucos_ssp/NW_022681471.1\n",
      "../zarr_data/Semnopithecus_hypoleucos_ssp/NW_022680469.1\n",
      "../zarr_data/Semnopithecus_hypoleucos_ssp/NW_022681432.1\n",
      "../zarr_data/Semnopithecus_hypoleucos_ssp/NW_022681445.1\n",
      "../zarr_data/Semnopithecus_hypoleucos_ssp/NW_022681437.1\n",
      "../zarr_data/Semnopithecus_hypoleucos_ssp/NW_022680465.1\n",
      "1 Brachyteles_hypoxanthus_ssp\n",
      "Skipping, sample size 1\n",
      "4 Chiropotes_sagulatus_ssp\n",
      "../zarr_data/Chiropotes_sagulatus_ssp/CM052624.1\n",
      "../zarr_data/Chiropotes_sagulatus_ssp/CM052647.1\n",
      "14 Aotus_azarai_ssp\n",
      "../zarr_data/Aotus_azarai_ssp/NW_026988412.1\n",
      "../zarr_data/Aotus_azarai_ssp/NW_026988496.1\n",
      "../zarr_data/Aotus_azarai_ssp/NW_026988472.1\n",
      "../zarr_data/Aotus_azarai_ssp/NW_026988415.1\n",
      "../zarr_data/Aotus_azarai_ssp/NW_026988488.1\n",
      "1 Saguinus_inustus_ssp\n",
      "Skipping, sample size 1\n",
      "2 Eulemur_fulvus_ssp\n",
      "../zarr_data/Eulemur_fulvus_ssp/NC_091012.1\n",
      "../zarr_data/Eulemur_fulvus_ssp/NC_090983.1\n",
      "2 Hapalemur_meridionalis_ssp\n",
      "../zarr_data/Hapalemur_meridionalis_ssp/NC_059155.1\n",
      "../zarr_data/Hapalemur_meridionalis_ssp/NC_059128.1\n",
      "0 Chlorocebus_tantalus_ssp\n",
      "Skipping, sample size 0\n",
      "1 Cercocebus_chrysogaster_ssp\n",
      "Skipping, sample size 1\n",
      "1 Hylobates_muelleri_ssp\n",
      "Skipping, sample size 1\n",
      "3 Allochrocebus_lhoesti_ssp\n",
      "../zarr_data/Allochrocebus_lhoesti_ssp/NC_132933.1\n",
      "../zarr_data/Allochrocebus_lhoesti_ssp/NC_132904.1\n",
      "3 Plecturocebus_dubius_ssp\n",
      "../zarr_data/Plecturocebus_dubius_ssp/CM080815.1\n",
      "../zarr_data/Plecturocebus_dubius_ssp/CM080837.1\n",
      "1 Miopithecus_talapoin_ssp\n",
      "Skipping, sample size 1\n",
      "2 Cercocebus_torquatus_ssp\n",
      "../zarr_data/Cercocebus_torquatus_ssp/HiC_scaffold_1\n",
      "../zarr_data/Cercocebus_torquatus_ssp/HiC_scaffold_111\n",
      "../zarr_data/Cercocebus_torquatus_ssp/HiC_scaffold_21\n",
      "7 Macaca_leonina_ssp\n",
      "../zarr_data/Macaca_leonina_ssp/NW_027257673.1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Macaca_leonina_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_leonina_ssp/NC_092146.1\n",
      "../zarr_data/Macaca_leonina_ssp/NC_092125.1\n",
      "7 Pygathrix_nemaeus_ssp\n",
      "../zarr_data/Pygathrix_nemaeus_ssp/NC_044549.1\n",
      "../zarr_data/Pygathrix_nemaeus_ssp/NC_044555.1\n",
      "3 Otolemur_crassicaudatus_ssp\n",
      "../zarr_data/Otolemur_crassicaudatus_ssp/HiC_scaffold_1\n",
      "3 Cercopithecus_cephus_ssp\n",
      "../zarr_data/Cercopithecus_cephus_ssp/CM053398.1\n",
      "../zarr_data/Cercopithecus_cephus_ssp/CM053363.1\n",
      "2 Loris_lydekkerianus_ssp\n",
      "../zarr_data/Loris_lydekkerianus_ssp/NC_069805.1\n",
      "../zarr_data/Loris_lydekkerianus_ssp/NC_069804.1\n",
      "../zarr_data/Loris_lydekkerianus_ssp/NC_069780.1\n",
      "80 Papio_cynocephalus_ssp\n",
      "Skipping, sample size 80\n",
      "1 Carlito_syrichta_ssp\n",
      "Skipping, sample size 1\n",
      "21 Alouatta_seniculus_ssp\n",
      "Skipping, sample size 21\n",
      "13 Cercocebus_atys_ssp\n",
      "../zarr_data/Cercocebus_atys_ssp/HiC_scaffold_21\n",
      "../zarr_data/Cercocebus_atys_ssp/HiC_scaffold_111\n",
      "../zarr_data/Cercocebus_atys_ssp/HiC_scaffold_1\n",
      "17 Pan_paniscus_ssp\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073267.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_085926.1\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073256.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073257.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_085927.1\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073266.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073264.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073255.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073254.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073265.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073261.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073271.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073270.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073260.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073259.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073262.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073272.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073253.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073268.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073249.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073269.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073252.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073273.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073263.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073258.2\n",
      "1 Presbytis_comata_ssp\n",
      "Skipping, sample size 1\n",
      "4 Eulemur_sanfordi_ssp\n",
      "../zarr_data/Eulemur_sanfordi_ssp/NC_090983.1\n",
      "../zarr_data/Eulemur_sanfordi_ssp/NC_091012.1\n",
      "2 Saimiri_sciureus_ssp\n",
      "../zarr_data/Saimiri_sciureus_ssp/NW_024100917.1\n",
      "../zarr_data/Saimiri_sciureus_ssp/NW_024100959.1\n",
      "../zarr_data/Saimiri_sciureus_ssp/NW_024100929.1\n",
      "4 Saguinus_midas_ssp\n",
      "../zarr_data/Saguinus_midas_ssp/CM063191.1\n",
      "../zarr_data/Saguinus_midas_ssp/JASSZA010000432.1\n",
      "../zarr_data/Saguinus_midas_ssp/CM063169.1\n",
      "1 Piliocolobus_kirkii_ssp\n",
      "Skipping, sample size 1\n",
      "2 Propithecus_verreauxi_ssp\n",
      "../zarr_data/Propithecus_verreauxi_ssp/HiC_scaffold_1\n",
      "../zarr_data/Propithecus_verreauxi_ssp/HiC_scaffold_2\n",
      "1 Nomascus_siki_ssp\n",
      "Skipping, sample size 1\n",
      "4 Trachypithecus_phayrei_ssp\n",
      "../zarr_data/Trachypithecus_phayrei_ssp/NW_022680465.1\n",
      "../zarr_data/Trachypithecus_phayrei_ssp/NW_022681437.1\n",
      "../zarr_data/Trachypithecus_phayrei_ssp/NW_022681445.1\n",
      "../zarr_data/Trachypithecus_phayrei_ssp/NW_022681432.1\n",
      "../zarr_data/Trachypithecus_phayrei_ssp/NW_022680469.1\n",
      "../zarr_data/Trachypithecus_phayrei_ssp/NW_022681471.1\n",
      "../zarr_data/Trachypithecus_phayrei_ssp/NW_022680461.1\n",
      "../zarr_data/Trachypithecus_phayrei_ssp/NW_022680471.1\n",
      "10 Propithecus_coquereli_ssp\n",
      "../zarr_data/Propithecus_coquereli_ssp/HiC_scaffold_2\n",
      "../zarr_data/Propithecus_coquereli_ssp/HiC_scaffold_1\n"
     ]
    }
   ],
   "source": [
    "# Fst implementation.\n",
    "for x in glob.glob(zarr_path+\"*\")[25:]:\n",
    "    # The name used to load all the files, short and long version\n",
    "    short_form = x.split(\"/\")[-1].split(\"_\")[0]\n",
    "    long_form = x.split(\"/\")[-1]\n",
    "    # Loading the various metadata files. Metadata, contig information, callability bed.\n",
    "    metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "    metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "    metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "    regions_df[\"chr_type\"] = [\"ChrX\" if x == 2 and y == 1 else \"ChrY\" if x == 0 and y == 1 else \"Aut\" for x, y in zip(regions_df.FEMALE_PLOIDY, regions_df.MALE_PLOIDY)]\n",
    "    large_contigs = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "    large_x = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID\n",
    "    # Skipping the large samples sizes and the singulars for the first Fst calc\n",
    "    # Also perform a 10X filter to get accurate samples sizes\n",
    "    sample_size = len(metadata_df.loc[(metadata_df.GVCF_FOLDER == long_form) &\n",
    "                                     (metadata_df.AVG_COVERAGE_A >= 10)])\n",
    "    print(sample_size, long_form)\n",
    "    if (sample_size > 20) or (sample_size <= 1):\n",
    "        print(\"Skipping, sample size\", sample_size)\n",
    "        continue\n",
    "    if os.path.exists(\"../results/window_stats/{}_Fst.txt\".format(long_form)):\n",
    "        print(\"Skipping as it's processed\")\n",
    "        continue\n",
    "    if len(glob.glob(x+\"/*\")) == 0:\n",
    "        print(\"Skipping, no completed zarrs\")\n",
    "        continue\n",
    "    # Loading the genetic data.\n",
    "    df_l = []\n",
    "    for c in glob.glob(x+\"/*\"):\n",
    "        print(c)\n",
    "        ds = sg.load_dataset(c)\n",
    "        # This implementation is the pi implementation.\n",
    "        # Probably problematic in some cases with population structure, but it is easier to implement\n",
    "        ds[\"sample_cohort\"] = ds[\"samples\"]\n",
    "        # Subsetting and windowing the sgkit dataset. The rechunking handles what otherwise would cause an error.\n",
    "        #ds[\"call_genotype\"] = ds[\"call_genotype\"].clip(0)\n",
    "        ds = ds.sel(contigs=[ds.variant_contig[0].values])\n",
    "        if c.split(\"/\")[-1] in list(large_x):\n",
    "            ds = haploid_double(ds, \"call_genotype\", \"samples\")\n",
    "        missing_rate = ds.call_genotype_mask[:,:,0].sum(axis=1).values/ds.call_genotype_mask[:,:,0].count(axis=1).values\n",
    "        ds = ds.isel(variants=(missing_rate <= 0))\n",
    "        ds = sg.window_by_genome(ds)\n",
    "        ds = (sg.Fst(ds.chunk({\"variants\": 10000})))\n",
    "        df_sub = pd.DataFrame(ds.stat_Fst[0,:,], columns=ds.sample_id)\n",
    "        df_sub[\"chrom\"] = c.split(\"/\")[-1]\n",
    "        df_sub[\"variants_used\"] = len(ds.variants)\n",
    "        df_sub[\"stat\"] = \"Fst\"\n",
    "        df_l.append(df_sub)\n",
    "        df_sub = pd.DataFrame(ds.stat_divergence[0,:,], columns=ds.sample_id)\n",
    "        df_sub[\"chrom\"] = c.split(\"/\")[-1]\n",
    "        df_sub[\"variants_used\"] = len(ds.variants)\n",
    "        df_sub[\"stat\"] = \"Divergence\"\n",
    "        df_l.append(df_sub)\n",
    "    output_df = pd.concat(df_l)\n",
    "    output_df[\"chr_type\"] = output_df[\"chrom\"].map(dict(zip(regions_df.CONTIG_ID, regions_df.chr_type)))\n",
    "    output_df[\"species\"] = long_form\n",
    "    output_df.to_csv(\"../results/window_stats/{}_Fst.txt\".format(long_form), sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "3ecc6ef2",
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy.spatial.distance import squareform\n",
    "from scipy.cluster import hierarchy\n",
    "from scipy.cluster.hierarchy import dendrogram\n",
    "from sklearn.cluster import AgglomerativeClustering\n",
    "\n",
    "def plot_dendrogram(model, **kwargs):\n",
    "    # Create linkage matrix and then plot the dendrogram\n",
    "\n",
    "    # create the counts of samples under each node\n",
    "    counts = np.zeros(model.children_.shape[0])\n",
    "    n_samples = len(model.labels_)\n",
    "    for i, merge in enumerate(model.children_):\n",
    "        current_count = 0\n",
    "        for child_idx in merge:\n",
    "            if child_idx < n_samples:\n",
    "                current_count += 1  # leaf node\n",
    "            else:\n",
    "                current_count += counts[child_idx - n_samples]\n",
    "        counts[i] = current_count\n",
    "\n",
    "    linkage_matrix = np.column_stack(\n",
    "        [model.children_, model.distances_, counts]\n",
    "    ).astype(float)\n",
    "\n",
    "    # Plot the corresponding dendrogram\n",
    "    dendrogram(linkage_matrix, **kwargs)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "780cec4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def divergence_divide(square_df):\n",
    "    square_out = np.zeros((len(square_df), len(square_df)))\n",
    "    for i in range(len(square_df)):\n",
    "        for j in range(len(square_df)):\n",
    "            square_out[i,j] = square_df.iloc[i,j]/((square_df.iloc[i,i]+square_df.iloc[j,j])/2)\n",
    "    return square_out"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2568731e",
   "metadata": {},
   "source": [
    "Flexible intervals:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "710d4ed8",
   "metadata": {},
   "outputs": [],
   "source": [
    "def interval_creator(bed_l, window_size):\n",
    "    # Input a bed file and the window size of intervals desired. Multiple chromosomes accepted.\n",
    "    df_l = []\n",
    "    for c in bed_l.chrom.unique():\n",
    "        print(c)\n",
    "        start_l, end_l = [], []\n",
    "        b = bed_l.loc[bed_l[\"chrom\"] == c].copy()\n",
    "        b[\"w_s\"] = b.end-b.start\n",
    "        w_start = b.start.iloc[0]\n",
    "        current_size = 0\n",
    "        for i, j, k in zip(b.start, b.end, b.w_s):\n",
    "            # Current window encapsulates the final stretch of the interval.\n",
    "            if current_size + k >= window_size:\n",
    "                start_l.append(w_start), end_l.append(i+(window_size-current_size))\n",
    "                w_start = i+(window_size-current_size)\n",
    "                # If the window still contains full intervals, contigous windows until it cant.\n",
    "                for x in range((k-window_size+current_size)//window_size):\n",
    "                    start_l.append(w_start), end_l.append(w_start+window_size)\n",
    "                    w_start += window_size\n",
    "                current_size = j-w_start\n",
    "            # Current window does not encapsulate per definition, so it has to be added to current size but nothing else.\n",
    "            else:\n",
    "                current_size += k\n",
    "        df_l.append(pd.DataFrame({\"chrom\": c, \"interval_start\": start_l, \"interval_end\": end_l}))\n",
    "    return pd.concat(df_l)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
