{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "551183f8",
   "metadata": {},
   "outputs": [],
   "source": [
    "%run ../scripts/notebook_settings.py\n",
    "import sgkit as sg\n",
    "import xarray as xr\n",
    "import glob"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "75c2c2ce",
   "metadata": {},
   "source": [
    "Workflow to load in the respective individuals and generate base statistics on them."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "d7fa0b04",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Functions\n",
    "\n",
    "def read_beds(long_form):\n",
    "    bed_path_x = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024/{}/filteredVCF/pos_bed_cov_based/{}_batch*_fploidy2_mploidy1.bed\".format(long_form, long_form)\n",
    "    bed_path_all = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024/{}/filteredVCF/pos_bed_cov_based/{}_batch*_fploidy2_mploidy2.bed\".format(long_form, long_form)\n",
    "    bed_l = []\n",
    "    for b in glob.glob(bed_path_all):\n",
    "        bed_file = pd.read_csv(b, sep=\"\\t\", names=[\"chrom\", \"start\", \"end\"])\n",
    "        bed_l.append(bed_file)\n",
    "    bed_files = pd.concat(bed_l)\n",
    "    bed_l = []\n",
    "    for b in glob.glob(bed_path_x):\n",
    "        #print(b)\n",
    "        bed_file = pd.read_csv(b, sep=\"\\t\", names=[\"chrom\", \"start\", \"end\"])\n",
    "        bed_l.append(bed_file)\n",
    "    if len(bed_l) > 0:\n",
    "        bed_x = pd.concat(bed_l)\n",
    "        bed_files = bed_files.loc[~(bed_files.chrom.isin(bed_x.chrom.unique()))]\n",
    "        bed_files = pd.concat([bed_files, bed_x]).sort_values(by=[\"chrom\", \"start\", \"end\"])\n",
    "    return bed_files\n",
    "\n",
    "def pos_windows(bed_l, window_size, chrom_order):\n",
    "    # Input a bed file and the window size of intervals desired. Multiple chromosomes accepted.\n",
    "    # It has to be sorted.\n",
    "    df_l = []\n",
    "    for c in chrom_order:\n",
    "        #print(c)\n",
    "        frac_l = []\n",
    "        b = bed_l.loc[bed_l[\"chrom\"] == c].copy()\n",
    "        b[\"w_s\"] = b.end-b.start\n",
    "        w_start = b.start.iloc[0]\n",
    "        current_pos, callable_bases = 0, 0\n",
    "        for i, j, k in zip(b.start, b.end, b.w_s):\n",
    "            # Nothing called in the current window under investigation.\n",
    "            while i-window_size >= current_pos:\n",
    "                frac_l.append(callable_bases/window_size)\n",
    "                callable_bases = 0\n",
    "                current_pos += window_size\n",
    "            # Window starts in current. We know this is true because of the previous while loop.\n",
    "            callable_bases += min(k, current_pos+window_size-i)\n",
    "            # Everything called in current.\n",
    "            while j-window_size >= current_pos:\n",
    "                frac_l.append(callable_bases/window_size)\n",
    "                callable_bases = 0\n",
    "                current_pos += window_size\n",
    "                if j-window_size >= current_pos:\n",
    "                    callable_bases += window_size\n",
    "                else:\n",
    "                # Window stops in current. Again, know this is true.\n",
    "                    callable_bases += j-current_pos\n",
    "        # Last window.\n",
    "        frac_l.append(callable_bases/(window_size))\n",
    "        df_l.append(pd.DataFrame({\"chrom\": c, \"window_start\": list(range(0, len(frac_l)*window_size, window_size)),\n",
    "                                  \"window_end\": list(range(window_size, (len(frac_l)+1)*window_size, window_size)),\n",
    "                                  \"callable_frac\": frac_l}))\n",
    "    return pd.concat(df_l)\n",
    "\n",
    "def haploid_double(ds, variable, dim):\n",
    "    unmasked = ~ds[f\"{variable}_mask\"]\n",
    "    overwrite = ds.call_genotype[:,:,0]\n",
    "    overwrite_2 = ds.call_genotype_mask[:,:,0]\n",
    "    return ds.assign(**{\n",
    "        f\"{variable}\": ds[variable].where(\n",
    "        unmasked, \n",
    "        overwrite),\n",
    "        f\"{variable}_mask\": ds[f\"{variable}_mask\"].where(\n",
    "        unmasked,\n",
    "        overwrite_2)})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "daf7b95d",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "zarr_path = \"../zarr_data/\"\n",
    "metadata_folders = glob.glob(metadata_path+\"*_individuals.txt\")\n",
    "\n",
    "size_cutoff = 1000000\n",
    "window_size = 100000\n",
    "missing_filter = 0.5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "f5a97f3f",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['../zarr_data/Macaca_maura_ssp/NC_092145.1',\n",
       " '../zarr_data/Macaca_maura_ssp/NW_027257673.1',\n",
       " '../zarr_data/Macaca_maura_ssp/NC_092146.1',\n",
       " '../zarr_data/Macaca_maura_ssp/NC_092125.1']"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "glob.glob(\"../zarr_data/Macaca_maura_ssp/*\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "7b80c137",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Loading metadata\n",
      "Loading bed files\n",
      "Loading genetic data\n",
      "../zarr_data/Macaca_maura_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_maura_ssp/NW_027257673.1\n",
      "../zarr_data/Macaca_maura_ssp/NC_092146.1\n",
      "../zarr_data/Macaca_maura_ssp/NC_092125.1\n"
     ]
    }
   ],
   "source": [
    "\n",
    "print(\"Loading metadata\")\n",
    "zarr_path = \"../zarr_data/Macaca_maura_ssp\"\n",
    "metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "\n",
    "short_form = zarr_path.split(\"/\")[-1].split(\"_\")[0]\n",
    "long_form = zarr_path.split(\"/\")[-1]\n",
    "# Loading the various metadata files. Metadata, contig information, callability bed.\n",
    "metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "regions_df[\"chr_type\"] = [\"chrX\" if x == 2 and y == 1 else \"aut\" for x, y in zip(regions_df.FEMALE_PLOIDY, regions_df.MALE_PLOIDY)]\n",
    "large_contigs = regions_df.loc[(regions_df.LENGTH >= 1000000) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "large_x = regions_df.loc[(regions_df.LENGTH >= 1000000) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID\n",
    "print(\"Loading bed files\")\n",
    "bed_files = read_beds(long_form)\n",
    "# Loading the genetic data.\n",
    "print(\"Loading genetic data\")\n",
    "df_l = []\n",
    "for c in glob.glob(zarr_path+\"/*\"):\n",
    "    ds = sg.load_dataset(c)\n",
    "    print(c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "45afc487",
   "metadata": {},
   "outputs": [],
   "source": [
    "ds_var_stats = sg.variant_stats(ds)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "47c5efb9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><svg style=\"position: absolute; width: 0; height: 0; overflow: hidden\">\n",
       "<defs>\n",
       "<symbol id=\"icon-database\" viewBox=\"0 0 32 32\">\n",
       "<path d=\"M16 0c-8.837 0-16 2.239-16 5v4c0 2.761 7.163 5 16 5s16-2.239 16-5v-4c0-2.761-7.163-5-16-5z\"></path>\n",
       "<path d=\"M16 17c-8.837 0-16-2.239-16-5v6c0 2.761 7.163 5 16 5s16-2.239 16-5v-6c0 2.761-7.163 5-16 5z\"></path>\n",
       "<path d=\"M16 26c-8.837 0-16-2.239-16-5v6c0 2.761 7.163 5 16 5s16-2.239 16-5v-6c0 2.761-7.163 5-16 5z\"></path>\n",
       "</symbol>\n",
       "<symbol id=\"icon-file-text2\" viewBox=\"0 0 32 32\">\n",
       "<path d=\"M28.681 7.159c-0.694-0.947-1.662-2.053-2.724-3.116s-2.169-2.030-3.116-2.724c-1.612-1.182-2.393-1.319-2.841-1.319h-15.5c-1.378 0-2.5 1.121-2.5 2.5v27c0 1.378 1.122 2.5 2.5 2.5h23c1.378 0 2.5-1.122 2.5-2.5v-19.5c0-0.448-0.137-1.23-1.319-2.841zM24.543 5.457c0.959 0.959 1.712 1.825 2.268 2.543h-4.811v-4.811c0.718 0.556 1.584 1.309 2.543 2.268zM28 29.5c0 0.271-0.229 0.5-0.5 0.5h-23c-0.271 0-0.5-0.229-0.5-0.5v-27c0-0.271 0.229-0.5 0.5-0.5 0 0 15.499-0 15.5 0v7c0 0.552 0.448 1 1 1h7v19.5z\"></path>\n",
       "<path d=\"M23 26h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z\"></path>\n",
       "<path d=\"M23 22h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z\"></path>\n",
       "<path d=\"M23 18h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z\"></path>\n",
       "</symbol>\n",
       "</defs>\n",
       "</svg>\n",
       "<style>/* CSS stylesheet for displaying xarray objects in jupyterlab.\n",
       " *\n",
       " */\n",
       "\n",
       ":root {\n",
       "  --xr-font-color0: var(--jp-content-font-color0, rgba(0, 0, 0, 1));\n",
       "  --xr-font-color2: var(--jp-content-font-color2, rgba(0, 0, 0, 0.54));\n",
       "  --xr-font-color3: var(--jp-content-font-color3, rgba(0, 0, 0, 0.38));\n",
       "  --xr-border-color: var(--jp-border-color2, #e0e0e0);\n",
       "  --xr-disabled-color: var(--jp-layout-color3, #bdbdbd);\n",
       "  --xr-background-color: var(--jp-layout-color0, white);\n",
       "  --xr-background-color-row-even: var(--jp-layout-color1, white);\n",
       "  --xr-background-color-row-odd: var(--jp-layout-color2, #eeeeee);\n",
       "}\n",
       "\n",
       "html[theme=\"dark\"],\n",
       "html[data-theme=\"dark\"],\n",
       "body[data-theme=\"dark\"],\n",
       "body.vscode-dark {\n",
       "  --xr-font-color0: rgba(255, 255, 255, 1);\n",
       "  --xr-font-color2: rgba(255, 255, 255, 0.54);\n",
       "  --xr-font-color3: rgba(255, 255, 255, 0.38);\n",
       "  --xr-border-color: #1f1f1f;\n",
       "  --xr-disabled-color: #515151;\n",
       "  --xr-background-color: #111111;\n",
       "  --xr-background-color-row-even: #111111;\n",
       "  --xr-background-color-row-odd: #313131;\n",
       "}\n",
       "\n",
       ".xr-wrap {\n",
       "  display: block !important;\n",
       "  min-width: 300px;\n",
       "  max-width: 700px;\n",
       "}\n",
       "\n",
       ".xr-text-repr-fallback {\n",
       "  /* fallback to plain text repr when CSS is not injected (untrusted notebook) */\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-header {\n",
       "  padding-top: 6px;\n",
       "  padding-bottom: 6px;\n",
       "  margin-bottom: 4px;\n",
       "  border-bottom: solid 1px var(--xr-border-color);\n",
       "}\n",
       "\n",
       ".xr-header > div,\n",
       ".xr-header > ul {\n",
       "  display: inline;\n",
       "  margin-top: 0;\n",
       "  margin-bottom: 0;\n",
       "}\n",
       "\n",
       ".xr-obj-type,\n",
       ".xr-array-name {\n",
       "  margin-left: 2px;\n",
       "  margin-right: 10px;\n",
       "}\n",
       "\n",
       ".xr-obj-type {\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-sections {\n",
       "  padding-left: 0 !important;\n",
       "  display: grid;\n",
       "  grid-template-columns: 150px auto auto 1fr 0 20px 0 20px;\n",
       "}\n",
       "\n",
       ".xr-section-item {\n",
       "  display: contents;\n",
       "}\n",
       "\n",
       ".xr-section-item input {\n",
       "  display: inline-block;\n",
       "  opacity: 0;\n",
       "  height: 0;\n",
       "}\n",
       "\n",
       ".xr-section-item input + label {\n",
       "  color: var(--xr-disabled-color);\n",
       "}\n",
       "\n",
       ".xr-section-item input:enabled + label {\n",
       "  cursor: pointer;\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-section-item input:focus + label {\n",
       "  border: 2px solid var(--xr-font-color0);\n",
       "}\n",
       "\n",
       ".xr-section-item input:enabled + label:hover {\n",
       "  color: var(--xr-font-color0);\n",
       "}\n",
       "\n",
       ".xr-section-summary {\n",
       "  grid-column: 1;\n",
       "  color: var(--xr-font-color2);\n",
       "  font-weight: 500;\n",
       "}\n",
       "\n",
       ".xr-section-summary > span {\n",
       "  display: inline-block;\n",
       "  padding-left: 0.5em;\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:disabled + label {\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-section-summary-in + label:before {\n",
       "  display: inline-block;\n",
       "  content: \"►\";\n",
       "  font-size: 11px;\n",
       "  width: 15px;\n",
       "  text-align: center;\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:disabled + label:before {\n",
       "  color: var(--xr-disabled-color);\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:checked + label:before {\n",
       "  content: \"▼\";\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:checked + label > span {\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-section-summary,\n",
       ".xr-section-inline-details {\n",
       "  padding-top: 4px;\n",
       "  padding-bottom: 4px;\n",
       "}\n",
       "\n",
       ".xr-section-inline-details {\n",
       "  grid-column: 2 / -1;\n",
       "}\n",
       "\n",
       ".xr-section-details {\n",
       "  display: none;\n",
       "  grid-column: 1 / -1;\n",
       "  margin-bottom: 5px;\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:checked ~ .xr-section-details {\n",
       "  display: contents;\n",
       "}\n",
       "\n",
       ".xr-array-wrap {\n",
       "  grid-column: 1 / -1;\n",
       "  display: grid;\n",
       "  grid-template-columns: 20px auto;\n",
       "}\n",
       "\n",
       ".xr-array-wrap > label {\n",
       "  grid-column: 1;\n",
       "  vertical-align: top;\n",
       "}\n",
       "\n",
       ".xr-preview {\n",
       "  color: var(--xr-font-color3);\n",
       "}\n",
       "\n",
       ".xr-array-preview,\n",
       ".xr-array-data {\n",
       "  padding: 0 5px !important;\n",
       "  grid-column: 2;\n",
       "}\n",
       "\n",
       ".xr-array-data,\n",
       ".xr-array-in:checked ~ .xr-array-preview {\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-array-in:checked ~ .xr-array-data,\n",
       ".xr-array-preview {\n",
       "  display: inline-block;\n",
       "}\n",
       "\n",
       ".xr-dim-list {\n",
       "  display: inline-block !important;\n",
       "  list-style: none;\n",
       "  padding: 0 !important;\n",
       "  margin: 0;\n",
       "}\n",
       "\n",
       ".xr-dim-list li {\n",
       "  display: inline-block;\n",
       "  padding: 0;\n",
       "  margin: 0;\n",
       "}\n",
       "\n",
       ".xr-dim-list:before {\n",
       "  content: \"(\";\n",
       "}\n",
       "\n",
       ".xr-dim-list:after {\n",
       "  content: \")\";\n",
       "}\n",
       "\n",
       ".xr-dim-list li:not(:last-child):after {\n",
       "  content: \",\";\n",
       "  padding-right: 5px;\n",
       "}\n",
       "\n",
       ".xr-has-index {\n",
       "  font-weight: bold;\n",
       "}\n",
       "\n",
       ".xr-var-list,\n",
       ".xr-var-item {\n",
       "  display: contents;\n",
       "}\n",
       "\n",
       ".xr-var-item > div,\n",
       ".xr-var-item label,\n",
       ".xr-var-item > .xr-var-name span {\n",
       "  background-color: var(--xr-background-color-row-even);\n",
       "  margin-bottom: 0;\n",
       "}\n",
       "\n",
       ".xr-var-item > .xr-var-name:hover span {\n",
       "  padding-right: 5px;\n",
       "}\n",
       "\n",
       ".xr-var-list > li:nth-child(odd) > div,\n",
       ".xr-var-list > li:nth-child(odd) > label,\n",
       ".xr-var-list > li:nth-child(odd) > .xr-var-name span {\n",
       "  background-color: var(--xr-background-color-row-odd);\n",
       "}\n",
       "\n",
       ".xr-var-name {\n",
       "  grid-column: 1;\n",
       "}\n",
       "\n",
       ".xr-var-dims {\n",
       "  grid-column: 2;\n",
       "}\n",
       "\n",
       ".xr-var-dtype {\n",
       "  grid-column: 3;\n",
       "  text-align: right;\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-var-preview {\n",
       "  grid-column: 4;\n",
       "}\n",
       "\n",
       ".xr-index-preview {\n",
       "  grid-column: 2 / 5;\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-var-name,\n",
       ".xr-var-dims,\n",
       ".xr-var-dtype,\n",
       ".xr-preview,\n",
       ".xr-attrs dt {\n",
       "  white-space: nowrap;\n",
       "  overflow: hidden;\n",
       "  text-overflow: ellipsis;\n",
       "  padding-right: 10px;\n",
       "}\n",
       "\n",
       ".xr-var-name:hover,\n",
       ".xr-var-dims:hover,\n",
       ".xr-var-dtype:hover,\n",
       ".xr-attrs dt:hover {\n",
       "  overflow: visible;\n",
       "  width: auto;\n",
       "  z-index: 1;\n",
       "}\n",
       "\n",
       ".xr-var-attrs,\n",
       ".xr-var-data,\n",
       ".xr-index-data {\n",
       "  display: none;\n",
       "  background-color: var(--xr-background-color) !important;\n",
       "  padding-bottom: 5px !important;\n",
       "}\n",
       "\n",
       ".xr-var-attrs-in:checked ~ .xr-var-attrs,\n",
       ".xr-var-data-in:checked ~ .xr-var-data,\n",
       ".xr-index-data-in:checked ~ .xr-index-data {\n",
       "  display: block;\n",
       "}\n",
       "\n",
       ".xr-var-data > table {\n",
       "  float: right;\n",
       "}\n",
       "\n",
       ".xr-var-name span,\n",
       ".xr-var-data,\n",
       ".xr-index-name div,\n",
       ".xr-index-data,\n",
       ".xr-attrs {\n",
       "  padding-left: 25px !important;\n",
       "}\n",
       "\n",
       ".xr-attrs,\n",
       ".xr-var-attrs,\n",
       ".xr-var-data,\n",
       ".xr-index-data {\n",
       "  grid-column: 1 / -1;\n",
       "}\n",
       "\n",
       "dl.xr-attrs {\n",
       "  padding: 0;\n",
       "  margin: 0;\n",
       "  display: grid;\n",
       "  grid-template-columns: 125px auto;\n",
       "}\n",
       "\n",
       ".xr-attrs dt,\n",
       ".xr-attrs dd {\n",
       "  padding: 0;\n",
       "  margin: 0;\n",
       "  float: left;\n",
       "  padding-right: 10px;\n",
       "  width: auto;\n",
       "}\n",
       "\n",
       ".xr-attrs dt {\n",
       "  font-weight: normal;\n",
       "  grid-column: 1;\n",
       "}\n",
       "\n",
       ".xr-attrs dt:hover span {\n",
       "  display: inline-block;\n",
       "  background: var(--xr-background-color);\n",
       "  padding-right: 10px;\n",
       "}\n",
       "\n",
       ".xr-attrs dd {\n",
       "  grid-column: 2;\n",
       "  white-space: pre-wrap;\n",
       "  word-break: break-all;\n",
       "}\n",
       "\n",
       ".xr-icon-database,\n",
       ".xr-icon-file-text2,\n",
       ".xr-no-icon {\n",
       "  display: inline-block;\n",
       "  vertical-align: middle;\n",
       "  width: 1em;\n",
       "  height: 1.5em !important;\n",
       "  stroke-width: 0;\n",
       "  stroke: currentColor;\n",
       "  fill: currentColor;\n",
       "}\n",
       "</style><pre class='xr-text-repr-fallback'>&lt;xarray.DataArray &#x27;variant_n_hom_alt&#x27; (variants: 20)&gt; Size: 160B\n",
       "array([0, 0, 0, 2, 5, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1])\n",
       "Dimensions without coordinates: variants</pre><div class='xr-wrap' style='display:none'><div class='xr-header'><div class='xr-obj-type'>xarray.DataArray</div><div class='xr-array-name'>'variant_n_hom_alt'</div><ul class='xr-dim-list'><li><span>variants</span>: 20</li></ul></div><ul class='xr-sections'><li class='xr-section-item'><div class='xr-array-wrap'><input id='section-9f2a5831-8fa7-41e1-8766-e867d618017f' class='xr-array-in' type='checkbox' checked><label for='section-9f2a5831-8fa7-41e1-8766-e867d618017f' title='Show/hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-array-preview xr-preview'><span>0 0 0 2 5 2 0 0 0 0 0 0 0 0 0 1 0 0 0 1</span></div><div class='xr-array-data'><pre>array([0, 0, 0, 2, 5, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1])</pre></div></div></li><li class='xr-section-item'><input id='section-df89cf6a-5407-41de-8acd-d6342f3d3a53' class='xr-section-summary-in' type='checkbox' disabled ><label for='section-df89cf6a-5407-41de-8acd-d6342f3d3a53' class='xr-section-summary'  title='Expand/collapse section'>Coordinates: <span>(0)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><ul class='xr-var-list'></ul></div></li><li class='xr-section-item'><input id='section-ada288f0-14db-4c7d-b5d8-0be2055d8c1a' class='xr-section-summary-in' type='checkbox' disabled ><label for='section-ada288f0-14db-4c7d-b5d8-0be2055d8c1a' class='xr-section-summary'  title='Expand/collapse section'>Indexes: <span>(0)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><ul class='xr-var-list'></ul></div></li><li class='xr-section-item'><input id='section-db93a92a-711b-4879-afe4-81e24ec7573e' class='xr-section-summary-in' type='checkbox' disabled ><label for='section-db93a92a-711b-4879-afe4-81e24ec7573e' class='xr-section-summary'  title='Expand/collapse section'>Attributes: <span>(0)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><dl class='xr-attrs'></dl></div></li></ul></div></div>"
      ],
      "text/plain": [
       "<xarray.DataArray 'variant_n_hom_alt' (variants: 20)> Size: 160B\n",
       "array([0, 0, 0, 2, 5, 2, 0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 0, 0, 1])\n",
       "Dimensions without coordinates: variants"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ds_var_stats.variant_n_hom_alt[:20].compute()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "6c7bda37",
   "metadata": {},
   "outputs": [],
   "source": [
    "from sgkit.cohorts import _cohorts_to_array\n",
    "from sgkit.stats.utils import assert_array_shape\n",
    "from sgkit.utils import (\n",
    "    conditional_merge_datasets,\n",
    "    create_dataset,\n",
    "    define_variable_if_absent,\n",
    ")\n",
    "from sgkit.window import has_windows, window_statistic\n",
    "\n",
    "from sgkit import variables\n",
    "from sgkit.stats.aggregation import (\n",
    "    count_cohort_alleles,\n",
    "    count_variant_alleles,\n",
    "    individual_heterozygosity,\n",
    ")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "18f7aeaf",
   "metadata": {},
   "outputs": [],
   "source": [
    "def alt_hom(ds):\n",
    "    alt_hom = ((ds.call_genotype[:,:,0] == ds.call_genotype[:,:,1]) & (ds.call_genotype[:,:,0] >= 1)).sum(axis=1)\n",
    "    div = window_statistic(\n",
    "            alt_hom,\n",
    "            np.sum,\n",
    "            ds.window_start.values,\n",
    "            ds.window_stop.values,\n",
    "            dtype=np.int64,\n",
    "            axis=0,\n",
    "    )\n",
    "    return div"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "db5aeb2a",
   "metadata": {},
   "outputs": [],
   "source": [
    "ds[\"sample_cohort\"] = ds[\"samples\"]\n",
    "ds = ds.sel(contigs=[ds.variant_contig[0].values])\n",
    "ds = sg.window_by_position(ds, size=100000)\n",
    "alt_hom(ds)[:10].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "1c530a44",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><svg style=\"position: absolute; width: 0; height: 0; overflow: hidden\">\n",
       "<defs>\n",
       "<symbol id=\"icon-database\" viewBox=\"0 0 32 32\">\n",
       "<path d=\"M16 0c-8.837 0-16 2.239-16 5v4c0 2.761 7.163 5 16 5s16-2.239 16-5v-4c0-2.761-7.163-5-16-5z\"></path>\n",
       "<path d=\"M16 17c-8.837 0-16-2.239-16-5v6c0 2.761 7.163 5 16 5s16-2.239 16-5v-6c0 2.761-7.163 5-16 5z\"></path>\n",
       "<path d=\"M16 26c-8.837 0-16-2.239-16-5v6c0 2.761 7.163 5 16 5s16-2.239 16-5v-6c0 2.761-7.163 5-16 5z\"></path>\n",
       "</symbol>\n",
       "<symbol id=\"icon-file-text2\" viewBox=\"0 0 32 32\">\n",
       "<path d=\"M28.681 7.159c-0.694-0.947-1.662-2.053-2.724-3.116s-2.169-2.030-3.116-2.724c-1.612-1.182-2.393-1.319-2.841-1.319h-15.5c-1.378 0-2.5 1.121-2.5 2.5v27c0 1.378 1.122 2.5 2.5 2.5h23c1.378 0 2.5-1.122 2.5-2.5v-19.5c0-0.448-0.137-1.23-1.319-2.841zM24.543 5.457c0.959 0.959 1.712 1.825 2.268 2.543h-4.811v-4.811c0.718 0.556 1.584 1.309 2.543 2.268zM28 29.5c0 0.271-0.229 0.5-0.5 0.5h-23c-0.271 0-0.5-0.229-0.5-0.5v-27c0-0.271 0.229-0.5 0.5-0.5 0 0 15.499-0 15.5 0v7c0 0.552 0.448 1 1 1h7v19.5z\"></path>\n",
       "<path d=\"M23 26h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z\"></path>\n",
       "<path d=\"M23 22h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z\"></path>\n",
       "<path d=\"M23 18h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z\"></path>\n",
       "</symbol>\n",
       "</defs>\n",
       "</svg>\n",
       "<style>/* CSS stylesheet for displaying xarray objects in jupyterlab.\n",
       " *\n",
       " */\n",
       "\n",
       ":root {\n",
       "  --xr-font-color0: var(--jp-content-font-color0, rgba(0, 0, 0, 1));\n",
       "  --xr-font-color2: var(--jp-content-font-color2, rgba(0, 0, 0, 0.54));\n",
       "  --xr-font-color3: var(--jp-content-font-color3, rgba(0, 0, 0, 0.38));\n",
       "  --xr-border-color: var(--jp-border-color2, #e0e0e0);\n",
       "  --xr-disabled-color: var(--jp-layout-color3, #bdbdbd);\n",
       "  --xr-background-color: var(--jp-layout-color0, white);\n",
       "  --xr-background-color-row-even: var(--jp-layout-color1, white);\n",
       "  --xr-background-color-row-odd: var(--jp-layout-color2, #eeeeee);\n",
       "}\n",
       "\n",
       "html[theme=\"dark\"],\n",
       "html[data-theme=\"dark\"],\n",
       "body[data-theme=\"dark\"],\n",
       "body.vscode-dark {\n",
       "  --xr-font-color0: rgba(255, 255, 255, 1);\n",
       "  --xr-font-color2: rgba(255, 255, 255, 0.54);\n",
       "  --xr-font-color3: rgba(255, 255, 255, 0.38);\n",
       "  --xr-border-color: #1f1f1f;\n",
       "  --xr-disabled-color: #515151;\n",
       "  --xr-background-color: #111111;\n",
       "  --xr-background-color-row-even: #111111;\n",
       "  --xr-background-color-row-odd: #313131;\n",
       "}\n",
       "\n",
       ".xr-wrap {\n",
       "  display: block !important;\n",
       "  min-width: 300px;\n",
       "  max-width: 700px;\n",
       "}\n",
       "\n",
       ".xr-text-repr-fallback {\n",
       "  /* fallback to plain text repr when CSS is not injected (untrusted notebook) */\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-header {\n",
       "  padding-top: 6px;\n",
       "  padding-bottom: 6px;\n",
       "  margin-bottom: 4px;\n",
       "  border-bottom: solid 1px var(--xr-border-color);\n",
       "}\n",
       "\n",
       ".xr-header > div,\n",
       ".xr-header > ul {\n",
       "  display: inline;\n",
       "  margin-top: 0;\n",
       "  margin-bottom: 0;\n",
       "}\n",
       "\n",
       ".xr-obj-type,\n",
       ".xr-array-name {\n",
       "  margin-left: 2px;\n",
       "  margin-right: 10px;\n",
       "}\n",
       "\n",
       ".xr-obj-type {\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-sections {\n",
       "  padding-left: 0 !important;\n",
       "  display: grid;\n",
       "  grid-template-columns: 150px auto auto 1fr 0 20px 0 20px;\n",
       "}\n",
       "\n",
       ".xr-section-item {\n",
       "  display: contents;\n",
       "}\n",
       "\n",
       ".xr-section-item input {\n",
       "  display: inline-block;\n",
       "  opacity: 0;\n",
       "  height: 0;\n",
       "}\n",
       "\n",
       ".xr-section-item input + label {\n",
       "  color: var(--xr-disabled-color);\n",
       "}\n",
       "\n",
       ".xr-section-item input:enabled + label {\n",
       "  cursor: pointer;\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-section-item input:focus + label {\n",
       "  border: 2px solid var(--xr-font-color0);\n",
       "}\n",
       "\n",
       ".xr-section-item input:enabled + label:hover {\n",
       "  color: var(--xr-font-color0);\n",
       "}\n",
       "\n",
       ".xr-section-summary {\n",
       "  grid-column: 1;\n",
       "  color: var(--xr-font-color2);\n",
       "  font-weight: 500;\n",
       "}\n",
       "\n",
       ".xr-section-summary > span {\n",
       "  display: inline-block;\n",
       "  padding-left: 0.5em;\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:disabled + label {\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-section-summary-in + label:before {\n",
       "  display: inline-block;\n",
       "  content: \"►\";\n",
       "  font-size: 11px;\n",
       "  width: 15px;\n",
       "  text-align: center;\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:disabled + label:before {\n",
       "  color: var(--xr-disabled-color);\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:checked + label:before {\n",
       "  content: \"▼\";\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:checked + label > span {\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-section-summary,\n",
       ".xr-section-inline-details {\n",
       "  padding-top: 4px;\n",
       "  padding-bottom: 4px;\n",
       "}\n",
       "\n",
       ".xr-section-inline-details {\n",
       "  grid-column: 2 / -1;\n",
       "}\n",
       "\n",
       ".xr-section-details {\n",
       "  display: none;\n",
       "  grid-column: 1 / -1;\n",
       "  margin-bottom: 5px;\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:checked ~ .xr-section-details {\n",
       "  display: contents;\n",
       "}\n",
       "\n",
       ".xr-array-wrap {\n",
       "  grid-column: 1 / -1;\n",
       "  display: grid;\n",
       "  grid-template-columns: 20px auto;\n",
       "}\n",
       "\n",
       ".xr-array-wrap > label {\n",
       "  grid-column: 1;\n",
       "  vertical-align: top;\n",
       "}\n",
       "\n",
       ".xr-preview {\n",
       "  color: var(--xr-font-color3);\n",
       "}\n",
       "\n",
       ".xr-array-preview,\n",
       ".xr-array-data {\n",
       "  padding: 0 5px !important;\n",
       "  grid-column: 2;\n",
       "}\n",
       "\n",
       ".xr-array-data,\n",
       ".xr-array-in:checked ~ .xr-array-preview {\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-array-in:checked ~ .xr-array-data,\n",
       ".xr-array-preview {\n",
       "  display: inline-block;\n",
       "}\n",
       "\n",
       ".xr-dim-list {\n",
       "  display: inline-block !important;\n",
       "  list-style: none;\n",
       "  padding: 0 !important;\n",
       "  margin: 0;\n",
       "}\n",
       "\n",
       ".xr-dim-list li {\n",
       "  display: inline-block;\n",
       "  padding: 0;\n",
       "  margin: 0;\n",
       "}\n",
       "\n",
       ".xr-dim-list:before {\n",
       "  content: \"(\";\n",
       "}\n",
       "\n",
       ".xr-dim-list:after {\n",
       "  content: \")\";\n",
       "}\n",
       "\n",
       ".xr-dim-list li:not(:last-child):after {\n",
       "  content: \",\";\n",
       "  padding-right: 5px;\n",
       "}\n",
       "\n",
       ".xr-has-index {\n",
       "  font-weight: bold;\n",
       "}\n",
       "\n",
       ".xr-var-list,\n",
       ".xr-var-item {\n",
       "  display: contents;\n",
       "}\n",
       "\n",
       ".xr-var-item > div,\n",
       ".xr-var-item label,\n",
       ".xr-var-item > .xr-var-name span {\n",
       "  background-color: var(--xr-background-color-row-even);\n",
       "  margin-bottom: 0;\n",
       "}\n",
       "\n",
       ".xr-var-item > .xr-var-name:hover span {\n",
       "  padding-right: 5px;\n",
       "}\n",
       "\n",
       ".xr-var-list > li:nth-child(odd) > div,\n",
       ".xr-var-list > li:nth-child(odd) > label,\n",
       ".xr-var-list > li:nth-child(odd) > .xr-var-name span {\n",
       "  background-color: var(--xr-background-color-row-odd);\n",
       "}\n",
       "\n",
       ".xr-var-name {\n",
       "  grid-column: 1;\n",
       "}\n",
       "\n",
       ".xr-var-dims {\n",
       "  grid-column: 2;\n",
       "}\n",
       "\n",
       ".xr-var-dtype {\n",
       "  grid-column: 3;\n",
       "  text-align: right;\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-var-preview {\n",
       "  grid-column: 4;\n",
       "}\n",
       "\n",
       ".xr-index-preview {\n",
       "  grid-column: 2 / 5;\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-var-name,\n",
       ".xr-var-dims,\n",
       ".xr-var-dtype,\n",
       ".xr-preview,\n",
       ".xr-attrs dt {\n",
       "  white-space: nowrap;\n",
       "  overflow: hidden;\n",
       "  text-overflow: ellipsis;\n",
       "  padding-right: 10px;\n",
       "}\n",
       "\n",
       ".xr-var-name:hover,\n",
       ".xr-var-dims:hover,\n",
       ".xr-var-dtype:hover,\n",
       ".xr-attrs dt:hover {\n",
       "  overflow: visible;\n",
       "  width: auto;\n",
       "  z-index: 1;\n",
       "}\n",
       "\n",
       ".xr-var-attrs,\n",
       ".xr-var-data,\n",
       ".xr-index-data {\n",
       "  display: none;\n",
       "  background-color: var(--xr-background-color) !important;\n",
       "  padding-bottom: 5px !important;\n",
       "}\n",
       "\n",
       ".xr-var-attrs-in:checked ~ .xr-var-attrs,\n",
       ".xr-var-data-in:checked ~ .xr-var-data,\n",
       ".xr-index-data-in:checked ~ .xr-index-data {\n",
       "  display: block;\n",
       "}\n",
       "\n",
       ".xr-var-data > table {\n",
       "  float: right;\n",
       "}\n",
       "\n",
       ".xr-var-name span,\n",
       ".xr-var-data,\n",
       ".xr-index-name div,\n",
       ".xr-index-data,\n",
       ".xr-attrs {\n",
       "  padding-left: 25px !important;\n",
       "}\n",
       "\n",
       ".xr-attrs,\n",
       ".xr-var-attrs,\n",
       ".xr-var-data,\n",
       ".xr-index-data {\n",
       "  grid-column: 1 / -1;\n",
       "}\n",
       "\n",
       "dl.xr-attrs {\n",
       "  padding: 0;\n",
       "  margin: 0;\n",
       "  display: grid;\n",
       "  grid-template-columns: 125px auto;\n",
       "}\n",
       "\n",
       ".xr-attrs dt,\n",
       ".xr-attrs dd {\n",
       "  padding: 0;\n",
       "  margin: 0;\n",
       "  float: left;\n",
       "  padding-right: 10px;\n",
       "  width: auto;\n",
       "}\n",
       "\n",
       ".xr-attrs dt {\n",
       "  font-weight: normal;\n",
       "  grid-column: 1;\n",
       "}\n",
       "\n",
       ".xr-attrs dt:hover span {\n",
       "  display: inline-block;\n",
       "  background: var(--xr-background-color);\n",
       "  padding-right: 10px;\n",
       "}\n",
       "\n",
       ".xr-attrs dd {\n",
       "  grid-column: 2;\n",
       "  white-space: pre-wrap;\n",
       "  word-break: break-all;\n",
       "}\n",
       "\n",
       ".xr-icon-database,\n",
       ".xr-icon-file-text2,\n",
       ".xr-no-icon {\n",
       "  display: inline-block;\n",
       "  vertical-align: middle;\n",
       "  width: 1em;\n",
       "  height: 1.5em !important;\n",
       "  stroke-width: 0;\n",
       "  stroke: currentColor;\n",
       "  fill: currentColor;\n",
       "}\n",
       "</style><pre class='xr-text-repr-fallback'>&lt;xarray.Dataset&gt; Size: 150MB\n",
       "Dimensions:               (variants: 2023479, samples: 7, ploidy: 2,\n",
       "                           contigs: 361, filters: 2, region_index_values: 2024,\n",
       "                           region_index_fields: 6, alleles: 2)\n",
       "Dimensions without coordinates: variants, samples, ploidy, contigs, filters,\n",
       "                                region_index_values, region_index_fields,\n",
       "                                alleles\n",
       "Data variables: (12/16)\n",
       "    call_genotype         (variants, samples, ploidy) int8 28MB dask.array&lt;chunksize=(1000, 7, 2), meta=np.ndarray&gt;\n",
       "    call_genotype_mask    (variants, samples, ploidy) bool 28MB dask.array&lt;chunksize=(1000, 7, 2), meta=np.ndarray&gt;\n",
       "    call_genotype_phased  (variants, samples) bool 14MB dask.array&lt;chunksize=(1000, 7), meta=np.ndarray&gt;\n",
       "    contig_id             (contigs) object 3kB dask.array&lt;chunksize=(361,), meta=np.ndarray&gt;\n",
       "    contig_length         (contigs) float64 3kB dask.array&lt;chunksize=(361,), meta=np.ndarray&gt;\n",
       "    filter_id             (filters) object 16B dask.array&lt;chunksize=(2,), meta=np.ndarray&gt;\n",
       "    ...                    ...\n",
       "    variant_filter        (variants, filters) bool 4MB dask.array&lt;chunksize=(1000, 2), meta=np.ndarray&gt;\n",
       "    variant_id            (variants) object 16MB dask.array&lt;chunksize=(1000,), meta=np.ndarray&gt;\n",
       "    variant_id_mask       (variants) bool 2MB dask.array&lt;chunksize=(1000,), meta=np.ndarray&gt;\n",
       "    variant_length        (variants) int16 4MB dask.array&lt;chunksize=(1000,), meta=np.ndarray&gt;\n",
       "    variant_position      (variants) int32 8MB dask.array&lt;chunksize=(1000,), meta=np.ndarray&gt;\n",
       "    variant_quality       (variants) float32 8MB dask.array&lt;chunksize=(1000,), meta=np.ndarray&gt;\n",
       "Attributes:\n",
       "    source:            bio2zarr-0.1.5\n",
       "    vcf_header:        ##fileformat=VCFv4.2\\n##FILTER=&lt;ID=PASS,Description=&quot;A...\n",
       "    vcf_zarr_version:  0.2</pre><div class='xr-wrap' style='display:none'><div class='xr-header'><div class='xr-obj-type'>xarray.Dataset</div></div><ul class='xr-sections'><li class='xr-section-item'><input id='section-e45f03de-5cea-455f-a7d1-57eb04906f5f' class='xr-section-summary-in' type='checkbox' disabled ><label for='section-e45f03de-5cea-455f-a7d1-57eb04906f5f' class='xr-section-summary'  title='Expand/collapse section'>Dimensions:</label><div class='xr-section-inline-details'><ul class='xr-dim-list'><li><span>variants</span>: 2023479</li><li><span>samples</span>: 7</li><li><span>ploidy</span>: 2</li><li><span>contigs</span>: 361</li><li><span>filters</span>: 2</li><li><span>region_index_values</span>: 2024</li><li><span>region_index_fields</span>: 6</li><li><span>alleles</span>: 2</li></ul></div><div class='xr-section-details'></div></li><li class='xr-section-item'><input id='section-721703b1-9954-468a-82d3-f16855796040' class='xr-section-summary-in' type='checkbox' disabled ><label for='section-721703b1-9954-468a-82d3-f16855796040' class='xr-section-summary'  title='Expand/collapse section'>Coordinates: <span>(0)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><ul class='xr-var-list'></ul></div></li><li class='xr-section-item'><input id='section-0b66600c-d2f7-4387-8b26-7f91c2854d05' class='xr-section-summary-in' type='checkbox'  ><label for='section-0b66600c-d2f7-4387-8b26-7f91c2854d05' class='xr-section-summary' >Data variables: <span>(16)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><ul class='xr-var-list'><li class='xr-var-item'><div class='xr-var-name'><span>call_genotype</span></div><div class='xr-var-dims'>(variants, samples, ploidy)</div><div class='xr-var-dtype'>int8</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(1000, 7, 2), meta=np.ndarray&gt;</div><input id='attrs-08bbb89a-21d7-41e1-ad88-f5be9eab7b1c' class='xr-var-attrs-in' type='checkbox' ><label for='attrs-08bbb89a-21d7-41e1-ad88-f5be9eab7b1c' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-98bacb62-6911-4af6-a6af-f54da99fcfff' class='xr-var-data-in' type='checkbox'><label for='data-98bacb62-6911-4af6-a6af-f54da99fcfff' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'><dt><span>description :</span></dt><dd></dd></dl></div><div class='xr-var-data'><table>\n",
       "    <tr>\n",
       "        <td>\n",
       "            <table style=\"border-collapse: collapse;\">\n",
       "                <thead>\n",
       "                    <tr>\n",
       "                        <td> </td>\n",
       "                        <th> Array </th>\n",
       "                        <th> Chunk </th>\n",
       "                    </tr>\n",
       "                </thead>\n",
       "                <tbody>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Bytes </th>\n",
       "                        <td> 27.02 MiB </td>\n",
       "                        <td> 13.67 kiB </td>\n",
       "                    </tr>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Shape </th>\n",
       "                        <td> (2023479, 7, 2) </td>\n",
       "                        <td> (1000, 7, 2) </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Dask graph </th>\n",
       "                        <td colspan=\"2\"> 2024 chunks in 2 graph layers </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Data type </th>\n",
       "                        <td colspan=\"2\"> int8 numpy.ndarray </td>\n",
       "                    </tr>\n",
       "                </tbody>\n",
       "            </table>\n",
       "        </td>\n",
       "        <td>\n",
       "        <svg width=\"156\" height=\"146\" style=\"stroke:rgb(0,0,0);stroke-width:1\" >\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"10\" y1=\"0\" x2=\"80\" y2=\"70\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"10\" y1=\"25\" x2=\"80\" y2=\"96\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"10\" y1=\"0\" x2=\"10\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"13\" y1=\"3\" x2=\"13\" y2=\"29\" />\n",
       "  <line x1=\"17\" y1=\"7\" x2=\"17\" y2=\"32\" />\n",
       "  <line x1=\"21\" y1=\"11\" x2=\"21\" y2=\"36\" />\n",
       "  <line x1=\"24\" y1=\"14\" x2=\"24\" y2=\"40\" />\n",
       "  <line x1=\"28\" y1=\"18\" x2=\"28\" y2=\"43\" />\n",
       "  <line x1=\"32\" y1=\"22\" x2=\"32\" y2=\"47\" />\n",
       "  <line x1=\"35\" y1=\"25\" x2=\"35\" y2=\"51\" />\n",
       "  <line x1=\"39\" y1=\"29\" x2=\"39\" y2=\"55\" />\n",
       "  <line x1=\"43\" y1=\"33\" x2=\"43\" y2=\"58\" />\n",
       "  <line x1=\"47\" y1=\"37\" x2=\"47\" y2=\"62\" />\n",
       "  <line x1=\"50\" y1=\"40\" x2=\"50\" y2=\"66\" />\n",
       "  <line x1=\"54\" y1=\"44\" x2=\"54\" y2=\"69\" />\n",
       "  <line x1=\"58\" y1=\"48\" x2=\"58\" y2=\"73\" />\n",
       "  <line x1=\"62\" y1=\"52\" x2=\"62\" y2=\"77\" />\n",
       "  <line x1=\"65\" y1=\"55\" x2=\"65\" y2=\"81\" />\n",
       "  <line x1=\"69\" y1=\"59\" x2=\"69\" y2=\"84\" />\n",
       "  <line x1=\"73\" y1=\"63\" x2=\"73\" y2=\"88\" />\n",
       "  <line x1=\"76\" y1=\"66\" x2=\"76\" y2=\"92\" />\n",
       "  <line x1=\"80\" y1=\"70\" x2=\"80\" y2=\"96\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"10.0,0.0 80.58823529411765,70.58823529411765 80.58823529411765,96.00085180870013 10.0,25.412616514582485\" style=\"fill:#8B4903A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"10\" y1=\"0\" x2=\"35\" y2=\"0\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"13\" y1=\"3\" x2=\"39\" y2=\"3\" />\n",
       "  <line x1=\"17\" y1=\"7\" x2=\"42\" y2=\"7\" />\n",
       "  <line x1=\"21\" y1=\"11\" x2=\"46\" y2=\"11\" />\n",
       "  <line x1=\"24\" y1=\"14\" x2=\"50\" y2=\"14\" />\n",
       "  <line x1=\"28\" y1=\"18\" x2=\"53\" y2=\"18\" />\n",
       "  <line x1=\"32\" y1=\"22\" x2=\"57\" y2=\"22\" />\n",
       "  <line x1=\"35\" y1=\"25\" x2=\"61\" y2=\"25\" />\n",
       "  <line x1=\"39\" y1=\"29\" x2=\"65\" y2=\"29\" />\n",
       "  <line x1=\"43\" y1=\"33\" x2=\"68\" y2=\"33\" />\n",
       "  <line x1=\"47\" y1=\"37\" x2=\"72\" y2=\"37\" />\n",
       "  <line x1=\"50\" y1=\"40\" x2=\"76\" y2=\"40\" />\n",
       "  <line x1=\"54\" y1=\"44\" x2=\"79\" y2=\"44\" />\n",
       "  <line x1=\"58\" y1=\"48\" x2=\"83\" y2=\"48\" />\n",
       "  <line x1=\"62\" y1=\"52\" x2=\"87\" y2=\"52\" />\n",
       "  <line x1=\"65\" y1=\"55\" x2=\"91\" y2=\"55\" />\n",
       "  <line x1=\"69\" y1=\"59\" x2=\"94\" y2=\"59\" />\n",
       "  <line x1=\"73\" y1=\"63\" x2=\"98\" y2=\"63\" />\n",
       "  <line x1=\"76\" y1=\"66\" x2=\"102\" y2=\"66\" />\n",
       "  <line x1=\"80\" y1=\"70\" x2=\"106\" y2=\"70\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"10\" y1=\"0\" x2=\"80\" y2=\"70\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"35\" y1=\"0\" x2=\"106\" y2=\"70\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"10.0,0.0 35.41261651458248,0.0 106.00085180870013,70.58823529411765 80.58823529411765,70.58823529411765\" style=\"fill:#8B4903A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"80\" y1=\"70\" x2=\"106\" y2=\"70\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"80\" y1=\"96\" x2=\"106\" y2=\"96\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"80\" y1=\"70\" x2=\"80\" y2=\"96\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"106\" y1=\"70\" x2=\"106\" y2=\"96\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"80.58823529411765,70.58823529411765 106.00085180870013,70.58823529411765 106.00085180870013,96.00085180870013 80.58823529411765,96.00085180870013\" style=\"fill:#ECB172A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Text -->\n",
       "  <text x=\"93.294544\" y=\"116.000852\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" >2</text>\n",
       "  <text x=\"126.000852\" y=\"83.294544\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(0,126.000852,83.294544)\">7</text>\n",
       "  <text x=\"35.294118\" y=\"80.706734\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(45,35.294118,80.706734)\">2023479</text>\n",
       "</svg>\n",
       "        </td>\n",
       "    </tr>\n",
       "</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>call_genotype_mask</span></div><div class='xr-var-dims'>(variants, samples, ploidy)</div><div class='xr-var-dtype'>bool</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(1000, 7, 2), meta=np.ndarray&gt;</div><input id='attrs-b76194eb-0668-4d0c-8670-d66ff2127ee2' class='xr-var-attrs-in' type='checkbox' ><label for='attrs-b76194eb-0668-4d0c-8670-d66ff2127ee2' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-a4416c5b-89e9-49e1-8011-9afc4310c393' class='xr-var-data-in' type='checkbox'><label for='data-a4416c5b-89e9-49e1-8011-9afc4310c393' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'><dt><span>description :</span></dt><dd></dd></dl></div><div class='xr-var-data'><table>\n",
       "    <tr>\n",
       "        <td>\n",
       "            <table style=\"border-collapse: collapse;\">\n",
       "                <thead>\n",
       "                    <tr>\n",
       "                        <td> </td>\n",
       "                        <th> Array </th>\n",
       "                        <th> Chunk </th>\n",
       "                    </tr>\n",
       "                </thead>\n",
       "                <tbody>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Bytes </th>\n",
       "                        <td> 27.02 MiB </td>\n",
       "                        <td> 13.67 kiB </td>\n",
       "                    </tr>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Shape </th>\n",
       "                        <td> (2023479, 7, 2) </td>\n",
       "                        <td> (1000, 7, 2) </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Dask graph </th>\n",
       "                        <td colspan=\"2\"> 2024 chunks in 2 graph layers </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Data type </th>\n",
       "                        <td colspan=\"2\"> bool numpy.ndarray </td>\n",
       "                    </tr>\n",
       "                </tbody>\n",
       "            </table>\n",
       "        </td>\n",
       "        <td>\n",
       "        <svg width=\"156\" height=\"146\" style=\"stroke:rgb(0,0,0);stroke-width:1\" >\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"10\" y1=\"0\" x2=\"80\" y2=\"70\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"10\" y1=\"25\" x2=\"80\" y2=\"96\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"10\" y1=\"0\" x2=\"10\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"13\" y1=\"3\" x2=\"13\" y2=\"29\" />\n",
       "  <line x1=\"17\" y1=\"7\" x2=\"17\" y2=\"32\" />\n",
       "  <line x1=\"21\" y1=\"11\" x2=\"21\" y2=\"36\" />\n",
       "  <line x1=\"24\" y1=\"14\" x2=\"24\" y2=\"40\" />\n",
       "  <line x1=\"28\" y1=\"18\" x2=\"28\" y2=\"43\" />\n",
       "  <line x1=\"32\" y1=\"22\" x2=\"32\" y2=\"47\" />\n",
       "  <line x1=\"35\" y1=\"25\" x2=\"35\" y2=\"51\" />\n",
       "  <line x1=\"39\" y1=\"29\" x2=\"39\" y2=\"55\" />\n",
       "  <line x1=\"43\" y1=\"33\" x2=\"43\" y2=\"58\" />\n",
       "  <line x1=\"47\" y1=\"37\" x2=\"47\" y2=\"62\" />\n",
       "  <line x1=\"50\" y1=\"40\" x2=\"50\" y2=\"66\" />\n",
       "  <line x1=\"54\" y1=\"44\" x2=\"54\" y2=\"69\" />\n",
       "  <line x1=\"58\" y1=\"48\" x2=\"58\" y2=\"73\" />\n",
       "  <line x1=\"62\" y1=\"52\" x2=\"62\" y2=\"77\" />\n",
       "  <line x1=\"65\" y1=\"55\" x2=\"65\" y2=\"81\" />\n",
       "  <line x1=\"69\" y1=\"59\" x2=\"69\" y2=\"84\" />\n",
       "  <line x1=\"73\" y1=\"63\" x2=\"73\" y2=\"88\" />\n",
       "  <line x1=\"76\" y1=\"66\" x2=\"76\" y2=\"92\" />\n",
       "  <line x1=\"80\" y1=\"70\" x2=\"80\" y2=\"96\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"10.0,0.0 80.58823529411765,70.58823529411765 80.58823529411765,96.00085180870013 10.0,25.412616514582485\" style=\"fill:#8B4903A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"10\" y1=\"0\" x2=\"35\" y2=\"0\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"13\" y1=\"3\" x2=\"39\" y2=\"3\" />\n",
       "  <line x1=\"17\" y1=\"7\" x2=\"42\" y2=\"7\" />\n",
       "  <line x1=\"21\" y1=\"11\" x2=\"46\" y2=\"11\" />\n",
       "  <line x1=\"24\" y1=\"14\" x2=\"50\" y2=\"14\" />\n",
       "  <line x1=\"28\" y1=\"18\" x2=\"53\" y2=\"18\" />\n",
       "  <line x1=\"32\" y1=\"22\" x2=\"57\" y2=\"22\" />\n",
       "  <line x1=\"35\" y1=\"25\" x2=\"61\" y2=\"25\" />\n",
       "  <line x1=\"39\" y1=\"29\" x2=\"65\" y2=\"29\" />\n",
       "  <line x1=\"43\" y1=\"33\" x2=\"68\" y2=\"33\" />\n",
       "  <line x1=\"47\" y1=\"37\" x2=\"72\" y2=\"37\" />\n",
       "  <line x1=\"50\" y1=\"40\" x2=\"76\" y2=\"40\" />\n",
       "  <line x1=\"54\" y1=\"44\" x2=\"79\" y2=\"44\" />\n",
       "  <line x1=\"58\" y1=\"48\" x2=\"83\" y2=\"48\" />\n",
       "  <line x1=\"62\" y1=\"52\" x2=\"87\" y2=\"52\" />\n",
       "  <line x1=\"65\" y1=\"55\" x2=\"91\" y2=\"55\" />\n",
       "  <line x1=\"69\" y1=\"59\" x2=\"94\" y2=\"59\" />\n",
       "  <line x1=\"73\" y1=\"63\" x2=\"98\" y2=\"63\" />\n",
       "  <line x1=\"76\" y1=\"66\" x2=\"102\" y2=\"66\" />\n",
       "  <line x1=\"80\" y1=\"70\" x2=\"106\" y2=\"70\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"10\" y1=\"0\" x2=\"80\" y2=\"70\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"35\" y1=\"0\" x2=\"106\" y2=\"70\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"10.0,0.0 35.41261651458248,0.0 106.00085180870013,70.58823529411765 80.58823529411765,70.58823529411765\" style=\"fill:#8B4903A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"80\" y1=\"70\" x2=\"106\" y2=\"70\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"80\" y1=\"96\" x2=\"106\" y2=\"96\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"80\" y1=\"70\" x2=\"80\" y2=\"96\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"106\" y1=\"70\" x2=\"106\" y2=\"96\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"80.58823529411765,70.58823529411765 106.00085180870013,70.58823529411765 106.00085180870013,96.00085180870013 80.58823529411765,96.00085180870013\" style=\"fill:#ECB172A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Text -->\n",
       "  <text x=\"93.294544\" y=\"116.000852\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" >2</text>\n",
       "  <text x=\"126.000852\" y=\"83.294544\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(0,126.000852,83.294544)\">7</text>\n",
       "  <text x=\"35.294118\" y=\"80.706734\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(45,35.294118,80.706734)\">2023479</text>\n",
       "</svg>\n",
       "        </td>\n",
       "    </tr>\n",
       "</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>call_genotype_phased</span></div><div class='xr-var-dims'>(variants, samples)</div><div class='xr-var-dtype'>bool</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(1000, 7), meta=np.ndarray&gt;</div><input id='attrs-26d639e2-ff11-4248-bac0-9c25a1cabe5b' class='xr-var-attrs-in' type='checkbox' ><label for='attrs-26d639e2-ff11-4248-bac0-9c25a1cabe5b' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-40bffff7-e7af-4f9c-9076-4a0d41bfe8e1' class='xr-var-data-in' type='checkbox'><label for='data-40bffff7-e7af-4f9c-9076-4a0d41bfe8e1' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'><dt><span>description :</span></dt><dd></dd></dl></div><div class='xr-var-data'><table>\n",
       "    <tr>\n",
       "        <td>\n",
       "            <table style=\"border-collapse: collapse;\">\n",
       "                <thead>\n",
       "                    <tr>\n",
       "                        <td> </td>\n",
       "                        <th> Array </th>\n",
       "                        <th> Chunk </th>\n",
       "                    </tr>\n",
       "                </thead>\n",
       "                <tbody>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Bytes </th>\n",
       "                        <td> 13.51 MiB </td>\n",
       "                        <td> 6.84 kiB </td>\n",
       "                    </tr>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Shape </th>\n",
       "                        <td> (2023479, 7) </td>\n",
       "                        <td> (1000, 7) </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Dask graph </th>\n",
       "                        <td colspan=\"2\"> 2024 chunks in 2 graph layers </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Data type </th>\n",
       "                        <td colspan=\"2\"> bool numpy.ndarray </td>\n",
       "                    </tr>\n",
       "                </tbody>\n",
       "            </table>\n",
       "        </td>\n",
       "        <td>\n",
       "        <svg width=\"75\" height=\"170\" style=\"stroke:rgb(0,0,0);stroke-width:1\" >\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"25\" y2=\"0\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"0\" y1=\"6\" x2=\"25\" y2=\"6\" />\n",
       "  <line x1=\"0\" y1=\"12\" x2=\"25\" y2=\"12\" />\n",
       "  <line x1=\"0\" y1=\"18\" x2=\"25\" y2=\"18\" />\n",
       "  <line x1=\"0\" y1=\"25\" x2=\"25\" y2=\"25\" />\n",
       "  <line x1=\"0\" y1=\"31\" x2=\"25\" y2=\"31\" />\n",
       "  <line x1=\"0\" y1=\"37\" x2=\"25\" y2=\"37\" />\n",
       "  <line x1=\"0\" y1=\"44\" x2=\"25\" y2=\"44\" />\n",
       "  <line x1=\"0\" y1=\"50\" x2=\"25\" y2=\"50\" />\n",
       "  <line x1=\"0\" y1=\"56\" x2=\"25\" y2=\"56\" />\n",
       "  <line x1=\"0\" y1=\"63\" x2=\"25\" y2=\"63\" />\n",
       "  <line x1=\"0\" y1=\"69\" x2=\"25\" y2=\"69\" />\n",
       "  <line x1=\"0\" y1=\"75\" x2=\"25\" y2=\"75\" />\n",
       "  <line x1=\"0\" y1=\"82\" x2=\"25\" y2=\"82\" />\n",
       "  <line x1=\"0\" y1=\"88\" x2=\"25\" y2=\"88\" />\n",
       "  <line x1=\"0\" y1=\"94\" x2=\"25\" y2=\"94\" />\n",
       "  <line x1=\"0\" y1=\"101\" x2=\"25\" y2=\"101\" />\n",
       "  <line x1=\"0\" y1=\"107\" x2=\"25\" y2=\"107\" />\n",
       "  <line x1=\"0\" y1=\"113\" x2=\"25\" y2=\"113\" />\n",
       "  <line x1=\"0\" y1=\"120\" x2=\"25\" y2=\"120\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"0\" y2=\"120\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"25\" y1=\"0\" x2=\"25\" y2=\"120\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"0.0,0.0 25.412616514582485,0.0 25.412616514582485,120.0 0.0,120.0\" style=\"fill:#8B4903A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Text -->\n",
       "  <text x=\"12.706308\" y=\"140.000000\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" >7</text>\n",
       "  <text x=\"45.412617\" y=\"60.000000\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(-90,45.412617,60.000000)\">2023479</text>\n",
       "</svg>\n",
       "        </td>\n",
       "    </tr>\n",
       "</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>contig_id</span></div><div class='xr-var-dims'>(contigs)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(361,), meta=np.ndarray&gt;</div><input id='attrs-11b38b40-e930-4d66-b706-c72ad88a63b5' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-11b38b40-e930-4d66-b706-c72ad88a63b5' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-d9757bb0-1a11-4dc7-ba5e-0b5cd9849eae' class='xr-var-data-in' type='checkbox'><label for='data-d9757bb0-1a11-4dc7-ba5e-0b5cd9849eae' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>\n",
       "    <tr>\n",
       "        <td>\n",
       "            <table style=\"border-collapse: collapse;\">\n",
       "                <thead>\n",
       "                    <tr>\n",
       "                        <td> </td>\n",
       "                        <th> Array </th>\n",
       "                        <th> Chunk </th>\n",
       "                    </tr>\n",
       "                </thead>\n",
       "                <tbody>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Bytes </th>\n",
       "                        <td> 2.82 kiB </td>\n",
       "                        <td> 2.82 kiB </td>\n",
       "                    </tr>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Shape </th>\n",
       "                        <td> (361,) </td>\n",
       "                        <td> (361,) </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Dask graph </th>\n",
       "                        <td colspan=\"2\"> 1 chunks in 2 graph layers </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Data type </th>\n",
       "                        <td colspan=\"2\"> object numpy.ndarray </td>\n",
       "                    </tr>\n",
       "                </tbody>\n",
       "            </table>\n",
       "        </td>\n",
       "        <td>\n",
       "        <svg width=\"170\" height=\"75\" style=\"stroke:rgb(0,0,0);stroke-width:1\" >\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"120\" y2=\"0\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"0\" y1=\"25\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"0\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"120\" y1=\"0\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485\" style=\"fill:#ECB172A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Text -->\n",
       "  <text x=\"60.000000\" y=\"45.412617\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" >361</text>\n",
       "  <text x=\"140.000000\" y=\"12.706308\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(0,140.000000,12.706308)\">1</text>\n",
       "</svg>\n",
       "        </td>\n",
       "    </tr>\n",
       "</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>contig_length</span></div><div class='xr-var-dims'>(contigs)</div><div class='xr-var-dtype'>float64</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(361,), meta=np.ndarray&gt;</div><input id='attrs-6f4575fb-8d99-46e7-a30b-8b855b7adc30' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-6f4575fb-8d99-46e7-a30b-8b855b7adc30' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-1d5cafe1-6bf3-4d9a-8657-960c165a7673' class='xr-var-data-in' type='checkbox'><label for='data-1d5cafe1-6bf3-4d9a-8657-960c165a7673' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>\n",
       "    <tr>\n",
       "        <td>\n",
       "            <table style=\"border-collapse: collapse;\">\n",
       "                <thead>\n",
       "                    <tr>\n",
       "                        <td> </td>\n",
       "                        <th> Array </th>\n",
       "                        <th> Chunk </th>\n",
       "                    </tr>\n",
       "                </thead>\n",
       "                <tbody>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Bytes </th>\n",
       "                        <td> 2.82 kiB </td>\n",
       "                        <td> 2.82 kiB </td>\n",
       "                    </tr>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Shape </th>\n",
       "                        <td> (361,) </td>\n",
       "                        <td> (361,) </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Dask graph </th>\n",
       "                        <td colspan=\"2\"> 1 chunks in 2 graph layers </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Data type </th>\n",
       "                        <td colspan=\"2\"> float64 numpy.ndarray </td>\n",
       "                    </tr>\n",
       "                </tbody>\n",
       "            </table>\n",
       "        </td>\n",
       "        <td>\n",
       "        <svg width=\"170\" height=\"75\" style=\"stroke:rgb(0,0,0);stroke-width:1\" >\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"120\" y2=\"0\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"0\" y1=\"25\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"0\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"120\" y1=\"0\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485\" style=\"fill:#ECB172A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Text -->\n",
       "  <text x=\"60.000000\" y=\"45.412617\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" >361</text>\n",
       "  <text x=\"140.000000\" y=\"12.706308\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(0,140.000000,12.706308)\">1</text>\n",
       "</svg>\n",
       "        </td>\n",
       "    </tr>\n",
       "</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>filter_id</span></div><div class='xr-var-dims'>(filters)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(2,), meta=np.ndarray&gt;</div><input id='attrs-dbaa43c9-7445-41bb-9184-7ebe27e5ba27' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-dbaa43c9-7445-41bb-9184-7ebe27e5ba27' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-df63bd83-f1d5-48e3-b1aa-a76a9fc97640' class='xr-var-data-in' type='checkbox'><label for='data-df63bd83-f1d5-48e3-b1aa-a76a9fc97640' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>\n",
       "    <tr>\n",
       "        <td>\n",
       "            <table style=\"border-collapse: collapse;\">\n",
       "                <thead>\n",
       "                    <tr>\n",
       "                        <td> </td>\n",
       "                        <th> Array </th>\n",
       "                        <th> Chunk </th>\n",
       "                    </tr>\n",
       "                </thead>\n",
       "                <tbody>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Bytes </th>\n",
       "                        <td> 16 B </td>\n",
       "                        <td> 16 B </td>\n",
       "                    </tr>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Shape </th>\n",
       "                        <td> (2,) </td>\n",
       "                        <td> (2,) </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Dask graph </th>\n",
       "                        <td colspan=\"2\"> 1 chunks in 2 graph layers </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Data type </th>\n",
       "                        <td colspan=\"2\"> object numpy.ndarray </td>\n",
       "                    </tr>\n",
       "                </tbody>\n",
       "            </table>\n",
       "        </td>\n",
       "        <td>\n",
       "        <svg width=\"170\" height=\"110\" style=\"stroke:rgb(0,0,0);stroke-width:1\" >\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"120\" y2=\"0\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"0\" y1=\"60\" x2=\"120\" y2=\"60\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"0\" y2=\"60\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"120\" y1=\"0\" x2=\"120\" y2=\"60\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"0.0,0.0 120.0,0.0 120.0,60.0 0.0,60.0\" style=\"fill:#ECB172A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Text -->\n",
       "  <text x=\"60.000000\" y=\"80.000000\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" >2</text>\n",
       "  <text x=\"140.000000\" y=\"30.000000\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(0,140.000000,30.000000)\">1</text>\n",
       "</svg>\n",
       "        </td>\n",
       "    </tr>\n",
       "</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>region_index</span></div><div class='xr-var-dims'>(region_index_values, region_index_fields)</div><div class='xr-var-dtype'>int32</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(2024, 6), meta=np.ndarray&gt;</div><input id='attrs-a9491c5e-8b3f-4d2a-96e1-13c10e95eebb' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-a9491c5e-8b3f-4d2a-96e1-13c10e95eebb' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-7f332b34-b75a-4f96-a550-bad055974524' class='xr-var-data-in' type='checkbox'><label for='data-7f332b34-b75a-4f96-a550-bad055974524' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>\n",
       "    <tr>\n",
       "        <td>\n",
       "            <table style=\"border-collapse: collapse;\">\n",
       "                <thead>\n",
       "                    <tr>\n",
       "                        <td> </td>\n",
       "                        <th> Array </th>\n",
       "                        <th> Chunk </th>\n",
       "                    </tr>\n",
       "                </thead>\n",
       "                <tbody>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Bytes </th>\n",
       "                        <td> 47.44 kiB </td>\n",
       "                        <td> 47.44 kiB </td>\n",
       "                    </tr>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Shape </th>\n",
       "                        <td> (2024, 6) </td>\n",
       "                        <td> (2024, 6) </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Dask graph </th>\n",
       "                        <td colspan=\"2\"> 1 chunks in 2 graph layers </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Data type </th>\n",
       "                        <td colspan=\"2\"> int32 numpy.ndarray </td>\n",
       "                    </tr>\n",
       "                </tbody>\n",
       "            </table>\n",
       "        </td>\n",
       "        <td>\n",
       "        <svg width=\"75\" height=\"170\" style=\"stroke:rgb(0,0,0);stroke-width:1\" >\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"25\" y2=\"0\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"0\" y1=\"120\" x2=\"25\" y2=\"120\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"0\" y2=\"120\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"25\" y1=\"0\" x2=\"25\" y2=\"120\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"0.0,0.0 25.412616514582485,0.0 25.412616514582485,120.0 0.0,120.0\" style=\"fill:#ECB172A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Text -->\n",
       "  <text x=\"12.706308\" y=\"140.000000\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" >6</text>\n",
       "  <text x=\"45.412617\" y=\"60.000000\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(-90,45.412617,60.000000)\">2024</text>\n",
       "</svg>\n",
       "        </td>\n",
       "    </tr>\n",
       "</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>sample_id</span></div><div class='xr-var-dims'>(samples)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(7,), meta=np.ndarray&gt;</div><input id='attrs-4950d230-43e2-4da0-9552-861fc8389e6f' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-4950d230-43e2-4da0-9552-861fc8389e6f' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-298921f5-4faa-450f-ba5d-c98a55814a99' class='xr-var-data-in' type='checkbox'><label for='data-298921f5-4faa-450f-ba5d-c98a55814a99' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><div class='xr-var-data'><table>\n",
       "    <tr>\n",
       "        <td>\n",
       "            <table style=\"border-collapse: collapse;\">\n",
       "                <thead>\n",
       "                    <tr>\n",
       "                        <td> </td>\n",
       "                        <th> Array </th>\n",
       "                        <th> Chunk </th>\n",
       "                    </tr>\n",
       "                </thead>\n",
       "                <tbody>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Bytes </th>\n",
       "                        <td> 56 B </td>\n",
       "                        <td> 56 B </td>\n",
       "                    </tr>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Shape </th>\n",
       "                        <td> (7,) </td>\n",
       "                        <td> (7,) </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Dask graph </th>\n",
       "                        <td colspan=\"2\"> 1 chunks in 2 graph layers </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Data type </th>\n",
       "                        <td colspan=\"2\"> object numpy.ndarray </td>\n",
       "                    </tr>\n",
       "                </tbody>\n",
       "            </table>\n",
       "        </td>\n",
       "        <td>\n",
       "        <svg width=\"170\" height=\"90\" style=\"stroke:rgb(0,0,0);stroke-width:1\" >\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"120\" y2=\"0\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"0\" y1=\"40\" x2=\"120\" y2=\"40\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"0\" y2=\"40\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"120\" y1=\"0\" x2=\"120\" y2=\"40\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"0.0,0.0 120.0,0.0 120.0,40.46844837769698 0.0,40.46844837769698\" style=\"fill:#ECB172A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Text -->\n",
       "  <text x=\"60.000000\" y=\"60.468448\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" >7</text>\n",
       "  <text x=\"140.000000\" y=\"20.234224\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(0,140.000000,20.234224)\">1</text>\n",
       "</svg>\n",
       "        </td>\n",
       "    </tr>\n",
       "</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>variant_allele</span></div><div class='xr-var-dims'>(variants, alleles)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(1000, 2), meta=np.ndarray&gt;</div><input id='attrs-56f2de1e-1b7c-4e81-a7f3-a178bae5ec30' class='xr-var-attrs-in' type='checkbox' ><label for='attrs-56f2de1e-1b7c-4e81-a7f3-a178bae5ec30' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-3c880e1c-6126-40ba-a789-84a21ef60469' class='xr-var-data-in' type='checkbox'><label for='data-3c880e1c-6126-40ba-a789-84a21ef60469' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'><dt><span>description :</span></dt><dd>List of the reference and alternate alleles</dd></dl></div><div class='xr-var-data'><table>\n",
       "    <tr>\n",
       "        <td>\n",
       "            <table style=\"border-collapse: collapse;\">\n",
       "                <thead>\n",
       "                    <tr>\n",
       "                        <td> </td>\n",
       "                        <th> Array </th>\n",
       "                        <th> Chunk </th>\n",
       "                    </tr>\n",
       "                </thead>\n",
       "                <tbody>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Bytes </th>\n",
       "                        <td> 30.88 MiB </td>\n",
       "                        <td> 15.62 kiB </td>\n",
       "                    </tr>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Shape </th>\n",
       "                        <td> (2023479, 2) </td>\n",
       "                        <td> (1000, 2) </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Dask graph </th>\n",
       "                        <td colspan=\"2\"> 2024 chunks in 2 graph layers </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Data type </th>\n",
       "                        <td colspan=\"2\"> object numpy.ndarray </td>\n",
       "                    </tr>\n",
       "                </tbody>\n",
       "            </table>\n",
       "        </td>\n",
       "        <td>\n",
       "        <svg width=\"75\" height=\"170\" style=\"stroke:rgb(0,0,0);stroke-width:1\" >\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"25\" y2=\"0\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"0\" y1=\"6\" x2=\"25\" y2=\"6\" />\n",
       "  <line x1=\"0\" y1=\"12\" x2=\"25\" y2=\"12\" />\n",
       "  <line x1=\"0\" y1=\"18\" x2=\"25\" y2=\"18\" />\n",
       "  <line x1=\"0\" y1=\"25\" x2=\"25\" y2=\"25\" />\n",
       "  <line x1=\"0\" y1=\"31\" x2=\"25\" y2=\"31\" />\n",
       "  <line x1=\"0\" y1=\"37\" x2=\"25\" y2=\"37\" />\n",
       "  <line x1=\"0\" y1=\"44\" x2=\"25\" y2=\"44\" />\n",
       "  <line x1=\"0\" y1=\"50\" x2=\"25\" y2=\"50\" />\n",
       "  <line x1=\"0\" y1=\"56\" x2=\"25\" y2=\"56\" />\n",
       "  <line x1=\"0\" y1=\"63\" x2=\"25\" y2=\"63\" />\n",
       "  <line x1=\"0\" y1=\"69\" x2=\"25\" y2=\"69\" />\n",
       "  <line x1=\"0\" y1=\"75\" x2=\"25\" y2=\"75\" />\n",
       "  <line x1=\"0\" y1=\"82\" x2=\"25\" y2=\"82\" />\n",
       "  <line x1=\"0\" y1=\"88\" x2=\"25\" y2=\"88\" />\n",
       "  <line x1=\"0\" y1=\"94\" x2=\"25\" y2=\"94\" />\n",
       "  <line x1=\"0\" y1=\"101\" x2=\"25\" y2=\"101\" />\n",
       "  <line x1=\"0\" y1=\"107\" x2=\"25\" y2=\"107\" />\n",
       "  <line x1=\"0\" y1=\"113\" x2=\"25\" y2=\"113\" />\n",
       "  <line x1=\"0\" y1=\"120\" x2=\"25\" y2=\"120\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"0\" y2=\"120\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"25\" y1=\"0\" x2=\"25\" y2=\"120\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"0.0,0.0 25.412616514582485,0.0 25.412616514582485,120.0 0.0,120.0\" style=\"fill:#8B4903A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Text -->\n",
       "  <text x=\"12.706308\" y=\"140.000000\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" >2</text>\n",
       "  <text x=\"45.412617\" y=\"60.000000\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(-90,45.412617,60.000000)\">2023479</text>\n",
       "</svg>\n",
       "        </td>\n",
       "    </tr>\n",
       "</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>variant_contig</span></div><div class='xr-var-dims'>(variants)</div><div class='xr-var-dtype'>int16</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(1000,), meta=np.ndarray&gt;</div><input id='attrs-6f6ce94b-5dac-437a-abc3-173b40e0b48f' class='xr-var-attrs-in' type='checkbox' ><label for='attrs-6f6ce94b-5dac-437a-abc3-173b40e0b48f' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-e0f245c9-73eb-4224-b44d-44586b8e9dfd' class='xr-var-data-in' type='checkbox'><label for='data-e0f245c9-73eb-4224-b44d-44586b8e9dfd' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'><dt><span>description :</span></dt><dd>An identifier from the reference genome or an angle-bracketed ID string pointing to a contig in the assembly file</dd></dl></div><div class='xr-var-data'><table>\n",
       "    <tr>\n",
       "        <td>\n",
       "            <table style=\"border-collapse: collapse;\">\n",
       "                <thead>\n",
       "                    <tr>\n",
       "                        <td> </td>\n",
       "                        <th> Array </th>\n",
       "                        <th> Chunk </th>\n",
       "                    </tr>\n",
       "                </thead>\n",
       "                <tbody>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Bytes </th>\n",
       "                        <td> 3.86 MiB </td>\n",
       "                        <td> 1.95 kiB </td>\n",
       "                    </tr>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Shape </th>\n",
       "                        <td> (2023479,) </td>\n",
       "                        <td> (1000,) </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Dask graph </th>\n",
       "                        <td colspan=\"2\"> 2024 chunks in 2 graph layers </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Data type </th>\n",
       "                        <td colspan=\"2\"> int16 numpy.ndarray </td>\n",
       "                    </tr>\n",
       "                </tbody>\n",
       "            </table>\n",
       "        </td>\n",
       "        <td>\n",
       "        <svg width=\"170\" height=\"75\" style=\"stroke:rgb(0,0,0);stroke-width:1\" >\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"120\" y2=\"0\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"0\" y1=\"25\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"0\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"6\" y1=\"0\" x2=\"6\" y2=\"25\" />\n",
       "  <line x1=\"12\" y1=\"0\" x2=\"12\" y2=\"25\" />\n",
       "  <line x1=\"18\" y1=\"0\" x2=\"18\" y2=\"25\" />\n",
       "  <line x1=\"25\" y1=\"0\" x2=\"25\" y2=\"25\" />\n",
       "  <line x1=\"31\" y1=\"0\" x2=\"31\" y2=\"25\" />\n",
       "  <line x1=\"37\" y1=\"0\" x2=\"37\" y2=\"25\" />\n",
       "  <line x1=\"44\" y1=\"0\" x2=\"44\" y2=\"25\" />\n",
       "  <line x1=\"50\" y1=\"0\" x2=\"50\" y2=\"25\" />\n",
       "  <line x1=\"56\" y1=\"0\" x2=\"56\" y2=\"25\" />\n",
       "  <line x1=\"63\" y1=\"0\" x2=\"63\" y2=\"25\" />\n",
       "  <line x1=\"69\" y1=\"0\" x2=\"69\" y2=\"25\" />\n",
       "  <line x1=\"75\" y1=\"0\" x2=\"75\" y2=\"25\" />\n",
       "  <line x1=\"82\" y1=\"0\" x2=\"82\" y2=\"25\" />\n",
       "  <line x1=\"88\" y1=\"0\" x2=\"88\" y2=\"25\" />\n",
       "  <line x1=\"94\" y1=\"0\" x2=\"94\" y2=\"25\" />\n",
       "  <line x1=\"101\" y1=\"0\" x2=\"101\" y2=\"25\" />\n",
       "  <line x1=\"107\" y1=\"0\" x2=\"107\" y2=\"25\" />\n",
       "  <line x1=\"113\" y1=\"0\" x2=\"113\" y2=\"25\" />\n",
       "  <line x1=\"120\" y1=\"0\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485\" style=\"fill:#8B4903A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Text -->\n",
       "  <text x=\"60.000000\" y=\"45.412617\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" >2023479</text>\n",
       "  <text x=\"140.000000\" y=\"12.706308\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(0,140.000000,12.706308)\">1</text>\n",
       "</svg>\n",
       "        </td>\n",
       "    </tr>\n",
       "</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>variant_filter</span></div><div class='xr-var-dims'>(variants, filters)</div><div class='xr-var-dtype'>bool</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(1000, 2), meta=np.ndarray&gt;</div><input id='attrs-04849a8b-7815-49dd-accd-6d43a139e98e' class='xr-var-attrs-in' type='checkbox' ><label for='attrs-04849a8b-7815-49dd-accd-6d43a139e98e' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-0bbbb851-e78a-4d0a-85c2-65816b20bab6' class='xr-var-data-in' type='checkbox'><label for='data-0bbbb851-e78a-4d0a-85c2-65816b20bab6' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'><dt><span>description :</span></dt><dd>Filter status of the variant</dd></dl></div><div class='xr-var-data'><table>\n",
       "    <tr>\n",
       "        <td>\n",
       "            <table style=\"border-collapse: collapse;\">\n",
       "                <thead>\n",
       "                    <tr>\n",
       "                        <td> </td>\n",
       "                        <th> Array </th>\n",
       "                        <th> Chunk </th>\n",
       "                    </tr>\n",
       "                </thead>\n",
       "                <tbody>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Bytes </th>\n",
       "                        <td> 3.86 MiB </td>\n",
       "                        <td> 1.95 kiB </td>\n",
       "                    </tr>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Shape </th>\n",
       "                        <td> (2023479, 2) </td>\n",
       "                        <td> (1000, 2) </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Dask graph </th>\n",
       "                        <td colspan=\"2\"> 2024 chunks in 2 graph layers </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Data type </th>\n",
       "                        <td colspan=\"2\"> bool numpy.ndarray </td>\n",
       "                    </tr>\n",
       "                </tbody>\n",
       "            </table>\n",
       "        </td>\n",
       "        <td>\n",
       "        <svg width=\"75\" height=\"170\" style=\"stroke:rgb(0,0,0);stroke-width:1\" >\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"25\" y2=\"0\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"0\" y1=\"6\" x2=\"25\" y2=\"6\" />\n",
       "  <line x1=\"0\" y1=\"12\" x2=\"25\" y2=\"12\" />\n",
       "  <line x1=\"0\" y1=\"18\" x2=\"25\" y2=\"18\" />\n",
       "  <line x1=\"0\" y1=\"25\" x2=\"25\" y2=\"25\" />\n",
       "  <line x1=\"0\" y1=\"31\" x2=\"25\" y2=\"31\" />\n",
       "  <line x1=\"0\" y1=\"37\" x2=\"25\" y2=\"37\" />\n",
       "  <line x1=\"0\" y1=\"44\" x2=\"25\" y2=\"44\" />\n",
       "  <line x1=\"0\" y1=\"50\" x2=\"25\" y2=\"50\" />\n",
       "  <line x1=\"0\" y1=\"56\" x2=\"25\" y2=\"56\" />\n",
       "  <line x1=\"0\" y1=\"63\" x2=\"25\" y2=\"63\" />\n",
       "  <line x1=\"0\" y1=\"69\" x2=\"25\" y2=\"69\" />\n",
       "  <line x1=\"0\" y1=\"75\" x2=\"25\" y2=\"75\" />\n",
       "  <line x1=\"0\" y1=\"82\" x2=\"25\" y2=\"82\" />\n",
       "  <line x1=\"0\" y1=\"88\" x2=\"25\" y2=\"88\" />\n",
       "  <line x1=\"0\" y1=\"94\" x2=\"25\" y2=\"94\" />\n",
       "  <line x1=\"0\" y1=\"101\" x2=\"25\" y2=\"101\" />\n",
       "  <line x1=\"0\" y1=\"107\" x2=\"25\" y2=\"107\" />\n",
       "  <line x1=\"0\" y1=\"113\" x2=\"25\" y2=\"113\" />\n",
       "  <line x1=\"0\" y1=\"120\" x2=\"25\" y2=\"120\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"0\" y2=\"120\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"25\" y1=\"0\" x2=\"25\" y2=\"120\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"0.0,0.0 25.412616514582485,0.0 25.412616514582485,120.0 0.0,120.0\" style=\"fill:#8B4903A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Text -->\n",
       "  <text x=\"12.706308\" y=\"140.000000\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" >2</text>\n",
       "  <text x=\"45.412617\" y=\"60.000000\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(-90,45.412617,60.000000)\">2023479</text>\n",
       "</svg>\n",
       "        </td>\n",
       "    </tr>\n",
       "</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>variant_id</span></div><div class='xr-var-dims'>(variants)</div><div class='xr-var-dtype'>object</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(1000,), meta=np.ndarray&gt;</div><input id='attrs-e6f92483-777d-4e6e-b9d4-3949a6dd1928' class='xr-var-attrs-in' type='checkbox' ><label for='attrs-e6f92483-777d-4e6e-b9d4-3949a6dd1928' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-649053e3-e591-49a2-9de3-035443d193aa' class='xr-var-data-in' type='checkbox'><label for='data-649053e3-e591-49a2-9de3-035443d193aa' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'><dt><span>description :</span></dt><dd>List of unique identifiers where applicable</dd></dl></div><div class='xr-var-data'><table>\n",
       "    <tr>\n",
       "        <td>\n",
       "            <table style=\"border-collapse: collapse;\">\n",
       "                <thead>\n",
       "                    <tr>\n",
       "                        <td> </td>\n",
       "                        <th> Array </th>\n",
       "                        <th> Chunk </th>\n",
       "                    </tr>\n",
       "                </thead>\n",
       "                <tbody>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Bytes </th>\n",
       "                        <td> 15.44 MiB </td>\n",
       "                        <td> 7.81 kiB </td>\n",
       "                    </tr>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Shape </th>\n",
       "                        <td> (2023479,) </td>\n",
       "                        <td> (1000,) </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Dask graph </th>\n",
       "                        <td colspan=\"2\"> 2024 chunks in 2 graph layers </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Data type </th>\n",
       "                        <td colspan=\"2\"> object numpy.ndarray </td>\n",
       "                    </tr>\n",
       "                </tbody>\n",
       "            </table>\n",
       "        </td>\n",
       "        <td>\n",
       "        <svg width=\"170\" height=\"75\" style=\"stroke:rgb(0,0,0);stroke-width:1\" >\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"120\" y2=\"0\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"0\" y1=\"25\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"0\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"6\" y1=\"0\" x2=\"6\" y2=\"25\" />\n",
       "  <line x1=\"12\" y1=\"0\" x2=\"12\" y2=\"25\" />\n",
       "  <line x1=\"18\" y1=\"0\" x2=\"18\" y2=\"25\" />\n",
       "  <line x1=\"25\" y1=\"0\" x2=\"25\" y2=\"25\" />\n",
       "  <line x1=\"31\" y1=\"0\" x2=\"31\" y2=\"25\" />\n",
       "  <line x1=\"37\" y1=\"0\" x2=\"37\" y2=\"25\" />\n",
       "  <line x1=\"44\" y1=\"0\" x2=\"44\" y2=\"25\" />\n",
       "  <line x1=\"50\" y1=\"0\" x2=\"50\" y2=\"25\" />\n",
       "  <line x1=\"56\" y1=\"0\" x2=\"56\" y2=\"25\" />\n",
       "  <line x1=\"63\" y1=\"0\" x2=\"63\" y2=\"25\" />\n",
       "  <line x1=\"69\" y1=\"0\" x2=\"69\" y2=\"25\" />\n",
       "  <line x1=\"75\" y1=\"0\" x2=\"75\" y2=\"25\" />\n",
       "  <line x1=\"82\" y1=\"0\" x2=\"82\" y2=\"25\" />\n",
       "  <line x1=\"88\" y1=\"0\" x2=\"88\" y2=\"25\" />\n",
       "  <line x1=\"94\" y1=\"0\" x2=\"94\" y2=\"25\" />\n",
       "  <line x1=\"101\" y1=\"0\" x2=\"101\" y2=\"25\" />\n",
       "  <line x1=\"107\" y1=\"0\" x2=\"107\" y2=\"25\" />\n",
       "  <line x1=\"113\" y1=\"0\" x2=\"113\" y2=\"25\" />\n",
       "  <line x1=\"120\" y1=\"0\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485\" style=\"fill:#8B4903A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Text -->\n",
       "  <text x=\"60.000000\" y=\"45.412617\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" >2023479</text>\n",
       "  <text x=\"140.000000\" y=\"12.706308\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(0,140.000000,12.706308)\">1</text>\n",
       "</svg>\n",
       "        </td>\n",
       "    </tr>\n",
       "</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>variant_id_mask</span></div><div class='xr-var-dims'>(variants)</div><div class='xr-var-dtype'>bool</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(1000,), meta=np.ndarray&gt;</div><input id='attrs-0c9915c4-2aad-4a5b-8a4a-01c5f29ac3ce' class='xr-var-attrs-in' type='checkbox' ><label for='attrs-0c9915c4-2aad-4a5b-8a4a-01c5f29ac3ce' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-6b1dd119-35a7-4661-8ded-abcaa1d2a76e' class='xr-var-data-in' type='checkbox'><label for='data-6b1dd119-35a7-4661-8ded-abcaa1d2a76e' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'><dt><span>description :</span></dt><dd></dd></dl></div><div class='xr-var-data'><table>\n",
       "    <tr>\n",
       "        <td>\n",
       "            <table style=\"border-collapse: collapse;\">\n",
       "                <thead>\n",
       "                    <tr>\n",
       "                        <td> </td>\n",
       "                        <th> Array </th>\n",
       "                        <th> Chunk </th>\n",
       "                    </tr>\n",
       "                </thead>\n",
       "                <tbody>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Bytes </th>\n",
       "                        <td> 1.93 MiB </td>\n",
       "                        <td> 0.98 kiB </td>\n",
       "                    </tr>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Shape </th>\n",
       "                        <td> (2023479,) </td>\n",
       "                        <td> (1000,) </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Dask graph </th>\n",
       "                        <td colspan=\"2\"> 2024 chunks in 2 graph layers </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Data type </th>\n",
       "                        <td colspan=\"2\"> bool numpy.ndarray </td>\n",
       "                    </tr>\n",
       "                </tbody>\n",
       "            </table>\n",
       "        </td>\n",
       "        <td>\n",
       "        <svg width=\"170\" height=\"75\" style=\"stroke:rgb(0,0,0);stroke-width:1\" >\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"120\" y2=\"0\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"0\" y1=\"25\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"0\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"6\" y1=\"0\" x2=\"6\" y2=\"25\" />\n",
       "  <line x1=\"12\" y1=\"0\" x2=\"12\" y2=\"25\" />\n",
       "  <line x1=\"18\" y1=\"0\" x2=\"18\" y2=\"25\" />\n",
       "  <line x1=\"25\" y1=\"0\" x2=\"25\" y2=\"25\" />\n",
       "  <line x1=\"31\" y1=\"0\" x2=\"31\" y2=\"25\" />\n",
       "  <line x1=\"37\" y1=\"0\" x2=\"37\" y2=\"25\" />\n",
       "  <line x1=\"44\" y1=\"0\" x2=\"44\" y2=\"25\" />\n",
       "  <line x1=\"50\" y1=\"0\" x2=\"50\" y2=\"25\" />\n",
       "  <line x1=\"56\" y1=\"0\" x2=\"56\" y2=\"25\" />\n",
       "  <line x1=\"63\" y1=\"0\" x2=\"63\" y2=\"25\" />\n",
       "  <line x1=\"69\" y1=\"0\" x2=\"69\" y2=\"25\" />\n",
       "  <line x1=\"75\" y1=\"0\" x2=\"75\" y2=\"25\" />\n",
       "  <line x1=\"82\" y1=\"0\" x2=\"82\" y2=\"25\" />\n",
       "  <line x1=\"88\" y1=\"0\" x2=\"88\" y2=\"25\" />\n",
       "  <line x1=\"94\" y1=\"0\" x2=\"94\" y2=\"25\" />\n",
       "  <line x1=\"101\" y1=\"0\" x2=\"101\" y2=\"25\" />\n",
       "  <line x1=\"107\" y1=\"0\" x2=\"107\" y2=\"25\" />\n",
       "  <line x1=\"113\" y1=\"0\" x2=\"113\" y2=\"25\" />\n",
       "  <line x1=\"120\" y1=\"0\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485\" style=\"fill:#8B4903A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Text -->\n",
       "  <text x=\"60.000000\" y=\"45.412617\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" >2023479</text>\n",
       "  <text x=\"140.000000\" y=\"12.706308\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(0,140.000000,12.706308)\">1</text>\n",
       "</svg>\n",
       "        </td>\n",
       "    </tr>\n",
       "</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>variant_length</span></div><div class='xr-var-dims'>(variants)</div><div class='xr-var-dtype'>int16</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(1000,), meta=np.ndarray&gt;</div><input id='attrs-f1f2f81a-fadf-4dd2-bfed-a6477d7cc40b' class='xr-var-attrs-in' type='checkbox' ><label for='attrs-f1f2f81a-fadf-4dd2-bfed-a6477d7cc40b' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-661d73f6-f80a-43ed-a24f-b7c017ab7692' class='xr-var-data-in' type='checkbox'><label for='data-661d73f6-f80a-43ed-a24f-b7c017ab7692' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'><dt><span>description :</span></dt><dd>The length of the variant measured in bases</dd></dl></div><div class='xr-var-data'><table>\n",
       "    <tr>\n",
       "        <td>\n",
       "            <table style=\"border-collapse: collapse;\">\n",
       "                <thead>\n",
       "                    <tr>\n",
       "                        <td> </td>\n",
       "                        <th> Array </th>\n",
       "                        <th> Chunk </th>\n",
       "                    </tr>\n",
       "                </thead>\n",
       "                <tbody>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Bytes </th>\n",
       "                        <td> 3.86 MiB </td>\n",
       "                        <td> 1.95 kiB </td>\n",
       "                    </tr>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Shape </th>\n",
       "                        <td> (2023479,) </td>\n",
       "                        <td> (1000,) </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Dask graph </th>\n",
       "                        <td colspan=\"2\"> 2024 chunks in 2 graph layers </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Data type </th>\n",
       "                        <td colspan=\"2\"> int16 numpy.ndarray </td>\n",
       "                    </tr>\n",
       "                </tbody>\n",
       "            </table>\n",
       "        </td>\n",
       "        <td>\n",
       "        <svg width=\"170\" height=\"75\" style=\"stroke:rgb(0,0,0);stroke-width:1\" >\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"120\" y2=\"0\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"0\" y1=\"25\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"0\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"6\" y1=\"0\" x2=\"6\" y2=\"25\" />\n",
       "  <line x1=\"12\" y1=\"0\" x2=\"12\" y2=\"25\" />\n",
       "  <line x1=\"18\" y1=\"0\" x2=\"18\" y2=\"25\" />\n",
       "  <line x1=\"25\" y1=\"0\" x2=\"25\" y2=\"25\" />\n",
       "  <line x1=\"31\" y1=\"0\" x2=\"31\" y2=\"25\" />\n",
       "  <line x1=\"37\" y1=\"0\" x2=\"37\" y2=\"25\" />\n",
       "  <line x1=\"44\" y1=\"0\" x2=\"44\" y2=\"25\" />\n",
       "  <line x1=\"50\" y1=\"0\" x2=\"50\" y2=\"25\" />\n",
       "  <line x1=\"56\" y1=\"0\" x2=\"56\" y2=\"25\" />\n",
       "  <line x1=\"63\" y1=\"0\" x2=\"63\" y2=\"25\" />\n",
       "  <line x1=\"69\" y1=\"0\" x2=\"69\" y2=\"25\" />\n",
       "  <line x1=\"75\" y1=\"0\" x2=\"75\" y2=\"25\" />\n",
       "  <line x1=\"82\" y1=\"0\" x2=\"82\" y2=\"25\" />\n",
       "  <line x1=\"88\" y1=\"0\" x2=\"88\" y2=\"25\" />\n",
       "  <line x1=\"94\" y1=\"0\" x2=\"94\" y2=\"25\" />\n",
       "  <line x1=\"101\" y1=\"0\" x2=\"101\" y2=\"25\" />\n",
       "  <line x1=\"107\" y1=\"0\" x2=\"107\" y2=\"25\" />\n",
       "  <line x1=\"113\" y1=\"0\" x2=\"113\" y2=\"25\" />\n",
       "  <line x1=\"120\" y1=\"0\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485\" style=\"fill:#8B4903A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Text -->\n",
       "  <text x=\"60.000000\" y=\"45.412617\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" >2023479</text>\n",
       "  <text x=\"140.000000\" y=\"12.706308\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(0,140.000000,12.706308)\">1</text>\n",
       "</svg>\n",
       "        </td>\n",
       "    </tr>\n",
       "</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>variant_position</span></div><div class='xr-var-dims'>(variants)</div><div class='xr-var-dtype'>int32</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(1000,), meta=np.ndarray&gt;</div><input id='attrs-3ccc0641-44f5-4e70-9523-013f99978901' class='xr-var-attrs-in' type='checkbox' ><label for='attrs-3ccc0641-44f5-4e70-9523-013f99978901' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-7420012b-18ee-4393-a33d-cd47d10f9fa8' class='xr-var-data-in' type='checkbox'><label for='data-7420012b-18ee-4393-a33d-cd47d10f9fa8' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'><dt><span>description :</span></dt><dd>The reference position</dd></dl></div><div class='xr-var-data'><table>\n",
       "    <tr>\n",
       "        <td>\n",
       "            <table style=\"border-collapse: collapse;\">\n",
       "                <thead>\n",
       "                    <tr>\n",
       "                        <td> </td>\n",
       "                        <th> Array </th>\n",
       "                        <th> Chunk </th>\n",
       "                    </tr>\n",
       "                </thead>\n",
       "                <tbody>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Bytes </th>\n",
       "                        <td> 7.72 MiB </td>\n",
       "                        <td> 3.91 kiB </td>\n",
       "                    </tr>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Shape </th>\n",
       "                        <td> (2023479,) </td>\n",
       "                        <td> (1000,) </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Dask graph </th>\n",
       "                        <td colspan=\"2\"> 2024 chunks in 2 graph layers </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Data type </th>\n",
       "                        <td colspan=\"2\"> int32 numpy.ndarray </td>\n",
       "                    </tr>\n",
       "                </tbody>\n",
       "            </table>\n",
       "        </td>\n",
       "        <td>\n",
       "        <svg width=\"170\" height=\"75\" style=\"stroke:rgb(0,0,0);stroke-width:1\" >\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"120\" y2=\"0\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"0\" y1=\"25\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"0\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"6\" y1=\"0\" x2=\"6\" y2=\"25\" />\n",
       "  <line x1=\"12\" y1=\"0\" x2=\"12\" y2=\"25\" />\n",
       "  <line x1=\"18\" y1=\"0\" x2=\"18\" y2=\"25\" />\n",
       "  <line x1=\"25\" y1=\"0\" x2=\"25\" y2=\"25\" />\n",
       "  <line x1=\"31\" y1=\"0\" x2=\"31\" y2=\"25\" />\n",
       "  <line x1=\"37\" y1=\"0\" x2=\"37\" y2=\"25\" />\n",
       "  <line x1=\"44\" y1=\"0\" x2=\"44\" y2=\"25\" />\n",
       "  <line x1=\"50\" y1=\"0\" x2=\"50\" y2=\"25\" />\n",
       "  <line x1=\"56\" y1=\"0\" x2=\"56\" y2=\"25\" />\n",
       "  <line x1=\"63\" y1=\"0\" x2=\"63\" y2=\"25\" />\n",
       "  <line x1=\"69\" y1=\"0\" x2=\"69\" y2=\"25\" />\n",
       "  <line x1=\"75\" y1=\"0\" x2=\"75\" y2=\"25\" />\n",
       "  <line x1=\"82\" y1=\"0\" x2=\"82\" y2=\"25\" />\n",
       "  <line x1=\"88\" y1=\"0\" x2=\"88\" y2=\"25\" />\n",
       "  <line x1=\"94\" y1=\"0\" x2=\"94\" y2=\"25\" />\n",
       "  <line x1=\"101\" y1=\"0\" x2=\"101\" y2=\"25\" />\n",
       "  <line x1=\"107\" y1=\"0\" x2=\"107\" y2=\"25\" />\n",
       "  <line x1=\"113\" y1=\"0\" x2=\"113\" y2=\"25\" />\n",
       "  <line x1=\"120\" y1=\"0\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485\" style=\"fill:#8B4903A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Text -->\n",
       "  <text x=\"60.000000\" y=\"45.412617\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" >2023479</text>\n",
       "  <text x=\"140.000000\" y=\"12.706308\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(0,140.000000,12.706308)\">1</text>\n",
       "</svg>\n",
       "        </td>\n",
       "    </tr>\n",
       "</table></div></li><li class='xr-var-item'><div class='xr-var-name'><span>variant_quality</span></div><div class='xr-var-dims'>(variants)</div><div class='xr-var-dtype'>float32</div><div class='xr-var-preview xr-preview'>dask.array&lt;chunksize=(1000,), meta=np.ndarray&gt;</div><input id='attrs-68ce6160-0a45-494d-85fc-1e6f75f59f4e' class='xr-var-attrs-in' type='checkbox' ><label for='attrs-68ce6160-0a45-494d-85fc-1e6f75f59f4e' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-37c36c12-322f-4f65-88f9-d9588630f5e4' class='xr-var-data-in' type='checkbox'><label for='data-37c36c12-322f-4f65-88f9-d9588630f5e4' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'><dt><span>description :</span></dt><dd>Phred-scaled quality score</dd></dl></div><div class='xr-var-data'><table>\n",
       "    <tr>\n",
       "        <td>\n",
       "            <table style=\"border-collapse: collapse;\">\n",
       "                <thead>\n",
       "                    <tr>\n",
       "                        <td> </td>\n",
       "                        <th> Array </th>\n",
       "                        <th> Chunk </th>\n",
       "                    </tr>\n",
       "                </thead>\n",
       "                <tbody>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Bytes </th>\n",
       "                        <td> 7.72 MiB </td>\n",
       "                        <td> 3.91 kiB </td>\n",
       "                    </tr>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Shape </th>\n",
       "                        <td> (2023479,) </td>\n",
       "                        <td> (1000,) </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Dask graph </th>\n",
       "                        <td colspan=\"2\"> 2024 chunks in 2 graph layers </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Data type </th>\n",
       "                        <td colspan=\"2\"> float32 numpy.ndarray </td>\n",
       "                    </tr>\n",
       "                </tbody>\n",
       "            </table>\n",
       "        </td>\n",
       "        <td>\n",
       "        <svg width=\"170\" height=\"75\" style=\"stroke:rgb(0,0,0);stroke-width:1\" >\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"120\" y2=\"0\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"0\" y1=\"25\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"0\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"6\" y1=\"0\" x2=\"6\" y2=\"25\" />\n",
       "  <line x1=\"12\" y1=\"0\" x2=\"12\" y2=\"25\" />\n",
       "  <line x1=\"18\" y1=\"0\" x2=\"18\" y2=\"25\" />\n",
       "  <line x1=\"25\" y1=\"0\" x2=\"25\" y2=\"25\" />\n",
       "  <line x1=\"31\" y1=\"0\" x2=\"31\" y2=\"25\" />\n",
       "  <line x1=\"37\" y1=\"0\" x2=\"37\" y2=\"25\" />\n",
       "  <line x1=\"44\" y1=\"0\" x2=\"44\" y2=\"25\" />\n",
       "  <line x1=\"50\" y1=\"0\" x2=\"50\" y2=\"25\" />\n",
       "  <line x1=\"56\" y1=\"0\" x2=\"56\" y2=\"25\" />\n",
       "  <line x1=\"63\" y1=\"0\" x2=\"63\" y2=\"25\" />\n",
       "  <line x1=\"69\" y1=\"0\" x2=\"69\" y2=\"25\" />\n",
       "  <line x1=\"75\" y1=\"0\" x2=\"75\" y2=\"25\" />\n",
       "  <line x1=\"82\" y1=\"0\" x2=\"82\" y2=\"25\" />\n",
       "  <line x1=\"88\" y1=\"0\" x2=\"88\" y2=\"25\" />\n",
       "  <line x1=\"94\" y1=\"0\" x2=\"94\" y2=\"25\" />\n",
       "  <line x1=\"101\" y1=\"0\" x2=\"101\" y2=\"25\" />\n",
       "  <line x1=\"107\" y1=\"0\" x2=\"107\" y2=\"25\" />\n",
       "  <line x1=\"113\" y1=\"0\" x2=\"113\" y2=\"25\" />\n",
       "  <line x1=\"120\" y1=\"0\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485\" style=\"fill:#8B4903A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Text -->\n",
       "  <text x=\"60.000000\" y=\"45.412617\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" >2023479</text>\n",
       "  <text x=\"140.000000\" y=\"12.706308\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(0,140.000000,12.706308)\">1</text>\n",
       "</svg>\n",
       "        </td>\n",
       "    </tr>\n",
       "</table></div></li></ul></div></li><li class='xr-section-item'><input id='section-1b2a1037-316a-4b4c-8812-ac8c25b4f167' class='xr-section-summary-in' type='checkbox' disabled ><label for='section-1b2a1037-316a-4b4c-8812-ac8c25b4f167' class='xr-section-summary'  title='Expand/collapse section'>Indexes: <span>(0)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><ul class='xr-var-list'></ul></div></li><li class='xr-section-item'><input id='section-bff4fb43-220a-40fb-9cfc-9d00c9a2fcb2' class='xr-section-summary-in' type='checkbox'  checked><label for='section-bff4fb43-220a-40fb-9cfc-9d00c9a2fcb2' class='xr-section-summary' >Attributes: <span>(3)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><dl class='xr-attrs'><dt><span>source :</span></dt><dd>bio2zarr-0.1.5</dd><dt><span>vcf_header :</span></dt><dd>##fileformat=VCFv4.2\n",
       "##FILTER=&lt;ID=PASS,Description=&quot;All filters passed&quot;&gt;\n",
       "##ALT=&lt;ID=NON_REF,Description=&quot;Represents any possible alternative allele not already represented at this location by REF and ALT&quot;&gt;\n",
       "##FILTER=&lt;ID=LowQual,Description=&quot;Low quality&quot;&gt;\n",
       "##FORMAT=&lt;ID=GT,Number=1,Type=String,Description=&quot;Genotype&quot;&gt;\n",
       "##GATKCommandLine=&lt;ID=GenomicsDBImport,CommandLine=&quot;GenomicsDBImport --genomicsdb-workspace-path /scratch/1776312/batch_0_fploidy_2_mploidy_2_subbatch_0 --genomicsdb-vcf-buffer-size 4194304 --batch-size 50 --sample-name-map /faststorage/project/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/GenomicsDB/batch_0_fploidy_2_mploidy_2/cohort.sample_map --reader-threads 1 --genomicsdb-shared-posixfs-optimizations true --intervals NC_092125.1:1-2000000 --tmp-dir /scratch/1776312 --genomicsdb-segment-size 1048576 --overwrite-existing-genomicsdb-workspace false --consolidate false --validate-sample-name-map false --merge-input-intervals false --max-num-intervals-to-import-in-parallel 1 --merge-contigs-into-num-partitions 0 --bypass-feature-reader false --genomicsdb-use-gcs-hdfs-connector false --interval-set-rule UNION --interval-padding 0 --interval-exclusion-padding 0 --interval-merging-rule ALL --read-validation-stringency SILENT --seconds-between-progress-updates 10.0 --disable-sequence-dictionary-validation false --create-output-bam-index true --create-output-bam-md5 false --create-output-variant-index true --create-output-variant-md5 false --max-variants-per-shard 0 --lenient false --add-output-sam-program-record true --add-output-vcf-command-line true --cloud-prefetch-buffer 0 --cloud-index-prefetch-buffer 0 --disable-bam-index-caching false --sites-only-vcf-output false --help false --version false --showHidden false --verbosity INFO --QUIET false --use-jdk-deflater false --use-jdk-inflater false --gcs-max-retries 20 --gcs-project-for-requester-pays  --disable-tool-default-read-filters false&quot;,Version=&quot;4.4.0.0&quot;,Date=&quot;August 5, 2025 at 8:42:55 AM CEST&quot;&gt;\n",
       "##contig=&lt;ID=NC_092125.1,length=233354531&gt;\n",
       "##contig=&lt;ID=NC_092126.1,length=203902748&gt;\n",
       "##contig=&lt;ID=NW_027257552.1,length=307142&gt;\n",
       "##contig=&lt;ID=NC_092127.1,length=192017243&gt;\n",
       "##contig=&lt;ID=NW_027257553.1,length=1114335&gt;\n",
       "##contig=&lt;ID=NC_092128.1,length=189967907&gt;\n",
       "##contig=&lt;ID=NW_027257554.1,length=1028371&gt;\n",
       "##contig=&lt;ID=NW_027257555.1,length=280435&gt;\n",
       "##contig=&lt;ID=NW_027257556.1,length=501422&gt;\n",
       "##contig=&lt;ID=NW_027257557.1,length=661072&gt;\n",
       "##contig=&lt;ID=NW_027257558.1,length=635413&gt;\n",
       "##contig=&lt;ID=NW_027257559.1,length=689531&gt;\n",
       "##contig=&lt;ID=NW_027257560.1,length=429852&gt;\n",
       "##contig=&lt;ID=NW_027257561.1,length=421469&gt;\n",
       "##contig=&lt;ID=NW_027257562.1,length=410642&gt;\n",
       "##contig=&lt;ID=NW_027257563.1,length=298804&gt;\n",
       "##contig=&lt;ID=NC_092129.1,length=184146409&gt;\n",
       "##contig=&lt;ID=NC_092130.1,length=181228022&gt;\n",
       "##contig=&lt;ID=NW_027257564.1,length=384283&gt;\n",
       "##contig=&lt;ID=NC_092131.1,length=179313751&gt;\n",
       "##contig=&lt;ID=NC_092132.1,length=154799650&gt;\n",
       "##contig=&lt;ID=NW_027257565.1,length=371445&gt;\n",
       "##contig=&lt;ID=NW_027257566.1,length=554562&gt;\n",
       "##contig=&lt;ID=NW_027257567.1,length=313440&gt;\n",
       "##contig=&lt;ID=NW_027257568.1,length=343533&gt;\n",
       "##contig=&lt;ID=NW_027257569.1,length=310383&gt;\n",
       "##contig=&lt;ID=NC_092133.1,length=144533704&gt;\n",
       "##contig=&lt;ID=NC_092134.1,length=143912906&gt;\n",
       "##contig=&lt;ID=NW_027257570.1,length=387861&gt;\n",
       "##contig=&lt;ID=NW_027257571.1,length=332351&gt;\n",
       "##contig=&lt;ID=NC_092135.1,length=140692836&gt;\n",
       "##contig=&lt;ID=NW_027257572.1,length=1046652&gt;\n",
       "##contig=&lt;ID=NW_027257573.1,length=441129&gt;\n",
       "##contig=&lt;ID=NW_027257574.1,length=505721&gt;\n",
       "##contig=&lt;ID=NW_027257575.1,length=396800&gt;\n",
       "##contig=&lt;ID=NC_092136.1,length=134121798&gt;\n",
       "##contig=&lt;ID=NC_092137.1,length=122685455&gt;\n",
       "##contig=&lt;ID=NW_027257576.1,length=371444&gt;\n",
       "##contig=&lt;ID=NC_092138.1,length=119675470&gt;\n",
       "##contig=&lt;ID=NW_027257577.1,length=577796&gt;\n",
       "##contig=&lt;ID=NW_027257578.1,length=909424&gt;\n",
       "##contig=&lt;ID=NW_027257579.1,length=371444&gt;\n",
       "##contig=&lt;ID=NC_092139.1,length=118469430&gt;\n",
       "##contig=&lt;ID=NW_027257580.1,length=324009&gt;\n",
       "##contig=&lt;ID=NC_092140.1,length=105885118&gt;\n",
       "##contig=&lt;ID=NC_092141.1,length=92152283&gt;\n",
       "##contig=&lt;ID=NC_092142.1,length=88956647&gt;\n",
       "##contig=&lt;ID=NC_092143.1,length=84033291&gt;\n",
       "##contig=&lt;ID=NC_092144.1,length=68621529&gt;\n",
       "##contig=&lt;ID=NC_092145.1,length=159757195&gt;\n",
       "##contig=&lt;ID=NW_027257581.1,length=557167&gt;\n",
       "##contig=&lt;ID=NC_092146.1,length=18418642&gt;\n",
       "##contig=&lt;ID=NW_027257582.1,length=495723&gt;\n",
       "##contig=&lt;ID=NW_027257583.1,length=228650&gt;\n",
       "##contig=&lt;ID=NW_027257584.1,length=225880&gt;\n",
       "##contig=&lt;ID=NW_027257585.1,length=225458&gt;\n",
       "##contig=&lt;ID=NW_027257586.1,length=223689&gt;\n",
       "##contig=&lt;ID=NW_027257587.1,length=224416&gt;\n",
       "##contig=&lt;ID=NW_027257588.1,length=222040&gt;\n",
       "##contig=&lt;ID=NW_027257589.1,length=218605&gt;\n",
       "##contig=&lt;ID=NW_027257590.1,length=218862&gt;\n",
       "##contig=&lt;ID=NW_027257591.1,length=217083&gt;\n",
       "##contig=&lt;ID=NW_027257592.1,length=215076&gt;\n",
       "##contig=&lt;ID=NW_027257593.1,length=214715&gt;\n",
       "##contig=&lt;ID=NW_027257594.1,length=212295&gt;\n",
       "##contig=&lt;ID=NW_027257595.1,length=211986&gt;\n",
       "##contig=&lt;ID=NW_027257596.1,length=209165&gt;\n",
       "##contig=&lt;ID=NW_027257597.1,length=202789&gt;\n",
       "##contig=&lt;ID=NW_027257598.1,length=198783&gt;\n",
       "##contig=&lt;ID=NW_027257599.1,length=194535&gt;\n",
       "##contig=&lt;ID=NW_027257600.1,length=187194&gt;\n",
       "##contig=&lt;ID=NW_027257601.1,length=185624&gt;\n",
       "##contig=&lt;ID=NW_027257602.1,length=188399&gt;\n",
       "##contig=&lt;ID=NW_027257603.1,length=184957&gt;\n",
       "##contig=&lt;ID=NW_027257604.1,length=184592&gt;\n",
       "##contig=&lt;ID=NW_027257605.1,length=180509&gt;\n",
       "##contig=&lt;ID=NW_027257606.1,length=179035&gt;\n",
       "##contig=&lt;ID=NW_027257607.1,length=178920&gt;\n",
       "##contig=&lt;ID=NW_027257608.1,length=176105&gt;\n",
       "##contig=&lt;ID=NW_027257609.1,length=169993&gt;\n",
       "##contig=&lt;ID=NW_027257610.1,length=175694&gt;\n",
       "##contig=&lt;ID=NW_027257611.1,length=165432&gt;\n",
       "##contig=&lt;ID=NW_027257612.1,length=164420&gt;\n",
       "##contig=&lt;ID=NW_027257613.1,length=161815&gt;\n",
       "##contig=&lt;ID=NW_027257614.1,length=158936&gt;\n",
       "##contig=&lt;ID=NW_027257615.1,length=157883&gt;\n",
       "##contig=&lt;ID=NW_027257616.1,length=156487&gt;\n",
       "##contig=&lt;ID=NW_027257617.1,length=155876&gt;\n",
       "##contig=&lt;ID=NW_027257618.1,length=151481&gt;\n",
       "##contig=&lt;ID=NW_027257619.1,length=151379&gt;\n",
       "##contig=&lt;ID=NW_027257620.1,length=148122&gt;\n",
       "##contig=&lt;ID=NW_027257621.1,length=147728&gt;\n",
       "##contig=&lt;ID=NW_027257622.1,length=144464&gt;\n",
       "##contig=&lt;ID=NW_027257623.1,length=136152&gt;\n",
       "##contig=&lt;ID=NW_027257624.1,length=127431&gt;\n",
       "##contig=&lt;ID=NW_027257625.1,length=124623&gt;\n",
       "##contig=&lt;ID=NW_027257626.1,length=121984&gt;\n",
       "##contig=&lt;ID=NW_027257627.1,length=115353&gt;\n",
       "##contig=&lt;ID=NW_027257628.1,length=114386&gt;\n",
       "##contig=&lt;ID=NW_027257629.1,length=110859&gt;\n",
       "##contig=&lt;ID=NW_027257630.1,length=110163&gt;\n",
       "##contig=&lt;ID=NW_027257631.1,length=106586&gt;\n",
       "##contig=&lt;ID=NW_027257632.1,length=106079&gt;\n",
       "##contig=&lt;ID=NW_027257633.1,length=105205&gt;\n",
       "##contig=&lt;ID=NW_027257634.1,length=97527&gt;\n",
       "##contig=&lt;ID=NW_027257635.1,length=96794&gt;\n",
       "##contig=&lt;ID=NW_027257636.1,length=92160&gt;\n",
       "##contig=&lt;ID=NW_027257637.1,length=85820&gt;\n",
       "##contig=&lt;ID=NW_027257638.1,length=83081&gt;\n",
       "##contig=&lt;ID=NW_027257639.1,length=81438&gt;\n",
       "##contig=&lt;ID=NW_027257640.1,length=76094&gt;\n",
       "##contig=&lt;ID=NW_027257641.1,length=69137&gt;\n",
       "##contig=&lt;ID=NW_027257642.1,length=62672&gt;\n",
       "##contig=&lt;ID=NW_027257643.1,length=58718&gt;\n",
       "##contig=&lt;ID=NW_027257644.1,length=57207&gt;\n",
       "##contig=&lt;ID=NW_027257645.1,length=56546&gt;\n",
       "##contig=&lt;ID=NW_027257646.1,length=52497&gt;\n",
       "##contig=&lt;ID=NW_027257647.1,length=51699&gt;\n",
       "##contig=&lt;ID=NW_027257648.1,length=50299&gt;\n",
       "##contig=&lt;ID=NW_027257649.1,length=44552&gt;\n",
       "##contig=&lt;ID=NW_027257650.1,length=43141&gt;\n",
       "##contig=&lt;ID=NW_027257651.1,length=43010&gt;\n",
       "##contig=&lt;ID=NW_027257652.1,length=41382&gt;\n",
       "##contig=&lt;ID=NW_027257653.1,length=40512&gt;\n",
       "##contig=&lt;ID=NW_027257654.1,length=38623&gt;\n",
       "##contig=&lt;ID=NW_027257655.1,length=36682&gt;\n",
       "##contig=&lt;ID=NW_027257656.1,length=35863&gt;\n",
       "##contig=&lt;ID=NW_027257657.1,length=34600&gt;\n",
       "##contig=&lt;ID=NW_027257658.1,length=33566&gt;\n",
       "##contig=&lt;ID=NW_027257659.1,length=31602&gt;\n",
       "##contig=&lt;ID=NW_027257660.1,length=30760&gt;\n",
       "##contig=&lt;ID=NW_027257661.1,length=29903&gt;\n",
       "##contig=&lt;ID=NW_027257662.1,length=29500&gt;\n",
       "##contig=&lt;ID=NW_027257663.1,length=28780&gt;\n",
       "##contig=&lt;ID=NW_027257664.1,length=28181&gt;\n",
       "##contig=&lt;ID=NW_027257665.1,length=26663&gt;\n",
       "##contig=&lt;ID=NW_027257666.1,length=25832&gt;\n",
       "##contig=&lt;ID=NW_027257667.1,length=24172&gt;\n",
       "##contig=&lt;ID=NW_027257668.1,length=22899&gt;\n",
       "##contig=&lt;ID=NW_027257669.1,length=21396&gt;\n",
       "##contig=&lt;ID=NW_027257670.1,length=19233&gt;\n",
       "##contig=&lt;ID=NW_027257671.1,length=16878&gt;\n",
       "##contig=&lt;ID=NW_027257672.1,length=723698&gt;\n",
       "##contig=&lt;ID=NW_027257673.1,length=1116256&gt;\n",
       "##contig=&lt;ID=NW_027257674.1,length=974037&gt;\n",
       "##contig=&lt;ID=NW_027257675.1,length=749393&gt;\n",
       "##contig=&lt;ID=NW_027257676.1,length=689987&gt;\n",
       "##contig=&lt;ID=NW_027257677.1,length=160670&gt;\n",
       "##contig=&lt;ID=NW_027257678.1,length=572209&gt;\n",
       "##contig=&lt;ID=NW_027257679.1,length=149469&gt;\n",
       "##contig=&lt;ID=NW_027257680.1,length=128679&gt;\n",
       "##contig=&lt;ID=NW_027257681.1,length=525238&gt;\n",
       "##contig=&lt;ID=NW_027257682.1,length=125976&gt;\n",
       "##contig=&lt;ID=NW_027257683.1,length=117793&gt;\n",
       "##contig=&lt;ID=NW_027257684.1,length=113170&gt;\n",
       "##contig=&lt;ID=NW_027257685.1,length=110263&gt;\n",
       "##contig=&lt;ID=NW_027257686.1,length=513081&gt;\n",
       "##contig=&lt;ID=NW_027257687.1,length=106844&gt;\n",
       "##contig=&lt;ID=NW_027257688.1,length=105653&gt;\n",
       "##contig=&lt;ID=NW_027257689.1,length=100742&gt;\n",
       "##contig=&lt;ID=NW_027257690.1,length=92676&gt;\n",
       "##contig=&lt;ID=NW_027257691.1,length=505758&gt;\n",
       "##contig=&lt;ID=NW_027257692.1,length=87674&gt;\n",
       "##contig=&lt;ID=NW_027257693.1,length=87123&gt;\n",
       "##contig=&lt;ID=NW_027257694.1,length=84978&gt;\n",
       "##contig=&lt;ID=NW_027257695.1,length=82084&gt;\n",
       "##contig=&lt;ID=NW_027257696.1,length=80670&gt;\n",
       "##contig=&lt;ID=NW_027257697.1,length=70156&gt;\n",
       "##contig=&lt;ID=NW_027257698.1,length=498504&gt;\n",
       "##contig=&lt;ID=NW_027257699.1,length=69679&gt;\n",
       "##contig=&lt;ID=NW_027257700.1,length=64389&gt;\n",
       "##contig=&lt;ID=NW_027257701.1,length=63978&gt;\n",
       "##contig=&lt;ID=NW_027257702.1,length=62477&gt;\n",
       "##contig=&lt;ID=NW_027257703.1,length=59947&gt;\n",
       "##contig=&lt;ID=NW_027257704.1,length=58906&gt;\n",
       "##contig=&lt;ID=NW_027257705.1,length=57798&gt;\n",
       "##contig=&lt;ID=NW_027257706.1,length=494366&gt;\n",
       "##contig=&lt;ID=NW_027257707.1,length=56960&gt;\n",
       "##contig=&lt;ID=NW_027257708.1,length=56602&gt;\n",
       "##contig=&lt;ID=NW_027257709.1,length=55585&gt;\n",
       "##contig=&lt;ID=NW_027257710.1,length=54159&gt;\n",
       "##contig=&lt;ID=NW_027257711.1,length=53915&gt;\n",
       "##contig=&lt;ID=NW_027257712.1,length=52458&gt;\n",
       "##contig=&lt;ID=NW_027257713.1,length=52329&gt;\n",
       "##contig=&lt;ID=NW_027257714.1,length=484854&gt;\n",
       "##contig=&lt;ID=NW_027257715.1,length=51298&gt;\n",
       "##contig=&lt;ID=NW_027257716.1,length=51255&gt;\n",
       "##contig=&lt;ID=NW_027257717.1,length=50588&gt;\n",
       "##contig=&lt;ID=NW_027257718.1,length=48114&gt;\n",
       "##contig=&lt;ID=NW_027257719.1,length=46996&gt;\n",
       "##contig=&lt;ID=NW_027257720.1,length=46028&gt;\n",
       "##contig=&lt;ID=NW_027257721.1,length=44612&gt;\n",
       "##contig=&lt;ID=NW_027257722.1,length=463209&gt;\n",
       "##contig=&lt;ID=NW_027257723.1,length=44401&gt;\n",
       "##contig=&lt;ID=NW_027257724.1,length=44086&gt;\n",
       "##contig=&lt;ID=NW_027257725.1,length=43089&gt;\n",
       "##contig=&lt;ID=NW_027257726.1,length=42840&gt;\n",
       "##contig=&lt;ID=NW_027257727.1,length=42516&gt;\n",
       "##contig=&lt;ID=NW_027257728.1,length=42472&gt;\n",
       "##contig=&lt;ID=NW_027257729.1,length=42095&gt;\n",
       "##contig=&lt;ID=NW_027257730.1,length=41897&gt;\n",
       "##contig=&lt;ID=NW_027257731.1,length=447793&gt;\n",
       "##contig=&lt;ID=NW_027257732.1,length=41391&gt;\n",
       "##contig=&lt;ID=NW_027257733.1,length=40747&gt;\n",
       "##contig=&lt;ID=NW_027257734.1,length=40714&gt;\n",
       "##contig=&lt;ID=NW_027257735.1,length=40569&gt;\n",
       "##contig=&lt;ID=NW_027257736.1,length=40242&gt;\n",
       "##contig=&lt;ID=NW_027257737.1,length=40226&gt;\n",
       "##contig=&lt;ID=NW_027257738.1,length=39567&gt;\n",
       "##contig=&lt;ID=NW_027257739.1,length=38911&gt;\n",
       "##contig=&lt;ID=NW_027257740.1,length=433435&gt;\n",
       "##contig=&lt;ID=NW_027257741.1,length=37894&gt;\n",
       "##contig=&lt;ID=NW_027257742.1,length=37689&gt;\n",
       "##contig=&lt;ID=NW_027257743.1,length=37649&gt;\n",
       "##contig=&lt;ID=NW_027257744.1,length=37376&gt;\n",
       "##contig=&lt;ID=NW_027257745.1,length=36640&gt;\n",
       "##contig=&lt;ID=NW_027257746.1,length=36585&gt;\n",
       "##contig=&lt;ID=NW_027257747.1,length=36134&gt;\n",
       "##contig=&lt;ID=NW_027257748.1,length=35982&gt;\n",
       "##contig=&lt;ID=NW_027257749.1,length=430909&gt;\n",
       "##contig=&lt;ID=NW_027257750.1,length=35580&gt;\n",
       "##contig=&lt;ID=NW_027257751.1,length=35326&gt;\n",
       "##contig=&lt;ID=NW_027257752.1,length=35017&gt;\n",
       "##contig=&lt;ID=NW_027257753.1,length=34717&gt;\n",
       "##contig=&lt;ID=NW_027257754.1,length=34591&gt;\n",
       "##contig=&lt;ID=NW_027257755.1,length=34501&gt;\n",
       "##contig=&lt;ID=NW_027257756.1,length=34399&gt;\n",
       "##contig=&lt;ID=NW_027257757.1,length=34193&gt;\n",
       "##contig=&lt;ID=NW_027257758.1,length=429448&gt;\n",
       "##contig=&lt;ID=NW_027257759.1,length=32898&gt;\n",
       "##contig=&lt;ID=NW_027257760.1,length=32839&gt;\n",
       "##contig=&lt;ID=NW_027257761.1,length=32618&gt;\n",
       "##contig=&lt;ID=NW_027257762.1,length=31999&gt;\n",
       "##contig=&lt;ID=NW_027257763.1,length=31856&gt;\n",
       "##contig=&lt;ID=NW_027257764.1,length=31712&gt;\n",
       "##contig=&lt;ID=NW_027257765.1,length=31317&gt;\n",
       "##contig=&lt;ID=NW_027257766.1,length=31170&gt;\n",
       "##contig=&lt;ID=NW_027257767.1,length=413645&gt;\n",
       "##contig=&lt;ID=NW_027257768.1,length=31108&gt;\n",
       "##contig=&lt;ID=NW_027257769.1,length=30989&gt;\n",
       "##contig=&lt;ID=NW_027257770.1,length=30989&gt;\n",
       "##contig=&lt;ID=NW_027257771.1,length=30661&gt;\n",
       "##contig=&lt;ID=NW_027257772.1,length=30597&gt;\n",
       "##contig=&lt;ID=NW_027257773.1,length=30576&gt;\n",
       "##contig=&lt;ID=NW_027257774.1,length=30162&gt;\n",
       "##contig=&lt;ID=NW_027257775.1,length=30001&gt;\n",
       "##contig=&lt;ID=NW_027257776.1,length=412353&gt;\n",
       "##contig=&lt;ID=NW_027257777.1,length=29838&gt;\n",
       "##contig=&lt;ID=NW_027257778.1,length=29815&gt;\n",
       "##contig=&lt;ID=NW_027257779.1,length=29677&gt;\n",
       "##contig=&lt;ID=NW_027257780.1,length=29650&gt;\n",
       "##contig=&lt;ID=NW_027257781.1,length=29501&gt;\n",
       "##contig=&lt;ID=NW_027257782.1,length=29343&gt;\n",
       "##contig=&lt;ID=NW_027257783.1,length=29150&gt;\n",
       "##contig=&lt;ID=NW_027257784.1,length=29126&gt;\n",
       "##contig=&lt;ID=NW_027257785.1,length=29116&gt;\n",
       "##contig=&lt;ID=NW_027257786.1,length=403413&gt;\n",
       "##contig=&lt;ID=NW_027257787.1,length=28956&gt;\n",
       "##contig=&lt;ID=NW_027257788.1,length=28615&gt;\n",
       "##contig=&lt;ID=NW_027257789.1,length=28510&gt;\n",
       "##contig=&lt;ID=NW_027257790.1,length=28444&gt;\n",
       "##contig=&lt;ID=NW_027257791.1,length=28329&gt;\n",
       "##contig=&lt;ID=NW_027257792.1,length=28254&gt;\n",
       "##contig=&lt;ID=NW_027257793.1,length=27933&gt;\n",
       "##contig=&lt;ID=NW_027257794.1,length=27763&gt;\n",
       "##contig=&lt;ID=NW_027257795.1,length=396989&gt;\n",
       "##contig=&lt;ID=NW_027257796.1,length=27697&gt;\n",
       "##contig=&lt;ID=NW_027257797.1,length=27235&gt;\n",
       "##contig=&lt;ID=NW_027257798.1,length=27155&gt;\n",
       "##contig=&lt;ID=NW_027257799.1,length=27142&gt;\n",
       "##contig=&lt;ID=NW_027257800.1,length=27053&gt;\n",
       "##contig=&lt;ID=NW_027257801.1,length=26346&gt;\n",
       "##contig=&lt;ID=NW_027257802.1,length=26286&gt;\n",
       "##contig=&lt;ID=NW_027257803.1,length=26185&gt;\n",
       "##contig=&lt;ID=NW_027257804.1,length=26185&gt;\n",
       "##contig=&lt;ID=NW_027257805.1,length=393971&gt;\n",
       "##contig=&lt;ID=NW_027257806.1,length=26046&gt;\n",
       "##contig=&lt;ID=NW_027257807.1,length=25736&gt;\n",
       "##contig=&lt;ID=NW_027257808.1,length=25376&gt;\n",
       "##contig=&lt;ID=NW_027257809.1,length=25180&gt;\n",
       "##contig=&lt;ID=NW_027257810.1,length=24913&gt;\n",
       "##contig=&lt;ID=NW_027257811.1,length=24697&gt;\n",
       "##contig=&lt;ID=NW_027257812.1,length=24566&gt;\n",
       "##contig=&lt;ID=NW_027257813.1,length=24223&gt;\n",
       "##contig=&lt;ID=NW_027257814.1,length=382580&gt;\n",
       "##contig=&lt;ID=NW_027257815.1,length=24016&gt;\n",
       "##contig=&lt;ID=NW_027257816.1,length=23706&gt;\n",
       "##contig=&lt;ID=NW_027257817.1,length=23500&gt;\n",
       "##contig=&lt;ID=NW_027257818.1,length=23265&gt;\n",
       "##contig=&lt;ID=NW_027257819.1,length=23234&gt;\n",
       "##contig=&lt;ID=NW_027257820.1,length=23103&gt;\n",
       "##contig=&lt;ID=NW_027257821.1,length=22920&gt;\n",
       "##contig=&lt;ID=NW_027257822.1,length=22242&gt;\n",
       "##contig=&lt;ID=NW_027257823.1,length=22131&gt;\n",
       "##contig=&lt;ID=NW_027257824.1,length=380176&gt;\n",
       "##contig=&lt;ID=NW_027257825.1,length=22041&gt;\n",
       "##contig=&lt;ID=NW_027257826.1,length=21920&gt;\n",
       "##contig=&lt;ID=NW_027257827.1,length=21770&gt;\n",
       "##contig=&lt;ID=NW_027257828.1,length=21415&gt;\n",
       "##contig=&lt;ID=NW_027257829.1,length=21411&gt;\n",
       "##contig=&lt;ID=NW_027257830.1,length=21229&gt;\n",
       "##contig=&lt;ID=NW_027257831.1,length=20906&gt;\n",
       "##contig=&lt;ID=NW_027257832.1,length=20572&gt;\n",
       "##contig=&lt;ID=NW_027257833.1,length=20449&gt;\n",
       "##contig=&lt;ID=NW_027257834.1,length=374384&gt;\n",
       "##contig=&lt;ID=NW_027257835.1,length=20395&gt;\n",
       "##contig=&lt;ID=NW_027257836.1,length=20022&gt;\n",
       "##contig=&lt;ID=NW_027257837.1,length=19926&gt;\n",
       "##contig=&lt;ID=NW_027257838.1,length=19416&gt;\n",
       "##contig=&lt;ID=NW_027257839.1,length=19287&gt;\n",
       "##contig=&lt;ID=NW_027257840.1,length=19064&gt;\n",
       "##contig=&lt;ID=NW_027257841.1,length=18818&gt;\n",
       "##contig=&lt;ID=NW_027257842.1,length=18617&gt;\n",
       "##contig=&lt;ID=NW_027257843.1,length=18384&gt;\n",
       "##contig=&lt;ID=NW_027257844.1,length=369106&gt;\n",
       "##contig=&lt;ID=NW_027257845.1,length=17836&gt;\n",
       "##contig=&lt;ID=NW_027257846.1,length=17776&gt;\n",
       "##contig=&lt;ID=NW_027257847.1,length=17573&gt;\n",
       "##contig=&lt;ID=NW_027257848.1,length=17356&gt;\n",
       "##contig=&lt;ID=NW_027257849.1,length=17051&gt;\n",
       "##contig=&lt;ID=NW_027257850.1,length=16578&gt;\n",
       "##contig=&lt;ID=NW_027257851.1,length=16424&gt;\n",
       "##contig=&lt;ID=NW_027257852.1,length=15835&gt;\n",
       "##contig=&lt;ID=NW_027257853.1,length=14835&gt;\n",
       "##contig=&lt;ID=NW_027257854.1,length=367272&gt;\n",
       "##contig=&lt;ID=NW_027257855.1,length=14525&gt;\n",
       "##contig=&lt;ID=NW_027257856.1,length=14018&gt;\n",
       "##contig=&lt;ID=NW_027257857.1,length=12860&gt;\n",
       "##contig=&lt;ID=NW_027257858.1,length=12854&gt;\n",
       "##contig=&lt;ID=NW_027257859.1,length=12797&gt;\n",
       "##contig=&lt;ID=NW_027257860.1,length=12482&gt;\n",
       "##contig=&lt;ID=NW_027257861.1,length=364980&gt;\n",
       "##contig=&lt;ID=NW_027257862.1,length=350906&gt;\n",
       "##contig=&lt;ID=NW_027257863.1,length=348713&gt;\n",
       "##contig=&lt;ID=NW_027257864.1,length=344693&gt;\n",
       "##contig=&lt;ID=NW_027257865.1,length=344002&gt;\n",
       "##contig=&lt;ID=NW_027257866.1,length=340795&gt;\n",
       "##contig=&lt;ID=NW_027257867.1,length=340604&gt;\n",
       "##contig=&lt;ID=NW_027257868.1,length=333007&gt;\n",
       "##contig=&lt;ID=NW_027257869.1,length=332876&gt;\n",
       "##contig=&lt;ID=NW_027257870.1,length=327501&gt;\n",
       "##contig=&lt;ID=NW_027257871.1,length=323058&gt;\n",
       "##contig=&lt;ID=NW_027257872.1,length=319914&gt;\n",
       "##contig=&lt;ID=NW_027257873.1,length=319619&gt;\n",
       "##contig=&lt;ID=NW_027257874.1,length=313559&gt;\n",
       "##contig=&lt;ID=NW_027257875.1,length=309471&gt;\n",
       "##contig=&lt;ID=NW_027257876.1,length=281635&gt;\n",
       "##contig=&lt;ID=NW_027257877.1,length=275372&gt;\n",
       "##contig=&lt;ID=NW_027257878.1,length=276282&gt;\n",
       "##contig=&lt;ID=NW_027257879.1,length=275270&gt;\n",
       "##contig=&lt;ID=NW_027257880.1,length=270626&gt;\n",
       "##contig=&lt;ID=NW_027257881.1,length=263712&gt;\n",
       "##contig=&lt;ID=NW_027257882.1,length=269516&gt;\n",
       "##contig=&lt;ID=NW_027257883.1,length=259494&gt;\n",
       "##contig=&lt;ID=NW_027257884.1,length=253501&gt;\n",
       "##contig=&lt;ID=NW_027257885.1,length=253216&gt;\n",
       "##contig=&lt;ID=NW_027257886.1,length=242677&gt;\n",
       "##contig=&lt;ID=NW_027257887.1,length=239217&gt;\n",
       "##contig=&lt;ID=NW_027257888.1,length=237873&gt;\n",
       "##contig=&lt;ID=NW_027257889.1,length=236110&gt;\n",
       "##contig=&lt;ID=NW_027257890.1,length=234054&gt;\n",
       "##source=GenomicsDBImport\n",
       "##bcftools_viewVersion=1.22+htslib-1.22\n",
       "##bcftools_viewCommand=view -S /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/Macaca_maura_ssp_Cov10_LenCov0.9.samples -R /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/pos_bed_cov_based/Macaca_maura_ssp_batch0_fploidy2_mploidy2.bed /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/gVCF/Macaca_maura_ssp_batch_0_fploidy_2_mploidy_2_gt.gvcf.gz; Date=Wed Aug  6 12:02:44 2025\n",
       "##bcftools_viewCommand=view -e FILTER=&#x27;LowQual&#x27;; Date=Wed Aug  6 12:02:44 2025\n",
       "##bcftools_normVersion=1.22+htslib-1.22\n",
       "##bcftools_normCommand=norm -m -any; Date=Wed Aug  6 12:02:44 2025\n",
       "##bcftools_viewCommand=view -v snps; Date=Wed Aug  6 12:02:44 2025\n",
       "##bcftools_filterVersion=1.22+htslib-1.22\n",
       "##bcftools_filterCommand=filter -e &#x27;QD &lt; 2 | FS &gt; 60 | MQ &lt; 40 | SOR &gt; 3 | ReadPosRankSum &lt; -8.0 | MQRankSum &lt; -12.5&#x27;; Date=Wed Aug  6 12:02:44 2025\n",
       "##bcftools_pluginVersion=1.22+htslib-1.22\n",
       "##bcftools_pluginCommand=plugin setGT -- -t q -n . -i &#x27;FMT/DP &lt; 5&#x27;; Date=Wed Aug  6 12:02:45 2025\n",
       "##bcftools_pluginCommand=plugin setGT -- -t q -n . -i &#x27;FMT/GQ &lt; 30&#x27;; Date=Wed Aug  6 12:02:46 2025\n",
       "##bcftools_pluginCommand=plugin setGT -- -t q -n . -i &#x27;FMT/GT=&quot;het&quot; &amp; FMT/AD[*:*] &lt; 3&#x27;; Date=Wed Aug  6 12:02:47 2025\n",
       "##bcftools_pluginCommand=plugin setGT -- -t q -n . -i &#x27;FMT/GT=&quot;het&quot; &amp; sMIN(FMT/AD[GT])/sMAX(FMT/AD[GT]) &lt; 0.3&#x27;; Date=Wed Aug  6 12:02:48 2025\n",
       "##bcftools_pluginCommand=plugin fill-tags -- -t all; Date=Wed Aug  6 12:02:49 2025\n",
       "##bcftools_viewCommand=view -i &#x27;INFO/AC &gt; 0&#x27;; Date=Wed Aug  6 12:02:49 2025\n",
       "##bcftools_viewCommand=view -Ob -o /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch0_fploidy2_mploidy2.bcf; Date=Wed Aug  6 12:02:50 2025\n",
       "##GATKCommandLine=&lt;ID=GenotypeGVCFs,CommandLine=&quot;GenotypeGVCFs --output /scratch/1778096/Macaca_maura_ssp_batch_22_fploidy_2_mploidy_2_gt.gvcf.gz --include-non-variant-sites true --variant gendb:///scratch/1778096/batch_22_fploidy_2_mploidy_2 --intervals /faststorage/project/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/gVCF/geno_beds_and_intervals/Macaca_maura_ssp_batch_22_fploidy_2_mploidy_2_loc_2000000_short_segments.intervals --reference /faststorage/project/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_nemestrina_ssp/ref/GCF_043159975.1_mMacNem.hap1_genomic_LargerThan1000bp.fasta --merge-input-intervals false --input-is-somatic false --tumor-lod-to-emit 3.5 --allele-fraction-error 0.001 --keep-combined-raw-annotations false --use-posteriors-to-calculate-qual false --dont-use-dragstr-priors false --use-new-qual-calculator true --annotate-with-num-discovered-alleles false --heterozygosity 0.001 --indel-heterozygosity 1.25E-4 --heterozygosity-stdev 0.01 --standard-min-confidence-threshold-for-calling 30.0 --max-alternate-alleles 6 --max-genotype-count 1024 --sample-ploidy 2 --num-reference-samples-if-no-call 0 --genotype-assignment-method USE_PLS_TO_ASSIGN --genomicsdb-max-alternate-alleles 50 --call-genotypes false --genomicsdb-use-bcf-codec false --genomicsdb-shared-posixfs-optimizations false --genomicsdb-use-gcs-hdfs-connector false --only-output-calls-starting-in-intervals false --interval-set-rule UNION --interval-padding 0 --interval-exclusion-padding 0 --interval-merging-rule ALL --read-validation-stringency SILENT --seconds-between-progress-updates 10.0 --disable-sequence-dictionary-validation false --create-output-bam-index true --create-output-bam-md5 false --create-output-variant-index true --create-output-variant-md5 false --max-variants-per-shard 0 --lenient false --add-output-sam-program-record true --add-output-vcf-command-line true --cloud-prefetch-buffer 40 --cloud-index-prefetch-buffer -1 --disable-bam-index-caching false --sites-only-vcf-output false --help false --version false --showHidden false --verbosity INFO --QUIET false --use-jdk-deflater false --use-jdk-inflater false --gcs-max-retries 20 --gcs-project-for-requester-pays  --disable-tool-default-read-filters false --disable-tool-default-annotations false --enable-all-annotations false --allow-old-rms-mapping-quality-annotation-data false&quot;,Version=&quot;4.4.0.0&quot;,Date=&quot;August 5, 2025 at 9:14:30 AM CEST&quot;&gt;\n",
       "##GATKCommandLine=&lt;ID=HaplotypeCaller,CommandLine=&quot;HaplotypeCaller --sample-ploidy 2 --native-pair-hmm-threads 8 --emit-ref-confidence BP_RESOLUTION --output /faststorage/project/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/gVCF/PD_0070_batch_22_ploidy_2.gvcf.gz --intervals /faststorage/project/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/gVCF/beds_and_intervals/PD_0070_batch_22_ploidy_2.intervals --input /scratch/1770811/PD_0070_batch_22_ploidy_2_readgroup.bam --reference /faststorage/project/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_nemestrina_ssp/ref/GCF_043159975.1_mMacNem.hap1_genomic_LargerThan1000bp.fasta --use-posteriors-to-calculate-qual false --dont-use-dragstr-priors false --use-new-qual-calculator true --annotate-with-num-discovered-alleles false --heterozygosity 0.001 --indel-heterozygosity 1.25E-4 --heterozygosity-stdev 0.01 --standard-min-confidence-threshold-for-calling 30.0 --max-alternate-alleles 6 --max-genotype-count 1024 --num-reference-samples-if-no-call 0 --genotype-assignment-method USE_PLS_TO_ASSIGN --contamination-fraction-to-filter 0.0 --output-mode EMIT_VARIANTS_ONLY --all-site-pls false --flow-likelihood-parallel-threads 0 --flow-likelihood-optimized-comp false --flow-use-t0-tag false --flow-probability-threshold 0.003 --flow-remove-non-single-base-pair-indels false --flow-remove-one-zero-probs false --flow-quantization-bins 121 --flow-fill-empty-bins-value 0.001 --flow-symmetric-indel-probs false --flow-report-insertion-or-deletion false --flow-disallow-probs-larger-than-call false --flow-lump-probs false --flow-retain-max-n-probs-base-format false --flow-probability-scaling-factor 10 --flow-order-cycle-length 4 --flow-number-of-uncertain-flows-to-clip 0 --flow-nucleotide-of-first-uncertain-flow T --keep-boundary-flows false --gvcf-gq-bands 1 --gvcf-gq-bands 2 --gvcf-gq-bands 3 --gvcf-gq-bands 4 --gvcf-gq-bands 5 --gvcf-gq-bands 6 --gvcf-gq-bands 7 --gvcf-gq-bands 8 --gvcf-gq-bands 9 --gvcf-gq-bands 10 --gvcf-gq-bands 11 --gvcf-gq-bands 12 --gvcf-gq-bands 13 --gvcf-gq-bands 14 --gvcf-gq-bands 15 --gvcf-gq-bands 16 --gvcf-gq-bands 17 --gvcf-gq-bands 18 --gvcf-gq-bands 19 --gvcf-gq-bands 20 --gvcf-gq-bands 21 --gvcf-gq-bands 22 --gvcf-gq-bands 23 --gvcf-gq-bands 24 --gvcf-gq-bands 25 --gvcf-gq-bands 26 --gvcf-gq-bands 27 --gvcf-gq-bands 28 --gvcf-gq-bands 29 --gvcf-gq-bands 30 --gvcf-gq-bands 31 --gvcf-gq-bands 32 --gvcf-gq-bands 33 --gvcf-gq-bands 34 --gvcf-gq-bands 35 --gvcf-gq-bands 36 --gvcf-gq-bands 37 --gvcf-gq-bands 38 --gvcf-gq-bands 39 --gvcf-gq-bands 40 --gvcf-gq-bands 41 --gvcf-gq-bands 42 --gvcf-gq-bands 43 --gvcf-gq-bands 44 --gvcf-gq-bands 45 --gvcf-gq-bands 46 --gvcf-gq-bands 47 --gvcf-gq-bands 48 --gvcf-gq-bands 49 --gvcf-gq-bands 50 --gvcf-gq-bands 51 --gvcf-gq-bands 52 --gvcf-gq-bands 53 --gvcf-gq-bands 54 --gvcf-gq-bands 55 --gvcf-gq-bands 56 --gvcf-gq-bands 57 --gvcf-gq-bands 58 --gvcf-gq-bands 59 --gvcf-gq-bands 60 --gvcf-gq-bands 70 --gvcf-gq-bands 80 --gvcf-gq-bands 90 --gvcf-gq-bands 99 --floor-blocks false --indel-size-to-eliminate-in-ref-model 10 --disable-optimizations false --dragen-mode false --flow-mode NONE --apply-bqd false --apply-frd false --disable-spanning-event-genotyping false --transform-dragen-mapping-quality false --mapping-quality-threshold-for-genotyping 20 --max-effective-depth-adjustment-for-frd 0 --just-determine-active-regions false --dont-genotype false --do-not-run-physical-phasing false --do-not-correct-overlapping-quality false --use-filtered-reads-for-annotations false --use-flow-aligner-for-stepwise-hc-filtering false --adaptive-pruning false --do-not-recover-dangling-branches false --recover-dangling-heads false --kmer-size 10 --kmer-size 25 --dont-increase-kmer-sizes-for-cycles false --allow-non-unique-kmers-in-ref false --num-pruning-samples 1 --min-dangling-branch-length 4 --recover-all-dangling-branches false --max-num-haplotypes-in-population 128 --min-pruning 2 --adaptive-pruning-initial-error-rate 0.001 --pruning-lod-threshold 2.302585092994046 --pruning-seeding-lod-threshold 9.210340371976184 --max-unpruned-variants 100 --linked-de-bruijn-graph false --disable-artificial-haplotype-recovery false --enable-legacy-graph-cycle-detection false --debug-assembly false --debug-graph-transformations false --capture-assembly-failure-bam false --num-matching-bases-in-dangling-end-to-recover -1 --error-correction-log-odds -Infinity --error-correct-reads false --kmer-length-for-read-error-correction 25 --min-observations-for-kmer-to-be-solid 20 --likelihood-calculation-engine PairHMM --base-quality-score-threshold 18 --dragstr-het-hom-ratio 2 --dont-use-dragstr-pair-hmm-scores false --pair-hmm-gap-continuation-penalty 10 --expected-mismatch-rate-for-read-disqualification 0.02 --pair-hmm-implementation FASTEST_AVAILABLE --pcr-indel-model CONSERVATIVE --phred-scaled-global-read-mismapping-rate 45 --disable-symmetric-hmm-normalizing false --disable-cap-base-qualities-to-map-quality false --enable-dynamic-read-disqualification-for-genotyping false --dynamic-read-disqualification-threshold 1.0 --native-pair-hmm-use-double-precision false --flow-hmm-engine-min-indel-adjust 6 --flow-hmm-engine-flat-insertion-penatly 45 --flow-hmm-engine-flat-deletion-penatly 45 --pileup-detection false --pileup-detection-enable-indel-pileup-calling false --num-artificial-haplotypes-to-add-per-allele 5 --artifical-haplotype-filtering-kmer-size 10 --pileup-detection-snp-alt-threshold 0.1 --pileup-detection-indel-alt-threshold 0.5 --pileup-detection-absolute-alt-depth 0.0 --pileup-detection-snp-adjacent-to-assembled-indel-range 5 --pileup-detection-bad-read-tolerance 0.0 --pileup-detection-proper-pair-read-badness true --pileup-detection-edit-distance-read-badness-threshold 0.08 --pileup-detection-chimeric-read-badness true --pileup-detection-template-mean-badness-threshold 0.0 --pileup-detection-template-std-badness-threshold 0.0 --bam-writer-type CALLED_HAPLOTYPES --dont-use-soft-clipped-bases false --override-fragment-softclip-check false --min-base-quality-score 10 --smith-waterman JAVA --max-mnp-distance 0 --force-call-filtered-alleles false --reference-model-deletion-quality 30 --soft-clip-low-quality-ends false --allele-informative-reads-overlap-margin 2 --smith-waterman-dangling-end-match-value 25 --smith-waterman-dangling-end-mismatch-penalty -50 --smith-waterman-dangling-end-gap-open-penalty -110 --smith-waterman-dangling-end-gap-extend-penalty -6 --smith-waterman-haplotype-to-reference-match-value 200 --smith-waterman-haplotype-to-reference-mismatch-penalty -150 --smith-waterman-haplotype-to-reference-gap-open-penalty -260 --smith-waterman-haplotype-to-reference-gap-extend-penalty -11 --smith-waterman-read-to-haplotype-match-value 10 --smith-waterman-read-to-haplotype-mismatch-penalty -15 --smith-waterman-read-to-haplotype-gap-open-penalty -30 --smith-waterman-read-to-haplotype-gap-extend-penalty -5 --flow-assembly-collapse-hmer-size 0 --flow-assembly-collapse-partial-mode false --flow-filter-alleles false --flow-filter-alleles-qual-threshold 30.0 --flow-filter-alleles-sor-threshold 3.0 --flow-filter-lone-alleles false --flow-filter-alleles-debug-graphs false --min-assembly-region-size 50 --max-assembly-region-size 300 --active-probability-threshold 0.002 --max-prob-propagation-distance 50 --force-active false --assembly-region-padding 100 --padding-around-indels 75 --padding-around-snps 20 --padding-around-strs 75 --max-extension-into-assembly-region-padding-legacy 25 --max-reads-per-alignment-start 50 --enable-legacy-assembly-region-trimming false --interval-set-rule UNION --interval-padding 0 --interval-exclusion-padding 0 --interval-merging-rule ALL --read-validation-stringency SILENT --seconds-between-progress-updates 10.0 --disable-sequence-dictionary-validation false --create-output-bam-index true --create-output-bam-md5 false --create-output-variant-index true --create-output-variant-md5 false --max-variants-per-shard 0 --lenient false --add-output-sam-program-record true --add-output-vcf-command-line true --cloud-prefetch-buffer 40 --cloud-index-prefetch-buffer -1 --disable-bam-index-caching false --sites-only-vcf-output false --help false --version false --showHidden false --verbosity INFO --QUIET false --use-jdk-deflater false --use-jdk-inflater false --gcs-max-retries 20 --gcs-project-for-requester-pays  --disable-tool-default-read-filters false --minimum-mapping-quality 20 --disable-tool-default-annotations false --enable-all-annotations false --allow-old-rms-mapping-quality-annotation-data false&quot;,Version=&quot;4.4.0.0&quot;,Date=&quot;August 4, 2025 at 6:21:49 PM CEST&quot;&gt;\n",
       "##bcftools_concatVersion=1.22+htslib-1.22\n",
       "##bcftools_concatCommand=concat /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch0_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch1_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch2_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch3_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch4_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch5_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch6_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch7_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch8_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch9_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch10_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch11_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch12_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch13_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch14_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch15_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch16_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch17_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch18_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch19_fploidy2_mploidy2.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch20_fploidy2_mploidy1.bcf /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_batch22_fploidy2_mploidy2.bcf; Date=Wed Aug  6 13:23:50 2025\n",
       "##bcftools_annotateVersion=1.22+htslib-1.22\n",
       "##bcftools_annotateCommand=annotate -x INFO,^FMT/GT -Ob -o /home/vasili/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_all_chr.bcf; Date=Wed Aug  6 13:23:50 2025\n",
       "##bcftools_viewCommand=view -Ou -r NC_092125.1 /home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024/Macaca_maura_ssp/filteredVCF/bcf_step1/Macaca_maura_ssp_all_chr.sorted.bcf; Date=Wed Aug 27 11:20:01 2025\n",
       "#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\tPD_0070\tSAMN07508142\tSAMN07508143\tSAMN07508144\tSAMN07508145\tSAMN07508146\tSAMN19736186\n",
       "</dd><dt><span>vcf_zarr_version :</span></dt><dd>0.2</dd></dl></div></li></ul></div></div>"
      ],
      "text/plain": [
       "<xarray.Dataset> Size: 150MB\n",
       "Dimensions:               (variants: 2023479, samples: 7, ploidy: 2,\n",
       "                           contigs: 361, filters: 2, region_index_values: 2024,\n",
       "                           region_index_fields: 6, alleles: 2)\n",
       "Dimensions without coordinates: variants, samples, ploidy, contigs, filters,\n",
       "                                region_index_values, region_index_fields,\n",
       "                                alleles\n",
       "Data variables: (12/16)\n",
       "    call_genotype         (variants, samples, ploidy) int8 28MB dask.array<chunksize=(1000, 7, 2), meta=np.ndarray>\n",
       "    call_genotype_mask    (variants, samples, ploidy) bool 28MB dask.array<chunksize=(1000, 7, 2), meta=np.ndarray>\n",
       "    call_genotype_phased  (variants, samples) bool 14MB dask.array<chunksize=(1000, 7), meta=np.ndarray>\n",
       "    contig_id             (contigs) object 3kB dask.array<chunksize=(361,), meta=np.ndarray>\n",
       "    contig_length         (contigs) float64 3kB dask.array<chunksize=(361,), meta=np.ndarray>\n",
       "    filter_id             (filters) object 16B dask.array<chunksize=(2,), meta=np.ndarray>\n",
       "    ...                    ...\n",
       "    variant_filter        (variants, filters) bool 4MB dask.array<chunksize=(1000, 2), meta=np.ndarray>\n",
       "    variant_id            (variants) object 16MB dask.array<chunksize=(1000,), meta=np.ndarray>\n",
       "    variant_id_mask       (variants) bool 2MB dask.array<chunksize=(1000,), meta=np.ndarray>\n",
       "    variant_length        (variants) int16 4MB dask.array<chunksize=(1000,), meta=np.ndarray>\n",
       "    variant_position      (variants) int32 8MB dask.array<chunksize=(1000,), meta=np.ndarray>\n",
       "    variant_quality       (variants) float32 8MB dask.array<chunksize=(1000,), meta=np.ndarray>\n",
       "Attributes:\n",
       "    source:            bio2zarr-0.1.5\n",
       "    vcf_header:        ##fileformat=VCFv4.2\\n##FILTER=<ID=PASS,Description=\"A...\n",
       "    vcf_zarr_version:  0.2"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ds"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "73edcdda",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([0, 0, 0, 2, 5, 2, 0, 0, 0, 0])"
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Count alt hom\n",
    "((ds.call_genotype[:,:,0] == ds.call_genotype[:,:,1]) & (ds.call_genotype[:,:,0] >= 1)).sum(axis=1)[:10].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "f8a626c7",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([2, 2, 2, 0, 0, 0, 6, 3, 3, 5])"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Count ref hom\n",
    "((ds.call_genotype[:,:,0] == ds.call_genotype[:,:,1]) & (ds.call_genotype[:,:,0] == 0)).sum(axis=1)[:10].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "3563d0e0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([2, 2, 2, 2, 5, 2, 6, 3, 3, 5, 5, 3, 0, 0, 0, 1, 2, 0, 2, 6])"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Count all hom\n",
    "((ds.call_genotype[:,:,0] == ds.call_genotype[:,:,1]) & (ds.call_genotype[:,:,0] >= 0)).sum(axis=1)[:20].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "id": "f09c4bcd",
   "metadata": {},
   "outputs": [
    {
     "ename": "NameError",
     "evalue": "name 'Hashable' is not defined",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mNameError\u001b[0m                                 Traceback (most recent call last)",
      "Cell \u001b[0;32mIn[42], line 1\u001b[0m\n\u001b[0;32m----> 1\u001b[0m cohort_allele_count: \u001b[43mHashable\u001b[49m \u001b[38;5;241m=\u001b[39m variables\u001b[38;5;241m.\u001b[39mcohort_allele_count,\n\u001b[1;32m      2\u001b[0m ds \u001b[38;5;241m=\u001b[39m define_variable_if_absent(\n\u001b[1;32m      3\u001b[0m         ds, variables\u001b[38;5;241m.\u001b[39mcohort_allele_count, cohort_allele_count, count_cohort_alleles\n\u001b[1;32m      4\u001b[0m     )\n",
      "\u001b[0;31mNameError\u001b[0m: name 'Hashable' is not defined"
     ]
    }
   ],
   "source": [
    "\n",
    "ds = define_variable_if_absent(\n",
    "        ds, variables.cohort_allele_count, cohort_allele_count, count_cohort_alleles\n",
    "    )\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "b0260bbe",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<function sgkit.utils.define_variable_if_absent(ds: xarray.core.dataset.Dataset, default_variable_name: Hashable, variable_name: Optional[Hashable], func: Callable[[xarray.core.dataset.Dataset], xarray.core.dataset.Dataset], **kwargs) -> xarray.core.dataset.Dataset>"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "define_variable_if_absent"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "858db6ba",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Avahi_laniger_ssp\n",
      "../zarr_data/Avahi_laniger_ssp/HiC_scaffold_2\n",
      "../zarr_data/Avahi_laniger_ssp/HiC_scaffold_1\n",
      "1 Lepilemur_sahamalazensis_ssp\n",
      "Skipping, sample size 1\n",
      "3 Allenopithecus_nigroviridis_ssp\n",
      "../zarr_data/Allenopithecus_nigroviridis_ssp/HiC_scaffold_1\n",
      "../zarr_data/Allenopithecus_nigroviridis_ssp/HiC_scaffold_2\n",
      "3 Alouatta_belzebul_ssp\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000060.1\n",
      "2 Cercopithecus_denti_ssp\n",
      "../zarr_data/Cercopithecus_denti_ssp/CM053398.1\n",
      "../zarr_data/Cercopithecus_denti_ssp/CM053363.1\n",
      "10 Saimiri_ustus_ssp\n",
      "../zarr_data/Saimiri_ustus_ssp/NW_024100917.1\n",
      "../zarr_data/Saimiri_ustus_ssp/NW_024100929.1\n",
      "../zarr_data/Saimiri_ustus_ssp/NW_024100959.1\n",
      "0 Trachypithecus_vetulus_ssp\n",
      "Skipping, sample size 0\n",
      "9 Trachypithecus_francoisi_ssp\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681432.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022680469.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681471.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022680461.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022680471.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022680465.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681437.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681445.1\n",
      "1 Nomascus_gabriellae_ssp\n",
      "Skipping, sample size 1\n",
      "2 Cercopithecus_hamlyni_ssp\n",
      "../zarr_data/Cercopithecus_hamlyni_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_hamlyni_ssp/CM053398.1\n",
      "1 Leontocebus_nigricollis_ssp\n",
      "Skipping, sample size 1\n",
      "4 Chiropotes_albinasus_ssp\n",
      "../zarr_data/Chiropotes_albinasus_ssp/CM052647.1\n",
      "../zarr_data/Chiropotes_albinasus_ssp/CM052624.1\n",
      "8 Macaca_tonkeana_ssp\n",
      "../zarr_data/Macaca_tonkeana_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_tonkeana_ssp/NC_092146.1\n",
      "../zarr_data/Macaca_tonkeana_ssp/NC_092125.1\n",
      "../zarr_data/Macaca_tonkeana_ssp/NW_027257673.1\n",
      "22 Gorilla_beringei_ssp\n",
      "Skipping, sample size 22\n",
      "5 Cebus_albifrons_ssp\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437031.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437119.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437042.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437094.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437043.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437085.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022436941.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437013.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437038.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437131.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437033.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437082.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437044.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437065.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437089.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022436957.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437055.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437104.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437015.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437081.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437116.1\n",
      "1 Mico_humeralifer_ssp\n",
      "Skipping, sample size 1\n",
      "19 Cacajao_calvus_ssp\n",
      "../zarr_data/Cacajao_calvus_ssp/CM052624.1\n",
      "../zarr_data/Cacajao_calvus_ssp/CM052647.1\n",
      "26 Macaca_nemestrina_ssp\n",
      "Skipping, sample size 26\n",
      "1 Macaca_sinica_ssp\n",
      "Skipping, sample size 1\n",
      "4 Aotus_griseimembra_ssp\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988472.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988412.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988496.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988488.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988415.1\n",
      "3 Cercopithecus_neglectus_ssp\n",
      "../zarr_data/Cercopithecus_neglectus_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_neglectus_ssp/CM053398.1\n",
      "1 Pithecia_vanzolinii_ssp\n",
      "Skipping, sample size 1\n",
      "2 Cebus_olivaceus_ssp\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437081.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437015.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437116.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022436957.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437089.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437104.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437055.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437082.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437065.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437044.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437033.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437013.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437131.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437038.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437043.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022436941.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437085.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437119.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437031.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437094.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437042.1\n",
      "1 Semnopithecus_priam_ssp\n",
      "Skipping, sample size 1\n",
      "9 Semnopithecus_entellus_ssp\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022680465.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681445.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681437.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022680469.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681471.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681432.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022680461.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022680471.1\n",
      "3 Hylobates_pileatus_ssp\n",
      "../zarr_data/Hylobates_pileatus_ssp/CM038521.1\n",
      "../zarr_data/Hylobates_pileatus_ssp/CM038516.1\n",
      "2 Galago_moholi_ssp\n",
      "../zarr_data/Galago_moholi_ssp/HiC_scaffold_31\n",
      "../zarr_data/Galago_moholi_ssp/HiC_scaffold_1\n",
      "3 Rhinopithecus_strykeri_ssp\n",
      "../zarr_data/Rhinopithecus_strykeri_ssp/NC_044549.1\n",
      "../zarr_data/Rhinopithecus_strykeri_ssp/NC_044555.1\n",
      "5 Otolemur_garnettii_ssp\n",
      "../zarr_data/Otolemur_garnettii_ssp/HiC_scaffold_1\n",
      "1 Colobus_polykomos_ssp\n",
      "Skipping, sample size 1\n",
      "2 Alouatta_discolor_ssp\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000017.1\n",
      "4 Macaca_leucogenys_ssp\n",
      "../zarr_data/Macaca_leucogenys_ssp/NC_065578.1\n",
      "../zarr_data/Macaca_leucogenys_ssp/NC_065597.1\n",
      "../zarr_data/Macaca_leucogenys_ssp/NC_065598.1\n",
      "../zarr_data/Macaca_leucogenys_ssp/NC_065599.1\n",
      "15 Cercopithecus_mona_ssp\n",
      "../zarr_data/Cercopithecus_mona_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_mona_ssp/CM053398.1\n",
      "1 Ateles_marginatus_ssp\n",
      "Skipping, sample size 1\n",
      "1 Callithrix_kuhlii_ssp\n",
      "Skipping, sample size 1\n",
      "7 Propithecus_diadema_ssp\n",
      "../zarr_data/Propithecus_diadema_ssp/HiC_scaffold_2\n",
      "../zarr_data/Propithecus_diadema_ssp/HiC_scaffold_1\n",
      "4 Rhinopithecus_brelichi_ssp\n",
      "../zarr_data/Rhinopithecus_brelichi_ssp/NC_044549.1\n",
      "../zarr_data/Rhinopithecus_brelichi_ssp/NC_044555.1\n",
      "6 Eulemur_coronatus_ssp\n",
      "../zarr_data/Eulemur_coronatus_ssp/HiC_scaffold_1\n",
      "1 Macaca_brunnescens_ssp\n",
      "Skipping, sample size 1\n",
      "5 Cercopithecus_ascanius_ssp\n",
      "../zarr_data/Cercopithecus_ascanius_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_ascanius_ssp/CM053398.1\n",
      "12 Callithrix_jacchus_ssp\n",
      "../zarr_data/Callithrix_jacchus_ssp/NC_071442.1\n",
      "../zarr_data/Callithrix_jacchus_ssp/NC_071464.1\n",
      "2 Avahi_peyrierasi_ssp\n",
      "../zarr_data/Avahi_peyrierasi_ssp/HiC_scaffold_2\n",
      "../zarr_data/Avahi_peyrierasi_ssp/HiC_scaffold_1\n",
      "1 Tarsius_dentatus_ssp\n",
      "Skipping, sample size 1\n",
      "2 Indri_indri_ssp\n",
      "../zarr_data/Indri_indri_ssp/HiC_scaffold_2\n",
      "../zarr_data/Indri_indri_ssp/HiC_scaffold_1\n",
      "17 Macaca_silenus_ssp\n",
      "../zarr_data/Macaca_silenus_ssp/NW_027257673.1\n",
      "../zarr_data/Macaca_silenus_ssp/NC_092146.1\n",
      "../zarr_data/Macaca_silenus_ssp/NC_092125.1\n",
      "../zarr_data/Macaca_silenus_ssp/NC_092145.1\n",
      "1 Allochrocebus_preussi_ssp\n",
      "Skipping, sample size 1\n",
      "1 Cercopithecus_petaurista_ssp\n",
      "Skipping, sample size 1\n",
      "4 Mico_argentatus_ssp\n",
      "../zarr_data/Mico_argentatus_ssp/NC_071464.1\n",
      "../zarr_data/Mico_argentatus_ssp/NC_071442.1\n",
      "1 Plecturocebus_hoffmannsi_ssp\n",
      "Skipping, sample size 1\n",
      "2 Alouatta_juara_ssp\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000001.1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000034.1\n",
      "1 Hylobates_klossii_ssp\n",
      "Skipping, sample size 1\n",
      "4 Cheracebus_lucifer_ssp\n",
      "../zarr_data/Cheracebus_lucifer_ssp/CM080837.1\n",
      "../zarr_data/Cheracebus_lucifer_ssp/CM080815.1\n",
      "1 Propithecus_deckenii_ssp\n",
      "Skipping, sample size 1\n",
      "2 Cheracebus_regulus_ssp\n",
      "../zarr_data/Cheracebus_regulus_ssp/CM080815.1\n",
      "../zarr_data/Cheracebus_regulus_ssp/CM080837.1\n",
      "2 Alouatta_palliata_ssp\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000040.1\n",
      "1 Colobus_angolensis_ssp\n",
      "Skipping, sample size 1\n",
      "0 Chlorocebus_cynosuros_ssp\n",
      "Skipping, sample size 0\n",
      "4 Hylobates_moloch_ssp\n",
      "../zarr_data/Hylobates_moloch_ssp/CM038516.1\n",
      "../zarr_data/Hylobates_moloch_ssp/CM038521.1\n",
      "18 Lemur_catta_ssp\n",
      "../zarr_data/Lemur_catta_ssp/NC_059155.1\n",
      "../zarr_data/Lemur_catta_ssp/NW_025423732.1\n",
      "../zarr_data/Lemur_catta_ssp/NC_059128.1\n",
      "23 Pongo_abelii_ssp\n",
      "Skipping, sample size 23\n",
      "1 Hapalemur_occidentalis_ssp\n",
      "Skipping, sample size 1\n",
      "1 Allochrocebus_solatus_ssp\n",
      "Skipping, sample size 1\n",
      "2 Cebuella_niveiventris_ssp\n",
      "../zarr_data/Cebuella_niveiventris_ssp/NC_071464.1\n",
      "../zarr_data/Cebuella_niveiventris_ssp/NC_071442.1\n",
      "11 Hoolock_hoolock_ssp\n",
      "../zarr_data/Hoolock_hoolock_ssp/CM104451.1\n",
      "../zarr_data/Hoolock_hoolock_ssp/CM104452.1\n",
      "../zarr_data/Hoolock_hoolock_ssp/CM104434.1\n",
      "2 Chlorocebus_pygerythrus_ssp\n",
      "../zarr_data/Chlorocebus_pygerythrus_ssp/NC_132904.1\n",
      "../zarr_data/Chlorocebus_pygerythrus_ssp/NC_132933.1\n",
      "1 Cercopithecus_roloway_ssp\n",
      "Skipping, sample size 1\n",
      "9 Colobus_guereza_ssp\n",
      "../zarr_data/Colobus_guereza_ssp/CM058074.1\n",
      "../zarr_data/Colobus_guereza_ssp/CM058096.1\n",
      "1 Callithrix_geoffroyi_ssp\n",
      "Skipping, sample size 1\n",
      "3 Trachypithecus_obscurus_ssp\n",
      "../zarr_data/Trachypithecus_obscurus_ssp/NW_022680469.1\n",
      "../zarr_data/Trachypithecus_obscurus_ssp/NW_022681471.1\n",
      "../zarr_data/Trachypithecus_obscurus_ssp/NW_022681432.1\n",
      "../zarr_data/Trachypithecus_obscurus_ssp/NW_022680461.1\n",
      "../zarr_data/Trachypithecus_obscurus_ssp/NW_022680471.1\n",
      "../zarr_data/Trachypithecus_obscurus_ssp/NW_022680465.1\n",
      "../zarr_data/Trachypithecus_obscurus_ssp/NW_022681445.1\n",
      "../zarr_data/Trachypithecus_obscurus_ssp/NW_022681437.1\n",
      "2 Saguinus_imperator_ssp\n",
      "../zarr_data/Saguinus_imperator_ssp/CM063169.1\n",
      "../zarr_data/Saguinus_imperator_ssp/JASSZA010000432.1\n",
      "../zarr_data/Saguinus_imperator_ssp/CM063191.1\n",
      "63 Gorilla_gorilla_ssp\n",
      "Skipping, sample size 63\n",
      "1 Microcebus_tavaratra_ssp\n",
      "Skipping, sample size 1\n",
      "9 Nycticebus_bengalensis_ssp\n",
      "../zarr_data/Nycticebus_bengalensis_ssp/NC_069804.1\n",
      "../zarr_data/Nycticebus_bengalensis_ssp/NC_069805.1\n",
      "../zarr_data/Nycticebus_bengalensis_ssp/NC_069780.1\n",
      "1 Trachypithecus_cristatus_ssp\n",
      "Skipping, sample size 1\n",
      "6 Eulemur_rufifrons_ssp\n",
      "../zarr_data/Eulemur_rufifrons_ssp/NC_090983.1\n",
      "../zarr_data/Eulemur_rufifrons_ssp/NC_091012.1\n",
      "1 Pithecia_albicans_ssp\n",
      "Skipping, sample size 1\n",
      "171 Macaca_mulatta_ssp\n",
      "Skipping, sample size 171\n",
      "2 Cacajao_melanocephalus_ssp\n",
      "../zarr_data/Cacajao_melanocephalus_ssp/CM052624.1\n",
      "../zarr_data/Cacajao_melanocephalus_ssp/CM052647.1\n",
      "1 Trachypithecus_hatinhensis_ssp\n",
      "Skipping, sample size 1\n",
      "1 Galagoides_demidoff_ssp\n",
      "Skipping, sample size 1\n",
      "1 Aotus_trivirgatus_ssp\n",
      "Skipping, sample size 1\n",
      "1 Pithecia_mittermeieri_ssp\n",
      "Skipping, sample size 1\n",
      "2 Hoolock_leuconedys_ssp\n",
      "../zarr_data/Hoolock_leuconedys_ssp/CM104452.1\n",
      "../zarr_data/Hoolock_leuconedys_ssp/CM104451.1\n",
      "../zarr_data/Hoolock_leuconedys_ssp/CM104434.1\n",
      "1 Rhinopithecus_avunculus_ssp\n",
      "Skipping, sample size 1\n",
      "1 Pongo_tapanuliensis_ssp\n",
      "Skipping, sample size 1\n",
      "1 Nomascus_annamensis_ssp\n",
      "Skipping, sample size 1\n",
      "4 Macaca_nigra_ssp\n",
      "../zarr_data/Macaca_nigra_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_nigra_ssp/NC_092146.1\n",
      "../zarr_data/Macaca_nigra_ssp/NC_092125.1\n",
      "../zarr_data/Macaca_nigra_ssp/NW_027257673.1\n",
      "1 Cercopithecus_diana_ssp\n",
      "Skipping, sample size 1\n",
      "4 Nomascus_leucogenys_ssp\n",
      "../zarr_data/Nomascus_leucogenys_ssp/NC_044381.1\n",
      "../zarr_data/Nomascus_leucogenys_ssp/NC_044406.1\n",
      "1 Plecturocebus_grovesi_ssp\n",
      "Skipping, sample size 1\n",
      "4 Hylobates_lar_ssp\n",
      "../zarr_data/Hylobates_lar_ssp/CM038516.1\n",
      "../zarr_data/Hylobates_lar_ssp/CM038521.1\n",
      "106 Papio_anubis_ssp\n",
      "Skipping, sample size 106\n",
      "4 Leontocebus_fuscicollis_ssp\n",
      "../zarr_data/Leontocebus_fuscicollis_ssp/CM063191.1\n",
      "../zarr_data/Leontocebus_fuscicollis_ssp/CM063169.1\n",
      "2 Macaca_nigrescens_ssp\n",
      "../zarr_data/Macaca_nigrescens_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_nigrescens_ssp/NC_092146.1\n",
      "../zarr_data/Macaca_nigrescens_ssp/NC_092125.1\n",
      "../zarr_data/Macaca_nigrescens_ssp/NW_027257673.1\n",
      "1 Trachypithecus_germaini_ssp\n",
      "Skipping, sample size 1\n",
      "3 Ateles_geoffroyi_ssp\n",
      "../zarr_data/Ateles_geoffroyi_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Ateles_geoffroyi_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Ateles_geoffroyi_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Ateles_geoffroyi_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Ateles_geoffroyi_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Ateles_geoffroyi_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Ateles_geoffroyi_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Ateles_geoffroyi_ssp/CAJZLT010000017.1\n",
      "1 Cephalopachus_bancanus_ssp\n",
      "Skipping, sample size 1\n",
      "8 Cheracebus_lugens_ssp\n",
      "../zarr_data/Cheracebus_lugens_ssp/CM080837.1\n",
      "../zarr_data/Cheracebus_lugens_ssp/CM080815.1\n",
      "5 Cercopithecus_mitis_ssp\n",
      "../zarr_data/Cercopithecus_mitis_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_mitis_ssp/CM053398.1\n",
      "1 Nomascus_concolor_ssp\n",
      "Skipping, sample size 1\n",
      "2 Plecturocebus_cinerascens_ssp\n",
      "../zarr_data/Plecturocebus_cinerascens_ssp/CM080815.1\n",
      "../zarr_data/Plecturocebus_cinerascens_ssp/CM080837.1\n",
      "2 Mandrillus_sphinx_ssp\n",
      "../zarr_data/Mandrillus_sphinx_ssp/HiC_scaffold_111\n",
      "../zarr_data/Mandrillus_sphinx_ssp/HiC_scaffold_1\n",
      "../zarr_data/Mandrillus_sphinx_ssp/HiC_scaffold_21\n",
      "0 Mico_humilis_ssp\n",
      "Skipping, sample size 0\n",
      "4 Cebus_unicolor_ssp\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437015.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437081.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437116.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437082.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437065.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437044.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022436957.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437089.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437055.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437104.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437013.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437038.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437131.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437033.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437031.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437119.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437094.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437042.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437043.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022437085.1\n",
      "../zarr_data/Cebus_unicolor_ssp/NW_022436941.1\n",
      "7 Eulemur_mongoz_ssp\n",
      "../zarr_data/Eulemur_mongoz_ssp/CM052853.1\n",
      "7 Mandrillus_leucophaeus_ssp\n",
      "../zarr_data/Mandrillus_leucophaeus_ssp/HiC_scaffold_1\n",
      "../zarr_data/Mandrillus_leucophaeus_ssp/HiC_scaffold_21\n",
      "../zarr_data/Mandrillus_leucophaeus_ssp/HiC_scaffold_111\n",
      "10 Pithecia_pithecia_ssp\n",
      "../zarr_data/Pithecia_pithecia_ssp/CM052624.1\n",
      "../zarr_data/Pithecia_pithecia_ssp/CM052647.1\n",
      "28 Papio_kindae_ssp\n",
      "Skipping, sample size 28\n",
      "1 Cheracebus_torquatus_ssp\n",
      "Skipping, sample size 1\n",
      "2 Lepilemur_septentrionalis_ssp\n",
      "../zarr_data/Lepilemur_septentrionalis_ssp/NC_033660.1\n",
      "../zarr_data/Lepilemur_septentrionalis_ssp/NC_033692.1\n",
      "1 Saguinus_bicolor_ssp\n",
      "Skipping, sample size 1\n",
      "6 Mirza_zaza_ssp\n",
      "../zarr_data/Mirza_zaza_ssp/NC_033660.1\n",
      "../zarr_data/Mirza_zaza_ssp/NC_033692.1\n",
      "67 Macaca_fuscata_ssp\n",
      "Skipping, sample size 67\n",
      "143 Pan_troglodytes_ssp\n",
      "Skipping, sample size 143\n",
      "14 Papio_papio_ssp\n",
      "../zarr_data/Papio_papio_ssp/NC_044997.1\n",
      "../zarr_data/Papio_papio_ssp/NC_044996.1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Papio_papio_ssp/NC_044976.1\n",
      "3 Galago_senegalensis_ssp\n",
      "../zarr_data/Galago_senegalensis_ssp/HiC_scaffold_1\n",
      "../zarr_data/Galago_senegalensis_ssp/HiC_scaffold_31\n",
      "2 Ateles_paniscus_ssp\n",
      "../zarr_data/Ateles_paniscus_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Ateles_paniscus_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Ateles_paniscus_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Ateles_paniscus_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Ateles_paniscus_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Ateles_paniscus_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Ateles_paniscus_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Ateles_paniscus_ssp/CAJZLT010000017.1\n",
      "5 Varecia_variegata_ssp\n",
      "../zarr_data/Varecia_variegata_ssp/CM052457.1\n",
      "../zarr_data/Varecia_variegata_ssp/CM052436.1\n",
      "1 Cercocebus_lunulatus_ssp\n",
      "Skipping, sample size 1\n",
      "1 Macaca_siberu_ssp\n",
      "Skipping, sample size 1\n",
      "2 Xanthonycticebus_pygmaeus_ssp\n",
      "../zarr_data/Xanthonycticebus_pygmaeus_ssp/NC_069780.1\n",
      "../zarr_data/Xanthonycticebus_pygmaeus_ssp/NC_069804.1\n",
      "../zarr_data/Xanthonycticebus_pygmaeus_ssp/NC_069805.1\n",
      "1 Avahi_occidentalis_ssp\n",
      "Skipping, sample size 1\n",
      "1 Microcebus_mittermeieri_ssp\n",
      "Skipping, sample size 1\n",
      "1 Propithecus_edwardsi_ssp\n",
      "Skipping, sample size 1\n",
      "7 Hapalemur_griseus_ssp\n",
      "../zarr_data/Hapalemur_griseus_ssp/NC_059128.1\n",
      "../zarr_data/Hapalemur_griseus_ssp/NC_059155.1\n",
      "2 Macaca_sylvanus_ssp\n",
      "../zarr_data/Macaca_sylvanus_ssp/NC_041774.1\n",
      "../zarr_data/Macaca_sylvanus_ssp/NC_041754.1\n",
      "../zarr_data/Macaca_sylvanus_ssp/NW_021161850.1\n",
      "1 Pithecia_chrysocephala_ssp\n",
      "Skipping, sample size 1\n",
      "31 Aotus_nancymaae_ssp\n",
      "Skipping, sample size 31\n",
      "5 Macaca_assamensis_ssp\n",
      "../zarr_data/Macaca_assamensis_ssp/NC_065599.1\n",
      "../zarr_data/Macaca_assamensis_ssp/NC_065598.1\n",
      "../zarr_data/Macaca_assamensis_ssp/NC_065597.1\n",
      "../zarr_data/Macaca_assamensis_ssp/NC_065578.1\n",
      "1 Plecturocebus_donacophilus_ssp\n",
      "Skipping, sample size 1\n",
      "3 Plecturocebus_bernhardi_ssp\n",
      "../zarr_data/Plecturocebus_bernhardi_ssp/CM080837.1\n",
      "../zarr_data/Plecturocebus_bernhardi_ssp/CM080815.1\n",
      "1 Trachypithecus_ebenus_ssp\n",
      "Skipping, sample size 1\n",
      "7 Macaca_maura_ssp\n",
      "../zarr_data/Macaca_maura_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_maura_ssp/NW_027257673.1\n",
      "../zarr_data/Macaca_maura_ssp/NC_092146.1\n",
      "../zarr_data/Macaca_maura_ssp/NC_092125.1\n",
      "5 Alouatta_macconnelli_ssp\n",
      "../zarr_data/Alouatta_macconnelli_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_macconnelli_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_macconnelli_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_macconnelli_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_macconnelli_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_macconnelli_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_macconnelli_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Alouatta_macconnelli_ssp/CAJZLT010000017.1\n",
      "1 Eulemur_cinereiceps_ssp\n",
      "Skipping, sample size 1\n",
      "11 Ateles_chamek_ssp\n",
      "../zarr_data/Ateles_chamek_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Ateles_chamek_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Ateles_chamek_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Ateles_chamek_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Ateles_chamek_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Ateles_chamek_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Ateles_chamek_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Ateles_chamek_ssp/CAJZLT010000017.1\n",
      "1 Plecturocebus_moloch_ssp\n",
      "Skipping, sample size 1\n",
      "6 Eulemur_macaco_ssp\n",
      "../zarr_data/Eulemur_macaco_ssp/HiC_scaffold_1\n",
      "2 Pithecia_hirsuta_ssp\n",
      "../zarr_data/Pithecia_hirsuta_ssp/CM052647.1\n",
      "../zarr_data/Pithecia_hirsuta_ssp/CM052624.1\n",
      "1 Hylobates_abbotti_ssp\n",
      "Skipping, sample size 1\n",
      "2 Cacajao_hosomi_ssp\n",
      "../zarr_data/Cacajao_hosomi_ssp/CM052624.1\n",
      "../zarr_data/Cacajao_hosomi_ssp/CM052647.1\n",
      "1 Saguinus_geoffroyi_ssp\n",
      "Skipping, sample size 1\n",
      "3 Pithecia_pissinatti_ssp\n",
      "Skipping, no completed zarrs\n",
      "3 Nasalis_larvatus_ssp\n",
      "../zarr_data/Nasalis_larvatus_ssp/CM106345.1\n",
      "../zarr_data/Nasalis_larvatus_ssp/CM106344.1\n",
      "2 Propithecus_perrieri_ssp\n",
      "../zarr_data/Propithecus_perrieri_ssp/HiC_scaffold_1\n",
      "../zarr_data/Propithecus_perrieri_ssp/HiC_scaffold_2\n",
      "13 Macaca_radiata_ssp\n",
      "../zarr_data/Macaca_radiata_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_radiata_ssp/NW_027257673.1\n",
      "../zarr_data/Macaca_radiata_ssp/NC_092146.1\n",
      "../zarr_data/Macaca_radiata_ssp/NC_092125.1\n",
      "1 Lepilemur_mustelinus_ssp\n",
      "Skipping, sample size 1\n",
      "8 Prolemur_simus_ssp\n",
      "../zarr_data/Prolemur_simus_ssp/NC_059128.1\n",
      "../zarr_data/Prolemur_simus_ssp/NC_059155.1\n",
      "1 Saguinus_oedipus_ssp\n",
      "Skipping, sample size 1\n",
      "1 Piliocolobus_gordonorum_ssp\n",
      "Skipping, sample size 1\n",
      "2 Alouatta_caraya_ssp\n",
      "../zarr_data/Alouatta_caraya_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_caraya_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_caraya_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_caraya_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_caraya_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Alouatta_caraya_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_caraya_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_caraya_ssp/CAJZLT010000034.1\n",
      "1 Lagothrix_lagotricha_ssp\n",
      "Skipping, sample size 1\n",
      "3 Leontopithecus_rosalia_ssp\n",
      "../zarr_data/Leontopithecus_rosalia_ssp/HiC_scaffold_23\n",
      "../zarr_data/Leontopithecus_rosalia_ssp/HiC_scaffold_1\n",
      "22 Daubentonia_madagascariensis_ssp\n",
      "Skipping, sample size 22\n",
      "8 Propithecus_tattersalli_ssp\n",
      "../zarr_data/Propithecus_tattersalli_ssp/HiC_scaffold_1\n",
      "../zarr_data/Propithecus_tattersalli_ssp/HiC_scaffold_2\n",
      "2 Hylobates_agilis_ssp\n",
      "../zarr_data/Hylobates_agilis_ssp/CM038521.1\n",
      "../zarr_data/Hylobates_agilis_ssp/CM038516.1\n",
      "3 Cacajao_ayresi_ssp\n",
      "../zarr_data/Cacajao_ayresi_ssp/CM052647.1\n",
      "../zarr_data/Cacajao_ayresi_ssp/CM052624.1\n",
      "2 Lepilemur_ruficaudatus_ssp\n",
      "../zarr_data/Lepilemur_ruficaudatus_ssp/NC_033660.1\n",
      "../zarr_data/Lepilemur_ruficaudatus_ssp/NC_033692.1\n",
      "10 Saimiri_boliviensis_ssp\n",
      "../zarr_data/Saimiri_boliviensis_ssp/NW_024100917.1\n",
      "../zarr_data/Saimiri_boliviensis_ssp/NW_024100929.1\n",
      "../zarr_data/Saimiri_boliviensis_ssp/NW_024100959.1\n",
      "5 Macaca_hecki_ssp\n",
      "../zarr_data/Macaca_hecki_ssp/NW_027257673.1\n",
      "../zarr_data/Macaca_hecki_ssp/NC_092125.1\n",
      "../zarr_data/Macaca_hecki_ssp/NC_092145.1\n",
      "3 Macaca_thibetana_ssp\n",
      "../zarr_data/Macaca_thibetana_ssp/NC_065598.1\n",
      "../zarr_data/Macaca_thibetana_ssp/NC_065578.1\n",
      "../zarr_data/Macaca_thibetana_ssp/NC_065597.1\n",
      "4 Varecia_rubra_ssp\n",
      "../zarr_data/Varecia_rubra_ssp/CM052436.1\n",
      "../zarr_data/Varecia_rubra_ssp/CM052457.1\n",
      "3 Cercopithecus_nictitans_ssp\n",
      "../zarr_data/Cercopithecus_nictitans_ssp/CM053398.1\n",
      "../zarr_data/Cercopithecus_nictitans_ssp/CM053363.1\n",
      "45 Theropithecus_gelada_ssp\n",
      "Skipping, sample size 45\n",
      "1 Microcebus_jonahi_ssp\n",
      "Skipping, sample size 1\n",
      "4 Semnopithecus_hypoleucos_ssp\n",
      "../zarr_data/Semnopithecus_hypoleucos_ssp/NW_022680471.1\n",
      "../zarr_data/Semnopithecus_hypoleucos_ssp/NW_022680461.1\n",
      "../zarr_data/Semnopithecus_hypoleucos_ssp/NW_022681471.1\n",
      "../zarr_data/Semnopithecus_hypoleucos_ssp/NW_022680469.1\n",
      "../zarr_data/Semnopithecus_hypoleucos_ssp/NW_022681432.1\n",
      "../zarr_data/Semnopithecus_hypoleucos_ssp/NW_022681445.1\n",
      "../zarr_data/Semnopithecus_hypoleucos_ssp/NW_022681437.1\n",
      "../zarr_data/Semnopithecus_hypoleucos_ssp/NW_022680465.1\n",
      "1 Brachyteles_hypoxanthus_ssp\n",
      "Skipping, sample size 1\n",
      "4 Chiropotes_sagulatus_ssp\n",
      "../zarr_data/Chiropotes_sagulatus_ssp/CM052624.1\n",
      "../zarr_data/Chiropotes_sagulatus_ssp/CM052647.1\n",
      "14 Aotus_azarai_ssp\n",
      "../zarr_data/Aotus_azarai_ssp/NW_026988412.1\n",
      "../zarr_data/Aotus_azarai_ssp/NW_026988496.1\n",
      "../zarr_data/Aotus_azarai_ssp/NW_026988472.1\n",
      "../zarr_data/Aotus_azarai_ssp/NW_026988415.1\n",
      "../zarr_data/Aotus_azarai_ssp/NW_026988488.1\n",
      "1 Saguinus_inustus_ssp\n",
      "Skipping, sample size 1\n",
      "2 Eulemur_fulvus_ssp\n",
      "../zarr_data/Eulemur_fulvus_ssp/NC_091012.1\n",
      "../zarr_data/Eulemur_fulvus_ssp/NC_090983.1\n",
      "2 Hapalemur_meridionalis_ssp\n",
      "../zarr_data/Hapalemur_meridionalis_ssp/NC_059155.1\n",
      "../zarr_data/Hapalemur_meridionalis_ssp/NC_059128.1\n",
      "0 Chlorocebus_tantalus_ssp\n",
      "Skipping, sample size 0\n",
      "1 Cercocebus_chrysogaster_ssp\n",
      "Skipping, sample size 1\n",
      "1 Hylobates_muelleri_ssp\n",
      "Skipping, sample size 1\n",
      "3 Allochrocebus_lhoesti_ssp\n",
      "../zarr_data/Allochrocebus_lhoesti_ssp/NC_132933.1\n",
      "../zarr_data/Allochrocebus_lhoesti_ssp/NC_132904.1\n",
      "3 Plecturocebus_dubius_ssp\n",
      "../zarr_data/Plecturocebus_dubius_ssp/CM080815.1\n",
      "../zarr_data/Plecturocebus_dubius_ssp/CM080837.1\n",
      "1 Miopithecus_talapoin_ssp\n",
      "Skipping, sample size 1\n",
      "2 Cercocebus_torquatus_ssp\n",
      "../zarr_data/Cercocebus_torquatus_ssp/HiC_scaffold_1\n",
      "../zarr_data/Cercocebus_torquatus_ssp/HiC_scaffold_111\n",
      "../zarr_data/Cercocebus_torquatus_ssp/HiC_scaffold_21\n",
      "7 Macaca_leonina_ssp\n",
      "../zarr_data/Macaca_leonina_ssp/NW_027257673.1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Macaca_leonina_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_leonina_ssp/NC_092146.1\n",
      "../zarr_data/Macaca_leonina_ssp/NC_092125.1\n",
      "7 Pygathrix_nemaeus_ssp\n",
      "../zarr_data/Pygathrix_nemaeus_ssp/NC_044549.1\n",
      "../zarr_data/Pygathrix_nemaeus_ssp/NC_044555.1\n",
      "3 Otolemur_crassicaudatus_ssp\n",
      "../zarr_data/Otolemur_crassicaudatus_ssp/HiC_scaffold_1\n",
      "3 Cercopithecus_cephus_ssp\n",
      "../zarr_data/Cercopithecus_cephus_ssp/CM053398.1\n",
      "../zarr_data/Cercopithecus_cephus_ssp/CM053363.1\n",
      "2 Loris_lydekkerianus_ssp\n",
      "../zarr_data/Loris_lydekkerianus_ssp/NC_069805.1\n",
      "../zarr_data/Loris_lydekkerianus_ssp/NC_069804.1\n",
      "../zarr_data/Loris_lydekkerianus_ssp/NC_069780.1\n",
      "80 Papio_cynocephalus_ssp\n",
      "Skipping, sample size 80\n",
      "1 Carlito_syrichta_ssp\n",
      "Skipping, sample size 1\n",
      "21 Alouatta_seniculus_ssp\n",
      "Skipping, sample size 21\n",
      "13 Cercocebus_atys_ssp\n",
      "../zarr_data/Cercocebus_atys_ssp/HiC_scaffold_21\n",
      "../zarr_data/Cercocebus_atys_ssp/HiC_scaffold_111\n",
      "../zarr_data/Cercocebus_atys_ssp/HiC_scaffold_1\n",
      "17 Pan_paniscus_ssp\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073267.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_085926.1\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073256.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073257.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_085927.1\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073266.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073264.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073255.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073254.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073265.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073261.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073271.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073270.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073260.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073259.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073262.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073272.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073253.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073268.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073249.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073269.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073252.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073273.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073263.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073258.2\n",
      "1 Presbytis_comata_ssp\n",
      "Skipping, sample size 1\n",
      "4 Eulemur_sanfordi_ssp\n",
      "../zarr_data/Eulemur_sanfordi_ssp/NC_090983.1\n",
      "../zarr_data/Eulemur_sanfordi_ssp/NC_091012.1\n",
      "2 Saimiri_sciureus_ssp\n",
      "../zarr_data/Saimiri_sciureus_ssp/NW_024100917.1\n",
      "../zarr_data/Saimiri_sciureus_ssp/NW_024100959.1\n",
      "../zarr_data/Saimiri_sciureus_ssp/NW_024100929.1\n",
      "4 Saguinus_midas_ssp\n",
      "../zarr_data/Saguinus_midas_ssp/CM063191.1\n",
      "../zarr_data/Saguinus_midas_ssp/JASSZA010000432.1\n",
      "../zarr_data/Saguinus_midas_ssp/CM063169.1\n",
      "1 Piliocolobus_kirkii_ssp\n",
      "Skipping, sample size 1\n",
      "2 Propithecus_verreauxi_ssp\n",
      "../zarr_data/Propithecus_verreauxi_ssp/HiC_scaffold_1\n",
      "../zarr_data/Propithecus_verreauxi_ssp/HiC_scaffold_2\n",
      "1 Nomascus_siki_ssp\n",
      "Skipping, sample size 1\n",
      "4 Trachypithecus_phayrei_ssp\n",
      "../zarr_data/Trachypithecus_phayrei_ssp/NW_022680465.1\n",
      "../zarr_data/Trachypithecus_phayrei_ssp/NW_022681437.1\n",
      "../zarr_data/Trachypithecus_phayrei_ssp/NW_022681445.1\n",
      "../zarr_data/Trachypithecus_phayrei_ssp/NW_022681432.1\n",
      "../zarr_data/Trachypithecus_phayrei_ssp/NW_022680469.1\n",
      "../zarr_data/Trachypithecus_phayrei_ssp/NW_022681471.1\n",
      "../zarr_data/Trachypithecus_phayrei_ssp/NW_022680461.1\n",
      "../zarr_data/Trachypithecus_phayrei_ssp/NW_022680471.1\n",
      "10 Propithecus_coquereli_ssp\n",
      "../zarr_data/Propithecus_coquereli_ssp/HiC_scaffold_2\n",
      "../zarr_data/Propithecus_coquereli_ssp/HiC_scaffold_1\n"
     ]
    }
   ],
   "source": [
    "# Fst implementation.\n",
    "for x in glob.glob(zarr_path+\"*\")[25:]:\n",
    "    # The name used to load all the files, short and long version\n",
    "    short_form = x.split(\"/\")[-1].split(\"_\")[0]\n",
    "    long_form = x.split(\"/\")[-1]\n",
    "    # Loading the various metadata files. Metadata, contig information, callability bed.\n",
    "    metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "    metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "    metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "    regions_df[\"chr_type\"] = [\"ChrX\" if x == 2 and y == 1 else \"ChrY\" if x == 0 and y == 1 else \"Aut\" for x, y in zip(regions_df.FEMALE_PLOIDY, regions_df.MALE_PLOIDY)]\n",
    "    large_contigs = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "    large_x = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID\n",
    "    # Skipping the large samples sizes and the singulars for the first Fst calc\n",
    "    # Also perform a 10X filter to get accurate samples sizes\n",
    "    sample_size = len(metadata_df.loc[(metadata_df.GVCF_FOLDER == long_form) &\n",
    "                                     (metadata_df.AVG_COVERAGE_A >= 10)])\n",
    "    print(sample_size, long_form)\n",
    "    if (sample_size > 20) or (sample_size <= 1):\n",
    "        print(\"Skipping, sample size\", sample_size)\n",
    "        continue\n",
    "    if os.path.exists(\"../results/window_stats/{}_Fst.txt\".format(long_form)):\n",
    "        print(\"Skipping as it's processed\")\n",
    "        continue\n",
    "    if len(glob.glob(x+\"/*\")) == 0:\n",
    "        print(\"Skipping, no completed zarrs\")\n",
    "        continue\n",
    "    # Loading the genetic data.\n",
    "    df_l = []\n",
    "    for c in glob.glob(x+\"/*\"):\n",
    "        print(c)\n",
    "        ds = sg.load_dataset(c)\n",
    "        # This implementation is the pi implementation.\n",
    "        # Probably problematic in some cases with population structure, but it is easier to implement\n",
    "        ds[\"sample_cohort\"] = ds[\"samples\"]\n",
    "        # Subsetting and windowing the sgkit dataset. The rechunking handles what otherwise would cause an error.\n",
    "        #ds[\"call_genotype\"] = ds[\"call_genotype\"].clip(0)\n",
    "        ds = ds.sel(contigs=[ds.variant_contig[0].values])\n",
    "        if c.split(\"/\")[-1] in list(large_x):\n",
    "            ds = haploid_double(ds, \"call_genotype\", \"samples\")\n",
    "        missing_rate = ds.call_genotype_mask[:,:,0].sum(axis=1).values/ds.call_genotype_mask[:,:,0].count(axis=1).values\n",
    "        ds = ds.isel(variants=(missing_rate <= 0))\n",
    "        ds = sg.window_by_genome(ds)\n",
    "        ds = (sg.Fst(ds.chunk({\"variants\": 10000})))\n",
    "        df_sub = pd.DataFrame(ds.stat_Fst[0,:,], columns=ds.sample_id)\n",
    "        df_sub[\"chrom\"] = c.split(\"/\")[-1]\n",
    "        df_sub[\"variants_used\"] = len(ds.variants)\n",
    "        df_sub[\"stat\"] = \"Fst\"\n",
    "        df_l.append(df_sub)\n",
    "        df_sub = pd.DataFrame(ds.stat_divergence[0,:,], columns=ds.sample_id)\n",
    "        df_sub[\"chrom\"] = c.split(\"/\")[-1]\n",
    "        df_sub[\"variants_used\"] = len(ds.variants)\n",
    "        df_sub[\"stat\"] = \"Divergence\"\n",
    "        df_l.append(df_sub)\n",
    "    output_df = pd.concat(df_l)\n",
    "    output_df[\"chr_type\"] = output_df[\"chrom\"].map(dict(zip(regions_df.CONTIG_ID, regions_df.chr_type)))\n",
    "    output_df[\"species\"] = long_form\n",
    "    output_df.to_csv(\"../results/window_stats/{}_Fst.txt\".format(long_form), sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "3ecc6ef2",
   "metadata": {},
   "outputs": [],
   "source": [
    "from scipy.spatial.distance import squareform\n",
    "from scipy.cluster import hierarchy\n",
    "from scipy.cluster.hierarchy import dendrogram\n",
    "from sklearn.cluster import AgglomerativeClustering\n",
    "\n",
    "def plot_dendrogram(model, **kwargs):\n",
    "    # Create linkage matrix and then plot the dendrogram\n",
    "\n",
    "    # create the counts of samples under each node\n",
    "    counts = np.zeros(model.children_.shape[0])\n",
    "    n_samples = len(model.labels_)\n",
    "    for i, merge in enumerate(model.children_):\n",
    "        current_count = 0\n",
    "        for child_idx in merge:\n",
    "            if child_idx < n_samples:\n",
    "                current_count += 1  # leaf node\n",
    "            else:\n",
    "                current_count += counts[child_idx - n_samples]\n",
    "        counts[i] = current_count\n",
    "\n",
    "    linkage_matrix = np.column_stack(\n",
    "        [model.children_, model.distances_, counts]\n",
    "    ).astype(float)\n",
    "\n",
    "    # Plot the corresponding dendrogram\n",
    "    dendrogram(linkage_matrix, **kwargs)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "780cec4e",
   "metadata": {},
   "outputs": [],
   "source": [
    "def divergence_divide(square_df):\n",
    "    square_out = np.zeros((len(square_df), len(square_df)))\n",
    "    for i in range(len(square_df)):\n",
    "        for j in range(len(square_df)):\n",
    "            square_out[i,j] = square_df.iloc[i,j]/((square_df.iloc[i,i]+square_df.iloc[j,j])/2)\n",
    "    return square_out"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "2568731e",
   "metadata": {},
   "source": [
    "Flexible intervals:"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "710d4ed8",
   "metadata": {},
   "outputs": [],
   "source": [
    "def interval_creator(bed_l, window_size):\n",
    "    # Input a bed file and the window size of intervals desired. Multiple chromosomes accepted.\n",
    "    df_l = []\n",
    "    for c in bed_l.chrom.unique():\n",
    "        print(c)\n",
    "        start_l, end_l = [], []\n",
    "        b = bed_l.loc[bed_l[\"chrom\"] == c].copy()\n",
    "        b[\"w_s\"] = b.end-b.start\n",
    "        w_start = b.start.iloc[0]\n",
    "        current_size = 0\n",
    "        for i, j, k in zip(b.start, b.end, b.w_s):\n",
    "            # Current window encapsulates the final stretch of the interval.\n",
    "            if current_size + k >= window_size:\n",
    "                start_l.append(w_start), end_l.append(i+(window_size-current_size))\n",
    "                w_start = i+(window_size-current_size)\n",
    "                # If the window still contains full intervals, contigous windows until it cant.\n",
    "                for x in range((k-window_size+current_size)//window_size):\n",
    "                    start_l.append(w_start), end_l.append(w_start+window_size)\n",
    "                    w_start += window_size\n",
    "                current_size = j-w_start\n",
    "            # Current window does not encapsulate per definition, so it has to be added to current size but nothing else.\n",
    "            else:\n",
    "                current_size += k\n",
    "        df_l.append(pd.DataFrame({\"chrom\": c, \"interval_start\": start_l, \"interval_end\": end_l}))\n",
    "    return pd.concat(df_l)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
