{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "18cd9075",
   "metadata": {},
   "outputs": [],
   "source": [
    "%run ../scripts/notebook_settings.py\n",
    "import sgkit as sg\n",
    "import xarray as xr\n",
    "import glob"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4302e3eb",
   "metadata": {},
   "source": [
    "Defining the current paths and sizes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "dc500ec8",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "zarr_path = \"../zarr_data/\"\n",
    "metadata_folders = glob.glob(metadata_path+\"*_individuals.txt\")\n",
    "\n",
    "size_cutoff = 1000000\n",
    "window_size = 100000\n",
    "missing_filter = 0.5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "df3190ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_beds(long_form):\n",
    "    bed_path_x = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024/{}/filteredVCF/pos_bed_cov_based/{}_batch*_fploidy2_mploidy1.bed\".format(long_form, long_form)\n",
    "    bed_path_all = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024/{}/filteredVCF/pos_bed_cov_based/{}_batch*_fploidy2_mploidy2.bed\".format(long_form, long_form)\n",
    "    bed_l = []\n",
    "    for b in glob.glob(bed_path_all):\n",
    "        bed_file = pd.read_csv(b, sep=\"\\t\", names=[\"chrom\", \"start\", \"end\"])\n",
    "        bed_l.append(bed_file)\n",
    "    bed_files = pd.concat(bed_l)\n",
    "    bed_l = []\n",
    "    for b in glob.glob(bed_path_x):\n",
    "        #print(b)\n",
    "        bed_file = pd.read_csv(b, sep=\"\\t\", names=[\"chrom\", \"start\", \"end\"])\n",
    "        bed_l.append(bed_file)\n",
    "    if len(bed_l) > 0:\n",
    "        bed_x = pd.concat(bed_l)\n",
    "        bed_files = bed_files.loc[~(bed_files.chrom.isin(bed_x.chrom.unique()))]\n",
    "        bed_files = pd.concat([bed_files, bed_x]).sort_values(by=[\"chrom\", \"start\", \"end\"])\n",
    "    return bed_files\n",
    "\n",
    "def pos_windows(bed_l, window_size, chrom_order):\n",
    "    # Input a bed file and the window size of intervals desired. Multiple chromosomes accepted.\n",
    "    # It has to be sorted.\n",
    "    df_l = []\n",
    "    for c in chrom_order:\n",
    "        #print(c)\n",
    "        frac_l = []\n",
    "        b = bed_l.loc[bed_l[\"chrom\"] == c].copy()\n",
    "        b[\"w_s\"] = b.end-b.start\n",
    "        w_start = b.start.iloc[0]\n",
    "        current_pos, callable_bases = 0, 0\n",
    "        for i, j, k in zip(b.start, b.end, b.w_s):\n",
    "            # Nothing called in the current window under investigation.\n",
    "            while i-window_size >= current_pos:\n",
    "                frac_l.append(callable_bases/window_size)\n",
    "                callable_bases = 0\n",
    "                current_pos += window_size\n",
    "            # Window starts in current. We know this is true because of the previous while loop.\n",
    "            callable_bases += min(k, current_pos+window_size-i)\n",
    "            # Everything called in current.\n",
    "            while j-window_size >= current_pos:\n",
    "                frac_l.append(callable_bases/window_size)\n",
    "                callable_bases = 0\n",
    "                current_pos += window_size\n",
    "                if j-window_size >= current_pos:\n",
    "                    callable_bases += window_size\n",
    "                else:\n",
    "                # Window stops in current. Again, know this is true.\n",
    "                    callable_bases += j-current_pos\n",
    "        # Last window.\n",
    "        frac_l.append(callable_bases/(window_size))\n",
    "        df_l.append(pd.DataFrame({\"chrom\": c, \"window_start\": list(range(0, len(frac_l)*window_size, window_size)),\n",
    "                                  \"window_end\": list(range(window_size, (len(frac_l)+1)*window_size, window_size)),\n",
    "                                  \"callable_frac\": frac_l}))\n",
    "    return pd.concat(df_l)\n",
    "\n",
    "def haploid_double(ds, variable, dim):\n",
    "    unmasked = ~ds[f\"{variable}_mask\"]\n",
    "    overwrite = ds.call_genotype[:,:,0]\n",
    "    overwrite_2 = ds.call_genotype_mask[:,:,0]\n",
    "    return ds.assign(**{\n",
    "        f\"{variable}\": ds[variable].where(\n",
    "        unmasked, \n",
    "        overwrite),\n",
    "        f\"{variable}_mask\": ds[f\"{variable}_mask\"].where(\n",
    "        unmasked,\n",
    "        overwrite_2)})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "9b0075b6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chlorocebus_aethiops_ssp 26\n",
      "Chlorocebus_aethiops_ssp\n",
      "Chlorocebus_cynosuros_ssp 16\n",
      "Chlorocebus_cynosuros_ssp\n",
      "Chlorocebus_tantalus_ssp 11\n",
      "Chlorocebus_tantalus_ssp\n",
      "Allenopithecus_nigroviridis_ssp 3\n",
      "Pithecia_pissinatti_ssp 3\n",
      "Pithecia_pissinatti_ssp\n",
      "Nasalis_larvatus_ssp 3\n"
     ]
    }
   ],
   "source": [
    "# Input checks: Is there a zarr/bed pair for all species?\n",
    "bed_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024/{}/filteredVCF/pos_bed_cov_based/{}_batch0_fploidy2_mploidy2.bed\"\n",
    "fvcf_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024/{}/filteredVCF/bcf_step1/{}_all_chr.sorted.bcf\"\n",
    "for x in glob.glob(metadata_path+\"*_individuals.txt\"):\n",
    "    short_form = x.split(\"/\")[-1].split(\"_\")[0]\n",
    "    long_form = x.split(\"/\")[-1]\n",
    "    # Loading the various metadata files. Metadata, contig information, callability bed.\n",
    "    metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "    metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "    metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    for f in metadata_df.GVCF_FOLDER.unique():\n",
    "        r_df = regions_df.loc[regions_df.REFERENCE_FOLDER == f]\n",
    "        if not os.path.exists(bed_path.format(f, f)):\n",
    "            print(f, len(metadata_df.loc[metadata_df.GVCF_FOLDER == f]))\n",
    "        if not os.path.exists(fvcf_path.format(f, f)):\n",
    "            print(f)\n",
    "        # for b, f, m in zip(r_df.BATCH, r_df.FEMALE_PLOIDY, r_df.MALE_PLOIDY):\n",
    "        #    print(b,f,m)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "3b1800f5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Leontopithecus_chrysomelas_ssp\n",
      "Leontopithecus_chrysomelas_ssp\n",
      "../zarr_data/Leontopithecus_chrysomelas_ssp/HiC_scaffold_23\n",
      "../zarr_data/Leontopithecus_chrysomelas_ssp/HiC_scaffold_1\n",
      "../zarr_data/Saguinus_labiatus_ssp\n",
      "Saguinus_labiatus_ssp\n",
      "../zarr_data/Saguinus_labiatus_ssp/CM063191.1\n",
      "../zarr_data/Saguinus_labiatus_ssp/JASSZA010000432.1\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "Dimension 0 has 3 blocks, adjust_chunks specified with 100 blocks",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mValueError\u001b[39m                                Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[5]\u001b[39m\u001b[32m, line 33\u001b[39m\n\u001b[32m     31\u001b[39m ds = ds.isel(variants=(missing_rate <= missing_filter))\n\u001b[32m     32\u001b[39m ds = sg.window_by_position(ds, size=window_size)\n\u001b[32m---> \u001b[39m\u001b[32m33\u001b[39m ds = (\u001b[43msg\u001b[49m\u001b[43m.\u001b[49m\u001b[43mdiversity\u001b[49m\u001b[43m(\u001b[49m\u001b[43mds\u001b[49m\u001b[43m.\u001b[49m\u001b[43mchunk\u001b[49m\u001b[43m(\u001b[49m\u001b[43m{\u001b[49m\u001b[33;43m\"\u001b[39;49m\u001b[33;43mvariants\u001b[39;49m\u001b[33;43m\"\u001b[39;49m\u001b[43m:\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mlen\u001b[39;49m\u001b[43m(\u001b[49m\u001b[43mds\u001b[49m\u001b[43m.\u001b[49m\u001b[43mvariants\u001b[49m\u001b[43m)\u001b[49m\u001b[43m/\u001b[49m\u001b[43m/\u001b[49m\u001b[32;43m100\u001b[39;49m\u001b[43m}\u001b[49m\u001b[43m)\u001b[49m\u001b[43m)\u001b[49m)\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/primatediversity2/lib/python3.12/site-packages/sgkit/stats/popgen.py:103\u001b[39m, in \u001b[36mdiversity\u001b[39m\u001b[34m(ds, cohort_allele_count, merge)\u001b[39m\n\u001b[32m    100\u001b[39m pi = n_diff / n_pairs_na\n\u001b[32m    102\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m has_windows(ds):\n\u001b[32m--> \u001b[39m\u001b[32m103\u001b[39m     div = \u001b[43mwindow_statistic\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    104\u001b[39m \u001b[43m        \u001b[49m\u001b[43mpi\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    105\u001b[39m \u001b[43m        \u001b[49m\u001b[43mnp\u001b[49m\u001b[43m.\u001b[49m\u001b[43msum\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    106\u001b[39m \u001b[43m        \u001b[49m\u001b[43mds\u001b[49m\u001b[43m.\u001b[49m\u001b[43mwindow_start\u001b[49m\u001b[43m.\u001b[49m\u001b[43mvalues\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    107\u001b[39m \u001b[43m        \u001b[49m\u001b[43mds\u001b[49m\u001b[43m.\u001b[49m\u001b[43mwindow_stop\u001b[49m\u001b[43m.\u001b[49m\u001b[43mvalues\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    108\u001b[39m \u001b[43m        \u001b[49m\u001b[43mdtype\u001b[49m\u001b[43m=\u001b[49m\u001b[43mpi\u001b[49m\u001b[43m.\u001b[49m\u001b[43mdtype\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    109\u001b[39m \u001b[43m        \u001b[49m\u001b[43maxis\u001b[49m\u001b[43m=\u001b[49m\u001b[32;43m0\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[32m    110\u001b[39m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    111\u001b[39m     new_ds = create_dataset(\n\u001b[32m    112\u001b[39m         {\n\u001b[32m    113\u001b[39m             variables.stat_diversity: (\n\u001b[32m   (...)\u001b[39m\u001b[32m    117\u001b[39m         }\n\u001b[32m    118\u001b[39m     )\n\u001b[32m    119\u001b[39m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/primatediversity2/lib/python3.12/site-packages/sgkit/window.py:570\u001b[39m, in \u001b[36mwindow_statistic\u001b[39m\u001b[34m(values, statistic, window_starts, window_stops, dtype, chunks, new_axis, **kwargs)\u001b[39m\n\u001b[32m    568\u001b[39m     \u001b[38;5;66;03m# new chunks are same except in first axis\u001b[39;00m\n\u001b[32m    569\u001b[39m     new_chunks = \u001b[38;5;28mtuple\u001b[39m([\u001b[38;5;28mtuple\u001b[39m(windows_per_chunk)] + \u001b[38;5;28mlist\u001b[39m(desired_chunks[\u001b[32m1\u001b[39m:]))  \u001b[38;5;66;03m# type: ignore\u001b[39;00m\n\u001b[32m--> \u001b[39m\u001b[32m570\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43mda\u001b[49m\u001b[43m.\u001b[49m\u001b[43mmap_overlap\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    571\u001b[39m \u001b[43m    \u001b[49m\u001b[43mblockwise_moving_stat\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    572\u001b[39m \u001b[43m    \u001b[49m\u001b[43mvalues\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    573\u001b[39m \u001b[43m    \u001b[49m\u001b[43mdtype\u001b[49m\u001b[43m=\u001b[49m\u001b[43mdtype\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    574\u001b[39m \u001b[43m    \u001b[49m\u001b[43mchunks\u001b[49m\u001b[43m=\u001b[49m\u001b[43mnew_chunks\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    575\u001b[39m \u001b[43m    \u001b[49m\u001b[43mdepth\u001b[49m\u001b[43m=\u001b[49m\u001b[43mdepth\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    576\u001b[39m \u001b[43m    \u001b[49m\u001b[43mboundary\u001b[49m\u001b[43m=\u001b[49m\u001b[32;43m0\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[32m    577\u001b[39m \u001b[43m    \u001b[49m\u001b[43mtrim\u001b[49m\u001b[43m=\u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[32m    578\u001b[39m \u001b[43m    \u001b[49m\u001b[43mnew_axis\u001b[49m\u001b[43m=\u001b[49m\u001b[43mnew_axis\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    579\u001b[39m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/primatediversity2/lib/python3.12/site-packages/dask/array/overlap.py:722\u001b[39m, in \u001b[36mmap_overlap\u001b[39m\u001b[34m(func, depth, boundary, trim, align_arrays, allow_rechunk, *args, **kwargs)\u001b[39m\n\u001b[32m    717\u001b[39m args = [\n\u001b[32m    718\u001b[39m     overlap(x, depth=d, boundary=b, allow_rechunk=allow_rechunk)\n\u001b[32m    719\u001b[39m     \u001b[38;5;28;01mfor\u001b[39;00m x, d, b \u001b[38;5;129;01min\u001b[39;00m \u001b[38;5;28mzip\u001b[39m(args, depth, boundary)\n\u001b[32m    720\u001b[39m ]\n\u001b[32m    721\u001b[39m assert_int_chunksize(args)\n\u001b[32m--> \u001b[39m\u001b[32m722\u001b[39m x = \u001b[43mmap_blocks\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfunc\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43margs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    723\u001b[39m assert_int_chunksize([x])\n\u001b[32m    724\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m trim:\n\u001b[32m    725\u001b[39m     \u001b[38;5;66;03m# Find index of array argument with maximum rank and break ties by choosing first provided\u001b[39;00m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/primatediversity2/lib/python3.12/site-packages/dask/array/core.py:866\u001b[39m, in \u001b[36mmap_blocks\u001b[39m\u001b[34m(func, name, token, dtype, chunks, drop_axis, new_axis, enforce_ndim, meta, *args, **kwargs)\u001b[39m\n\u001b[32m    850\u001b[39m     out = blockwise(\n\u001b[32m    851\u001b[39m         apply_and_enforce,\n\u001b[32m    852\u001b[39m         out_ind,\n\u001b[32m   (...)\u001b[39m\u001b[32m    863\u001b[39m         **kwargs,\n\u001b[32m    864\u001b[39m     )\n\u001b[32m    865\u001b[39m \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[32m--> \u001b[39m\u001b[32m866\u001b[39m     out = \u001b[43mblockwise\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    867\u001b[39m \u001b[43m        \u001b[49m\u001b[43mfunc\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    868\u001b[39m \u001b[43m        \u001b[49m\u001b[43mout_ind\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    869\u001b[39m \u001b[43m        \u001b[49m\u001b[43m*\u001b[49m\u001b[43mconcat\u001b[49m\u001b[43m(\u001b[49m\u001b[43margpairs\u001b[49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    870\u001b[39m \u001b[43m        \u001b[49m\u001b[43mname\u001b[49m\u001b[43m=\u001b[49m\u001b[43mname\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    871\u001b[39m \u001b[43m        \u001b[49m\u001b[43mnew_axes\u001b[49m\u001b[43m=\u001b[49m\u001b[43mnew_axes\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    872\u001b[39m \u001b[43m        \u001b[49m\u001b[43mdtype\u001b[49m\u001b[43m=\u001b[49m\u001b[43mdtype\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    873\u001b[39m \u001b[43m        \u001b[49m\u001b[43mconcatenate\u001b[49m\u001b[43m=\u001b[49m\u001b[38;5;28;43;01mTrue\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[32m    874\u001b[39m \u001b[43m        \u001b[49m\u001b[43malign_arrays\u001b[49m\u001b[43m=\u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m,\u001b[49m\n\u001b[32m    875\u001b[39m \u001b[43m        \u001b[49m\u001b[43madjust_chunks\u001b[49m\u001b[43m=\u001b[49m\u001b[43madjust_chunks\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    876\u001b[39m \u001b[43m        \u001b[49m\u001b[43mmeta\u001b[49m\u001b[43m=\u001b[49m\u001b[43mmeta\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    877\u001b[39m \u001b[43m        \u001b[49m\u001b[43m*\u001b[49m\u001b[43m*\u001b[49m\u001b[43mkwargs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    878\u001b[39m \u001b[43m    \u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    880\u001b[39m extra_argpairs = []\n\u001b[32m    881\u001b[39m extra_names = []\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/primatediversity2/lib/python3.12/site-packages/dask/array/blockwise.py:272\u001b[39m, in \u001b[36mblockwise\u001b[39m\u001b[34m(func, out_ind, name, token, dtype, adjust_chunks, new_axes, align_arrays, concatenate, meta, *args, **kwargs)\u001b[39m\n\u001b[32m    270\u001b[39m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(adjust_chunks[ind], (\u001b[38;5;28mtuple\u001b[39m, \u001b[38;5;28mlist\u001b[39m)):\n\u001b[32m    271\u001b[39m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(adjust_chunks[ind]) != \u001b[38;5;28mlen\u001b[39m(chunks[i]):\n\u001b[32m--> \u001b[39m\u001b[32m272\u001b[39m         \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\n\u001b[32m    273\u001b[39m             \u001b[33mf\u001b[39m\u001b[33m\"\u001b[39m\u001b[33mDimension \u001b[39m\u001b[38;5;132;01m{\u001b[39;00mi\u001b[38;5;132;01m}\u001b[39;00m\u001b[33m has \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mlen\u001b[39m(chunks[i])\u001b[38;5;132;01m}\u001b[39;00m\u001b[33m blocks, adjust_chunks \u001b[39m\u001b[33m\"\u001b[39m\n\u001b[32m    274\u001b[39m             \u001b[33mf\u001b[39m\u001b[33m\"\u001b[39m\u001b[33mspecified with \u001b[39m\u001b[38;5;132;01m{\u001b[39;00m\u001b[38;5;28mlen\u001b[39m(adjust_chunks[ind])\u001b[38;5;132;01m}\u001b[39;00m\u001b[33m blocks\u001b[39m\u001b[33m\"\u001b[39m\n\u001b[32m    275\u001b[39m         )\n\u001b[32m    276\u001b[39m     chunks[i] = \u001b[38;5;28mtuple\u001b[39m(adjust_chunks[ind])\n\u001b[32m    277\u001b[39m \u001b[38;5;28;01melse\u001b[39;00m:\n",
      "\u001b[31mValueError\u001b[39m: Dimension 0 has 3 blocks, adjust_chunks specified with 100 blocks"
     ]
    }
   ],
   "source": [
    "# Pi implementation\n",
    "for x in glob.glob(zarr_path+\"*\"):\n",
    "    print(x)\n",
    "    # The name used to load all the files, short and long version\n",
    "    short_form = x.split(\"/\")[-1].split(\"_\")[0]\n",
    "    long_form = x.split(\"/\")[-1]\n",
    "    print(long_form)\n",
    "    # Loading the various metadata files. Metadata, contig information, callability bed.\n",
    "    metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "    metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "    metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "    regions_df[\"chr_type\"] = [\"chrX\" if x == 2 and y == 1 else \"aut\" for x, y in zip(regions_df.FEMALE_PLOIDY, regions_df.MALE_PLOIDY)]\n",
    "    large_contigs = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "    large_x = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID\n",
    "    bed_files = read_beds(long_form)\n",
    "    # Loading the genetic data.\n",
    "    df_l = []\n",
    "    for c in glob.glob(x+\"/*\"):\n",
    "        print(c)\n",
    "        ds = sg.load_dataset(c)\n",
    "        # This implementation is the pi implementation.\n",
    "        # Probably problematic in some cases with population structure, but it is easier to implement\n",
    "        ds[\"sample_cohort\"] = np.repeat([0], ds.sizes[\"samples\"])\n",
    "        # Subsetting and windowing the sgkit dataset. The rechunking handles what otherwise would cause an error.\n",
    "        #ds[\"call_genotype\"] = ds[\"call_genotype\"].clip(0)\n",
    "        ds = ds.sel(contigs=[ds.variant_contig[0].values])\n",
    "        missing_rate = ds.call_genotype_mask[:,:,0].sum(axis=1).values/ds.call_genotype_mask[:,:,0].count(axis=1).values\n",
    "        ds = ds.isel(variants=(missing_rate <= missing_filter))\n",
    "        ds = sg.window_by_position(ds, size=window_size)\n",
    "        ds = (sg.diversity(ds.chunk({\"variants\": len(ds.variants)//100})))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "id": "1ccd4956",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<bound method DataArrayAggregations.count of <xarray.DataArray 'variants' (variants: 5748)> Size: 46kB\n",
       "array([   0,    1,    2, ..., 5745, 5746, 5747])\n",
       "Dimensions without coordinates: variants>"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ds.variants.chunk().count"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "2ebaf6e9",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "20    NC_044996.1\n",
       "Name: CONTIG_ID, dtype: object"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "large_x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "1337890e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array(['NC_044976.1', 'NC_044977.1', 'NC_044978.1', 'NC_044979.1',\n",
       "       'NC_044980.1', 'NC_044981.1', 'NC_044982.1', 'NC_044983.1',\n",
       "       'NC_044984.1', 'NC_044985.1', 'NC_044986.1', 'NC_044987.1',\n",
       "       'NC_044988.1', 'NC_044989.1', 'NC_044990.1', 'NC_044991.1',\n",
       "       'NC_044992.1', 'NC_044993.1', 'NC_044994.1', 'NC_044995.1',\n",
       "       'NC_044996.1'], dtype=object)"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "bed_files.chrom.unique()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "id": "7005dccf",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/eriks/miniconda3/envs/primatediversity2/lib/python3.12/site-packages/sgkit/utils.py:222: MergeWarning: The following variables in the input dataset will be replaced in the output: stat_diversity\n",
      "  warnings.warn(\n"
     ]
    }
   ],
   "source": [
    "ds = (sg.diversity(ds.chunk({\"variants\": len(ds.variants)//50})))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "bbe7906e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>REFERENCE_FOLDER</th>\n",
       "      <th>REGION</th>\n",
       "      <th>CONTIG_ID</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "      <th>BATCH</th>\n",
       "      <th>FEMALE_PLOIDY</th>\n",
       "      <th>MALE_PLOIDY</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Trachypithecus_francoisi_ssp</td>\n",
       "      <td>NW_022680469.1</td>\n",
       "      <td>NW_022680469.1</td>\n",
       "      <td>0</td>\n",
       "      <td>1129610</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               REFERENCE_FOLDER          REGION       CONTIG_ID  START  \\\n",
       "9  Trachypithecus_francoisi_ssp  NW_022680469.1  NW_022680469.1      0   \n",
       "\n",
       "       END  BATCH  FEMALE_PLOIDY  MALE_PLOIDY  \n",
       "9  1129610      0              0            1  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "regions_df = pd.read_csv(metadata_path+\"Semnopithecus_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "regions_df.loc[regions_df.FEMALE_PLOIDY == 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "b822180c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Leontopithecus_chrysomelas_ssp\n",
      "Leontopithecus_chrysomelas_ssp\n",
      "../zarr_data/Leontopithecus_chrysomelas_ssp/HiC_scaffold_23\n",
      "../zarr_data/Leontopithecus_chrysomelas_ssp/HiC_scaffold_1\n",
      "../zarr_data/Saguinus_labiatus_ssp\n",
      "Saguinus_labiatus_ssp\n",
      "../zarr_data/Saguinus_labiatus_ssp/CM063191.1\n",
      "../zarr_data/Saguinus_labiatus_ssp/JASSZA010000432.1\n",
      "../zarr_data/Saguinus_labiatus_ssp/CM063169.1\n"
     ]
    }
   ],
   "source": [
    "# Pi implementation\n",
    "for x in glob.glob(zarr_path+\"*\"):\n",
    "    print(x)\n",
    "    # The name used to load all the files, short and long version\n",
    "    short_form = x.split(\"/\")[-1].split(\"_\")[0]\n",
    "    long_form = x.split(\"/\")[-1]\n",
    "    print(long_form)\n",
    "    # Loading the various metadata files. Metadata, contig information, callability bed.\n",
    "    metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "    metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "    metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "    regions_df[\"chr_type\"] = [\"chrX\" if x == 2 and y == 1 else \"aut\" for x, y in zip(regions_df.FEMALE_PLOIDY, regions_df.MALE_PLOIDY)]\n",
    "    large_contigs = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "    large_x = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID\n",
    "    bed_files = read_beds(long_form)\n",
    "    # Loading the genetic data.\n",
    "    df_l = []\n",
    "    for c in glob.glob(x+\"/*\"):\n",
    "        print(c)\n",
    "        ds = sg.load_dataset(c)\n",
    "        # This implementation is the pi implementation.\n",
    "        # Probably problematic in some cases with population structure, but it is easier to implement\n",
    "        ds[\"sample_cohort\"] = np.repeat([0], ds.sizes[\"samples\"])\n",
    "        # Subsetting and windowing the sgkit dataset. The rechunking handles what otherwise would cause an error.\n",
    "        #ds[\"call_genotype\"] = ds[\"call_genotype\"].clip(0)\n",
    "        ds = ds.sel(contigs=[ds.variant_contig[0].values])\n",
    "        missing_rate = ds.call_genotype_mask[:,:,0].sum(axis=1).values/ds.call_genotype_mask[:,:,0].count(axis=1).values\n",
    "        ds = ds.isel(variants=(missing_rate <= missing_filter))\n",
    "        ds = sg.window_by_position(ds, size=window_size)\n",
    "        ds = (sg.diversity(ds.chunk({\"variants\": 50000})))\n",
    "        df_sub = pd.DataFrame({\"window_start\": list(range(0, len(ds.window_start)*window_size, window_size)),\n",
    "                               \"pi\": ds.stat_diversity[:,0]})\n",
    "        df_sub[\"chrom\"] = c.split(\"/\")[-1]\n",
    "        df_l.append(df_sub)\n",
    "    df_het = pd.concat(df_l)\n",
    "    bed_files = read_beds(long_form)\n",
    "    intervals_callable = pos_windows(bed_files, window_size, df_het[\"chrom\"].unique())\n",
    "    output_df = pd.merge(df_het, intervals_callable, on=[\"chrom\", \"window_start\"])\n",
    "    output_df[\"chr_type\"] = output_df[\"chrom\"].map(dict(zip(regions_df.CONTIG_ID, regions_df.chr_type)))\n",
    "    output_df[\"species\"] = long_form\n",
    "    output_df.to_csv(\"../results/window_stats/{}_100kb_pi.txt\".format(long_form), sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "80b23a53",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([0.  , 0.25, 0.25, ..., 0.  , 0.  , 0.25])"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "ds = sg.load_dataset(\"../zarr_data/Leontopithecus_chrysomelas_ssp/HiC_scaffold_1\")\n",
    "        # This implementation is the pi implementation.\n",
    "        # Probably problematic in some cases with population structure, but it is easier to implement\n",
    "ds[\"sample_cohort\"] = np.repeat([0], ds.sizes[\"samples\"])\n",
    "        # Subsetting and windowing the sgkit dataset. The rechunking handles what otherwise would cause an error.\n",
    "        #ds[\"call_genotype\"] = ds[\"call_genotype\"].clip(0)\n",
    "ds = ds.sel(contigs=[ds.variant_contig[0].values])\n",
    "missing_rate = ds.call_genotype_mask[:,:,0].sum(axis=1).values/ds.call_genotype_mask[:,:,0].count(axis=1).values\n",
    "missing_rate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "8eaa8e10",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.75"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missing_rate.max()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "fa582d41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([0., 0., 0., ..., 0., 0., 0.])"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missing_rate = ds.call_genotype_mask[:,:,0].sum(axis=1).values/ds.call_genotype_mask[:,:,0].count(axis=1).values\n",
    "missing_rate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "11424372",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[[nan]],\n",
       "\n",
       "       [[nan]],\n",
       "\n",
       "       [[nan]],\n",
       "\n",
       "       ...,\n",
       "\n",
       "       [[nan]],\n",
       "\n",
       "       [[nan]],\n",
       "\n",
       "       [[nan]]])"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sg.Fst(ds)[\"stat_Fst\"].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "bd1ae499",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Leontopithecus_chrysomelas_ssp\n",
      "Leontopithecus_chrysomelas_ssp\n",
      "../zarr_data/Leontopithecus_chrysomelas_ssp/HiC_scaffold_23\n",
      "../zarr_data/Leontopithecus_chrysomelas_ssp/HiC_scaffold_1\n"
     ]
    }
   ],
   "source": [
    "# Heterozygosity implementation. No missing filter - I count all sites which have a het call.\n",
    "for x in glob.glob(zarr_path+\"*\")[:1]:\n",
    "    print(x)\n",
    "    # The name used to load all the files, short and long version\n",
    "    short_form = x.split(\"/\")[-1].split(\"_\")[0]\n",
    "    long_form = x.split(\"/\")[-1]\n",
    "    print(long_form)\n",
    "    # Loading the various metadata files. Metadata, contig information, callability bed.\n",
    "    metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "    metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "    metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "    regions_df[\"chr_type\"] = [\"chrX\" if x == 2 and y == 1 else \"aut\" for x, y in zip(regions_df.FEMALE_PLOIDY, regions_df.MALE_PLOIDY)]\n",
    "    large_contigs = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "    large_x = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID\n",
    "    bed_files = read_beds(long_form)\n",
    "    # Loading the genetic data.\n",
    "    df_l = []\n",
    "    for c in glob.glob(x+\"/*\"):\n",
    "        print(c)\n",
    "        ds = sg.load_dataset(c)\n",
    "        # Probably problematic in some cases with population structure, but it is easier to implement\n",
    "        ds[\"sample_cohort\"] = ds[\"samples\"]\n",
    "        # Subsetting and windowing the sgkit dataset. The rechunking handles what otherwise would cause an error.\n",
    "        if c.split(\"/\")[-1] in list(large_x):\n",
    "            ds = haploid_double(ds, \"call_genotype\", \"samples\")\n",
    "        ds[\"call_genotype\"] = ds[\"call_genotype\"].clip(0)\n",
    "        ds = ds.sel(contigs=[ds.variant_contig[0].values])\n",
    "        ds = sg.window_by_position(ds, size=window_size)\n",
    "        ds = (sg.diversity(ds.chunk({\"variants\": 50000})))\n",
    "        df_sub = pd.DataFrame(ds.stat_diversity, columns=ds.sample_id)\n",
    "        df_sub[\"window_start\"] = list(range(0, len(ds.window_start)*window_size, window_size))\n",
    "        df_sub[\"chrom\"] = c.split(\"/\")[-1]\n",
    "        df_l.append(df_sub)\n",
    "    df_het = pd.concat(df_l)\n",
    "    bed_files = read_beds(long_form)\n",
    "    intervals_callable = pos_windows(bed_files, window_size, df_het[\"chrom\"].unique())\n",
    "    output_df = pd.merge(df_het, intervals_callable, on=[\"chrom\", \"window_start\"])\n",
    "    output_df[\"chr_type\"] = output_df[\"chrom\"].map(dict(zip(regions_df.CONTIG_ID, regions_df.chr_type)))\n",
    "    output_df[\"species\"] = long_form\n",
    "    output_df.to_csv(\"../results/window_stats/{}_100kb_het.txt\".format(long_form), sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "692ecaec",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div><svg style=\"position: absolute; width: 0; height: 0; overflow: hidden\">\n",
       "<defs>\n",
       "<symbol id=\"icon-database\" viewBox=\"0 0 32 32\">\n",
       "<path d=\"M16 0c-8.837 0-16 2.239-16 5v4c0 2.761 7.163 5 16 5s16-2.239 16-5v-4c0-2.761-7.163-5-16-5z\"></path>\n",
       "<path d=\"M16 17c-8.837 0-16-2.239-16-5v6c0 2.761 7.163 5 16 5s16-2.239 16-5v-6c0 2.761-7.163 5-16 5z\"></path>\n",
       "<path d=\"M16 26c-8.837 0-16-2.239-16-5v6c0 2.761 7.163 5 16 5s16-2.239 16-5v-6c0 2.761-7.163 5-16 5z\"></path>\n",
       "</symbol>\n",
       "<symbol id=\"icon-file-text2\" viewBox=\"0 0 32 32\">\n",
       "<path d=\"M28.681 7.159c-0.694-0.947-1.662-2.053-2.724-3.116s-2.169-2.030-3.116-2.724c-1.612-1.182-2.393-1.319-2.841-1.319h-15.5c-1.378 0-2.5 1.121-2.5 2.5v27c0 1.378 1.122 2.5 2.5 2.5h23c1.378 0 2.5-1.122 2.5-2.5v-19.5c0-0.448-0.137-1.23-1.319-2.841zM24.543 5.457c0.959 0.959 1.712 1.825 2.268 2.543h-4.811v-4.811c0.718 0.556 1.584 1.309 2.543 2.268zM28 29.5c0 0.271-0.229 0.5-0.5 0.5h-23c-0.271 0-0.5-0.229-0.5-0.5v-27c0-0.271 0.229-0.5 0.5-0.5 0 0 15.499-0 15.5 0v7c0 0.552 0.448 1 1 1h7v19.5z\"></path>\n",
       "<path d=\"M23 26h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z\"></path>\n",
       "<path d=\"M23 22h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z\"></path>\n",
       "<path d=\"M23 18h-14c-0.552 0-1-0.448-1-1s0.448-1 1-1h14c0.552 0 1 0.448 1 1s-0.448 1-1 1z\"></path>\n",
       "</symbol>\n",
       "</defs>\n",
       "</svg>\n",
       "<style>/* CSS stylesheet for displaying xarray objects in jupyterlab.\n",
       " *\n",
       " */\n",
       "\n",
       ":root {\n",
       "  --xr-font-color0: var(--jp-content-font-color0, rgba(0, 0, 0, 1));\n",
       "  --xr-font-color2: var(--jp-content-font-color2, rgba(0, 0, 0, 0.54));\n",
       "  --xr-font-color3: var(--jp-content-font-color3, rgba(0, 0, 0, 0.38));\n",
       "  --xr-border-color: var(--jp-border-color2, #e0e0e0);\n",
       "  --xr-disabled-color: var(--jp-layout-color3, #bdbdbd);\n",
       "  --xr-background-color: var(--jp-layout-color0, white);\n",
       "  --xr-background-color-row-even: var(--jp-layout-color1, white);\n",
       "  --xr-background-color-row-odd: var(--jp-layout-color2, #eeeeee);\n",
       "}\n",
       "\n",
       "html[theme=\"dark\"],\n",
       "html[data-theme=\"dark\"],\n",
       "body[data-theme=\"dark\"],\n",
       "body.vscode-dark {\n",
       "  --xr-font-color0: rgba(255, 255, 255, 1);\n",
       "  --xr-font-color2: rgba(255, 255, 255, 0.54);\n",
       "  --xr-font-color3: rgba(255, 255, 255, 0.38);\n",
       "  --xr-border-color: #1f1f1f;\n",
       "  --xr-disabled-color: #515151;\n",
       "  --xr-background-color: #111111;\n",
       "  --xr-background-color-row-even: #111111;\n",
       "  --xr-background-color-row-odd: #313131;\n",
       "}\n",
       "\n",
       ".xr-wrap {\n",
       "  display: block !important;\n",
       "  min-width: 300px;\n",
       "  max-width: 700px;\n",
       "}\n",
       "\n",
       ".xr-text-repr-fallback {\n",
       "  /* fallback to plain text repr when CSS is not injected (untrusted notebook) */\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-header {\n",
       "  padding-top: 6px;\n",
       "  padding-bottom: 6px;\n",
       "  margin-bottom: 4px;\n",
       "  border-bottom: solid 1px var(--xr-border-color);\n",
       "}\n",
       "\n",
       ".xr-header > div,\n",
       ".xr-header > ul {\n",
       "  display: inline;\n",
       "  margin-top: 0;\n",
       "  margin-bottom: 0;\n",
       "}\n",
       "\n",
       ".xr-obj-type,\n",
       ".xr-array-name {\n",
       "  margin-left: 2px;\n",
       "  margin-right: 10px;\n",
       "}\n",
       "\n",
       ".xr-obj-type {\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-sections {\n",
       "  padding-left: 0 !important;\n",
       "  display: grid;\n",
       "  grid-template-columns: 150px auto auto 1fr 0 20px 0 20px;\n",
       "}\n",
       "\n",
       ".xr-section-item {\n",
       "  display: contents;\n",
       "}\n",
       "\n",
       ".xr-section-item input {\n",
       "  display: inline-block;\n",
       "  opacity: 0;\n",
       "  height: 0;\n",
       "}\n",
       "\n",
       ".xr-section-item input + label {\n",
       "  color: var(--xr-disabled-color);\n",
       "}\n",
       "\n",
       ".xr-section-item input:enabled + label {\n",
       "  cursor: pointer;\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-section-item input:focus + label {\n",
       "  border: 2px solid var(--xr-font-color0);\n",
       "}\n",
       "\n",
       ".xr-section-item input:enabled + label:hover {\n",
       "  color: var(--xr-font-color0);\n",
       "}\n",
       "\n",
       ".xr-section-summary {\n",
       "  grid-column: 1;\n",
       "  color: var(--xr-font-color2);\n",
       "  font-weight: 500;\n",
       "}\n",
       "\n",
       ".xr-section-summary > span {\n",
       "  display: inline-block;\n",
       "  padding-left: 0.5em;\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:disabled + label {\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-section-summary-in + label:before {\n",
       "  display: inline-block;\n",
       "  content: \"►\";\n",
       "  font-size: 11px;\n",
       "  width: 15px;\n",
       "  text-align: center;\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:disabled + label:before {\n",
       "  color: var(--xr-disabled-color);\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:checked + label:before {\n",
       "  content: \"▼\";\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:checked + label > span {\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-section-summary,\n",
       ".xr-section-inline-details {\n",
       "  padding-top: 4px;\n",
       "  padding-bottom: 4px;\n",
       "}\n",
       "\n",
       ".xr-section-inline-details {\n",
       "  grid-column: 2 / -1;\n",
       "}\n",
       "\n",
       ".xr-section-details {\n",
       "  display: none;\n",
       "  grid-column: 1 / -1;\n",
       "  margin-bottom: 5px;\n",
       "}\n",
       "\n",
       ".xr-section-summary-in:checked ~ .xr-section-details {\n",
       "  display: contents;\n",
       "}\n",
       "\n",
       ".xr-array-wrap {\n",
       "  grid-column: 1 / -1;\n",
       "  display: grid;\n",
       "  grid-template-columns: 20px auto;\n",
       "}\n",
       "\n",
       ".xr-array-wrap > label {\n",
       "  grid-column: 1;\n",
       "  vertical-align: top;\n",
       "}\n",
       "\n",
       ".xr-preview {\n",
       "  color: var(--xr-font-color3);\n",
       "}\n",
       "\n",
       ".xr-array-preview,\n",
       ".xr-array-data {\n",
       "  padding: 0 5px !important;\n",
       "  grid-column: 2;\n",
       "}\n",
       "\n",
       ".xr-array-data,\n",
       ".xr-array-in:checked ~ .xr-array-preview {\n",
       "  display: none;\n",
       "}\n",
       "\n",
       ".xr-array-in:checked ~ .xr-array-data,\n",
       ".xr-array-preview {\n",
       "  display: inline-block;\n",
       "}\n",
       "\n",
       ".xr-dim-list {\n",
       "  display: inline-block !important;\n",
       "  list-style: none;\n",
       "  padding: 0 !important;\n",
       "  margin: 0;\n",
       "}\n",
       "\n",
       ".xr-dim-list li {\n",
       "  display: inline-block;\n",
       "  padding: 0;\n",
       "  margin: 0;\n",
       "}\n",
       "\n",
       ".xr-dim-list:before {\n",
       "  content: \"(\";\n",
       "}\n",
       "\n",
       ".xr-dim-list:after {\n",
       "  content: \")\";\n",
       "}\n",
       "\n",
       ".xr-dim-list li:not(:last-child):after {\n",
       "  content: \",\";\n",
       "  padding-right: 5px;\n",
       "}\n",
       "\n",
       ".xr-has-index {\n",
       "  font-weight: bold;\n",
       "}\n",
       "\n",
       ".xr-var-list,\n",
       ".xr-var-item {\n",
       "  display: contents;\n",
       "}\n",
       "\n",
       ".xr-var-item > div,\n",
       ".xr-var-item label,\n",
       ".xr-var-item > .xr-var-name span {\n",
       "  background-color: var(--xr-background-color-row-even);\n",
       "  margin-bottom: 0;\n",
       "}\n",
       "\n",
       ".xr-var-item > .xr-var-name:hover span {\n",
       "  padding-right: 5px;\n",
       "}\n",
       "\n",
       ".xr-var-list > li:nth-child(odd) > div,\n",
       ".xr-var-list > li:nth-child(odd) > label,\n",
       ".xr-var-list > li:nth-child(odd) > .xr-var-name span {\n",
       "  background-color: var(--xr-background-color-row-odd);\n",
       "}\n",
       "\n",
       ".xr-var-name {\n",
       "  grid-column: 1;\n",
       "}\n",
       "\n",
       ".xr-var-dims {\n",
       "  grid-column: 2;\n",
       "}\n",
       "\n",
       ".xr-var-dtype {\n",
       "  grid-column: 3;\n",
       "  text-align: right;\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-var-preview {\n",
       "  grid-column: 4;\n",
       "}\n",
       "\n",
       ".xr-index-preview {\n",
       "  grid-column: 2 / 5;\n",
       "  color: var(--xr-font-color2);\n",
       "}\n",
       "\n",
       ".xr-var-name,\n",
       ".xr-var-dims,\n",
       ".xr-var-dtype,\n",
       ".xr-preview,\n",
       ".xr-attrs dt {\n",
       "  white-space: nowrap;\n",
       "  overflow: hidden;\n",
       "  text-overflow: ellipsis;\n",
       "  padding-right: 10px;\n",
       "}\n",
       "\n",
       ".xr-var-name:hover,\n",
       ".xr-var-dims:hover,\n",
       ".xr-var-dtype:hover,\n",
       ".xr-attrs dt:hover {\n",
       "  overflow: visible;\n",
       "  width: auto;\n",
       "  z-index: 1;\n",
       "}\n",
       "\n",
       ".xr-var-attrs,\n",
       ".xr-var-data,\n",
       ".xr-index-data {\n",
       "  display: none;\n",
       "  background-color: var(--xr-background-color) !important;\n",
       "  padding-bottom: 5px !important;\n",
       "}\n",
       "\n",
       ".xr-var-attrs-in:checked ~ .xr-var-attrs,\n",
       ".xr-var-data-in:checked ~ .xr-var-data,\n",
       ".xr-index-data-in:checked ~ .xr-index-data {\n",
       "  display: block;\n",
       "}\n",
       "\n",
       ".xr-var-data > table {\n",
       "  float: right;\n",
       "}\n",
       "\n",
       ".xr-var-name span,\n",
       ".xr-var-data,\n",
       ".xr-index-name div,\n",
       ".xr-index-data,\n",
       ".xr-attrs {\n",
       "  padding-left: 25px !important;\n",
       "}\n",
       "\n",
       ".xr-attrs,\n",
       ".xr-var-attrs,\n",
       ".xr-var-data,\n",
       ".xr-index-data {\n",
       "  grid-column: 1 / -1;\n",
       "}\n",
       "\n",
       "dl.xr-attrs {\n",
       "  padding: 0;\n",
       "  margin: 0;\n",
       "  display: grid;\n",
       "  grid-template-columns: 125px auto;\n",
       "}\n",
       "\n",
       ".xr-attrs dt,\n",
       ".xr-attrs dd {\n",
       "  padding: 0;\n",
       "  margin: 0;\n",
       "  float: left;\n",
       "  padding-right: 10px;\n",
       "  width: auto;\n",
       "}\n",
       "\n",
       ".xr-attrs dt {\n",
       "  font-weight: normal;\n",
       "  grid-column: 1;\n",
       "}\n",
       "\n",
       ".xr-attrs dt:hover span {\n",
       "  display: inline-block;\n",
       "  background: var(--xr-background-color);\n",
       "  padding-right: 10px;\n",
       "}\n",
       "\n",
       ".xr-attrs dd {\n",
       "  grid-column: 2;\n",
       "  white-space: pre-wrap;\n",
       "  word-break: break-all;\n",
       "}\n",
       "\n",
       ".xr-icon-database,\n",
       ".xr-icon-file-text2,\n",
       ".xr-no-icon {\n",
       "  display: inline-block;\n",
       "  vertical-align: middle;\n",
       "  width: 1em;\n",
       "  height: 1.5em !important;\n",
       "  stroke-width: 0;\n",
       "  stroke: currentColor;\n",
       "  fill: currentColor;\n",
       "}\n",
       "</style><pre class='xr-text-repr-fallback'>&lt;xarray.DataArray &#x27;stat_diversity&#x27; (windows: 1941)&gt; Size: 16kB\n",
       "dask.array&lt;getitem, shape=(1941,), dtype=float64, chunksize=(175,), chunktype=numpy.ndarray&gt;\n",
       "Dimensions without coordinates: windows\n",
       "Attributes:\n",
       "    comment:  Genetic diversity (also known as &quot;Tajima’s pi&quot;) for cohorts.</pre><div class='xr-wrap' style='display:none'><div class='xr-header'><div class='xr-obj-type'>xarray.DataArray</div><div class='xr-array-name'>'stat_diversity'</div><ul class='xr-dim-list'><li><span>windows</span>: 1941</li></ul></div><ul class='xr-sections'><li class='xr-section-item'><div class='xr-array-wrap'><input id='section-8c6bd9fc-8b33-44e3-8b12-ada2207f7f83' class='xr-array-in' type='checkbox' checked><label for='section-8c6bd9fc-8b33-44e3-8b12-ada2207f7f83' title='Show/hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-array-preview xr-preview'><span>dask.array&lt;chunksize=(125,), meta=np.ndarray&gt;</span></div><div class='xr-array-data'><table>\n",
       "    <tr>\n",
       "        <td>\n",
       "            <table style=\"border-collapse: collapse;\">\n",
       "                <thead>\n",
       "                    <tr>\n",
       "                        <td> </td>\n",
       "                        <th> Array </th>\n",
       "                        <th> Chunk </th>\n",
       "                    </tr>\n",
       "                </thead>\n",
       "                <tbody>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Bytes </th>\n",
       "                        <td> 15.16 kiB </td>\n",
       "                        <td> 1.37 kiB </td>\n",
       "                    </tr>\n",
       "                    \n",
       "                    <tr>\n",
       "                        <th> Shape </th>\n",
       "                        <td> (1941,) </td>\n",
       "                        <td> (175,) </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Dask graph </th>\n",
       "                        <td colspan=\"2\"> 13 chunks in 32 graph layers </td>\n",
       "                    </tr>\n",
       "                    <tr>\n",
       "                        <th> Data type </th>\n",
       "                        <td colspan=\"2\"> float64 numpy.ndarray </td>\n",
       "                    </tr>\n",
       "                </tbody>\n",
       "            </table>\n",
       "        </td>\n",
       "        <td>\n",
       "        <svg width=\"170\" height=\"75\" style=\"stroke:rgb(0,0,0);stroke-width:1\" >\n",
       "\n",
       "  <!-- Horizontal lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"120\" y2=\"0\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"0\" y1=\"25\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Vertical lines -->\n",
       "  <line x1=\"0\" y1=\"0\" x2=\"0\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "  <line x1=\"7\" y1=\"0\" x2=\"7\" y2=\"25\" />\n",
       "  <line x1=\"18\" y1=\"0\" x2=\"18\" y2=\"25\" />\n",
       "  <line x1=\"29\" y1=\"0\" x2=\"29\" y2=\"25\" />\n",
       "  <line x1=\"37\" y1=\"0\" x2=\"37\" y2=\"25\" />\n",
       "  <line x1=\"47\" y1=\"0\" x2=\"47\" y2=\"25\" />\n",
       "  <line x1=\"57\" y1=\"0\" x2=\"57\" y2=\"25\" />\n",
       "  <line x1=\"67\" y1=\"0\" x2=\"67\" y2=\"25\" />\n",
       "  <line x1=\"78\" y1=\"0\" x2=\"78\" y2=\"25\" />\n",
       "  <line x1=\"86\" y1=\"0\" x2=\"86\" y2=\"25\" />\n",
       "  <line x1=\"94\" y1=\"0\" x2=\"94\" y2=\"25\" />\n",
       "  <line x1=\"103\" y1=\"0\" x2=\"103\" y2=\"25\" />\n",
       "  <line x1=\"112\" y1=\"0\" x2=\"112\" y2=\"25\" />\n",
       "  <line x1=\"120\" y1=\"0\" x2=\"120\" y2=\"25\" style=\"stroke-width:2\" />\n",
       "\n",
       "  <!-- Colored Rectangle -->\n",
       "  <polygon points=\"0.0,0.0 120.0,0.0 120.0,25.412616514582485 0.0,25.412616514582485\" style=\"fill:#ECB172A0;stroke-width:0\"/>\n",
       "\n",
       "  <!-- Text -->\n",
       "  <text x=\"60.000000\" y=\"45.412617\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" >1941</text>\n",
       "  <text x=\"140.000000\" y=\"12.706308\" font-size=\"1.0rem\" font-weight=\"100\" text-anchor=\"middle\" transform=\"rotate(0,140.000000,12.706308)\">1</text>\n",
       "</svg>\n",
       "        </td>\n",
       "    </tr>\n",
       "</table></div></div></li><li class='xr-section-item'><input id='section-ec7f134f-e76a-4f44-9a11-35db63e88600' class='xr-section-summary-in' type='checkbox' disabled ><label for='section-ec7f134f-e76a-4f44-9a11-35db63e88600' class='xr-section-summary'  title='Expand/collapse section'>Coordinates: <span>(0)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><ul class='xr-var-list'></ul></div></li><li class='xr-section-item'><input id='section-f51295c9-2fa8-4f2b-ae5c-416cd2532ac0' class='xr-section-summary-in' type='checkbox' disabled ><label for='section-f51295c9-2fa8-4f2b-ae5c-416cd2532ac0' class='xr-section-summary'  title='Expand/collapse section'>Indexes: <span>(0)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><ul class='xr-var-list'></ul></div></li><li class='xr-section-item'><input id='section-1d096013-52a4-442f-a8a1-6a0333d9b66d' class='xr-section-summary-in' type='checkbox'  checked><label for='section-1d096013-52a4-442f-a8a1-6a0333d9b66d' class='xr-section-summary' >Attributes: <span>(1)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><dl class='xr-attrs'><dt><span>comment :</span></dt><dd>Genetic diversity (also known as &quot;Tajima’s pi&quot;) for cohorts.</dd></dl></div></li></ul></div></div>"
      ],
      "text/plain": [
       "<xarray.DataArray 'stat_diversity' (windows: 1941)> Size: 16kB\n",
       "dask.array<getitem, shape=(1941,), dtype=float64, chunksize=(175,), chunktype=numpy.ndarray>\n",
       "Dimensions without coordinates: windows\n",
       "Attributes:\n",
       "    comment:  Genetic diversity (also known as \"Tajima’s pi\") for cohorts."
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ds.stat_diversity[:,0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "id": "ce1e7ce0",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>het</th>\n",
       "      <th>GVCF_ID</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>154.0</td>\n",
       "      <td>PD_0840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>83.0</td>\n",
       "      <td>PD_0840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>100.0</td>\n",
       "      <td>PD_0840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>39.0</td>\n",
       "      <td>PD_0840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>110.0</td>\n",
       "      <td>PD_0840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1936</th>\n",
       "      <td>97.0</td>\n",
       "      <td>PD_0840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1937</th>\n",
       "      <td>92.0</td>\n",
       "      <td>PD_0840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1938</th>\n",
       "      <td>96.0</td>\n",
       "      <td>PD_0840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1939</th>\n",
       "      <td>111.0</td>\n",
       "      <td>PD_0840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1940</th>\n",
       "      <td>79.0</td>\n",
       "      <td>PD_0840</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1941 rows × 2 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "        het  GVCF_ID\n",
       "0     154.0  PD_0840\n",
       "1      83.0  PD_0840\n",
       "2     100.0  PD_0840\n",
       "3      39.0  PD_0840\n",
       "4     110.0  PD_0840\n",
       "...     ...      ...\n",
       "1936   97.0  PD_0840\n",
       "1937   92.0  PD_0840\n",
       "1938   96.0  PD_0840\n",
       "1939  111.0  PD_0840\n",
       "1940   79.0  PD_0840\n",
       "\n",
       "[1941 rows x 2 columns]"
      ]
     },
     "execution_count": 15,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "pd.DataFrame({\"het\": ds.stat_diversity[:,0], \"GVCF_ID\": ds.sample_id[0].values})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "70666f1b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PD_0840</th>\n",
       "      <th>PD_0841</th>\n",
       "      <th>PD_0844</th>\n",
       "      <th>SAMN31952328</th>\n",
       "      <th>window_start</th>\n",
       "      <th>chrom</th>\n",
       "      <th>window_end</th>\n",
       "      <th>callable_frac</th>\n",
       "      <th>chr_type</th>\n",
       "      <th>species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.0</td>\n",
       "      <td>235.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>HiC_scaffold_23</td>\n",
       "      <td>100000</td>\n",
       "      <td>0.93982</td>\n",
       "      <td>chrX</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>194.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100000</td>\n",
       "      <td>HiC_scaffold_23</td>\n",
       "      <td>200000</td>\n",
       "      <td>0.94975</td>\n",
       "      <td>chrX</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>194.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>200000</td>\n",
       "      <td>HiC_scaffold_23</td>\n",
       "      <td>300000</td>\n",
       "      <td>0.96536</td>\n",
       "      <td>chrX</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>300000</td>\n",
       "      <td>HiC_scaffold_23</td>\n",
       "      <td>400000</td>\n",
       "      <td>0.95552</td>\n",
       "      <td>chrX</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>222.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>400000</td>\n",
       "      <td>HiC_scaffold_23</td>\n",
       "      <td>500000</td>\n",
       "      <td>0.94898</td>\n",
       "      <td>chrX</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3227</th>\n",
       "      <td>97.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>193600000</td>\n",
       "      <td>HiC_scaffold_1</td>\n",
       "      <td>193700000</td>\n",
       "      <td>0.96172</td>\n",
       "      <td>aut</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3228</th>\n",
       "      <td>92.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>129.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>193700000</td>\n",
       "      <td>HiC_scaffold_1</td>\n",
       "      <td>193800000</td>\n",
       "      <td>0.95889</td>\n",
       "      <td>aut</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3229</th>\n",
       "      <td>96.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>193800000</td>\n",
       "      <td>HiC_scaffold_1</td>\n",
       "      <td>193900000</td>\n",
       "      <td>0.96756</td>\n",
       "      <td>aut</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3230</th>\n",
       "      <td>111.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>93.0</td>\n",
       "      <td>102.0</td>\n",
       "      <td>193900000</td>\n",
       "      <td>HiC_scaffold_1</td>\n",
       "      <td>194000000</td>\n",
       "      <td>0.96973</td>\n",
       "      <td>aut</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3231</th>\n",
       "      <td>79.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>194000000</td>\n",
       "      <td>HiC_scaffold_1</td>\n",
       "      <td>194100000</td>\n",
       "      <td>0.93039</td>\n",
       "      <td>aut</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3232 rows × 10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      PD_0840  PD_0841  PD_0844  SAMN31952328  window_start            chrom  \\\n",
       "0         0.0    235.0      0.0           0.0             0  HiC_scaffold_23   \n",
       "1         0.0    194.0      0.0           0.0        100000  HiC_scaffold_23   \n",
       "2         0.0    194.0      0.0           0.0        200000  HiC_scaffold_23   \n",
       "3         0.0    170.0      0.0           0.0        300000  HiC_scaffold_23   \n",
       "4         0.0    222.0      0.0           0.0        400000  HiC_scaffold_23   \n",
       "...       ...      ...      ...           ...           ...              ...   \n",
       "3227     97.0     27.0     49.0          34.0     193600000   HiC_scaffold_1   \n",
       "3228     92.0     21.0    129.0          24.0     193700000   HiC_scaffold_1   \n",
       "3229     96.0     15.0    124.0          20.0     193800000   HiC_scaffold_1   \n",
       "3230    111.0     15.0     93.0         102.0     193900000   HiC_scaffold_1   \n",
       "3231     79.0      1.0      8.0           7.0     194000000   HiC_scaffold_1   \n",
       "\n",
       "      window_end  callable_frac chr_type                         species  \n",
       "0         100000        0.93982     chrX  Leontopithecus_chrysomelas_ssp  \n",
       "1         200000        0.94975     chrX  Leontopithecus_chrysomelas_ssp  \n",
       "2         300000        0.96536     chrX  Leontopithecus_chrysomelas_ssp  \n",
       "3         400000        0.95552     chrX  Leontopithecus_chrysomelas_ssp  \n",
       "4         500000        0.94898     chrX  Leontopithecus_chrysomelas_ssp  \n",
       "...          ...            ...      ...                             ...  \n",
       "3227   193700000        0.96172      aut  Leontopithecus_chrysomelas_ssp  \n",
       "3228   193800000        0.95889      aut  Leontopithecus_chrysomelas_ssp  \n",
       "3229   193900000        0.96756      aut  Leontopithecus_chrysomelas_ssp  \n",
       "3230   194000000        0.96973      aut  Leontopithecus_chrysomelas_ssp  \n",
       "3231   194100000        0.93039      aut  Leontopithecus_chrysomelas_ssp  \n",
       "\n",
       "[3232 rows x 10 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "output_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c38d6fd6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Leontopithecus_chrysomelas_ssp\n",
      "Leontopithecus_chrysomelas_ssp\n",
      "4 Leontopithecus_chrysomelas_ssp\n",
      "../zarr_data/Leontopithecus_chrysomelas_ssp/HiC_scaffold_23\n",
      "../zarr_data/Leontopithecus_chrysomelas_ssp/HiC_scaffold_1\n",
      "../zarr_data/Saguinus_labiatus_ssp\n",
      "Saguinus_labiatus_ssp\n",
      "2 Saguinus_labiatus_ssp\n",
      "../zarr_data/Saguinus_labiatus_ssp/CM063191.1\n",
      "../zarr_data/Saguinus_labiatus_ssp/JASSZA010000432.1\n",
      "../zarr_data/Saguinus_labiatus_ssp/CM063169.1\n",
      "../zarr_data/Plecturocebus_cupreus_ssp\n",
      "Plecturocebus_cupreus_ssp\n",
      "5 Plecturocebus_cupreus_ssp\n",
      "../zarr_data/Plecturocebus_cupreus_ssp/CM080837.1\n",
      "../zarr_data/Plecturocebus_cupreus_ssp/CM080815.1\n",
      "../zarr_data/Papio_hamadryas_ssp\n",
      "Papio_hamadryas_ssp\n",
      "47 Papio_hamadryas_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_fascicularis_ssp\n",
      "Macaca_fascicularis_ssp\n",
      "159 Macaca_fascicularis_ssp\n",
      "Skipping\n",
      "../zarr_data/Saimiri_macrodon_ssp\n",
      "Saimiri_macrodon_ssp\n",
      "3 Saimiri_macrodon_ssp\n",
      "../zarr_data/Saimiri_macrodon_ssp/NW_024100917.1\n",
      "../zarr_data/Saimiri_macrodon_ssp/NW_024100929.1\n",
      "../zarr_data/Saimiri_macrodon_ssp/NW_024100959.1\n",
      "../zarr_data/Cheirogaleus_crossleyi_ssp\n",
      "Cheirogaleus_crossleyi_ssp\n",
      "1 Cheirogaleus_crossleyi_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheirogaleus_medius_ssp\n",
      "Cheirogaleus_medius_ssp\n",
      "5 Cheirogaleus_medius_ssp\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01001483.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01002707.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01002350.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01000656.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01003151.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01000568.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01000937.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01001044.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01003018.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01003185.1\n",
      "../zarr_data/Plecturocebus_caligatus_ssp\n",
      "Plecturocebus_caligatus_ssp\n",
      "3 Plecturocebus_caligatus_ssp\n",
      "../zarr_data/Plecturocebus_caligatus_ssp/CM080815.1\n",
      "../zarr_data/Plecturocebus_caligatus_ssp/CM080837.1\n",
      "../zarr_data/Callimico_goeldii_ssp\n",
      "Callimico_goeldii_ssp\n",
      "2 Callimico_goeldii_ssp\n",
      "../zarr_data/Callimico_goeldii_ssp/HiC_scaffold_1\n",
      "../zarr_data/Trachypithecus_laotum_ssp\n",
      "Trachypithecus_laotum_ssp\n",
      "1 Trachypithecus_laotum_ssp\n",
      "Skipping\n",
      "../zarr_data/Lepilemur_dorsalis_ssp\n",
      "Lepilemur_dorsalis_ssp\n",
      "2 Lepilemur_dorsalis_ssp\n",
      "../zarr_data/Lepilemur_dorsalis_ssp/NC_033692.1\n",
      "../zarr_data/Lepilemur_dorsalis_ssp/NC_033660.1\n",
      "../zarr_data/Miopithecus_ogouensis_ssp\n",
      "Miopithecus_ogouensis_ssp\n",
      "1 Miopithecus_ogouensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Tarsius_lariang_ssp\n",
      "Tarsius_lariang_ssp\n",
      "1 Tarsius_lariang_ssp\n",
      "Skipping\n",
      "../zarr_data/Mirza_coquereli_ssp\n",
      "Mirza_coquereli_ssp\n",
      "1 Mirza_coquereli_ssp\n",
      "Skipping\n",
      "../zarr_data/Ateles_belzebuth_ssp\n",
      "Ateles_belzebuth_ssp\n",
      "2 Ateles_belzebuth_ssp\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp\n",
      "Semnopithecus_schistaceus_ssp\n",
      "2 Semnopithecus_schistaceus_ssp\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022680471.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022680461.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022681471.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022681432.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022681445.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022681437.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022680465.1\n",
      "../zarr_data/Saimiri_oerstedii_ssp\n",
      "Saimiri_oerstedii_ssp\n",
      "1 Saimiri_oerstedii_ssp\n",
      "Skipping\n",
      "../zarr_data/Piliocolobus_badius_ssp\n",
      "Piliocolobus_badius_ssp\n",
      "1 Piliocolobus_badius_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_brunneus_ssp\n",
      "Plecturocebus_brunneus_ssp\n",
      "2 Plecturocebus_brunneus_ssp\n",
      "../zarr_data/Plecturocebus_brunneus_ssp/CM080815.1\n",
      "../zarr_data/Plecturocebus_brunneus_ssp/CM080837.1\n",
      "../zarr_data/Cebus_imitator_ssp\n",
      "Cebus_imitator_ssp\n",
      "2 Cebus_imitator_ssp\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437082.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437065.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437044.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022436957.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437089.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437104.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437055.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437081.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437015.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437116.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437119.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437031.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437094.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437042.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437043.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022436941.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437085.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437013.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437131.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437038.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437033.1\n",
      "../zarr_data/Perodicticus_potto_ssp\n",
      "Perodicticus_potto_ssp\n",
      "3 Perodicticus_potto_ssp\n",
      "../zarr_data/Perodicticus_potto_ssp/NC_069780.1\n",
      "../zarr_data/Perodicticus_potto_ssp/NC_069804.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp\n",
      "Trachypithecus_poliocephalus_ssp\n",
      "12 Trachypithecus_poliocephalus_ssp\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022681471.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022681432.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022680471.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022680461.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022680465.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022681445.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022681437.1\n",
      "../zarr_data/Symphalangus_syndactylus_ssp\n",
      "Symphalangus_syndactylus_ssp\n",
      "5 Symphalangus_syndactylus_ssp\n",
      "../zarr_data/Symphalangus_syndactylus_ssp/NC_072423.2\n",
      "../zarr_data/Symphalangus_syndactylus_ssp/NC_072447.2\n",
      "../zarr_data/Pongo_pygmaeus_ssp\n",
      "Pongo_pygmaeus_ssp\n",
      "26 Pongo_pygmaeus_ssp\n",
      "Skipping\n",
      "../zarr_data/Avahi_laniger_ssp\n",
      "Avahi_laniger_ssp\n",
      "2 Avahi_laniger_ssp\n",
      "../zarr_data/Avahi_laniger_ssp/HiC_scaffold_2\n",
      "../zarr_data/Avahi_laniger_ssp/HiC_scaffold_1\n",
      "../zarr_data/Lepilemur_sahamalazensis_ssp\n",
      "Lepilemur_sahamalazensis_ssp\n",
      "1 Lepilemur_sahamalazensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Allenopithecus_nigroviridis_ssp\n",
      "Allenopithecus_nigroviridis_ssp\n",
      "3 Allenopithecus_nigroviridis_ssp\n",
      "../zarr_data/Allenopithecus_nigroviridis_ssp/HiC_scaffold_1\n",
      "../zarr_data/Allenopithecus_nigroviridis_ssp/HiC_scaffold_2\n",
      "../zarr_data/Alouatta_belzebul_ssp\n",
      "Alouatta_belzebul_ssp\n",
      "3 Alouatta_belzebul_ssp\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Cercopithecus_denti_ssp\n",
      "Cercopithecus_denti_ssp\n",
      "2 Cercopithecus_denti_ssp\n",
      "../zarr_data/Cercopithecus_denti_ssp/CM053398.1\n",
      "../zarr_data/Cercopithecus_denti_ssp/CM053363.1\n",
      "../zarr_data/Saimiri_ustus_ssp\n",
      "Saimiri_ustus_ssp\n",
      "10 Saimiri_ustus_ssp\n",
      "../zarr_data/Saimiri_ustus_ssp/NW_024100917.1\n",
      "../zarr_data/Saimiri_ustus_ssp/NW_024100929.1\n",
      "../zarr_data/Saimiri_ustus_ssp/NW_024100959.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp\n",
      "Trachypithecus_vetulus_ssp\n",
      "3 Trachypithecus_vetulus_ssp\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022680461.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022680471.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022681432.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022681471.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022681437.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022681445.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022680465.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp\n",
      "Trachypithecus_francoisi_ssp\n",
      "9 Trachypithecus_francoisi_ssp\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681432.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681471.1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022680461.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022680471.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022680465.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681437.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681445.1\n",
      "../zarr_data/Nomascus_gabriellae_ssp\n",
      "Nomascus_gabriellae_ssp\n",
      "1 Nomascus_gabriellae_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_hamlyni_ssp\n",
      "Cercopithecus_hamlyni_ssp\n",
      "2 Cercopithecus_hamlyni_ssp\n",
      "../zarr_data/Cercopithecus_hamlyni_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_hamlyni_ssp/CM053398.1\n",
      "../zarr_data/Leontocebus_nigricollis_ssp\n",
      "Leontocebus_nigricollis_ssp\n",
      "1 Leontocebus_nigricollis_ssp\n",
      "Skipping\n",
      "../zarr_data/Chiropotes_albinasus_ssp\n",
      "Chiropotes_albinasus_ssp\n",
      "4 Chiropotes_albinasus_ssp\n",
      "../zarr_data/Chiropotes_albinasus_ssp/CM052647.1\n",
      "../zarr_data/Chiropotes_albinasus_ssp/CM052624.1\n",
      "../zarr_data/Macaca_tonkeana_ssp\n",
      "Macaca_tonkeana_ssp\n",
      "8 Macaca_tonkeana_ssp\n",
      "../zarr_data/Macaca_tonkeana_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_tonkeana_ssp/NC_092125.1\n",
      "../zarr_data/Macaca_tonkeana_ssp/NW_027257673.1\n",
      "../zarr_data/Gorilla_beringei_ssp\n",
      "Gorilla_beringei_ssp\n",
      "28 Gorilla_beringei_ssp\n",
      "Skipping\n",
      "../zarr_data/Cebus_albifrons_ssp\n",
      "Cebus_albifrons_ssp\n",
      "5 Cebus_albifrons_ssp\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437031.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437119.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437042.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437094.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437043.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437085.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022436941.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437013.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437038.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437131.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437033.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437082.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437044.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437065.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437089.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022436957.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437055.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437104.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437015.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437081.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437116.1\n",
      "../zarr_data/Mico_humeralifer_ssp\n",
      "Mico_humeralifer_ssp\n",
      "1 Mico_humeralifer_ssp\n",
      "Skipping\n",
      "../zarr_data/Cacajao_calvus_ssp\n",
      "Cacajao_calvus_ssp\n",
      "19 Cacajao_calvus_ssp\n",
      "../zarr_data/Cacajao_calvus_ssp/CM052624.1\n",
      "../zarr_data/Cacajao_calvus_ssp/CM052647.1\n",
      "../zarr_data/Macaca_nemestrina_ssp\n",
      "Macaca_nemestrina_ssp\n",
      "26 Macaca_nemestrina_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_sinica_ssp\n",
      "Macaca_sinica_ssp\n",
      "1 Macaca_sinica_ssp\n",
      "Skipping\n",
      "../zarr_data/Aotus_griseimembra_ssp\n",
      "Aotus_griseimembra_ssp\n",
      "4 Aotus_griseimembra_ssp\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988472.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988412.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988496.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988488.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988415.1\n",
      "../zarr_data/Cercopithecus_neglectus_ssp\n",
      "Cercopithecus_neglectus_ssp\n",
      "3 Cercopithecus_neglectus_ssp\n",
      "../zarr_data/Cercopithecus_neglectus_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_neglectus_ssp/CM053398.1\n",
      "../zarr_data/Pithecia_vanzolinii_ssp\n",
      "Pithecia_vanzolinii_ssp\n",
      "1 Pithecia_vanzolinii_ssp\n",
      "Skipping\n",
      "../zarr_data/Cebus_olivaceus_ssp\n",
      "Cebus_olivaceus_ssp\n",
      "2 Cebus_olivaceus_ssp\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437081.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437015.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437116.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022436957.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437089.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437104.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437055.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437082.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437065.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437044.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437033.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437013.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437131.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437038.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437043.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022436941.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437085.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437119.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437031.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437094.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437042.1\n",
      "../zarr_data/Semnopithecus_priam_ssp\n",
      "Semnopithecus_priam_ssp\n",
      "1 Semnopithecus_priam_ssp\n",
      "Skipping\n",
      "../zarr_data/Semnopithecus_entellus_ssp\n",
      "Semnopithecus_entellus_ssp\n",
      "9 Semnopithecus_entellus_ssp\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022680465.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681445.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681437.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681471.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681432.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022680461.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022680471.1\n",
      "../zarr_data/Hylobates_pileatus_ssp\n",
      "Hylobates_pileatus_ssp\n",
      "3 Hylobates_pileatus_ssp\n",
      "../zarr_data/Hylobates_pileatus_ssp/CM038521.1\n",
      "../zarr_data/Hylobates_pileatus_ssp/CM038516.1\n",
      "../zarr_data/Galago_moholi_ssp\n",
      "Galago_moholi_ssp\n",
      "2 Galago_moholi_ssp\n",
      "../zarr_data/Galago_moholi_ssp/HiC_scaffold_31\n",
      "../zarr_data/Galago_moholi_ssp/HiC_scaffold_1\n",
      "../zarr_data/Rhinopithecus_strykeri_ssp\n",
      "Rhinopithecus_strykeri_ssp\n",
      "3 Rhinopithecus_strykeri_ssp\n",
      "../zarr_data/Rhinopithecus_strykeri_ssp/NC_044549.1\n",
      "../zarr_data/Rhinopithecus_strykeri_ssp/NC_044555.1\n",
      "../zarr_data/Otolemur_garnettii_ssp\n",
      "Otolemur_garnettii_ssp\n",
      "40 Otolemur_garnettii_ssp\n",
      "Skipping\n",
      "../zarr_data/Colobus_polykomos_ssp\n",
      "Colobus_polykomos_ssp\n",
      "1 Colobus_polykomos_ssp\n",
      "Skipping\n",
      "../zarr_data/Alouatta_discolor_ssp\n",
      "Alouatta_discolor_ssp\n",
      "2 Alouatta_discolor_ssp\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Macaca_leucogenys_ssp\n",
      "Macaca_leucogenys_ssp\n",
      "4 Macaca_leucogenys_ssp\n",
      "../zarr_data/Macaca_leucogenys_ssp/NC_065578.1\n",
      "../zarr_data/Macaca_leucogenys_ssp/NC_065597.1\n",
      "../zarr_data/Macaca_leucogenys_ssp/NC_065598.1\n",
      "../zarr_data/Cercopithecus_mona_ssp\n",
      "Cercopithecus_mona_ssp\n",
      "15 Cercopithecus_mona_ssp\n",
      "../zarr_data/Cercopithecus_mona_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_mona_ssp/CM053398.1\n",
      "../zarr_data/Ateles_marginatus_ssp\n",
      "Ateles_marginatus_ssp\n",
      "1 Ateles_marginatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Callithrix_kuhlii_ssp\n",
      "Callithrix_kuhlii_ssp\n",
      "1 Callithrix_kuhlii_ssp\n",
      "Skipping\n",
      "../zarr_data/Propithecus_diadema_ssp\n",
      "Propithecus_diadema_ssp\n",
      "7 Propithecus_diadema_ssp\n",
      "../zarr_data/Propithecus_diadema_ssp/HiC_scaffold_2\n",
      "../zarr_data/Propithecus_diadema_ssp/HiC_scaffold_1\n",
      "../zarr_data/Rhinopithecus_brelichi_ssp\n",
      "Rhinopithecus_brelichi_ssp\n",
      "5 Rhinopithecus_brelichi_ssp\n",
      "../zarr_data/Rhinopithecus_brelichi_ssp/NC_044549.1\n",
      "../zarr_data/Rhinopithecus_brelichi_ssp/NC_044555.1\n",
      "../zarr_data/Eulemur_coronatus_ssp\n",
      "Eulemur_coronatus_ssp\n",
      "6 Eulemur_coronatus_ssp\n",
      "../zarr_data/Eulemur_coronatus_ssp/HiC_scaffold_1\n",
      "../zarr_data/Macaca_brunnescens_ssp\n",
      "Macaca_brunnescens_ssp\n",
      "1 Macaca_brunnescens_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_ascanius_ssp\n",
      "Cercopithecus_ascanius_ssp\n",
      "5 Cercopithecus_ascanius_ssp\n",
      "../zarr_data/Cercopithecus_ascanius_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_ascanius_ssp/CM053398.1\n",
      "../zarr_data/Callithrix_jacchus_ssp\n",
      "Callithrix_jacchus_ssp\n",
      "12 Callithrix_jacchus_ssp\n",
      "../zarr_data/Callithrix_jacchus_ssp/NC_071442.1\n",
      "../zarr_data/Callithrix_jacchus_ssp/NC_071464.1\n",
      "../zarr_data/Avahi_peyrierasi_ssp\n",
      "Avahi_peyrierasi_ssp\n",
      "2 Avahi_peyrierasi_ssp\n",
      "../zarr_data/Avahi_peyrierasi_ssp/HiC_scaffold_2\n",
      "../zarr_data/Avahi_peyrierasi_ssp/HiC_scaffold_1\n",
      "../zarr_data/Tarsius_dentatus_ssp\n",
      "Tarsius_dentatus_ssp\n",
      "1 Tarsius_dentatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Indri_indri_ssp\n",
      "Indri_indri_ssp\n",
      "2 Indri_indri_ssp\n",
      "../zarr_data/Indri_indri_ssp/HiC_scaffold_2\n",
      "../zarr_data/Indri_indri_ssp/HiC_scaffold_1\n",
      "../zarr_data/Macaca_silenus_ssp\n",
      "Macaca_silenus_ssp\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Macaca_silenus_ssp\n",
      "../zarr_data/Macaca_silenus_ssp/NW_027257673.1\n",
      "../zarr_data/Macaca_silenus_ssp/NC_092125.1\n",
      "../zarr_data/Macaca_silenus_ssp/NC_092145.1\n",
      "../zarr_data/Allochrocebus_preussi_ssp\n",
      "Allochrocebus_preussi_ssp\n",
      "1 Allochrocebus_preussi_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_petaurista_ssp\n",
      "Cercopithecus_petaurista_ssp\n",
      "1 Cercopithecus_petaurista_ssp\n",
      "Skipping\n",
      "../zarr_data/Mico_argentatus_ssp\n",
      "Mico_argentatus_ssp\n",
      "4 Mico_argentatus_ssp\n",
      "../zarr_data/Mico_argentatus_ssp/NC_071464.1\n",
      "../zarr_data/Mico_argentatus_ssp/NC_071442.1\n",
      "../zarr_data/Plecturocebus_hoffmannsi_ssp\n",
      "Plecturocebus_hoffmannsi_ssp\n",
      "1 Plecturocebus_hoffmannsi_ssp\n",
      "Skipping\n",
      "../zarr_data/Alouatta_juara_ssp\n",
      "Alouatta_juara_ssp\n",
      "2 Alouatta_juara_ssp\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Hylobates_klossii_ssp\n",
      "Hylobates_klossii_ssp\n",
      "1 Hylobates_klossii_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheracebus_lucifer_ssp\n",
      "Cheracebus_lucifer_ssp\n",
      "4 Cheracebus_lucifer_ssp\n",
      "../zarr_data/Cheracebus_lucifer_ssp/CM080837.1\n",
      "../zarr_data/Cheracebus_lucifer_ssp/CM080815.1\n",
      "../zarr_data/Propithecus_deckenii_ssp\n",
      "Propithecus_deckenii_ssp\n",
      "1 Propithecus_deckenii_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheracebus_regulus_ssp\n",
      "Cheracebus_regulus_ssp\n",
      "2 Cheracebus_regulus_ssp\n",
      "../zarr_data/Cheracebus_regulus_ssp/CM080815.1\n",
      "../zarr_data/Cheracebus_regulus_ssp/CM080837.1\n",
      "../zarr_data/Alouatta_palliata_ssp\n",
      "Alouatta_palliata_ssp\n",
      "6 Alouatta_palliata_ssp\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Colobus_angolensis_ssp\n",
      "Colobus_angolensis_ssp\n",
      "1 Colobus_angolensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Chlorocebus_cynosuros_ssp\n",
      "Chlorocebus_cynosuros_ssp\n",
      "16 Chlorocebus_cynosuros_ssp\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "No objects to concatenate",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mValueError\u001b[39m                                Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[11]\u001b[39m\u001b[32m, line 45\u001b[39m\n\u001b[32m     43\u001b[39m     df_sub[\u001b[33m\"\u001b[39m\u001b[33mvariants_used\u001b[39m\u001b[33m\"\u001b[39m] = \u001b[38;5;28mlen\u001b[39m(ds.variants)\n\u001b[32m     44\u001b[39m     df_l.append(df_sub)\n\u001b[32m---> \u001b[39m\u001b[32m45\u001b[39m output_df = \u001b[43mpd\u001b[49m\u001b[43m.\u001b[49m\u001b[43mconcat\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdf_l\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m     46\u001b[39m output_df[\u001b[33m\"\u001b[39m\u001b[33mchr_type\u001b[39m\u001b[33m\"\u001b[39m] = output_df[\u001b[33m\"\u001b[39m\u001b[33mchrom\u001b[39m\u001b[33m\"\u001b[39m].map(\u001b[38;5;28mdict\u001b[39m(\u001b[38;5;28mzip\u001b[39m(regions_df.CONTIG_ID, regions_df.chr_type)))\n\u001b[32m     47\u001b[39m output_df[\u001b[33m\"\u001b[39m\u001b[33mspecies\u001b[39m\u001b[33m\"\u001b[39m] = long_form\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/primatediversity2/lib/python3.12/site-packages/pandas/core/reshape/concat.py:382\u001b[39m, in \u001b[36mconcat\u001b[39m\u001b[34m(objs, axis, join, ignore_index, keys, levels, names, verify_integrity, sort, copy)\u001b[39m\n\u001b[32m    379\u001b[39m \u001b[38;5;28;01melif\u001b[39;00m copy \u001b[38;5;129;01mand\u001b[39;00m using_copy_on_write():\n\u001b[32m    380\u001b[39m     copy = \u001b[38;5;28;01mFalse\u001b[39;00m\n\u001b[32m--> \u001b[39m\u001b[32m382\u001b[39m op = \u001b[43m_Concatenator\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    383\u001b[39m \u001b[43m    \u001b[49m\u001b[43mobjs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    384\u001b[39m \u001b[43m    \u001b[49m\u001b[43maxis\u001b[49m\u001b[43m=\u001b[49m\u001b[43maxis\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    385\u001b[39m \u001b[43m    \u001b[49m\u001b[43mignore_index\u001b[49m\u001b[43m=\u001b[49m\u001b[43mignore_index\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    386\u001b[39m \u001b[43m    \u001b[49m\u001b[43mjoin\u001b[49m\u001b[43m=\u001b[49m\u001b[43mjoin\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    387\u001b[39m \u001b[43m    \u001b[49m\u001b[43mkeys\u001b[49m\u001b[43m=\u001b[49m\u001b[43mkeys\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    388\u001b[39m \u001b[43m    \u001b[49m\u001b[43mlevels\u001b[49m\u001b[43m=\u001b[49m\u001b[43mlevels\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    389\u001b[39m \u001b[43m    \u001b[49m\u001b[43mnames\u001b[49m\u001b[43m=\u001b[49m\u001b[43mnames\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    390\u001b[39m \u001b[43m    \u001b[49m\u001b[43mverify_integrity\u001b[49m\u001b[43m=\u001b[49m\u001b[43mverify_integrity\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    391\u001b[39m \u001b[43m    \u001b[49m\u001b[43mcopy\u001b[49m\u001b[43m=\u001b[49m\u001b[43mcopy\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    392\u001b[39m \u001b[43m    \u001b[49m\u001b[43msort\u001b[49m\u001b[43m=\u001b[49m\u001b[43msort\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    393\u001b[39m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    395\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m op.get_result()\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/primatediversity2/lib/python3.12/site-packages/pandas/core/reshape/concat.py:445\u001b[39m, in \u001b[36m_Concatenator.__init__\u001b[39m\u001b[34m(self, objs, axis, join, keys, levels, names, ignore_index, verify_integrity, copy, sort)\u001b[39m\n\u001b[32m    442\u001b[39m \u001b[38;5;28mself\u001b[39m.verify_integrity = verify_integrity\n\u001b[32m    443\u001b[39m \u001b[38;5;28mself\u001b[39m.copy = copy\n\u001b[32m--> \u001b[39m\u001b[32m445\u001b[39m objs, keys = \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_clean_keys_and_objs\u001b[49m\u001b[43m(\u001b[49m\u001b[43mobjs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkeys\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    447\u001b[39m \u001b[38;5;66;03m# figure out what our result ndim is going to be\u001b[39;00m\n\u001b[32m    448\u001b[39m ndims = \u001b[38;5;28mself\u001b[39m._get_ndims(objs)\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/primatediversity2/lib/python3.12/site-packages/pandas/core/reshape/concat.py:507\u001b[39m, in \u001b[36m_Concatenator._clean_keys_and_objs\u001b[39m\u001b[34m(self, objs, keys)\u001b[39m\n\u001b[32m    504\u001b[39m     objs_list = \u001b[38;5;28mlist\u001b[39m(objs)\n\u001b[32m    506\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(objs_list) == \u001b[32m0\u001b[39m:\n\u001b[32m--> \u001b[39m\u001b[32m507\u001b[39m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[33m\"\u001b[39m\u001b[33mNo objects to concatenate\u001b[39m\u001b[33m\"\u001b[39m)\n\u001b[32m    509\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m keys \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[32m    510\u001b[39m     objs_list = \u001b[38;5;28mlist\u001b[39m(com.not_none(*objs_list))\n",
      "\u001b[31mValueError\u001b[39m: No objects to concatenate"
     ]
    }
   ],
   "source": [
    "# Fst implementation.\n",
    "for x in glob.glob(zarr_path+\"*\"):\n",
    "    print(x)\n",
    "    # The name used to load all the files, short and long version\n",
    "    short_form = x.split(\"/\")[-1].split(\"_\")[0]\n",
    "    long_form = x.split(\"/\")[-1]\n",
    "    print(long_form)\n",
    "    # Loading the various metadata files. Metadata, contig information, callability bed.\n",
    "    metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "    metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "    metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "    regions_df[\"chr_type\"] = [\"chrX\" if x == 2 and y == 1 else \"aut\" for x, y in zip(regions_df.FEMALE_PLOIDY, regions_df.MALE_PLOIDY)]\n",
    "    large_contigs = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "    large_x = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID\n",
    "    # Skipping the large samples sizes and the singulars for the Fst calc\n",
    "    print(len(metadata_df.loc[metadata_df.GVCF_FOLDER == long_form]), long_form)\n",
    "    if ((len(metadata_df.loc[metadata_df.GVCF_FOLDER == long_form]) > 20) or\n",
    "    (len(metadata_df.loc[metadata_df.GVCF_FOLDER == long_form]) == 1)):\n",
    "        print(\"Skipping\")\n",
    "        continue\n",
    "    # Loading the genetic data.\n",
    "    df_l = []\n",
    "    for c in glob.glob(x+\"/*\"):\n",
    "        print(c)\n",
    "        ds = sg.load_dataset(c)\n",
    "        # This implementation is the pi implementation.\n",
    "        # Probably problematic in some cases with population structure, but it is easier to implement\n",
    "        ds[\"sample_cohort\"] = ds[\"samples\"]\n",
    "        # Subsetting and windowing the sgkit dataset. The rechunking handles what otherwise would cause an error.\n",
    "        #ds[\"call_genotype\"] = ds[\"call_genotype\"].clip(0)\n",
    "        ds = ds.sel(contigs=[ds.variant_contig[0].values])\n",
    "        if c.split(\"/\")[-1] in list(large_x):\n",
    "            ds = haploid_double(ds, \"call_genotype\", \"samples\")\n",
    "        missing_rate = ds.call_genotype_mask[:,:,0].sum(axis=1).values/ds.call_genotype_mask[:,:,0].count(axis=1).values\n",
    "        ds = ds.isel(variants=(missing_rate <= 0))\n",
    "        ds = sg.window_by_genome(ds)\n",
    "        ds = (sg.Fst(ds.chunk({\"variants\": 50000})))\n",
    "        df_sub = pd.DataFrame(ds.stat_Fst[0,:,], columns=ds.sample_id)\n",
    "        df_sub[\"chrom\"] = c.split(\"/\")[-1]\n",
    "        df_sub[\"variants_used\"] = len(ds.variants)\n",
    "        df_l.append(df_sub)\n",
    "    output_df = pd.concat(df_l)\n",
    "    output_df[\"chr_type\"] = output_df[\"chrom\"].map(dict(zip(regions_df.CONTIG_ID, regions_df.chr_type)))\n",
    "    output_df[\"species\"] = long_form\n",
    "    output_df.to_csv(\"../results/window_stats/{}_Fst.txt\".format(long_form), sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "59a431b6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Leontopithecus_chrysomelas_ssp\n",
      "Leontopithecus_chrysomelas_ssp\n",
      "4 Leontopithecus_chrysomelas_ssp\n",
      "Skipping\n",
      "../zarr_data/Saguinus_labiatus_ssp\n",
      "Saguinus_labiatus_ssp\n",
      "2 Saguinus_labiatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_cupreus_ssp\n",
      "Plecturocebus_cupreus_ssp\n",
      "5 Plecturocebus_cupreus_ssp\n",
      "Skipping\n",
      "../zarr_data/Papio_hamadryas_ssp\n",
      "Papio_hamadryas_ssp\n",
      "47 Papio_hamadryas_ssp\n",
      "../zarr_data/Papio_hamadryas_ssp/NC_044976.1\n",
      "../zarr_data/Papio_hamadryas_ssp/NC_044996.1\n",
      "../zarr_data/Macaca_fascicularis_ssp\n",
      "Macaca_fascicularis_ssp\n",
      "159 Macaca_fascicularis_ssp\n",
      "Skipping\n",
      "../zarr_data/Saimiri_macrodon_ssp\n",
      "Saimiri_macrodon_ssp\n",
      "3 Saimiri_macrodon_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheirogaleus_crossleyi_ssp\n",
      "Cheirogaleus_crossleyi_ssp\n",
      "1 Cheirogaleus_crossleyi_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheirogaleus_medius_ssp\n",
      "Cheirogaleus_medius_ssp\n",
      "5 Cheirogaleus_medius_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_caligatus_ssp\n",
      "Plecturocebus_caligatus_ssp\n",
      "3 Plecturocebus_caligatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Callimico_goeldii_ssp\n",
      "Callimico_goeldii_ssp\n",
      "2 Callimico_goeldii_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_laotum_ssp\n",
      "Trachypithecus_laotum_ssp\n",
      "1 Trachypithecus_laotum_ssp\n",
      "Skipping\n",
      "../zarr_data/Lepilemur_dorsalis_ssp\n",
      "Lepilemur_dorsalis_ssp\n",
      "2 Lepilemur_dorsalis_ssp\n",
      "Skipping\n",
      "../zarr_data/Miopithecus_ogouensis_ssp\n",
      "Miopithecus_ogouensis_ssp\n",
      "1 Miopithecus_ogouensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Tarsius_lariang_ssp\n",
      "Tarsius_lariang_ssp\n",
      "1 Tarsius_lariang_ssp\n",
      "Skipping\n",
      "../zarr_data/Mirza_coquereli_ssp\n",
      "Mirza_coquereli_ssp\n",
      "1 Mirza_coquereli_ssp\n",
      "Skipping\n",
      "../zarr_data/Ateles_belzebuth_ssp\n",
      "Ateles_belzebuth_ssp\n",
      "2 Ateles_belzebuth_ssp\n",
      "Skipping\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp\n",
      "Semnopithecus_schistaceus_ssp\n",
      "2 Semnopithecus_schistaceus_ssp\n",
      "Skipping\n",
      "../zarr_data/Saimiri_oerstedii_ssp\n",
      "Saimiri_oerstedii_ssp\n",
      "1 Saimiri_oerstedii_ssp\n",
      "Skipping\n",
      "../zarr_data/Piliocolobus_badius_ssp\n",
      "Piliocolobus_badius_ssp\n",
      "1 Piliocolobus_badius_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_brunneus_ssp\n",
      "Plecturocebus_brunneus_ssp\n",
      "2 Plecturocebus_brunneus_ssp\n",
      "Skipping\n",
      "../zarr_data/Cebus_imitator_ssp\n",
      "Cebus_imitator_ssp\n",
      "2 Cebus_imitator_ssp\n",
      "Skipping\n",
      "../zarr_data/Perodicticus_potto_ssp\n",
      "Perodicticus_potto_ssp\n",
      "3 Perodicticus_potto_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp\n",
      "Trachypithecus_poliocephalus_ssp\n",
      "12 Trachypithecus_poliocephalus_ssp\n",
      "Skipping\n",
      "../zarr_data/Symphalangus_syndactylus_ssp\n",
      "Symphalangus_syndactylus_ssp\n",
      "5 Symphalangus_syndactylus_ssp\n",
      "Skipping\n",
      "../zarr_data/Pongo_pygmaeus_ssp\n",
      "Pongo_pygmaeus_ssp\n",
      "26 Pongo_pygmaeus_ssp\n",
      "../zarr_data/Pongo_pygmaeus_ssp/NC_072396.2\n",
      "../zarr_data/Pongo_pygmaeus_ssp/NC_072373.2\n",
      "../zarr_data/Avahi_laniger_ssp\n",
      "Avahi_laniger_ssp\n",
      "2 Avahi_laniger_ssp\n",
      "Skipping\n",
      "../zarr_data/Lepilemur_sahamalazensis_ssp\n",
      "Lepilemur_sahamalazensis_ssp\n",
      "1 Lepilemur_sahamalazensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Allenopithecus_nigroviridis_ssp\n",
      "Allenopithecus_nigroviridis_ssp\n",
      "3 Allenopithecus_nigroviridis_ssp\n",
      "Skipping\n",
      "../zarr_data/Alouatta_belzebul_ssp\n",
      "Alouatta_belzebul_ssp\n",
      "3 Alouatta_belzebul_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_denti_ssp\n",
      "Cercopithecus_denti_ssp\n",
      "2 Cercopithecus_denti_ssp\n",
      "Skipping\n",
      "../zarr_data/Saimiri_ustus_ssp\n",
      "Saimiri_ustus_ssp\n",
      "10 Saimiri_ustus_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_vetulus_ssp\n",
      "Trachypithecus_vetulus_ssp\n",
      "3 Trachypithecus_vetulus_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_francoisi_ssp\n",
      "Trachypithecus_francoisi_ssp\n",
      "9 Trachypithecus_francoisi_ssp\n",
      "Skipping\n",
      "../zarr_data/Nomascus_gabriellae_ssp\n",
      "Nomascus_gabriellae_ssp\n",
      "1 Nomascus_gabriellae_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_hamlyni_ssp\n",
      "Cercopithecus_hamlyni_ssp\n",
      "2 Cercopithecus_hamlyni_ssp\n",
      "Skipping\n",
      "../zarr_data/Leontocebus_nigricollis_ssp\n",
      "Leontocebus_nigricollis_ssp\n",
      "1 Leontocebus_nigricollis_ssp\n",
      "Skipping\n",
      "../zarr_data/Chiropotes_albinasus_ssp\n",
      "Chiropotes_albinasus_ssp\n",
      "4 Chiropotes_albinasus_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_tonkeana_ssp\n",
      "Macaca_tonkeana_ssp\n",
      "8 Macaca_tonkeana_ssp\n",
      "Skipping\n",
      "../zarr_data/Gorilla_beringei_ssp\n",
      "Gorilla_beringei_ssp\n",
      "28 Gorilla_beringei_ssp\n",
      "../zarr_data/Gorilla_beringei_ssp/NC_073224.2\n",
      "../zarr_data/Gorilla_beringei_ssp/NC_073247.2\n",
      "../zarr_data/Cebus_albifrons_ssp\n",
      "Cebus_albifrons_ssp\n",
      "5 Cebus_albifrons_ssp\n",
      "Skipping\n",
      "../zarr_data/Mico_humeralifer_ssp\n",
      "Mico_humeralifer_ssp\n",
      "1 Mico_humeralifer_ssp\n",
      "Skipping\n",
      "../zarr_data/Cacajao_calvus_ssp\n",
      "Cacajao_calvus_ssp\n",
      "19 Cacajao_calvus_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_nemestrina_ssp\n",
      "Macaca_nemestrina_ssp\n",
      "26 Macaca_nemestrina_ssp\n",
      "../zarr_data/Macaca_nemestrina_ssp/NC_092125.1\n",
      "../zarr_data/Macaca_nemestrina_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_nemestrina_ssp/NW_027257673.1\n",
      "../zarr_data/Macaca_sinica_ssp\n",
      "Macaca_sinica_ssp\n",
      "1 Macaca_sinica_ssp\n",
      "Skipping\n",
      "../zarr_data/Aotus_griseimembra_ssp\n",
      "Aotus_griseimembra_ssp\n",
      "4 Aotus_griseimembra_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_neglectus_ssp\n",
      "Cercopithecus_neglectus_ssp\n",
      "3 Cercopithecus_neglectus_ssp\n",
      "Skipping\n",
      "../zarr_data/Pithecia_vanzolinii_ssp\n",
      "Pithecia_vanzolinii_ssp\n",
      "1 Pithecia_vanzolinii_ssp\n",
      "Skipping\n",
      "../zarr_data/Cebus_olivaceus_ssp\n",
      "Cebus_olivaceus_ssp\n",
      "2 Cebus_olivaceus_ssp\n",
      "Skipping\n",
      "../zarr_data/Semnopithecus_priam_ssp\n",
      "Semnopithecus_priam_ssp\n",
      "1 Semnopithecus_priam_ssp\n",
      "Skipping\n",
      "../zarr_data/Semnopithecus_entellus_ssp\n",
      "Semnopithecus_entellus_ssp\n",
      "9 Semnopithecus_entellus_ssp\n",
      "Skipping\n",
      "../zarr_data/Hylobates_pileatus_ssp\n",
      "Hylobates_pileatus_ssp\n",
      "3 Hylobates_pileatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Galago_moholi_ssp\n",
      "Galago_moholi_ssp\n",
      "2 Galago_moholi_ssp\n",
      "Skipping\n",
      "../zarr_data/Rhinopithecus_strykeri_ssp\n",
      "Rhinopithecus_strykeri_ssp\n",
      "3 Rhinopithecus_strykeri_ssp\n",
      "Skipping\n",
      "../zarr_data/Otolemur_garnettii_ssp\n",
      "Otolemur_garnettii_ssp\n",
      "40 Otolemur_garnettii_ssp\n",
      "../zarr_data/Otolemur_garnettii_ssp/HiC_scaffold_1\n",
      "../zarr_data/Colobus_polykomos_ssp\n",
      "Colobus_polykomos_ssp\n",
      "1 Colobus_polykomos_ssp\n",
      "Skipping\n",
      "../zarr_data/Alouatta_discolor_ssp\n",
      "Alouatta_discolor_ssp\n",
      "2 Alouatta_discolor_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_leucogenys_ssp\n",
      "Macaca_leucogenys_ssp\n",
      "4 Macaca_leucogenys_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_mona_ssp\n",
      "Cercopithecus_mona_ssp\n",
      "15 Cercopithecus_mona_ssp\n",
      "Skipping\n",
      "../zarr_data/Ateles_marginatus_ssp\n",
      "Ateles_marginatus_ssp\n",
      "1 Ateles_marginatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Callithrix_kuhlii_ssp\n",
      "Callithrix_kuhlii_ssp\n",
      "1 Callithrix_kuhlii_ssp\n",
      "Skipping\n",
      "../zarr_data/Propithecus_diadema_ssp\n",
      "Propithecus_diadema_ssp\n",
      "7 Propithecus_diadema_ssp\n",
      "Skipping\n",
      "../zarr_data/Rhinopithecus_brelichi_ssp\n",
      "Rhinopithecus_brelichi_ssp\n",
      "5 Rhinopithecus_brelichi_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_coronatus_ssp\n",
      "Eulemur_coronatus_ssp\n",
      "6 Eulemur_coronatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_brunnescens_ssp\n",
      "Macaca_brunnescens_ssp\n",
      "1 Macaca_brunnescens_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_ascanius_ssp\n",
      "Cercopithecus_ascanius_ssp\n",
      "5 Cercopithecus_ascanius_ssp\n",
      "Skipping\n",
      "../zarr_data/Callithrix_jacchus_ssp\n",
      "Callithrix_jacchus_ssp\n",
      "12 Callithrix_jacchus_ssp\n",
      "Skipping\n",
      "../zarr_data/Avahi_peyrierasi_ssp\n",
      "Avahi_peyrierasi_ssp\n",
      "2 Avahi_peyrierasi_ssp\n",
      "Skipping\n",
      "../zarr_data/Tarsius_dentatus_ssp\n",
      "Tarsius_dentatus_ssp\n",
      "1 Tarsius_dentatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Indri_indri_ssp\n",
      "Indri_indri_ssp\n",
      "2 Indri_indri_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_silenus_ssp\n",
      "Macaca_silenus_ssp\n",
      "17 Macaca_silenus_ssp\n",
      "Skipping\n",
      "../zarr_data/Allochrocebus_preussi_ssp\n",
      "Allochrocebus_preussi_ssp\n",
      "1 Allochrocebus_preussi_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_petaurista_ssp\n",
      "Cercopithecus_petaurista_ssp\n",
      "1 Cercopithecus_petaurista_ssp\n",
      "Skipping\n",
      "../zarr_data/Mico_argentatus_ssp\n",
      "Mico_argentatus_ssp\n",
      "4 Mico_argentatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_hoffmannsi_ssp\n",
      "Plecturocebus_hoffmannsi_ssp\n",
      "1 Plecturocebus_hoffmannsi_ssp\n",
      "Skipping\n",
      "../zarr_data/Alouatta_juara_ssp\n",
      "Alouatta_juara_ssp\n",
      "2 Alouatta_juara_ssp\n",
      "Skipping\n",
      "../zarr_data/Hylobates_klossii_ssp\n",
      "Hylobates_klossii_ssp\n",
      "1 Hylobates_klossii_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheracebus_lucifer_ssp\n",
      "Cheracebus_lucifer_ssp\n",
      "4 Cheracebus_lucifer_ssp\n",
      "Skipping\n",
      "../zarr_data/Propithecus_deckenii_ssp\n",
      "Propithecus_deckenii_ssp\n",
      "1 Propithecus_deckenii_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheracebus_regulus_ssp\n",
      "Cheracebus_regulus_ssp\n",
      "2 Cheracebus_regulus_ssp\n",
      "Skipping\n",
      "../zarr_data/Alouatta_palliata_ssp\n",
      "Alouatta_palliata_ssp\n",
      "6 Alouatta_palliata_ssp\n",
      "Skipping\n",
      "../zarr_data/Colobus_angolensis_ssp\n",
      "Colobus_angolensis_ssp\n",
      "1 Colobus_angolensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Chlorocebus_cynosuros_ssp\n",
      "Chlorocebus_cynosuros_ssp\n",
      "16 Chlorocebus_cynosuros_ssp\n",
      "Skipping\n",
      "../zarr_data/Hylobates_moloch_ssp\n",
      "Hylobates_moloch_ssp\n",
      "4 Hylobates_moloch_ssp\n",
      "Skipping\n",
      "../zarr_data/Lemur_catta_ssp\n",
      "Lemur_catta_ssp\n",
      "18 Lemur_catta_ssp\n",
      "Skipping\n",
      "../zarr_data/Pongo_abelii_ssp\n",
      "Pongo_abelii_ssp\n",
      "27 Pongo_abelii_ssp\n",
      "../zarr_data/Pongo_abelii_ssp/NC_071985.2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Pongo_abelii_ssp/NC_072008.2\n",
      "../zarr_data/Hapalemur_occidentalis_ssp\n",
      "Hapalemur_occidentalis_ssp\n",
      "1 Hapalemur_occidentalis_ssp\n",
      "Skipping\n",
      "../zarr_data/Allochrocebus_solatus_ssp\n",
      "Allochrocebus_solatus_ssp\n",
      "1 Allochrocebus_solatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Cebuella_niveiventris_ssp\n",
      "Cebuella_niveiventris_ssp\n",
      "2 Cebuella_niveiventris_ssp\n",
      "Skipping\n",
      "../zarr_data/Hoolock_hoolock_ssp\n",
      "Hoolock_hoolock_ssp\n",
      "11 Hoolock_hoolock_ssp\n",
      "Skipping\n",
      "../zarr_data/Chlorocebus_pygerythrus_ssp\n",
      "Chlorocebus_pygerythrus_ssp\n",
      "58 Chlorocebus_pygerythrus_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_roloway_ssp\n",
      "Cercopithecus_roloway_ssp\n",
      "1 Cercopithecus_roloway_ssp\n",
      "Skipping\n",
      "../zarr_data/Colobus_guereza_ssp\n",
      "Colobus_guereza_ssp\n",
      "9 Colobus_guereza_ssp\n",
      "Skipping\n",
      "../zarr_data/Callithrix_geoffroyi_ssp\n",
      "Callithrix_geoffroyi_ssp\n",
      "1 Callithrix_geoffroyi_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_obscurus_ssp\n",
      "Trachypithecus_obscurus_ssp\n",
      "3 Trachypithecus_obscurus_ssp\n",
      "Skipping\n",
      "../zarr_data/Saguinus_imperator_ssp\n",
      "Saguinus_imperator_ssp\n",
      "2 Saguinus_imperator_ssp\n",
      "Skipping\n",
      "../zarr_data/Gorilla_gorilla_ssp\n",
      "Gorilla_gorilla_ssp\n",
      "179 Gorilla_gorilla_ssp\n",
      "Skipping\n",
      "../zarr_data/Microcebus_tavaratra_ssp\n",
      "Microcebus_tavaratra_ssp\n",
      "1 Microcebus_tavaratra_ssp\n",
      "Skipping\n",
      "../zarr_data/Nycticebus_bengalensis_ssp\n",
      "Nycticebus_bengalensis_ssp\n",
      "9 Nycticebus_bengalensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_cristatus_ssp\n",
      "Trachypithecus_cristatus_ssp\n",
      "1 Trachypithecus_cristatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_rufifrons_ssp\n",
      "Eulemur_rufifrons_ssp\n",
      "6 Eulemur_rufifrons_ssp\n",
      "Skipping\n",
      "../zarr_data/Pithecia_albicans_ssp\n",
      "Pithecia_albicans_ssp\n",
      "1 Pithecia_albicans_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_mulatta_ssp\n",
      "Macaca_mulatta_ssp\n",
      "426 Macaca_mulatta_ssp\n",
      "Skipping\n",
      "../zarr_data/Cacajao_melanocephalus_ssp\n",
      "Cacajao_melanocephalus_ssp\n",
      "2 Cacajao_melanocephalus_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_hatinhensis_ssp\n",
      "Trachypithecus_hatinhensis_ssp\n",
      "1 Trachypithecus_hatinhensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Galagoides_demidoff_ssp\n",
      "Galagoides_demidoff_ssp\n",
      "1 Galagoides_demidoff_ssp\n",
      "Skipping\n",
      "../zarr_data/Aotus_trivirgatus_ssp\n",
      "Aotus_trivirgatus_ssp\n",
      "1 Aotus_trivirgatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Pithecia_mittermeieri_ssp\n",
      "Pithecia_mittermeieri_ssp\n",
      "2 Pithecia_mittermeieri_ssp\n",
      "Skipping\n",
      "../zarr_data/Hoolock_leuconedys_ssp\n",
      "Hoolock_leuconedys_ssp\n",
      "2 Hoolock_leuconedys_ssp\n",
      "Skipping\n",
      "../zarr_data/Rhinopithecus_avunculus_ssp\n",
      "Rhinopithecus_avunculus_ssp\n",
      "1 Rhinopithecus_avunculus_ssp\n",
      "Skipping\n",
      "../zarr_data/Pongo_tapanuliensis_ssp\n",
      "Pongo_tapanuliensis_ssp\n",
      "2 Pongo_tapanuliensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Nomascus_annamensis_ssp\n",
      "Nomascus_annamensis_ssp\n",
      "1 Nomascus_annamensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_nigra_ssp\n",
      "Macaca_nigra_ssp\n",
      "7 Macaca_nigra_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_diana_ssp\n",
      "Cercopithecus_diana_ssp\n",
      "1 Cercopithecus_diana_ssp\n",
      "Skipping\n",
      "../zarr_data/Nomascus_leucogenys_ssp\n",
      "Nomascus_leucogenys_ssp\n",
      "4 Nomascus_leucogenys_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_grovesi_ssp\n",
      "Plecturocebus_grovesi_ssp\n",
      "1 Plecturocebus_grovesi_ssp\n",
      "Skipping\n",
      "../zarr_data/Hylobates_lar_ssp\n",
      "Hylobates_lar_ssp\n",
      "4 Hylobates_lar_ssp\n",
      "Skipping\n",
      "../zarr_data/Papio_anubis_ssp\n",
      "Papio_anubis_ssp\n",
      "107 Papio_anubis_ssp\n",
      "Skipping\n",
      "../zarr_data/Leontocebus_fuscicollis_ssp\n",
      "Leontocebus_fuscicollis_ssp\n",
      "4 Leontocebus_fuscicollis_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_nigrescens_ssp\n",
      "Macaca_nigrescens_ssp\n",
      "2 Macaca_nigrescens_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_germaini_ssp\n",
      "Trachypithecus_germaini_ssp\n",
      "1 Trachypithecus_germaini_ssp\n",
      "Skipping\n",
      "../zarr_data/Ateles_geoffroyi_ssp\n",
      "Ateles_geoffroyi_ssp\n",
      "3 Ateles_geoffroyi_ssp\n",
      "Skipping\n",
      "../zarr_data/Cephalopachus_bancanus_ssp\n",
      "Cephalopachus_bancanus_ssp\n",
      "1 Cephalopachus_bancanus_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheracebus_lugens_ssp\n",
      "Cheracebus_lugens_ssp\n",
      "8 Cheracebus_lugens_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_mitis_ssp\n",
      "Cercopithecus_mitis_ssp\n",
      "5 Cercopithecus_mitis_ssp\n",
      "Skipping\n",
      "../zarr_data/Nomascus_concolor_ssp\n",
      "Nomascus_concolor_ssp\n",
      "1 Nomascus_concolor_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_cinerascens_ssp\n",
      "Plecturocebus_cinerascens_ssp\n",
      "2 Plecturocebus_cinerascens_ssp\n",
      "Skipping\n",
      "../zarr_data/Mandrillus_sphinx_ssp\n",
      "Mandrillus_sphinx_ssp\n",
      "3 Mandrillus_sphinx_ssp\n",
      "Skipping\n",
      "../zarr_data/Mico_humilis_ssp\n",
      "Mico_humilis_ssp\n",
      "1 Mico_humilis_ssp\n",
      "Skipping\n",
      "../zarr_data/Cebus_unicolor_ssp\n",
      "Cebus_unicolor_ssp\n",
      "4 Cebus_unicolor_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_mongoz_ssp\n",
      "Eulemur_mongoz_ssp\n",
      "7 Eulemur_mongoz_ssp\n",
      "Skipping\n",
      "../zarr_data/Mandrillus_leucophaeus_ssp\n",
      "Mandrillus_leucophaeus_ssp\n",
      "7 Mandrillus_leucophaeus_ssp\n",
      "Skipping\n",
      "../zarr_data/Pithecia_pithecia_ssp\n",
      "Pithecia_pithecia_ssp\n",
      "10 Pithecia_pithecia_ssp\n",
      "Skipping\n",
      "../zarr_data/Papio_kindae_ssp\n",
      "Papio_kindae_ssp\n",
      "28 Papio_kindae_ssp\n",
      "../zarr_data/Papio_kindae_ssp/NC_044976.1\n",
      "../zarr_data/Papio_kindae_ssp/NC_044996.1\n",
      "../zarr_data/Cheracebus_torquatus_ssp\n",
      "Cheracebus_torquatus_ssp\n",
      "1 Cheracebus_torquatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Lepilemur_septentrionalis_ssp\n",
      "Lepilemur_septentrionalis_ssp\n",
      "2 Lepilemur_septentrionalis_ssp\n",
      "Skipping\n",
      "../zarr_data/Saguinus_bicolor_ssp\n",
      "Saguinus_bicolor_ssp\n",
      "3 Saguinus_bicolor_ssp\n",
      "Skipping\n",
      "../zarr_data/Mirza_zaza_ssp\n",
      "Mirza_zaza_ssp\n",
      "6 Mirza_zaza_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_fuscata_ssp\n",
      "Macaca_fuscata_ssp\n",
      "67 Macaca_fuscata_ssp\n",
      "Skipping\n",
      "../zarr_data/Pan_troglodytes_ssp\n",
      "Pan_troglodytes_ssp\n",
      "199 Pan_troglodytes_ssp\n",
      "Skipping\n",
      "../zarr_data/Papio_papio_ssp\n",
      "Papio_papio_ssp\n",
      "14 Papio_papio_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_nemestrina_ssp_pca\n",
      "Macaca_nemestrina_ssp_pca\n",
      "0 Macaca_nemestrina_ssp_pca\n",
      "Skipping\n",
      "../zarr_data/Galago_senegalensis_ssp\n",
      "Galago_senegalensis_ssp\n",
      "3 Galago_senegalensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Ateles_paniscus_ssp\n",
      "Ateles_paniscus_ssp\n",
      "2 Ateles_paniscus_ssp\n",
      "Skipping\n",
      "../zarr_data/Varecia_variegata_ssp\n",
      "Varecia_variegata_ssp\n",
      "5 Varecia_variegata_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercocebus_lunulatus_ssp\n",
      "Cercocebus_lunulatus_ssp\n",
      "1 Cercocebus_lunulatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_siberu_ssp\n",
      "Macaca_siberu_ssp\n",
      "1 Macaca_siberu_ssp\n",
      "Skipping\n",
      "../zarr_data/Xanthonycticebus_pygmaeus_ssp\n",
      "Xanthonycticebus_pygmaeus_ssp\n",
      "3 Xanthonycticebus_pygmaeus_ssp\n",
      "Skipping\n",
      "../zarr_data/Avahi_occidentalis_ssp\n",
      "Avahi_occidentalis_ssp\n",
      "1 Avahi_occidentalis_ssp\n",
      "Skipping\n",
      "../zarr_data/Microcebus_mittermeieri_ssp\n",
      "Microcebus_mittermeieri_ssp\n",
      "1 Microcebus_mittermeieri_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_thibetana_thibetana_pca\n",
      "Macaca_thibetana_thibetana_pca\n",
      "0 Macaca_thibetana_thibetana_pca\n",
      "Skipping\n",
      "../zarr_data/Propithecus_edwardsi_ssp\n",
      "Propithecus_edwardsi_ssp\n",
      "1 Propithecus_edwardsi_ssp\n",
      "Skipping\n",
      "../zarr_data/Hapalemur_griseus_ssp\n",
      "Hapalemur_griseus_ssp\n",
      "8 Hapalemur_griseus_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_sylvanus_ssp\n",
      "Macaca_sylvanus_ssp\n",
      "2 Macaca_sylvanus_ssp\n",
      "Skipping\n",
      "../zarr_data/Pithecia_chrysocephala_ssp\n",
      "Pithecia_chrysocephala_ssp\n",
      "1 Pithecia_chrysocephala_ssp\n",
      "Skipping\n",
      "../zarr_data/Aotus_nancymaae_ssp\n",
      "Aotus_nancymaae_ssp\n",
      "31 Aotus_nancymaae_ssp\n",
      "../zarr_data/Aotus_nancymaae_ssp/NW_026988488.1\n",
      "../zarr_data/Aotus_nancymaae_ssp/NW_026988415.1\n",
      "../zarr_data/Aotus_nancymaae_ssp/NW_026988472.1\n",
      "../zarr_data/Aotus_nancymaae_ssp/NW_026988496.1\n",
      "../zarr_data/Aotus_nancymaae_ssp/NW_026988412.1\n",
      "../zarr_data/Macaca_assamensis_ssp\n",
      "Macaca_assamensis_ssp\n",
      "5 Macaca_assamensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_donacophilus_ssp\n",
      "Plecturocebus_donacophilus_ssp\n",
      "1 Plecturocebus_donacophilus_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_bernhardi_ssp\n",
      "Plecturocebus_bernhardi_ssp\n",
      "3 Plecturocebus_bernhardi_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_ebenus_ssp\n",
      "Trachypithecus_ebenus_ssp\n",
      "1 Trachypithecus_ebenus_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_maura_ssp\n",
      "Macaca_maura_ssp\n",
      "7 Macaca_maura_ssp\n",
      "Skipping\n",
      "../zarr_data/Alouatta_macconnelli_ssp\n",
      "Alouatta_macconnelli_ssp\n",
      "5 Alouatta_macconnelli_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_cinereiceps_ssp\n",
      "Eulemur_cinereiceps_ssp\n",
      "1 Eulemur_cinereiceps_ssp\n",
      "Skipping\n",
      "../zarr_data/Ateles_chamek_ssp\n",
      "Ateles_chamek_ssp\n",
      "11 Ateles_chamek_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_moloch_ssp\n",
      "Plecturocebus_moloch_ssp\n",
      "1 Plecturocebus_moloch_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_macaco_ssp\n",
      "Eulemur_macaco_ssp\n",
      "9 Eulemur_macaco_ssp\n",
      "Skipping\n",
      "../zarr_data/Pithecia_hirsuta_ssp\n",
      "Pithecia_hirsuta_ssp\n",
      "2 Pithecia_hirsuta_ssp\n",
      "Skipping\n",
      "../zarr_data/Hylobates_abbotti_ssp\n",
      "Hylobates_abbotti_ssp\n",
      "1 Hylobates_abbotti_ssp\n",
      "Skipping\n",
      "../zarr_data/Cacajao_hosomi_ssp\n",
      "Cacajao_hosomi_ssp\n",
      "2 Cacajao_hosomi_ssp\n",
      "Skipping\n",
      "../zarr_data/Saguinus_geoffroyi_ssp\n",
      "Saguinus_geoffroyi_ssp\n",
      "1 Saguinus_geoffroyi_ssp\n",
      "Skipping\n",
      "../zarr_data/Pithecia_pissinatti_ssp\n",
      "Pithecia_pissinatti_ssp\n",
      "3 Pithecia_pissinatti_ssp\n",
      "Skipping\n",
      "../zarr_data/Nasalis_larvatus_ssp\n",
      "Nasalis_larvatus_ssp\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Nasalis_larvatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Propithecus_perrieri_ssp\n",
      "Propithecus_perrieri_ssp\n",
      "2 Propithecus_perrieri_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_radiata_ssp\n",
      "Macaca_radiata_ssp\n",
      "13 Macaca_radiata_ssp\n",
      "Skipping\n",
      "../zarr_data/Lepilemur_mustelinus_ssp\n",
      "Lepilemur_mustelinus_ssp\n",
      "1 Lepilemur_mustelinus_ssp\n",
      "Skipping\n",
      "../zarr_data/Prolemur_simus_ssp\n",
      "Prolemur_simus_ssp\n",
      "8 Prolemur_simus_ssp\n",
      "Skipping\n",
      "../zarr_data/Saguinus_oedipus_ssp\n",
      "Saguinus_oedipus_ssp\n",
      "1 Saguinus_oedipus_ssp\n",
      "Skipping\n",
      "../zarr_data/Piliocolobus_gordonorum_ssp\n",
      "Piliocolobus_gordonorum_ssp\n",
      "1 Piliocolobus_gordonorum_ssp\n",
      "Skipping\n",
      "../zarr_data/Alouatta_caraya_ssp\n",
      "Alouatta_caraya_ssp\n",
      "2 Alouatta_caraya_ssp\n",
      "Skipping\n",
      "../zarr_data/Lagothrix_lagotricha_ssp\n",
      "Lagothrix_lagotricha_ssp\n",
      "1 Lagothrix_lagotricha_ssp\n",
      "Skipping\n",
      "../zarr_data/Leontopithecus_rosalia_ssp\n",
      "Leontopithecus_rosalia_ssp\n",
      "3 Leontopithecus_rosalia_ssp\n",
      "Skipping\n",
      "../zarr_data/Daubentonia_madagascariensis_ssp\n",
      "Daubentonia_madagascariensis_ssp\n",
      "35 Daubentonia_madagascariensis_ssp\n",
      "../zarr_data/Daubentonia_madagascariensis_ssp/JBFSEQ010000009.1\n",
      "../zarr_data/Daubentonia_madagascariensis_ssp/JBFSEQ010000001.1\n",
      "../zarr_data/Propithecus_tattersalli_ssp\n",
      "Propithecus_tattersalli_ssp\n",
      "8 Propithecus_tattersalli_ssp\n",
      "Skipping\n",
      "../zarr_data/Hylobates_agilis_ssp\n",
      "Hylobates_agilis_ssp\n",
      "2 Hylobates_agilis_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_fascicularis_ssp_pca\n",
      "Macaca_fascicularis_ssp_pca\n",
      "0 Macaca_fascicularis_ssp_pca\n",
      "Skipping\n",
      "../zarr_data/Cacajao_ayresi_ssp\n",
      "Cacajao_ayresi_ssp\n",
      "3 Cacajao_ayresi_ssp\n",
      "Skipping\n",
      "../zarr_data/Lepilemur_ruficaudatus_ssp\n",
      "Lepilemur_ruficaudatus_ssp\n",
      "2 Lepilemur_ruficaudatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Saimiri_boliviensis_ssp\n",
      "Saimiri_boliviensis_ssp\n",
      "12 Saimiri_boliviensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_hecki_ssp\n",
      "Macaca_hecki_ssp\n",
      "5 Macaca_hecki_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_thibetana_ssp\n",
      "Macaca_thibetana_ssp\n",
      "3 Macaca_thibetana_ssp\n",
      "Skipping\n",
      "../zarr_data/Varecia_rubra_ssp\n",
      "Varecia_rubra_ssp\n",
      "4 Varecia_rubra_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_nictitans_ssp\n",
      "Cercopithecus_nictitans_ssp\n",
      "3 Cercopithecus_nictitans_ssp\n",
      "Skipping\n",
      "../zarr_data/Theropithecus_gelada_ssp\n",
      "Theropithecus_gelada_ssp\n",
      "73 Theropithecus_gelada_ssp\n",
      "Skipping\n",
      "../zarr_data/Microcebus_jonahi_ssp\n",
      "Microcebus_jonahi_ssp\n",
      "1 Microcebus_jonahi_ssp\n",
      "Skipping\n",
      "../zarr_data/Semnopithecus_hypoleucos_ssp\n",
      "Semnopithecus_hypoleucos_ssp\n",
      "4 Semnopithecus_hypoleucos_ssp\n",
      "Skipping\n",
      "../zarr_data/Brachyteles_hypoxanthus_ssp\n",
      "Brachyteles_hypoxanthus_ssp\n",
      "1 Brachyteles_hypoxanthus_ssp\n",
      "Skipping\n",
      "../zarr_data/Chiropotes_sagulatus_ssp\n",
      "Chiropotes_sagulatus_ssp\n",
      "4 Chiropotes_sagulatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Aotus_azarai_ssp\n",
      "Aotus_azarai_ssp\n",
      "14 Aotus_azarai_ssp\n",
      "Skipping\n",
      "../zarr_data/Papio_anubis_ssp_pca\n",
      "Papio_anubis_ssp_pca\n",
      "0 Papio_anubis_ssp_pca\n",
      "Skipping\n",
      "../zarr_data/Saguinus_inustus_ssp\n",
      "Saguinus_inustus_ssp\n",
      "1 Saguinus_inustus_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_fulvus_ssp\n",
      "Eulemur_fulvus_ssp\n",
      "2 Eulemur_fulvus_ssp\n",
      "Skipping\n",
      "../zarr_data/Hapalemur_meridionalis_ssp\n",
      "Hapalemur_meridionalis_ssp\n",
      "2 Hapalemur_meridionalis_ssp\n",
      "Skipping\n",
      "../zarr_data/Chlorocebus_tantalus_ssp\n",
      "Chlorocebus_tantalus_ssp\n",
      "11 Chlorocebus_tantalus_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercocebus_chrysogaster_ssp\n",
      "Cercocebus_chrysogaster_ssp\n",
      "1 Cercocebus_chrysogaster_ssp\n",
      "Skipping\n",
      "../zarr_data/Hylobates_muelleri_ssp\n",
      "Hylobates_muelleri_ssp\n",
      "1 Hylobates_muelleri_ssp\n",
      "Skipping\n",
      "../zarr_data/Allochrocebus_lhoesti_ssp\n",
      "Allochrocebus_lhoesti_ssp\n",
      "3 Allochrocebus_lhoesti_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_dubius_ssp\n",
      "Plecturocebus_dubius_ssp\n",
      "3 Plecturocebus_dubius_ssp\n",
      "Skipping\n",
      "../zarr_data/Miopithecus_talapoin_ssp\n",
      "Miopithecus_talapoin_ssp\n",
      "1 Miopithecus_talapoin_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercocebus_torquatus_ssp\n",
      "Cercocebus_torquatus_ssp\n",
      "2 Cercocebus_torquatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_leonina_ssp\n",
      "Macaca_leonina_ssp\n",
      "8 Macaca_leonina_ssp\n",
      "Skipping\n",
      "../zarr_data/Pygathrix_nemaeus_ssp\n",
      "Pygathrix_nemaeus_ssp\n",
      "7 Pygathrix_nemaeus_ssp\n",
      "Skipping\n",
      "../zarr_data/Otolemur_crassicaudatus_ssp\n",
      "Otolemur_crassicaudatus_ssp\n",
      "3 Otolemur_crassicaudatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_cephus_ssp\n",
      "Cercopithecus_cephus_ssp\n",
      "3 Cercopithecus_cephus_ssp\n",
      "Skipping\n",
      "../zarr_data/Loris_lydekkerianus_ssp\n",
      "Loris_lydekkerianus_ssp\n",
      "2 Loris_lydekkerianus_ssp\n",
      "Skipping\n",
      "../zarr_data/Papio_cynocephalus_ssp\n",
      "Papio_cynocephalus_ssp\n",
      "537 Papio_cynocephalus_ssp\n",
      "Skipping\n",
      "../zarr_data/Carlito_syrichta_ssp\n",
      "Carlito_syrichta_ssp\n",
      "1 Carlito_syrichta_ssp\n",
      "Skipping\n",
      "../zarr_data/Alouatta_seniculus_ssp\n",
      "Alouatta_seniculus_ssp\n",
      "21 Alouatta_seniculus_ssp\n",
      "../zarr_data/Alouatta_seniculus_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_seniculus_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_seniculus_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_seniculus_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_seniculus_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_seniculus_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_seniculus_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Alouatta_seniculus_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Cercocebus_atys_ssp\n",
      "Cercocebus_atys_ssp\n",
      "15 Cercocebus_atys_ssp\n",
      "Skipping\n",
      "../zarr_data/Pan_paniscus_ssp\n",
      "Pan_paniscus_ssp\n",
      "23 Pan_paniscus_ssp\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073267.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_085926.1\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073256.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073257.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_085927.1\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073266.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073264.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073255.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073254.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073265.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073261.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073271.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073270.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073260.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073259.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073262.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073272.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073253.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073268.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073249.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073269.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073252.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073263.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073258.2\n",
      "../zarr_data/Presbytis_comata_ssp\n",
      "Presbytis_comata_ssp\n",
      "1 Presbytis_comata_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_sanfordi_ssp\n",
      "Eulemur_sanfordi_ssp\n",
      "5 Eulemur_sanfordi_ssp\n",
      "Skipping\n",
      "../zarr_data/Saimiri_sciureus_ssp\n",
      "Saimiri_sciureus_ssp\n",
      "2 Saimiri_sciureus_ssp\n",
      "Skipping\n",
      "../zarr_data/Saguinus_midas_ssp\n",
      "Saguinus_midas_ssp\n",
      "4 Saguinus_midas_ssp\n",
      "Skipping\n",
      "../zarr_data/Piliocolobus_kirkii_ssp\n",
      "Piliocolobus_kirkii_ssp\n",
      "1 Piliocolobus_kirkii_ssp\n",
      "Skipping\n",
      "../zarr_data/Propithecus_verreauxi_ssp\n",
      "Propithecus_verreauxi_ssp\n",
      "2 Propithecus_verreauxi_ssp\n",
      "Skipping\n",
      "../zarr_data/Nomascus_siki_ssp\n",
      "Nomascus_siki_ssp\n",
      "1 Nomascus_siki_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_phayrei_ssp\n",
      "Trachypithecus_phayrei_ssp\n",
      "5 Trachypithecus_phayrei_ssp\n",
      "Skipping\n",
      "../zarr_data/Propithecus_coquereli_ssp\n",
      "Propithecus_coquereli_ssp\n",
      "10 Propithecus_coquereli_ssp\n",
      "Skipping\n",
      "../zarr_data/Rhinopithecus_bieti_ssp\n",
      "Rhinopithecus_bieti_ssp\n",
      "36 Rhinopithecus_bieti_ssp\n",
      "../zarr_data/Rhinopithecus_bieti_ssp/NC_044549.1\n",
      "../zarr_data/Rhinopithecus_bieti_ssp/NC_044555.1\n",
      "../zarr_data/Eulemur_collaris_ssp\n",
      "Eulemur_collaris_ssp\n",
      "7 Eulemur_collaris_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheirogaleus_major_ssp\n",
      "Cheirogaleus_major_ssp\n",
      "3 Cheirogaleus_major_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheirogaleus_sibreei_ssp\n",
      "Cheirogaleus_sibreei_ssp\n",
      "1 Cheirogaleus_sibreei_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_mulatta_ssp_pca\n",
      "Macaca_mulatta_ssp_pca\n",
      "0 Macaca_mulatta_ssp_pca\n",
      "Skipping\n",
      "../zarr_data/Lophocebus_aterrimus_ssp\n",
      "Lophocebus_aterrimus_ssp\n",
      "3 Lophocebus_aterrimus_ssp\n",
      "Skipping\n",
      "../zarr_data/Chlorocebus_dryas_ssp\n",
      "Chlorocebus_dryas_ssp\n",
      "1 Chlorocebus_dryas_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_arctoides_ssp\n",
      "Macaca_arctoides_ssp\n",
      "11 Macaca_arctoides_ssp\n",
      "Skipping\n",
      "../zarr_data/Chiropotes_israelita_ssp\n",
      "Chiropotes_israelita_ssp\n",
      "4 Chiropotes_israelita_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_johnii_ssp\n",
      "Trachypithecus_johnii_ssp\n",
      "2 Trachypithecus_johnii_ssp\n",
      "Skipping\n",
      "../zarr_data/Chlorocebus_sabaeus_ssp\n",
      "Chlorocebus_sabaeus_ssp\n",
      "53 Chlorocebus_sabaeus_ssp\n",
      "Skipping\n",
      "../zarr_data/Presbytis_melalophos_ssp\n",
      "Presbytis_melalophos_ssp\n",
      "1 Presbytis_melalophos_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_albifrons_ssp\n",
      "Eulemur_albifrons_ssp\n",
      "3 Eulemur_albifrons_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_rubriventer_ssp\n",
      "Eulemur_rubriventer_ssp\n",
      "7 Eulemur_rubriventer_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_pogonias_ssp\n",
      "Cercopithecus_pogonias_ssp\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Cercopithecus_pogonias_ssp\n",
      "Skipping\n",
      "../zarr_data/Microcebus_griseorufus_ssp\n",
      "Microcebus_griseorufus_ssp\n",
      "1 Microcebus_griseorufus_ssp\n",
      "Skipping\n",
      "../zarr_data/Saguinus_mystax_ssp\n",
      "Saguinus_mystax_ssp\n",
      "1 Saguinus_mystax_ssp\n",
      "Skipping\n",
      "../zarr_data/Tarsius_wallacei_ssp\n",
      "Tarsius_wallacei_ssp\n",
      "1 Tarsius_wallacei_ssp\n",
      "Skipping\n",
      "../zarr_data/Arctocebus_calabarensis_ssp\n",
      "Arctocebus_calabarensis_ssp\n",
      "1 Arctocebus_calabarensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_auratus_ssp\n",
      "Trachypithecus_auratus_ssp\n",
      "2 Trachypithecus_auratus_ssp\n",
      "Skipping\n",
      "../zarr_data/Papio_ursinus_ssp\n",
      "Papio_ursinus_ssp\n",
      "8 Papio_ursinus_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_miltoni_ssp\n",
      "Plecturocebus_miltoni_ssp\n",
      "1 Plecturocebus_miltoni_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_flavifrons_ssp\n",
      "Eulemur_flavifrons_ssp\n",
      "10 Eulemur_flavifrons_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_wolfi_ssp\n",
      "Cercopithecus_wolfi_ssp\n",
      "3 Cercopithecus_wolfi_ssp\n",
      "Skipping\n",
      "../zarr_data/Aotus_vociferans_ssp\n",
      "Aotus_vociferans_ssp\n",
      "5 Aotus_vociferans_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_pileatus_ssp\n",
      "Trachypithecus_pileatus_ssp\n",
      "1 Trachypithecus_pileatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_geei_ssp\n",
      "Trachypithecus_geei_ssp\n",
      "6 Trachypithecus_geei_ssp\n",
      "Skipping\n",
      "../zarr_data/Microcebus_ravelobensis_ssp\n",
      "Microcebus_ravelobensis_ssp\n",
      "3 Microcebus_ravelobensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Erythrocebus_patas_ssp\n",
      "Erythrocebus_patas_ssp\n",
      "6 Erythrocebus_patas_ssp\n",
      "Skipping\n",
      "../zarr_data/Rhinopithecus_roxellana_ssp\n",
      "Rhinopithecus_roxellana_ssp\n",
      "57 Rhinopithecus_roxellana_ssp\n",
      "Skipping\n",
      "../zarr_data/Piliocolobus_tephrosceles_ssp\n",
      "Piliocolobus_tephrosceles_ssp\n",
      "1 Piliocolobus_tephrosceles_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_rufus_ssp\n",
      "Eulemur_rufus_ssp\n",
      "1 Eulemur_rufus_ssp\n",
      "Skipping\n",
      "../zarr_data/Lepilemur_ankaranensis_ssp\n",
      "Lepilemur_ankaranensis_ssp\n",
      "5 Lepilemur_ankaranensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Loris_tardigradus_ssp\n",
      "Loris_tardigradus_ssp\n",
      "2 Loris_tardigradus_ssp\n",
      "Skipping\n",
      "../zarr_data/Lepilemur_mittermeieri_ssp\n",
      "Lepilemur_mittermeieri_ssp\n",
      "1 Lepilemur_mittermeieri_ssp\n",
      "Skipping\n",
      "../zarr_data/Cebuella_pygmaea_ssp\n",
      "Cebuella_pygmaea_ssp\n",
      "2 Cebuella_pygmaea_ssp\n",
      "Skipping\n",
      "../zarr_data/Nycticebus_coucang_ssp\n",
      "Nycticebus_coucang_ssp\n",
      "3 Nycticebus_coucang_ssp\n",
      "Skipping\n",
      "../zarr_data/Sapajus_apella_ssp\n",
      "Sapajus_apella_ssp\n",
      "15 Sapajus_apella_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_campbelli_ssp\n",
      "Cercopithecus_campbelli_ssp\n",
      "2 Cercopithecus_campbelli_ssp\n",
      "Skipping\n",
      "../zarr_data/Microcebus_murinus_ssp\n",
      "Microcebus_murinus_ssp\n",
      "26 Microcebus_murinus_ssp\n",
      "../zarr_data/Microcebus_murinus_ssp/NC_033692.1\n",
      "../zarr_data/Microcebus_murinus_ssp/NC_033660.1\n",
      "../zarr_data/Pygathrix_cinerea_ssp\n",
      "Pygathrix_cinerea_ssp\n",
      "5 Pygathrix_cinerea_ssp\n",
      "Skipping\n",
      "../zarr_data/Microcebus_arnholdi_ssp\n",
      "Microcebus_arnholdi_ssp\n",
      "2 Microcebus_arnholdi_ssp\n",
      "Skipping\n",
      "../zarr_data/Pygathrix_nigripes_ssp\n",
      "Pygathrix_nigripes_ssp\n",
      "6 Pygathrix_nigripes_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_cyclopis_ssp\n",
      "Macaca_cyclopis_ssp\n",
      "4 Macaca_cyclopis_ssp\n",
      "Skipping\n",
      "../zarr_data/Saimiri_cassiquiarensis_ssp\n",
      "Saimiri_cassiquiarensis_ssp\n",
      "9 Saimiri_cassiquiarensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Hapalemur_aureus_ssp\n",
      "Hapalemur_aureus_ssp\n",
      "1 Hapalemur_aureus_ssp\n",
      "Skipping\n"
     ]
    }
   ],
   "source": [
    "# Fst implementation medium\n",
    "for x in glob.glob(zarr_path+\"*\"):\n",
    "    print(x)\n",
    "    # The name used to load all the files, short and long version\n",
    "    short_form = x.split(\"/\")[-1].split(\"_\")[0]\n",
    "    long_form = x.split(\"/\")[-1]\n",
    "    print(long_form)\n",
    "    # Loading the various metadata files. Metadata, contig information, callability bed.\n",
    "    metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "    metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "    metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "    regions_df[\"chr_type\"] = [\"chrX\" if x == 2 and y == 1 else \"aut\" for x, y in zip(regions_df.FEMALE_PLOIDY, regions_df.MALE_PLOIDY)]\n",
    "    large_contigs = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "    large_x = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID\n",
    "    # Skipping the large samples sizes and the singulars for the Fst calc\n",
    "    print(len(metadata_df.loc[metadata_df.GVCF_FOLDER == long_form]), long_form)\n",
    "    if ((len(metadata_df.loc[metadata_df.GVCF_FOLDER == long_form]) > 50) or\n",
    "    (len(metadata_df.loc[metadata_df.GVCF_FOLDER == long_form]) <= 20)):\n",
    "        print(\"Skipping\")\n",
    "        continue\n",
    "    # Loading the genetic data.\n",
    "    df_l = []\n",
    "    for c in glob.glob(x+\"/*\"):\n",
    "        print(c)\n",
    "        ds = sg.load_dataset(c)\n",
    "        # This implementation is the pi implementation.\n",
    "        # Probably problematic in some cases with population structure, but it is easier to implement\n",
    "        ds[\"sample_cohort\"] = ds[\"samples\"]\n",
    "        # Subsetting and windowing the sgkit dataset. The rechunking handles what otherwise would cause an error.\n",
    "        #ds[\"call_genotype\"] = ds[\"call_genotype\"].clip(0)\n",
    "        ds = ds.sel(contigs=[ds.variant_contig[0].values])\n",
    "        if c.split(\"/\")[-1] in list(large_x):\n",
    "            ds = haploid_double(ds, \"call_genotype\", \"samples\")\n",
    "        missing_rate = ds.call_genotype_mask[:,:,0].sum(axis=1).values/ds.call_genotype_mask[:,:,0].count(axis=1).values\n",
    "        ds = ds.isel(variants=(missing_rate <= 0))\n",
    "        ds = sg.window_by_genome(ds)\n",
    "        ds = (sg.Fst(ds.chunk({\"variants\": 50000})))\n",
    "        df_sub = pd.DataFrame(ds.stat_Fst[0,:,], columns=ds.sample_id)\n",
    "        df_sub[\"chrom\"] = c.split(\"/\")[-1]\n",
    "        df_sub[\"variants_used\"] = len(ds.variants)\n",
    "        df_l.append(df_sub)\n",
    "    output_df = pd.concat(df_l)\n",
    "    output_df[\"chr_type\"] = output_df[\"chrom\"].map(dict(zip(regions_df.CONTIG_ID, regions_df.chr_type)))\n",
    "    output_df[\"species\"] = long_form\n",
    "    output_df.to_csv(\"../results/window_stats/{}_Fst.txt\".format(long_form), sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "bcc90c66",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PD_0120</th>\n",
       "      <th>PD_0348</th>\n",
       "      <th>chrom</th>\n",
       "      <th>variants_used</th>\n",
       "      <th>chr_type</th>\n",
       "      <th>species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CM063191.1</td>\n",
       "      <td>1535859</td>\n",
       "      <td>chrX</td>\n",
       "      <td>Saguinus_labiatus_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CM063191.1</td>\n",
       "      <td>1535859</td>\n",
       "      <td>chrX</td>\n",
       "      <td>Saguinus_labiatus_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.517100</td>\n",
       "      <td>JASSZA010000432.1</td>\n",
       "      <td>4587</td>\n",
       "      <td>aut</td>\n",
       "      <td>Saguinus_labiatus_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>-0.517100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JASSZA010000432.1</td>\n",
       "      <td>4587</td>\n",
       "      <td>aut</td>\n",
       "      <td>Saguinus_labiatus_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.184585</td>\n",
       "      <td>CM063169.1</td>\n",
       "      <td>3030007</td>\n",
       "      <td>aut</td>\n",
       "      <td>Saguinus_labiatus_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.184585</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CM063169.1</td>\n",
       "      <td>3030007</td>\n",
       "      <td>aut</td>\n",
       "      <td>Saguinus_labiatus_ssp</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    PD_0120   PD_0348              chrom  variants_used chr_type  \\\n",
       "0       NaN       NaN         CM063191.1        1535859     chrX   \n",
       "1       NaN       NaN         CM063191.1        1535859     chrX   \n",
       "0       NaN -0.517100  JASSZA010000432.1           4587      aut   \n",
       "1 -0.517100       NaN  JASSZA010000432.1           4587      aut   \n",
       "0       NaN  0.184585         CM063169.1        3030007      aut   \n",
       "1  0.184585       NaN         CM063169.1        3030007      aut   \n",
       "\n",
       "                 species  \n",
       "0  Saguinus_labiatus_ssp  \n",
       "1  Saguinus_labiatus_ssp  \n",
       "0  Saguinus_labiatus_ssp  \n",
       "1  Saguinus_labiatus_ssp  \n",
       "0  Saguinus_labiatus_ssp  \n",
       "1  Saguinus_labiatus_ssp  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "output_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "a8699942",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Macaca_fascicularis_ssp\n",
      "Macaca_fascicularis_ssp\n",
      "../zarr_data/Macaca_fascicularis_ssp/CP141342.1\n",
      "../zarr_data/Macaca_fascicularis_ssp/CP141341.1\n",
      "../zarr_data/Macaca_fascicularis_ssp/CP141361.1\n"
     ]
    }
   ],
   "source": [
    "for x in glob.glob(zarr_path+\"*\")[4:5]:\n",
    "    print(x)\n",
    "    # The name used to load all the files, short and long version\n",
    "    short_form = x.split(\"/\")[-1].split(\"_\")[0]\n",
    "    long_form = x.split(\"/\")[-1]\n",
    "    print(long_form)\n",
    "    # Loading the various metadata files. Metadata, contig information, callability bed.\n",
    "    metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "    metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "    metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "    regions_df[\"chr_type\"] = [\"chrX\" if x == 2 and y == 1 else \"aut\" for x, y in zip(regions_df.FEMALE_PLOIDY, regions_df.MALE_PLOIDY)]\n",
    "    large_contigs = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "    large_x = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID\n",
    "    # Loading the genetic data.\n",
    "    df_l = []\n",
    "    for c in glob.glob(x+\"/*\"):\n",
    "        print(c)\n",
    "        ds = sg.load_dataset(c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f70a7cb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f85c614",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "zarr_path = \"../zarr_data/\"\n",
    "metadata_folders = glob.glob(metadata_path+\"*_individuals.txt\")\n",
    "\n",
    "for folder in metadata_folders[:0]:\n",
    "    metadata_df = pd.read_csv(folder, sep=\"\\t\")\n",
    "    short_form = folder.split(\"/\")[-1].split(\"_\")[0]\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "    female_df = metadata_df[pd.to_numeric(metadata_df['AVG_COVERAGE_X'], errors='coerce').notnull()]\n",
    "    female_df = female_df.loc[(female_df.GENETIC_SEX == \"F\") & (female_df.AVG_COVERAGE_A >= 10)].sort_values(by=\"AVG_COVERAGE_A\", ascending=False)\n",
    "    for GVCF_FOLDER in female_df.GVCF_FOLDER.unique():\n",
    "        # Check that zarr data exists.\n",
    "        zarr_chroms = glob.glob(zarr_path+GVCF_FOLDER+\"/*\")\n",
    "        for c in zarr_chroms:\n",
    "            ds = sg.load_dataset(c, drop_variables=[\"variant_allele\"])\n",
    "            print(c)\n",
    "        reference = metadata_df.loc[metadata_df.GVCF_FOLDER == GVCF_FOLDER].REFERENCE_FOLDER.unique()[0]\n",
    "        regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "        regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "        large_contigs = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "        large_x = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID\n",
    "        bed_files = read_beds(GVCF_FOLDER)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "81f81bfd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/Pongo_individuals.txt'"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_path+\"{}_individuals.txt\".format(short_form)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9462dfdb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Loading the genetic data.\n",
    "    ds = sg.load_dataset(\"../data/{}\".format(long_form))\n",
    "    sex_map = dict(zip(metadata_df.GVCF_ID, metadata_df.SEX_I))\n",
    "    # If you want to split based on sex. I will make every individual unique\n",
    "    #ds[\"sample_cohort\"] = xr.DataArray(pd.Series(ds.sample_id.values).map(sex_map), dims=\"samples\")\n",
    "    ds[\"sample_cohort\"] = ds[\"samples\"]\n",
    "    # Only keeping contigs which are above the cutoff (1Mb or more)\n",
    "    kept_contigs =  [x for x in ds.contig_id.values if (x == large_contigs).any()]\n",
    "    contig_IDs = pd.Series(kept_contigs).map(dict(zip(ds.contig_id.values, range(len(ds.contig_id.values))))).values\n",
    "    # Subsetting and windowing the sgkit dataset. The rechunking handles what otherwise would cause an error.\n",
    "    ds_autx = ds.sel(variants=(ds.variant_contig.isin(contig_IDs).compute()), contigs=contig_IDs)\n",
    "    ds_autx = sg.window_by_position(ds_autx, size=window_size)\n",
    "    ds_autx[\"call_genotype\"] = ds_autx[\"call_genotype\"].clip(0)\n",
    "    ds_autx_diversity = sg.diversity(ds_autx.chunk({\"variants\": 50000}))\n",
    "    # Generating window callability\n",
    "    contig_l, window_start_l = [], []\n",
    "    v_c = pd.Series(ds_autx.window_contig).value_counts(sort=False)\n",
    "    for i in range(len(v_c)):\n",
    "        window_start_l.extend(list(range(0, v_c[i]*window_size, window_size)))\n",
    "        contig_l.extend([kept_contigs[i]]*v_c[i])\n",
    "    window_df = pd.DataFrame({\"chrom\": contig_l, \"window_start\": window_start_l})\n",
    "    intervals_callable = pos_windows(bed_files, window_size, kept_contigs)\n",
    "    window_df_call = window_df.merge(intervals_callable, on=[\"chrom\", \"window_start\"])\n",
    "    # Merge and save\n",
    "    output_df = pd.merge(window_df_call, pd.DataFrame(ds_autx_diversity[\"stat_diversity\"],\n",
    "                                      columns=ds_autx_diversity[\"sample_id\"]),\n",
    "                         left_index=True, right_index=True)\n",
    "    output_df.to_csv(\"../results/window_stats/{}_10kb_het.txt\".format(long_form), sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f315c4e4",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_l = []\n",
    "for p in df_paths:\n",
    "    d = pd.read_csv(p, sep=\"\\t\", index_col=[0])\n",
    "    d[\"species\"] = p.split(\"/\")[-1].split(\"_1\")[0]\n",
    "    df_l.append(pd.melt(d, id_vars=[\"chrom\", \"window_start\", \"window_end\", \"callable_frac\", \"species\"], var_name=\"GVCF_ID\", value_name=\"het\"))\n",
    "het_df = pd.concat(df_l)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5de8c014",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Standardized parts.\n",
    "short_form = \"Papio\"\n",
    "long_form = \"Papio_papio_ssp\"\n",
    "size_cutoff = 1000000\n",
    "window_size = 100000 # Should maybe decrease to 25 or 10kb.\n",
    "\n",
    "metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "\n",
    "regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "large_contigs = regions_df.loc[(regions_df.LENGTH >= 1000000) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "large_x = regions_df.loc[(regions_df.LENGTH >= 1000000) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0fa50281",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "20    NC_044996.1\n",
       "Name: CONTIG_ID, dtype: object"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "large_x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "c3ea2789",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Standardized parts.\n",
    "short_form = \"Gorilla\"\n",
    "long_form = \"Gorilla_Gorilla_ssp\"\n",
    "size_cutoff = 1000000\n",
    "window_size = 100000 # Should maybe decrease to 25 or 10kb.\n",
    "\n",
    "metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "\n",
    "regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "large_contigs = regions_df.loc[(regions_df.LENGTH >= 1000000) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "large_x = regions_df.loc[(regions_df.LENGTH >= 1000000) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "531a1b87",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "23    NC_073247.2\n",
       "25    NC_073247.2\n",
       "Name: CONTIG_ID, dtype: object"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "large_x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "4f5c4322",
   "metadata": {},
   "outputs": [],
   "source": [
    "het_df[\"chr_type\"] = [\"aut\" if x not in [\"NC_044996.1\", \"NC_073247.2\"] else \"chrX\" for x in het_df[\"chrom\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "94d187a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "het_df_high_call = het_df.loc[het_df.callable_frac >= 0.95]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "87ffdf3f",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "metadata_df_gorilla = pd.read_csv(metadata_path+\"Gorilla_individuals.txt\".format(short_form), sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "ac38c294",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_mean_het = het_df_high_call.groupby([\"GVCF_ID\", \"species\", \"chr_type\"])[\"het\"].mean().reset_index()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
