{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "18cd9075",
   "metadata": {},
   "outputs": [],
   "source": [
    "%run ../scripts/notebook_settings.py\n",
    "import sgkit as sg\n",
    "import xarray as xr\n",
    "import glob"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "4302e3eb",
   "metadata": {},
   "source": [
    "Defining the current paths and sizes."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "dc500ec8",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "zarr_path = \"../zarr_data/\"\n",
    "metadata_folders = glob.glob(metadata_path+\"*_individuals.txt\")\n",
    "\n",
    "size_cutoff = 1000000\n",
    "window_size = 100000\n",
    "missing_filter = 0.5"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "df3190ac",
   "metadata": {},
   "outputs": [],
   "source": [
    "def read_beds(long_form):\n",
    "    bed_path_x = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024/{}/filteredVCF/pos_bed_cov_based/{}_batch*_fploidy2_mploidy1.bed\".format(long_form, long_form)\n",
    "    bed_path_all = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024/{}/filteredVCF/pos_bed_cov_based/{}_batch*_fploidy2_mploidy2.bed\".format(long_form, long_form)\n",
    "    bed_l = []\n",
    "    for b in glob.glob(bed_path_all):\n",
    "        bed_file = pd.read_csv(b, sep=\"\\t\", names=[\"chrom\", \"start\", \"end\"])\n",
    "        bed_l.append(bed_file)\n",
    "    bed_files = pd.concat(bed_l)\n",
    "    bed_l = []\n",
    "    for b in glob.glob(bed_path_x):\n",
    "        #print(b)\n",
    "        bed_file = pd.read_csv(b, sep=\"\\t\", names=[\"chrom\", \"start\", \"end\"])\n",
    "        bed_l.append(bed_file)\n",
    "    if len(bed_l) > 0:\n",
    "        bed_x = pd.concat(bed_l)\n",
    "        bed_files = bed_files.loc[~(bed_files.chrom.isin(bed_x.chrom.unique()))]\n",
    "        bed_files = pd.concat([bed_files, bed_x]).sort_values(by=[\"chrom\", \"start\", \"end\"])\n",
    "    return bed_files\n",
    "\n",
    "def pos_windows(bed_l, window_size, chrom_order):\n",
    "    # Input a bed file and the window size of intervals desired. Multiple chromosomes accepted.\n",
    "    # It has to be sorted.\n",
    "    df_l = []\n",
    "    for c in chrom_order:\n",
    "        #print(c)\n",
    "        frac_l = []\n",
    "        b = bed_l.loc[bed_l[\"chrom\"] == c].copy()\n",
    "        b[\"w_s\"] = b.end-b.start\n",
    "        w_start = b.start.iloc[0]\n",
    "        current_pos, callable_bases = 0, 0\n",
    "        for i, j, k in zip(b.start, b.end, b.w_s):\n",
    "            # Nothing called in the current window under investigation.\n",
    "            while i-window_size >= current_pos:\n",
    "                frac_l.append(callable_bases/window_size)\n",
    "                callable_bases = 0\n",
    "                current_pos += window_size\n",
    "            # Window starts in current. We know this is true because of the previous while loop.\n",
    "            callable_bases += min(k, current_pos+window_size-i)\n",
    "            # Everything called in current.\n",
    "            while j-window_size >= current_pos:\n",
    "                frac_l.append(callable_bases/window_size)\n",
    "                callable_bases = 0\n",
    "                current_pos += window_size\n",
    "                if j-window_size >= current_pos:\n",
    "                    callable_bases += window_size\n",
    "                else:\n",
    "                # Window stops in current. Again, know this is true.\n",
    "                    callable_bases += j-current_pos\n",
    "        # Last window.\n",
    "        frac_l.append(callable_bases/(window_size))\n",
    "        df_l.append(pd.DataFrame({\"chrom\": c, \"window_start\": list(range(0, len(frac_l)*window_size, window_size)),\n",
    "                                  \"window_end\": list(range(window_size, (len(frac_l)+1)*window_size, window_size)),\n",
    "                                  \"callable_frac\": frac_l}))\n",
    "    return pd.concat(df_l)\n",
    "\n",
    "def haploid_double(ds, variable, dim):\n",
    "    unmasked = ~ds[f\"{variable}_mask\"]\n",
    "    overwrite = ds.call_genotype[:,:,0]\n",
    "    overwrite_2 = ds.call_genotype_mask[:,:,0]\n",
    "    return ds.assign(**{\n",
    "        f\"{variable}\": ds[variable].where(\n",
    "        unmasked, \n",
    "        overwrite),\n",
    "        f\"{variable}_mask\": ds[f\"{variable}_mask\"].where(\n",
    "        unmasked,\n",
    "        overwrite_2)})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "9b0075b6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Chlorocebus_aethiops_ssp 26\n",
      "Chlorocebus_aethiops_ssp\n",
      "Chlorocebus_cynosuros_ssp 16\n",
      "Chlorocebus_cynosuros_ssp\n",
      "Chlorocebus_tantalus_ssp 11\n",
      "Chlorocebus_tantalus_ssp\n",
      "Allenopithecus_nigroviridis_ssp 3\n",
      "Pithecia_pissinatti_ssp 3\n",
      "Pithecia_pissinatti_ssp\n",
      "Nasalis_larvatus_ssp 3\n"
     ]
    }
   ],
   "source": [
    "# Input checks: Is there a zarr/bed pair for all species?\n",
    "bed_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024/{}/filteredVCF/pos_bed_cov_based/{}_batch0_fploidy2_mploidy2.bed\"\n",
    "fvcf_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024/{}/filteredVCF/bcf_step1/{}_all_chr.sorted.bcf\"\n",
    "for x in glob.glob(metadata_path+\"*_individuals.txt\"):\n",
    "    short_form = x.split(\"/\")[-1].split(\"_\")[0]\n",
    "    long_form = x.split(\"/\")[-1]\n",
    "    # Loading the various metadata files. Metadata, contig information, callability bed.\n",
    "    metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "    metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "    metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    for f in metadata_df.GVCF_FOLDER.unique():\n",
    "        r_df = regions_df.loc[regions_df.REFERENCE_FOLDER == f]\n",
    "        if not os.path.exists(bed_path.format(f, f)):\n",
    "            print(f, len(metadata_df.loc[metadata_df.GVCF_FOLDER == f]))\n",
    "        if not os.path.exists(fvcf_path.format(f, f)):\n",
    "            print(f)\n",
    "        # for b, f, m in zip(r_df.BATCH, r_df.FEMALE_PLOIDY, r_df.MALE_PLOIDY):\n",
    "        #    print(b,f,m)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "3b1800f5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>BIOSAMPLE_ID</th>\n",
       "      <th>PDGP_ID</th>\n",
       "      <th>GVCF_ID</th>\n",
       "      <th>SRA_SAMPLE_ID</th>\n",
       "      <th>BIOPROJECT_ID</th>\n",
       "      <th>BAM_FOLDER</th>\n",
       "      <th>GVCF_FOLDER</th>\n",
       "      <th>REFERENCE_FOLDER</th>\n",
       "      <th>GENUS</th>\n",
       "      <th>SPECIES</th>\n",
       "      <th>SUBSPECIES</th>\n",
       "      <th>REPORTED_SEX</th>\n",
       "      <th>GENETIC_SEX</th>\n",
       "      <th>AVG_COVERAGE_A</th>\n",
       "      <th>PROP_COVERED_A</th>\n",
       "      <th>AVG_COVERAGE_X</th>\n",
       "      <th>PROP_COVERED_X</th>\n",
       "      <th>AVG_COVERAGE_Y</th>\n",
       "      <th>PROP_COVERED_Y</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>SAMEA112482877</td>\n",
       "      <td>PD_0102</td>\n",
       "      <td>PD_0102</td>\n",
       "      <td>ERS14600228</td>\n",
       "      <td>PRJEB59576</td>\n",
       "      <td>Semnopithecus_entellus_ssp</td>\n",
       "      <td>Semnopithecus_entellus_ssp</td>\n",
       "      <td>Trachypithecus_francoisi_ssp</td>\n",
       "      <td>Semnopithecus</td>\n",
       "      <td>entellus</td>\n",
       "      <td>ssp</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>67.956</td>\n",
       "      <td>0.990</td>\n",
       "      <td>62.760</td>\n",
       "      <td>0.969</td>\n",
       "      <td>20.987</td>\n",
       "      <td>0.120</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>SAMEA112483111</td>\n",
       "      <td>PD_0582</td>\n",
       "      <td>PD_0582</td>\n",
       "      <td>ERS14600462</td>\n",
       "      <td>PRJEB59576</td>\n",
       "      <td>Semnopithecus_entellus_ssp</td>\n",
       "      <td>Semnopithecus_entellus_ssp</td>\n",
       "      <td>Trachypithecus_francoisi_ssp</td>\n",
       "      <td>Semnopithecus</td>\n",
       "      <td>entellus</td>\n",
       "      <td>ssp</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>29.608</td>\n",
       "      <td>0.988</td>\n",
       "      <td>26.250</td>\n",
       "      <td>0.964</td>\n",
       "      <td>10.238</td>\n",
       "      <td>0.098</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>SAMEA112483112</td>\n",
       "      <td>PD_0583</td>\n",
       "      <td>PD_0583</td>\n",
       "      <td>ERS14600463</td>\n",
       "      <td>PRJEB59576</td>\n",
       "      <td>Semnopithecus_entellus_ssp</td>\n",
       "      <td>Semnopithecus_entellus_ssp</td>\n",
       "      <td>Trachypithecus_francoisi_ssp</td>\n",
       "      <td>Semnopithecus</td>\n",
       "      <td>entellus</td>\n",
       "      <td>ssp</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>24.638</td>\n",
       "      <td>0.988</td>\n",
       "      <td>22.673</td>\n",
       "      <td>0.963</td>\n",
       "      <td>8.737</td>\n",
       "      <td>0.094</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>SAMEA112483113</td>\n",
       "      <td>PD_0584</td>\n",
       "      <td>PD_0584</td>\n",
       "      <td>ERS14600464</td>\n",
       "      <td>PRJEB59576</td>\n",
       "      <td>Semnopithecus_entellus_ssp</td>\n",
       "      <td>Semnopithecus_entellus_ssp</td>\n",
       "      <td>Trachypithecus_francoisi_ssp</td>\n",
       "      <td>Semnopithecus</td>\n",
       "      <td>entellus</td>\n",
       "      <td>ssp</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>26.570</td>\n",
       "      <td>0.988</td>\n",
       "      <td>24.080</td>\n",
       "      <td>0.963</td>\n",
       "      <td>9.425</td>\n",
       "      <td>0.095</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>SAMEA112483114</td>\n",
       "      <td>PD_0585</td>\n",
       "      <td>PD_0585</td>\n",
       "      <td>ERS14600465</td>\n",
       "      <td>PRJEB59576</td>\n",
       "      <td>Semnopithecus_entellus_ssp</td>\n",
       "      <td>Semnopithecus_entellus_ssp</td>\n",
       "      <td>Trachypithecus_francoisi_ssp</td>\n",
       "      <td>Semnopithecus</td>\n",
       "      <td>entellus</td>\n",
       "      <td>ssp</td>\n",
       "      <td>M</td>\n",
       "      <td>M</td>\n",
       "      <td>24.672</td>\n",
       "      <td>0.988</td>\n",
       "      <td>11.608</td>\n",
       "      <td>0.959</td>\n",
       "      <td>11.266</td>\n",
       "      <td>0.890</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>SAMEA112483115</td>\n",
       "      <td>PD_0586</td>\n",
       "      <td>PD_0586</td>\n",
       "      <td>ERS14600466</td>\n",
       "      <td>PRJEB59576</td>\n",
       "      <td>Semnopithecus_entellus_ssp</td>\n",
       "      <td>Semnopithecus_entellus_ssp</td>\n",
       "      <td>Trachypithecus_francoisi_ssp</td>\n",
       "      <td>Semnopithecus</td>\n",
       "      <td>entellus</td>\n",
       "      <td>ssp</td>\n",
       "      <td>M</td>\n",
       "      <td>M</td>\n",
       "      <td>43.908</td>\n",
       "      <td>0.989</td>\n",
       "      <td>21.059</td>\n",
       "      <td>0.965</td>\n",
       "      <td>20.575</td>\n",
       "      <td>0.901</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>SAMEA112483116</td>\n",
       "      <td>PD_0587</td>\n",
       "      <td>PD_0587</td>\n",
       "      <td>ERS14600467</td>\n",
       "      <td>PRJEB59576</td>\n",
       "      <td>Semnopithecus_entellus_ssp</td>\n",
       "      <td>Semnopithecus_entellus_ssp</td>\n",
       "      <td>Trachypithecus_francoisi_ssp</td>\n",
       "      <td>Semnopithecus</td>\n",
       "      <td>entellus</td>\n",
       "      <td>ssp</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>40.447</td>\n",
       "      <td>0.960</td>\n",
       "      <td>40.239</td>\n",
       "      <td>0.909</td>\n",
       "      <td>11.883</td>\n",
       "      <td>0.111</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>SAMEA112483117</td>\n",
       "      <td>PD_0588</td>\n",
       "      <td>PD_0588</td>\n",
       "      <td>ERS14600468</td>\n",
       "      <td>PRJEB59576</td>\n",
       "      <td>Semnopithecus_entellus_ssp</td>\n",
       "      <td>Semnopithecus_entellus_ssp</td>\n",
       "      <td>Trachypithecus_francoisi_ssp</td>\n",
       "      <td>Semnopithecus</td>\n",
       "      <td>entellus</td>\n",
       "      <td>ssp</td>\n",
       "      <td>M</td>\n",
       "      <td>M</td>\n",
       "      <td>25.000</td>\n",
       "      <td>0.990</td>\n",
       "      <td>13.820</td>\n",
       "      <td>0.961</td>\n",
       "      <td>13.301</td>\n",
       "      <td>0.905</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>SAMN07678111</td>\n",
       "      <td>SAMN07678111</td>\n",
       "      <td>SAMN07678111</td>\n",
       "      <td>SRS6386391</td>\n",
       "      <td>PRJNA399419</td>\n",
       "      <td>Semnopithecus_entellus_ssp</td>\n",
       "      <td>Semnopithecus_entellus_ssp</td>\n",
       "      <td>Trachypithecus_francoisi_ssp</td>\n",
       "      <td>Semnopithecus</td>\n",
       "      <td>entellus</td>\n",
       "      <td>ssp</td>\n",
       "      <td>Unknown</td>\n",
       "      <td>M</td>\n",
       "      <td>24.526</td>\n",
       "      <td>0.985</td>\n",
       "      <td>12.080</td>\n",
       "      <td>0.942</td>\n",
       "      <td>11.061</td>\n",
       "      <td>0.851</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>SAMEA112483118</td>\n",
       "      <td>PD_0589</td>\n",
       "      <td>PD_0589</td>\n",
       "      <td>ERS14600469</td>\n",
       "      <td>PRJEB59576</td>\n",
       "      <td>Semnopithecus_hypoleucos_ssp</td>\n",
       "      <td>Semnopithecus_hypoleucos_ssp</td>\n",
       "      <td>Trachypithecus_francoisi_ssp</td>\n",
       "      <td>Semnopithecus</td>\n",
       "      <td>hypoleucos</td>\n",
       "      <td>ssp</td>\n",
       "      <td>M</td>\n",
       "      <td>M</td>\n",
       "      <td>60.926</td>\n",
       "      <td>0.989</td>\n",
       "      <td>30.039</td>\n",
       "      <td>0.967</td>\n",
       "      <td>27.987</td>\n",
       "      <td>0.893</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>SAMEA112483119</td>\n",
       "      <td>PD_0590</td>\n",
       "      <td>PD_0590</td>\n",
       "      <td>ERS14600470</td>\n",
       "      <td>PRJEB59576</td>\n",
       "      <td>Semnopithecus_hypoleucos_ssp</td>\n",
       "      <td>Semnopithecus_hypoleucos_ssp</td>\n",
       "      <td>Trachypithecus_francoisi_ssp</td>\n",
       "      <td>Semnopithecus</td>\n",
       "      <td>hypoleucos</td>\n",
       "      <td>ssp</td>\n",
       "      <td>M</td>\n",
       "      <td>M</td>\n",
       "      <td>33.565</td>\n",
       "      <td>0.988</td>\n",
       "      <td>16.505</td>\n",
       "      <td>0.962</td>\n",
       "      <td>15.443</td>\n",
       "      <td>0.884</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>SAMEA112483120</td>\n",
       "      <td>PD_0591</td>\n",
       "      <td>PD_0591</td>\n",
       "      <td>ERS14600471</td>\n",
       "      <td>PRJEB59576</td>\n",
       "      <td>Semnopithecus_hypoleucos_ssp</td>\n",
       "      <td>Semnopithecus_hypoleucos_ssp</td>\n",
       "      <td>Trachypithecus_francoisi_ssp</td>\n",
       "      <td>Semnopithecus</td>\n",
       "      <td>hypoleucos</td>\n",
       "      <td>ssp</td>\n",
       "      <td>M</td>\n",
       "      <td>M</td>\n",
       "      <td>28.303</td>\n",
       "      <td>0.988</td>\n",
       "      <td>13.750</td>\n",
       "      <td>0.961</td>\n",
       "      <td>12.819</td>\n",
       "      <td>0.891</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>SAMEA112483121</td>\n",
       "      <td>PD_0592</td>\n",
       "      <td>PD_0592</td>\n",
       "      <td>ERS14600472</td>\n",
       "      <td>PRJEB59576</td>\n",
       "      <td>Semnopithecus_hypoleucos_ssp</td>\n",
       "      <td>Semnopithecus_hypoleucos_ssp</td>\n",
       "      <td>Trachypithecus_francoisi_ssp</td>\n",
       "      <td>Semnopithecus</td>\n",
       "      <td>hypoleucos</td>\n",
       "      <td>ssp</td>\n",
       "      <td>M</td>\n",
       "      <td>M</td>\n",
       "      <td>47.397</td>\n",
       "      <td>0.990</td>\n",
       "      <td>23.279</td>\n",
       "      <td>0.967</td>\n",
       "      <td>21.751</td>\n",
       "      <td>0.903</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>SAMEA112482875</td>\n",
       "      <td>PD_0100</td>\n",
       "      <td>PD_0100</td>\n",
       "      <td>ERS14600226</td>\n",
       "      <td>PRJEB59576</td>\n",
       "      <td>Semnopithecus_priam_ssp</td>\n",
       "      <td>Semnopithecus_priam_ssp</td>\n",
       "      <td>Trachypithecus_francoisi_ssp</td>\n",
       "      <td>Semnopithecus</td>\n",
       "      <td>priam</td>\n",
       "      <td>ssp</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>36.372</td>\n",
       "      <td>0.989</td>\n",
       "      <td>34.536</td>\n",
       "      <td>0.967</td>\n",
       "      <td>13.514</td>\n",
       "      <td>0.116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>SAMEA112483122</td>\n",
       "      <td>PD_0593</td>\n",
       "      <td>PD_0593</td>\n",
       "      <td>ERS14600473</td>\n",
       "      <td>PRJEB59576</td>\n",
       "      <td>Semnopithecus_schistaceus_ssp</td>\n",
       "      <td>Semnopithecus_schistaceus_ssp</td>\n",
       "      <td>Trachypithecus_francoisi_ssp</td>\n",
       "      <td>Semnopithecus</td>\n",
       "      <td>schistaceus</td>\n",
       "      <td>ssp</td>\n",
       "      <td>M</td>\n",
       "      <td>M</td>\n",
       "      <td>26.362</td>\n",
       "      <td>0.988</td>\n",
       "      <td>12.739</td>\n",
       "      <td>0.961</td>\n",
       "      <td>11.843</td>\n",
       "      <td>0.894</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>SAMEA112483123</td>\n",
       "      <td>PD_0594</td>\n",
       "      <td>PD_0594</td>\n",
       "      <td>ERS14600474</td>\n",
       "      <td>PRJEB59576</td>\n",
       "      <td>Semnopithecus_schistaceus_ssp</td>\n",
       "      <td>Semnopithecus_schistaceus_ssp</td>\n",
       "      <td>Trachypithecus_francoisi_ssp</td>\n",
       "      <td>Semnopithecus</td>\n",
       "      <td>schistaceus</td>\n",
       "      <td>ssp</td>\n",
       "      <td>F</td>\n",
       "      <td>F</td>\n",
       "      <td>41.272</td>\n",
       "      <td>0.988</td>\n",
       "      <td>32.357</td>\n",
       "      <td>0.965</td>\n",
       "      <td>12.012</td>\n",
       "      <td>0.104</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      BIOSAMPLE_ID       PDGP_ID       GVCF_ID SRA_SAMPLE_ID BIOPROJECT_ID  \\\n",
       "0   SAMEA112482877       PD_0102       PD_0102   ERS14600228    PRJEB59576   \n",
       "1   SAMEA112483111       PD_0582       PD_0582   ERS14600462    PRJEB59576   \n",
       "2   SAMEA112483112       PD_0583       PD_0583   ERS14600463    PRJEB59576   \n",
       "3   SAMEA112483113       PD_0584       PD_0584   ERS14600464    PRJEB59576   \n",
       "4   SAMEA112483114       PD_0585       PD_0585   ERS14600465    PRJEB59576   \n",
       "5   SAMEA112483115       PD_0586       PD_0586   ERS14600466    PRJEB59576   \n",
       "6   SAMEA112483116       PD_0587       PD_0587   ERS14600467    PRJEB59576   \n",
       "7   SAMEA112483117       PD_0588       PD_0588   ERS14600468    PRJEB59576   \n",
       "8     SAMN07678111  SAMN07678111  SAMN07678111    SRS6386391   PRJNA399419   \n",
       "9   SAMEA112483118       PD_0589       PD_0589   ERS14600469    PRJEB59576   \n",
       "10  SAMEA112483119       PD_0590       PD_0590   ERS14600470    PRJEB59576   \n",
       "11  SAMEA112483120       PD_0591       PD_0591   ERS14600471    PRJEB59576   \n",
       "12  SAMEA112483121       PD_0592       PD_0592   ERS14600472    PRJEB59576   \n",
       "13  SAMEA112482875       PD_0100       PD_0100   ERS14600226    PRJEB59576   \n",
       "14  SAMEA112483122       PD_0593       PD_0593   ERS14600473    PRJEB59576   \n",
       "15  SAMEA112483123       PD_0594       PD_0594   ERS14600474    PRJEB59576   \n",
       "\n",
       "                       BAM_FOLDER                    GVCF_FOLDER  \\\n",
       "0      Semnopithecus_entellus_ssp     Semnopithecus_entellus_ssp   \n",
       "1      Semnopithecus_entellus_ssp     Semnopithecus_entellus_ssp   \n",
       "2      Semnopithecus_entellus_ssp     Semnopithecus_entellus_ssp   \n",
       "3      Semnopithecus_entellus_ssp     Semnopithecus_entellus_ssp   \n",
       "4      Semnopithecus_entellus_ssp     Semnopithecus_entellus_ssp   \n",
       "5      Semnopithecus_entellus_ssp     Semnopithecus_entellus_ssp   \n",
       "6      Semnopithecus_entellus_ssp     Semnopithecus_entellus_ssp   \n",
       "7      Semnopithecus_entellus_ssp     Semnopithecus_entellus_ssp   \n",
       "8      Semnopithecus_entellus_ssp     Semnopithecus_entellus_ssp   \n",
       "9    Semnopithecus_hypoleucos_ssp   Semnopithecus_hypoleucos_ssp   \n",
       "10   Semnopithecus_hypoleucos_ssp   Semnopithecus_hypoleucos_ssp   \n",
       "11   Semnopithecus_hypoleucos_ssp   Semnopithecus_hypoleucos_ssp   \n",
       "12   Semnopithecus_hypoleucos_ssp   Semnopithecus_hypoleucos_ssp   \n",
       "13        Semnopithecus_priam_ssp        Semnopithecus_priam_ssp   \n",
       "14  Semnopithecus_schistaceus_ssp  Semnopithecus_schistaceus_ssp   \n",
       "15  Semnopithecus_schistaceus_ssp  Semnopithecus_schistaceus_ssp   \n",
       "\n",
       "                REFERENCE_FOLDER          GENUS      SPECIES SUBSPECIES  \\\n",
       "0   Trachypithecus_francoisi_ssp  Semnopithecus     entellus        ssp   \n",
       "1   Trachypithecus_francoisi_ssp  Semnopithecus     entellus        ssp   \n",
       "2   Trachypithecus_francoisi_ssp  Semnopithecus     entellus        ssp   \n",
       "3   Trachypithecus_francoisi_ssp  Semnopithecus     entellus        ssp   \n",
       "4   Trachypithecus_francoisi_ssp  Semnopithecus     entellus        ssp   \n",
       "5   Trachypithecus_francoisi_ssp  Semnopithecus     entellus        ssp   \n",
       "6   Trachypithecus_francoisi_ssp  Semnopithecus     entellus        ssp   \n",
       "7   Trachypithecus_francoisi_ssp  Semnopithecus     entellus        ssp   \n",
       "8   Trachypithecus_francoisi_ssp  Semnopithecus     entellus        ssp   \n",
       "9   Trachypithecus_francoisi_ssp  Semnopithecus   hypoleucos        ssp   \n",
       "10  Trachypithecus_francoisi_ssp  Semnopithecus   hypoleucos        ssp   \n",
       "11  Trachypithecus_francoisi_ssp  Semnopithecus   hypoleucos        ssp   \n",
       "12  Trachypithecus_francoisi_ssp  Semnopithecus   hypoleucos        ssp   \n",
       "13  Trachypithecus_francoisi_ssp  Semnopithecus        priam        ssp   \n",
       "14  Trachypithecus_francoisi_ssp  Semnopithecus  schistaceus        ssp   \n",
       "15  Trachypithecus_francoisi_ssp  Semnopithecus  schistaceus        ssp   \n",
       "\n",
       "   REPORTED_SEX GENETIC_SEX  AVG_COVERAGE_A  PROP_COVERED_A  AVG_COVERAGE_X  \\\n",
       "0             F           F          67.956           0.990          62.760   \n",
       "1             F           F          29.608           0.988          26.250   \n",
       "2             F           F          24.638           0.988          22.673   \n",
       "3             F           F          26.570           0.988          24.080   \n",
       "4             M           M          24.672           0.988          11.608   \n",
       "5             M           M          43.908           0.989          21.059   \n",
       "6             F           F          40.447           0.960          40.239   \n",
       "7             M           M          25.000           0.990          13.820   \n",
       "8       Unknown           M          24.526           0.985          12.080   \n",
       "9             M           M          60.926           0.989          30.039   \n",
       "10            M           M          33.565           0.988          16.505   \n",
       "11            M           M          28.303           0.988          13.750   \n",
       "12            M           M          47.397           0.990          23.279   \n",
       "13            F           F          36.372           0.989          34.536   \n",
       "14            M           M          26.362           0.988          12.739   \n",
       "15            F           F          41.272           0.988          32.357   \n",
       "\n",
       "    PROP_COVERED_X  AVG_COVERAGE_Y  PROP_COVERED_Y  \n",
       "0            0.969          20.987           0.120  \n",
       "1            0.964          10.238           0.098  \n",
       "2            0.963           8.737           0.094  \n",
       "3            0.963           9.425           0.095  \n",
       "4            0.959          11.266           0.890  \n",
       "5            0.965          20.575           0.901  \n",
       "6            0.909          11.883           0.111  \n",
       "7            0.961          13.301           0.905  \n",
       "8            0.942          11.061           0.851  \n",
       "9            0.967          27.987           0.893  \n",
       "10           0.962          15.443           0.884  \n",
       "11           0.961          12.819           0.891  \n",
       "12           0.967          21.751           0.903  \n",
       "13           0.967          13.514           0.116  \n",
       "14           0.961          11.843           0.894  \n",
       "15           0.965          12.012           0.104  "
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_df = pd.read_csv(metadata_path+\"Semnopithecus_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "metadata_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "id": "bbe7906e",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>REFERENCE_FOLDER</th>\n",
       "      <th>REGION</th>\n",
       "      <th>CONTIG_ID</th>\n",
       "      <th>START</th>\n",
       "      <th>END</th>\n",
       "      <th>BATCH</th>\n",
       "      <th>FEMALE_PLOIDY</th>\n",
       "      <th>MALE_PLOIDY</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>Trachypithecus_francoisi_ssp</td>\n",
       "      <td>NW_022680469.1</td>\n",
       "      <td>NW_022680469.1</td>\n",
       "      <td>0</td>\n",
       "      <td>1129610</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "               REFERENCE_FOLDER          REGION       CONTIG_ID  START  \\\n",
       "9  Trachypithecus_francoisi_ssp  NW_022680469.1  NW_022680469.1      0   \n",
       "\n",
       "       END  BATCH  FEMALE_PLOIDY  MALE_PLOIDY  \n",
       "9  1129610      0              0            1  "
      ]
     },
     "execution_count": 13,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "regions_df = pd.read_csv(metadata_path+\"Semnopithecus_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "regions_df.loc[regions_df.FEMALE_PLOIDY == 0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "id": "b822180c",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Leontopithecus_chrysomelas_ssp\n",
      "Leontopithecus_chrysomelas_ssp\n",
      "../zarr_data/Leontopithecus_chrysomelas_ssp/HiC_scaffold_23\n",
      "../zarr_data/Leontopithecus_chrysomelas_ssp/HiC_scaffold_1\n",
      "../zarr_data/Saguinus_labiatus_ssp\n",
      "Saguinus_labiatus_ssp\n",
      "../zarr_data/Saguinus_labiatus_ssp/CM063191.1\n",
      "../zarr_data/Saguinus_labiatus_ssp/JASSZA010000432.1\n",
      "../zarr_data/Saguinus_labiatus_ssp/CM063169.1\n"
     ]
    }
   ],
   "source": [
    "# Pi implementation\n",
    "for x in glob.glob(zarr_path+\"*\"):\n",
    "    print(x)\n",
    "    # The name used to load all the files, short and long version\n",
    "    short_form = x.split(\"/\")[-1].split(\"_\")[0]\n",
    "    long_form = x.split(\"/\")[-1]\n",
    "    print(long_form)\n",
    "    # Loading the various metadata files. Metadata, contig information, callability bed.\n",
    "    metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "    metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "    metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "    regions_df[\"chr_type\"] = [\"chrX\" if x == 2 and y == 1 else \"aut\" for x, y in zip(regions_df.FEMALE_PLOIDY, regions_df.MALE_PLOIDY)]\n",
    "    large_contigs = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "    large_x = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID\n",
    "    bed_files = read_beds(long_form)\n",
    "    # Loading the genetic data.\n",
    "    df_l = []\n",
    "    for c in glob.glob(x+\"/*\"):\n",
    "        print(c)\n",
    "        ds = sg.load_dataset(c)\n",
    "        # This implementation is the pi implementation.\n",
    "        # Probably problematic in some cases with population structure, but it is easier to implement\n",
    "        ds[\"sample_cohort\"] = np.repeat([0], ds.sizes[\"samples\"])\n",
    "        # Subsetting and windowing the sgkit dataset. The rechunking handles what otherwise would cause an error.\n",
    "        #ds[\"call_genotype\"] = ds[\"call_genotype\"].clip(0)\n",
    "        ds = ds.sel(contigs=[ds.variant_contig[0].values])\n",
    "        missing_rate = ds.call_genotype_mask[:,:,0].sum(axis=1).values/ds.call_genotype_mask[:,:,0].count(axis=1).values\n",
    "        ds = ds.isel(variants=(missing_rate <= missing_filter))\n",
    "        ds = sg.window_by_position(ds, size=window_size)\n",
    "        ds = (sg.diversity(ds.chunk({\"variants\": 50000})))\n",
    "        df_sub = pd.DataFrame({\"window_start\": list(range(0, len(ds.window_start)*window_size, window_size)),\n",
    "                               \"pi\": ds.stat_diversity[:,0]})\n",
    "        df_sub[\"chrom\"] = c.split(\"/\")[-1]\n",
    "        df_l.append(df_sub)\n",
    "    df_het = pd.concat(df_l)\n",
    "    bed_files = read_beds(long_form)\n",
    "    intervals_callable = pos_windows(bed_files, window_size, df_het[\"chrom\"].unique())\n",
    "    output_df = pd.merge(df_het, intervals_callable, on=[\"chrom\", \"window_start\"])\n",
    "    output_df[\"chr_type\"] = output_df[\"chrom\"].map(dict(zip(regions_df.CONTIG_ID, regions_df.chr_type)))\n",
    "    output_df[\"species\"] = long_form\n",
    "    output_df.to_csv(\"../results/window_stats/{}_100kb_pi.txt\".format(long_form), sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "id": "80b23a53",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([0.  , 0.25, 0.25, ..., 0.  , 0.  , 0.25])"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "\n",
    "ds = sg.load_dataset(\"../zarr_data/Leontopithecus_chrysomelas_ssp/HiC_scaffold_1\")\n",
    "        # This implementation is the pi implementation.\n",
    "        # Probably problematic in some cases with population structure, but it is easier to implement\n",
    "ds[\"sample_cohort\"] = np.repeat([0], ds.sizes[\"samples\"])\n",
    "        # Subsetting and windowing the sgkit dataset. The rechunking handles what otherwise would cause an error.\n",
    "        #ds[\"call_genotype\"] = ds[\"call_genotype\"].clip(0)\n",
    "ds = ds.sel(contigs=[ds.variant_contig[0].values])\n",
    "missing_rate = ds.call_genotype_mask[:,:,0].sum(axis=1).values/ds.call_genotype_mask[:,:,0].count(axis=1).values\n",
    "missing_rate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "id": "8eaa8e10",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "0.75"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missing_rate.max()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "id": "fa582d41",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([0., 0., 0., ..., 0., 0., 0.])"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "missing_rate = ds.call_genotype_mask[:,:,0].sum(axis=1).values/ds.call_genotype_mask[:,:,0].count(axis=1).values\n",
    "missing_rate"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "id": "11424372",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([[[nan]],\n",
       "\n",
       "       [[nan]],\n",
       "\n",
       "       [[nan]],\n",
       "\n",
       "       ...,\n",
       "\n",
       "       [[nan]],\n",
       "\n",
       "       [[nan]],\n",
       "\n",
       "       [[nan]]])"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "sg.Fst(ds)[\"stat_Fst\"].values"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "id": "bd1ae499",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Leontopithecus_chrysomelas_ssp\n",
      "Leontopithecus_chrysomelas_ssp\n",
      "../zarr_data/Leontopithecus_chrysomelas_ssp/HiC_scaffold_23\n",
      "../zarr_data/Leontopithecus_chrysomelas_ssp/HiC_scaffold_1\n",
      "../zarr_data/Saguinus_labiatus_ssp\n",
      "Saguinus_labiatus_ssp\n",
      "../zarr_data/Saguinus_labiatus_ssp/CM063191.1\n",
      "../zarr_data/Saguinus_labiatus_ssp/JASSZA010000432.1\n",
      "../zarr_data/Saguinus_labiatus_ssp/CM063169.1\n",
      "../zarr_data/Plecturocebus_cupreus_ssp\n",
      "Plecturocebus_cupreus_ssp\n",
      "../zarr_data/Plecturocebus_cupreus_ssp/CM080837.1\n",
      "../zarr_data/Plecturocebus_cupreus_ssp/CM080815.1\n",
      "../zarr_data/Papio_hamadryas_ssp\n",
      "Papio_hamadryas_ssp\n",
      "../zarr_data/Papio_hamadryas_ssp/NC_044976.1\n",
      "../zarr_data/Papio_hamadryas_ssp/NC_044996.1\n",
      "../zarr_data/Macaca_fascicularis_ssp\n",
      "Macaca_fascicularis_ssp\n",
      "../zarr_data/Macaca_fascicularis_ssp/CP141342.1\n",
      "../zarr_data/Macaca_fascicularis_ssp/CP141341.1\n",
      "../zarr_data/Macaca_fascicularis_ssp/CP141361.1\n",
      "../zarr_data/Saimiri_macrodon_ssp\n",
      "Saimiri_macrodon_ssp\n",
      "../zarr_data/Saimiri_macrodon_ssp/NW_024100917.1\n",
      "../zarr_data/Saimiri_macrodon_ssp/NW_024100929.1\n",
      "../zarr_data/Saimiri_macrodon_ssp/NW_024100959.1\n",
      "../zarr_data/Cheirogaleus_crossleyi_ssp\n",
      "Cheirogaleus_crossleyi_ssp\n",
      "../zarr_data/Cheirogaleus_crossleyi_ssp/VBSM01000656.1\n",
      "../zarr_data/Cheirogaleus_crossleyi_ssp/VBSM01003151.1\n",
      "../zarr_data/Cheirogaleus_crossleyi_ssp/VBSM01000568.1\n",
      "../zarr_data/Cheirogaleus_crossleyi_ssp/VBSM01000937.1\n",
      "../zarr_data/Cheirogaleus_crossleyi_ssp/VBSM01001044.1\n",
      "../zarr_data/Cheirogaleus_crossleyi_ssp/VBSM01003018.1\n",
      "../zarr_data/Cheirogaleus_crossleyi_ssp/VBSM01003185.1\n",
      "../zarr_data/Cheirogaleus_crossleyi_ssp/VBSM01001483.1\n",
      "../zarr_data/Cheirogaleus_crossleyi_ssp/VBSM01002707.1\n",
      "../zarr_data/Cheirogaleus_crossleyi_ssp/VBSM01002350.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp\n",
      "Cheirogaleus_medius_ssp\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01001483.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01002707.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01002350.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01000656.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01003151.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01000568.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01000937.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01001044.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01003018.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01003185.1\n",
      "../zarr_data/Plecturocebus_caligatus_ssp\n",
      "Plecturocebus_caligatus_ssp\n",
      "../zarr_data/Plecturocebus_caligatus_ssp/CM080815.1\n",
      "../zarr_data/Plecturocebus_caligatus_ssp/CM080837.1\n",
      "../zarr_data/Callimico_goeldii_ssp\n",
      "Callimico_goeldii_ssp\n",
      "../zarr_data/Callimico_goeldii_ssp/HiC_scaffold_1\n",
      "../zarr_data/Trachypithecus_laotum_ssp\n",
      "Trachypithecus_laotum_ssp\n",
      "../zarr_data/Trachypithecus_laotum_ssp/NW_022681437.1\n",
      "../zarr_data/Trachypithecus_laotum_ssp/NW_022681445.1\n",
      "../zarr_data/Trachypithecus_laotum_ssp/NW_022680465.1\n",
      "../zarr_data/Trachypithecus_laotum_ssp/NW_022680471.1\n",
      "../zarr_data/Trachypithecus_laotum_ssp/NW_022680461.1\n",
      "../zarr_data/Trachypithecus_laotum_ssp/NW_022681432.1\n",
      "../zarr_data/Trachypithecus_laotum_ssp/NW_022681471.1\n",
      "../zarr_data/Lepilemur_dorsalis_ssp\n",
      "Lepilemur_dorsalis_ssp\n",
      "../zarr_data/Lepilemur_dorsalis_ssp/NC_033692.1\n",
      "../zarr_data/Lepilemur_dorsalis_ssp/NC_033660.1\n",
      "../zarr_data/Miopithecus_ogouensis_ssp\n",
      "Miopithecus_ogouensis_ssp\n",
      "../zarr_data/Miopithecus_ogouensis_ssp/CM052571.1\n",
      "../zarr_data/Miopithecus_ogouensis_ssp/JAPYXX010000028.1\n",
      "../zarr_data/Miopithecus_ogouensis_ssp/CM052597.1\n",
      "../zarr_data/Tarsius_lariang_ssp\n",
      "Tarsius_lariang_ssp\n",
      "../zarr_data/Tarsius_lariang_ssp/JAPVLO010000001.1\n",
      "../zarr_data/Mirza_coquereli_ssp\n",
      "Mirza_coquereli_ssp\n",
      "../zarr_data/Mirza_coquereli_ssp/NC_033692.1\n",
      "../zarr_data/Mirza_coquereli_ssp/NC_033660.1\n",
      "../zarr_data/Ateles_belzebuth_ssp\n",
      "Ateles_belzebuth_ssp\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp\n",
      "Semnopithecus_schistaceus_ssp\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022680471.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022680461.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022681471.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022681432.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022681445.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022681437.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022680465.1\n",
      "../zarr_data/Saimiri_oerstedii_ssp\n",
      "Saimiri_oerstedii_ssp\n",
      "../zarr_data/Saimiri_oerstedii_ssp/NW_024100917.1\n",
      "../zarr_data/Saimiri_oerstedii_ssp/NW_024100929.1\n",
      "../zarr_data/Saimiri_oerstedii_ssp/NW_024100959.1\n",
      "../zarr_data/Piliocolobus_badius_ssp\n",
      "Piliocolobus_badius_ssp\n",
      "../zarr_data/Piliocolobus_badius_ssp/NC_045434.1\n",
      "../zarr_data/Piliocolobus_badius_ssp/NC_045455.1\n",
      "../zarr_data/Plecturocebus_brunneus_ssp\n",
      "Plecturocebus_brunneus_ssp\n",
      "../zarr_data/Plecturocebus_brunneus_ssp/CM080815.1\n",
      "../zarr_data/Plecturocebus_brunneus_ssp/CM080837.1\n",
      "../zarr_data/Cebus_imitator_ssp\n",
      "Cebus_imitator_ssp\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437082.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437065.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437044.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022436957.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437089.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437104.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437055.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437081.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437015.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437116.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437119.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437031.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437094.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437042.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437043.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022436941.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437085.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437013.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437131.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437038.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437033.1\n",
      "../zarr_data/Perodicticus_potto_ssp\n",
      "Perodicticus_potto_ssp\n",
      "../zarr_data/Perodicticus_potto_ssp/NC_069780.1\n",
      "../zarr_data/Perodicticus_potto_ssp/NC_069804.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp\n",
      "Trachypithecus_poliocephalus_ssp\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022681471.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022681432.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022680471.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022680461.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022680465.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022681445.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022681437.1\n",
      "../zarr_data/Symphalangus_syndactylus_ssp\n",
      "Symphalangus_syndactylus_ssp\n",
      "../zarr_data/Symphalangus_syndactylus_ssp/NC_072423.2\n",
      "../zarr_data/Symphalangus_syndactylus_ssp/NC_072447.2\n",
      "../zarr_data/Pongo_pygmaeus_ssp\n",
      "Pongo_pygmaeus_ssp\n",
      "../zarr_data/Pongo_pygmaeus_ssp/NC_072396.2\n",
      "../zarr_data/Pongo_pygmaeus_ssp/NC_072373.2\n",
      "../zarr_data/Avahi_laniger_ssp\n",
      "Avahi_laniger_ssp\n",
      "../zarr_data/Avahi_laniger_ssp/HiC_scaffold_2\n",
      "../zarr_data/Avahi_laniger_ssp/HiC_scaffold_1\n",
      "../zarr_data/Lepilemur_sahamalazensis_ssp\n",
      "Lepilemur_sahamalazensis_ssp\n",
      "../zarr_data/Lepilemur_sahamalazensis_ssp/NC_033660.1\n",
      "../zarr_data/Lepilemur_sahamalazensis_ssp/NC_033692.1\n",
      "../zarr_data/Allenopithecus_nigroviridis_ssp\n",
      "Allenopithecus_nigroviridis_ssp\n",
      "../zarr_data/Allenopithecus_nigroviridis_ssp/HiC_scaffold_1\n",
      "../zarr_data/Allenopithecus_nigroviridis_ssp/HiC_scaffold_2\n",
      "../zarr_data/Alouatta_belzebul_ssp\n",
      "Alouatta_belzebul_ssp\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Cercopithecus_denti_ssp\n",
      "Cercopithecus_denti_ssp\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Cercopithecus_denti_ssp/CM053398.1\n",
      "../zarr_data/Cercopithecus_denti_ssp/CM053363.1\n",
      "../zarr_data/Saimiri_ustus_ssp\n",
      "Saimiri_ustus_ssp\n",
      "../zarr_data/Saimiri_ustus_ssp/NW_024100917.1\n",
      "../zarr_data/Saimiri_ustus_ssp/NW_024100929.1\n",
      "../zarr_data/Saimiri_ustus_ssp/NW_024100959.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp\n",
      "Trachypithecus_vetulus_ssp\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022680461.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022680471.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022681432.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022681471.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022681437.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022681445.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022680465.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp\n",
      "Trachypithecus_francoisi_ssp\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681432.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681471.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022680461.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022680471.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022680465.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681437.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681445.1\n",
      "../zarr_data/Nomascus_gabriellae_ssp\n",
      "Nomascus_gabriellae_ssp\n",
      "../zarr_data/Nomascus_gabriellae_ssp/NC_044381.1\n",
      "../zarr_data/Nomascus_gabriellae_ssp/NC_044406.1\n",
      "../zarr_data/Cercopithecus_hamlyni_ssp\n",
      "Cercopithecus_hamlyni_ssp\n",
      "../zarr_data/Cercopithecus_hamlyni_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_hamlyni_ssp/CM053398.1\n",
      "../zarr_data/Leontocebus_nigricollis_ssp\n",
      "Leontocebus_nigricollis_ssp\n",
      "../zarr_data/Leontocebus_nigricollis_ssp/CM063191.1\n",
      "../zarr_data/Leontocebus_nigricollis_ssp/CM063169.1\n",
      "../zarr_data/Chiropotes_albinasus_ssp\n",
      "Chiropotes_albinasus_ssp\n",
      "../zarr_data/Chiropotes_albinasus_ssp/CM052647.1\n",
      "../zarr_data/Chiropotes_albinasus_ssp/CM052624.1\n",
      "../zarr_data/Macaca_tonkeana_ssp\n",
      "Macaca_tonkeana_ssp\n",
      "../zarr_data/Macaca_tonkeana_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_tonkeana_ssp/NC_092125.1\n",
      "../zarr_data/Macaca_tonkeana_ssp/NW_027257673.1\n",
      "../zarr_data/Gorilla_beringei_ssp\n",
      "Gorilla_beringei_ssp\n",
      "../zarr_data/Gorilla_beringei_ssp/NC_073224.2\n",
      "../zarr_data/Gorilla_beringei_ssp/NC_073247.2\n",
      "../zarr_data/Cebus_albifrons_ssp\n",
      "Cebus_albifrons_ssp\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437031.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437119.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437042.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437094.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437043.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437085.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022436941.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437013.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437038.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437131.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437033.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437082.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437044.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437065.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437089.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022436957.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437055.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437104.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437015.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437081.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437116.1\n",
      "../zarr_data/Mico_humeralifer_ssp\n",
      "Mico_humeralifer_ssp\n",
      "../zarr_data/Mico_humeralifer_ssp/NC_071442.1\n",
      "../zarr_data/Mico_humeralifer_ssp/NC_071464.1\n",
      "../zarr_data/Cacajao_calvus_ssp\n",
      "Cacajao_calvus_ssp\n",
      "../zarr_data/Cacajao_calvus_ssp/CM052624.1\n",
      "../zarr_data/Cacajao_calvus_ssp/CM052647.1\n",
      "../zarr_data/Macaca_nemestrina_ssp\n",
      "Macaca_nemestrina_ssp\n",
      "../zarr_data/Macaca_nemestrina_ssp/NC_092125.1\n",
      "../zarr_data/Macaca_nemestrina_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_nemestrina_ssp/NW_027257673.1\n",
      "../zarr_data/Macaca_sinica_ssp\n",
      "Macaca_sinica_ssp\n",
      "../zarr_data/Macaca_sinica_ssp/NC_065598.1\n",
      "../zarr_data/Macaca_sinica_ssp/NC_065597.1\n",
      "../zarr_data/Macaca_sinica_ssp/NC_065578.1\n",
      "../zarr_data/Aotus_griseimembra_ssp\n",
      "Aotus_griseimembra_ssp\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988472.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988412.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988496.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988488.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988415.1\n",
      "../zarr_data/Cercopithecus_neglectus_ssp\n",
      "Cercopithecus_neglectus_ssp\n",
      "../zarr_data/Cercopithecus_neglectus_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_neglectus_ssp/CM053398.1\n",
      "../zarr_data/Pithecia_vanzolinii_ssp\n",
      "Pithecia_vanzolinii_ssp\n",
      "../zarr_data/Pithecia_vanzolinii_ssp/CM052647.1\n",
      "../zarr_data/Pithecia_vanzolinii_ssp/CM052624.1\n",
      "../zarr_data/Cebus_olivaceus_ssp\n",
      "Cebus_olivaceus_ssp\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437081.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437015.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437116.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022436957.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437089.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437104.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437055.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437082.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437065.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437044.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437033.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437013.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437131.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437038.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437043.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022436941.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437085.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437119.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437031.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437094.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437042.1\n",
      "../zarr_data/Semnopithecus_priam_ssp\n",
      "Semnopithecus_priam_ssp\n",
      "../zarr_data/Semnopithecus_priam_ssp/NW_022681471.1\n",
      "../zarr_data/Semnopithecus_priam_ssp/NW_022681432.1\n",
      "../zarr_data/Semnopithecus_priam_ssp/NW_022680461.1\n",
      "../zarr_data/Semnopithecus_priam_ssp/NW_022680471.1\n",
      "../zarr_data/Semnopithecus_priam_ssp/NW_022680465.1\n",
      "../zarr_data/Semnopithecus_priam_ssp/NW_022681445.1\n",
      "../zarr_data/Semnopithecus_priam_ssp/NW_022681437.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp\n",
      "Semnopithecus_entellus_ssp\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022680465.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681445.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681437.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681471.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681432.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022680461.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022680471.1\n",
      "../zarr_data/Hylobates_pileatus_ssp\n",
      "Hylobates_pileatus_ssp\n",
      "../zarr_data/Hylobates_pileatus_ssp/CM038521.1\n",
      "../zarr_data/Hylobates_pileatus_ssp/CM038516.1\n",
      "../zarr_data/Galago_moholi_ssp\n",
      "Galago_moholi_ssp\n",
      "../zarr_data/Galago_moholi_ssp/HiC_scaffold_31\n",
      "../zarr_data/Galago_moholi_ssp/HiC_scaffold_1\n",
      "../zarr_data/Rhinopithecus_strykeri_ssp\n",
      "Rhinopithecus_strykeri_ssp\n",
      "../zarr_data/Rhinopithecus_strykeri_ssp/NC_044549.1\n",
      "../zarr_data/Rhinopithecus_strykeri_ssp/NC_044555.1\n",
      "../zarr_data/Otolemur_garnettii_ssp\n",
      "Otolemur_garnettii_ssp\n",
      "../zarr_data/Otolemur_garnettii_ssp/HiC_scaffold_1\n",
      "../zarr_data/Colobus_polykomos_ssp\n",
      "Colobus_polykomos_ssp\n",
      "../zarr_data/Colobus_polykomos_ssp/CM058074.1\n",
      "../zarr_data/Colobus_polykomos_ssp/CM058096.1\n",
      "../zarr_data/Alouatta_discolor_ssp\n",
      "Alouatta_discolor_ssp\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Macaca_leucogenys_ssp\n",
      "Macaca_leucogenys_ssp\n",
      "../zarr_data/Macaca_leucogenys_ssp/NC_065578.1\n",
      "../zarr_data/Macaca_leucogenys_ssp/NC_065597.1\n",
      "../zarr_data/Macaca_leucogenys_ssp/NC_065598.1\n",
      "../zarr_data/Cercopithecus_mona_ssp\n",
      "Cercopithecus_mona_ssp\n",
      "../zarr_data/Cercopithecus_mona_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_mona_ssp/CM053398.1\n",
      "../zarr_data/Ateles_marginatus_ssp\n",
      "Ateles_marginatus_ssp\n",
      "../zarr_data/Ateles_marginatus_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Ateles_marginatus_ssp/CAJZLT010000040.1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Ateles_marginatus_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Ateles_marginatus_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Ateles_marginatus_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Ateles_marginatus_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Ateles_marginatus_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Ateles_marginatus_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Callithrix_kuhlii_ssp\n",
      "Callithrix_kuhlii_ssp\n",
      "../zarr_data/Callithrix_kuhlii_ssp/NC_071464.1\n",
      "../zarr_data/Callithrix_kuhlii_ssp/NC_071442.1\n",
      "../zarr_data/Propithecus_diadema_ssp\n",
      "Propithecus_diadema_ssp\n",
      "../zarr_data/Propithecus_diadema_ssp/HiC_scaffold_2\n",
      "../zarr_data/Propithecus_diadema_ssp/HiC_scaffold_1\n",
      "../zarr_data/Rhinopithecus_brelichi_ssp\n",
      "Rhinopithecus_brelichi_ssp\n",
      "../zarr_data/Rhinopithecus_brelichi_ssp/NC_044549.1\n",
      "../zarr_data/Rhinopithecus_brelichi_ssp/NC_044555.1\n",
      "../zarr_data/Eulemur_coronatus_ssp\n",
      "Eulemur_coronatus_ssp\n",
      "../zarr_data/Eulemur_coronatus_ssp/HiC_scaffold_1\n",
      "../zarr_data/Macaca_brunnescens_ssp\n",
      "Macaca_brunnescens_ssp\n",
      "../zarr_data/Macaca_brunnescens_ssp/NW_027257673.1\n",
      "../zarr_data/Macaca_brunnescens_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_brunnescens_ssp/NC_092125.1\n",
      "../zarr_data/Cercopithecus_ascanius_ssp\n",
      "Cercopithecus_ascanius_ssp\n",
      "../zarr_data/Cercopithecus_ascanius_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_ascanius_ssp/CM053398.1\n",
      "../zarr_data/Callithrix_jacchus_ssp\n",
      "Callithrix_jacchus_ssp\n",
      "../zarr_data/Callithrix_jacchus_ssp/NC_071442.1\n",
      "../zarr_data/Callithrix_jacchus_ssp/NC_071464.1\n",
      "../zarr_data/Avahi_peyrierasi_ssp\n",
      "Avahi_peyrierasi_ssp\n",
      "../zarr_data/Avahi_peyrierasi_ssp/HiC_scaffold_2\n",
      "../zarr_data/Avahi_peyrierasi_ssp/HiC_scaffold_1\n",
      "../zarr_data/Tarsius_dentatus_ssp\n",
      "Tarsius_dentatus_ssp\n",
      "../zarr_data/Tarsius_dentatus_ssp/JAPVLO010000001.1\n",
      "../zarr_data/Indri_indri_ssp\n",
      "Indri_indri_ssp\n",
      "../zarr_data/Indri_indri_ssp/HiC_scaffold_2\n",
      "../zarr_data/Indri_indri_ssp/HiC_scaffold_1\n",
      "../zarr_data/Macaca_silenus_ssp\n",
      "Macaca_silenus_ssp\n",
      "../zarr_data/Macaca_silenus_ssp/NW_027257673.1\n",
      "../zarr_data/Macaca_silenus_ssp/NC_092125.1\n",
      "../zarr_data/Macaca_silenus_ssp/NC_092145.1\n",
      "../zarr_data/Allochrocebus_preussi_ssp\n",
      "Allochrocebus_preussi_ssp\n",
      "../zarr_data/Allochrocebus_preussi_ssp/NC_132904.1\n",
      "../zarr_data/Allochrocebus_preussi_ssp/NC_132933.1\n",
      "../zarr_data/Cercopithecus_petaurista_ssp\n",
      "Cercopithecus_petaurista_ssp\n",
      "../zarr_data/Cercopithecus_petaurista_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_petaurista_ssp/CM053398.1\n",
      "../zarr_data/Mico_argentatus_ssp\n",
      "Mico_argentatus_ssp\n",
      "../zarr_data/Mico_argentatus_ssp/NC_071464.1\n",
      "../zarr_data/Mico_argentatus_ssp/NC_071442.1\n",
      "../zarr_data/Plecturocebus_hoffmannsi_ssp\n",
      "Plecturocebus_hoffmannsi_ssp\n",
      "../zarr_data/Plecturocebus_hoffmannsi_ssp/CM080815.1\n",
      "../zarr_data/Plecturocebus_hoffmannsi_ssp/CM080837.1\n",
      "../zarr_data/Alouatta_juara_ssp\n",
      "Alouatta_juara_ssp\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Hylobates_klossii_ssp\n",
      "Hylobates_klossii_ssp\n",
      "../zarr_data/Hylobates_klossii_ssp/CM038521.1\n",
      "../zarr_data/Hylobates_klossii_ssp/CM038516.1\n",
      "../zarr_data/Cheracebus_lucifer_ssp\n",
      "Cheracebus_lucifer_ssp\n",
      "../zarr_data/Cheracebus_lucifer_ssp/CM080837.1\n",
      "../zarr_data/Cheracebus_lucifer_ssp/CM080815.1\n",
      "../zarr_data/Propithecus_deckenii_ssp\n",
      "Propithecus_deckenii_ssp\n",
      "../zarr_data/Propithecus_deckenii_ssp/HiC_scaffold_2\n",
      "../zarr_data/Propithecus_deckenii_ssp/HiC_scaffold_1\n",
      "../zarr_data/Cheracebus_regulus_ssp\n",
      "Cheracebus_regulus_ssp\n",
      "../zarr_data/Cheracebus_regulus_ssp/CM080815.1\n",
      "../zarr_data/Cheracebus_regulus_ssp/CM080837.1\n",
      "../zarr_data/Alouatta_palliata_ssp\n",
      "Alouatta_palliata_ssp\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Colobus_angolensis_ssp\n",
      "Colobus_angolensis_ssp\n",
      "../zarr_data/Colobus_angolensis_ssp/CM058074.1\n",
      "../zarr_data/Colobus_angolensis_ssp/CM058096.1\n",
      "../zarr_data/Chlorocebus_cynosuros_ssp\n",
      "Chlorocebus_cynosuros_ssp\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "No objects to concatenate",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mValueError\u001b[39m                                Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[22]\u001b[39m\u001b[32m, line 18\u001b[39m\n\u001b[32m     15\u001b[39m large_contigs = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == \u001b[32m2\u001b[39m)].CONTIG_ID.unique()\n\u001b[32m     16\u001b[39m large_x = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == \u001b[32m2\u001b[39m) &\n\u001b[32m     17\u001b[39m                     (regions_df.MALE_PLOIDY == \u001b[32m1\u001b[39m)].CONTIG_ID\n\u001b[32m---> \u001b[39m\u001b[32m18\u001b[39m bed_files = \u001b[43mread_beds\u001b[49m\u001b[43m(\u001b[49m\u001b[43mlong_form\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m     19\u001b[39m \u001b[38;5;66;03m# Loading the genetic data.\u001b[39;00m\n\u001b[32m     20\u001b[39m df_l = []\n",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[5]\u001b[39m\u001b[32m, line 8\u001b[39m, in \u001b[36mread_beds\u001b[39m\u001b[34m(long_form)\u001b[39m\n\u001b[32m      6\u001b[39m     bed_file = pd.read_csv(b, sep=\u001b[33m\"\u001b[39m\u001b[38;5;130;01m\\t\u001b[39;00m\u001b[33m\"\u001b[39m, names=[\u001b[33m\"\u001b[39m\u001b[33mchrom\u001b[39m\u001b[33m\"\u001b[39m, \u001b[33m\"\u001b[39m\u001b[33mstart\u001b[39m\u001b[33m\"\u001b[39m, \u001b[33m\"\u001b[39m\u001b[33mend\u001b[39m\u001b[33m\"\u001b[39m])\n\u001b[32m      7\u001b[39m     bed_l.append(bed_file)\n\u001b[32m----> \u001b[39m\u001b[32m8\u001b[39m bed_files = \u001b[43mpd\u001b[49m\u001b[43m.\u001b[49m\u001b[43mconcat\u001b[49m\u001b[43m(\u001b[49m\u001b[43mbed_l\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m      9\u001b[39m bed_l = []\n\u001b[32m     10\u001b[39m \u001b[38;5;28;01mfor\u001b[39;00m b \u001b[38;5;129;01min\u001b[39;00m glob.glob(bed_path_x):\n\u001b[32m     11\u001b[39m     \u001b[38;5;66;03m#print(b)\u001b[39;00m\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/primatediversity2/lib/python3.12/site-packages/pandas/core/reshape/concat.py:382\u001b[39m, in \u001b[36mconcat\u001b[39m\u001b[34m(objs, axis, join, ignore_index, keys, levels, names, verify_integrity, sort, copy)\u001b[39m\n\u001b[32m    379\u001b[39m \u001b[38;5;28;01melif\u001b[39;00m copy \u001b[38;5;129;01mand\u001b[39;00m using_copy_on_write():\n\u001b[32m    380\u001b[39m     copy = \u001b[38;5;28;01mFalse\u001b[39;00m\n\u001b[32m--> \u001b[39m\u001b[32m382\u001b[39m op = \u001b[43m_Concatenator\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    383\u001b[39m \u001b[43m    \u001b[49m\u001b[43mobjs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    384\u001b[39m \u001b[43m    \u001b[49m\u001b[43maxis\u001b[49m\u001b[43m=\u001b[49m\u001b[43maxis\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    385\u001b[39m \u001b[43m    \u001b[49m\u001b[43mignore_index\u001b[49m\u001b[43m=\u001b[49m\u001b[43mignore_index\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    386\u001b[39m \u001b[43m    \u001b[49m\u001b[43mjoin\u001b[49m\u001b[43m=\u001b[49m\u001b[43mjoin\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    387\u001b[39m \u001b[43m    \u001b[49m\u001b[43mkeys\u001b[49m\u001b[43m=\u001b[49m\u001b[43mkeys\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    388\u001b[39m \u001b[43m    \u001b[49m\u001b[43mlevels\u001b[49m\u001b[43m=\u001b[49m\u001b[43mlevels\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    389\u001b[39m \u001b[43m    \u001b[49m\u001b[43mnames\u001b[49m\u001b[43m=\u001b[49m\u001b[43mnames\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    390\u001b[39m \u001b[43m    \u001b[49m\u001b[43mverify_integrity\u001b[49m\u001b[43m=\u001b[49m\u001b[43mverify_integrity\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    391\u001b[39m \u001b[43m    \u001b[49m\u001b[43mcopy\u001b[49m\u001b[43m=\u001b[49m\u001b[43mcopy\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    392\u001b[39m \u001b[43m    \u001b[49m\u001b[43msort\u001b[49m\u001b[43m=\u001b[49m\u001b[43msort\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    393\u001b[39m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    395\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m op.get_result()\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/primatediversity2/lib/python3.12/site-packages/pandas/core/reshape/concat.py:445\u001b[39m, in \u001b[36m_Concatenator.__init__\u001b[39m\u001b[34m(self, objs, axis, join, keys, levels, names, ignore_index, verify_integrity, copy, sort)\u001b[39m\n\u001b[32m    442\u001b[39m \u001b[38;5;28mself\u001b[39m.verify_integrity = verify_integrity\n\u001b[32m    443\u001b[39m \u001b[38;5;28mself\u001b[39m.copy = copy\n\u001b[32m--> \u001b[39m\u001b[32m445\u001b[39m objs, keys = \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_clean_keys_and_objs\u001b[49m\u001b[43m(\u001b[49m\u001b[43mobjs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkeys\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    447\u001b[39m \u001b[38;5;66;03m# figure out what our result ndim is going to be\u001b[39;00m\n\u001b[32m    448\u001b[39m ndims = \u001b[38;5;28mself\u001b[39m._get_ndims(objs)\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/primatediversity2/lib/python3.12/site-packages/pandas/core/reshape/concat.py:507\u001b[39m, in \u001b[36m_Concatenator._clean_keys_and_objs\u001b[39m\u001b[34m(self, objs, keys)\u001b[39m\n\u001b[32m    504\u001b[39m     objs_list = \u001b[38;5;28mlist\u001b[39m(objs)\n\u001b[32m    506\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(objs_list) == \u001b[32m0\u001b[39m:\n\u001b[32m--> \u001b[39m\u001b[32m507\u001b[39m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[33m\"\u001b[39m\u001b[33mNo objects to concatenate\u001b[39m\u001b[33m\"\u001b[39m)\n\u001b[32m    509\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m keys \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[32m    510\u001b[39m     objs_list = \u001b[38;5;28mlist\u001b[39m(com.not_none(*objs_list))\n",
      "\u001b[31mValueError\u001b[39m: No objects to concatenate"
     ]
    }
   ],
   "source": [
    "# Heterozygosity implementation. No missing filter - I count all sites which have a het call.\n",
    "for x in glob.glob(zarr_path+\"*\"):\n",
    "    print(x)\n",
    "    # The name used to load all the files, short and long version\n",
    "    short_form = x.split(\"/\")[-1].split(\"_\")[0]\n",
    "    long_form = x.split(\"/\")[-1]\n",
    "    print(long_form)\n",
    "    # Loading the various metadata files. Metadata, contig information, callability bed.\n",
    "    metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "    metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "    metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "    regions_df[\"chr_type\"] = [\"chrX\" if x == 2 and y == 1 else \"aut\" for x, y in zip(regions_df.FEMALE_PLOIDY, regions_df.MALE_PLOIDY)]\n",
    "    large_contigs = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "    large_x = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID\n",
    "    bed_files = read_beds(long_form)\n",
    "    # Loading the genetic data.\n",
    "    df_l = []\n",
    "    for c in glob.glob(x+\"/*\"):\n",
    "        print(c)\n",
    "        ds = sg.load_dataset(c)\n",
    "        # This implementation is the pi implementation.\n",
    "        # Probably problematic in some cases with population structure, but it is easier to implement\n",
    "        ds[\"sample_cohort\"] = ds[\"samples\"]\n",
    "        # Subsetting and windowing the sgkit dataset. The rechunking handles what otherwise would cause an error.\n",
    "        if c.split(\"/\")[-1] in list(large_x):\n",
    "            ds = haploid_double(ds, \"call_genotype\", \"samples\")\n",
    "        ds[\"call_genotype\"] = ds[\"call_genotype\"].clip(0)\n",
    "        ds = ds.sel(contigs=[ds.variant_contig[0].values])\n",
    "        ds = sg.window_by_position(ds, size=window_size)\n",
    "        ds = (sg.diversity(ds.chunk({\"variants\": 50000})))\n",
    "        df_sub = pd.DataFrame(ds.stat_diversity, columns=ds.sample_id)\n",
    "        df_sub[\"window_start\"] = list(range(0, len(ds.window_start)*window_size, window_size))\n",
    "        df_sub[\"chrom\"] = c.split(\"/\")[-1]\n",
    "        df_l.append(df_sub)\n",
    "    df_het = pd.concat(df_l)\n",
    "    bed_files = read_beds(long_form)\n",
    "    intervals_callable = pos_windows(bed_files, window_size, df_het[\"chrom\"].unique())\n",
    "    output_df = pd.merge(df_het, intervals_callable, on=[\"chrom\", \"window_start\"])\n",
    "    output_df[\"chr_type\"] = output_df[\"chrom\"].map(dict(zip(regions_df.CONTIG_ID, regions_df.chr_type)))\n",
    "    output_df[\"species\"] = long_form\n",
    "    output_df.to_csv(\"../results/window_stats/{}_100kb_het.txt\".format(long_form), sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "id": "70666f1b",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PD_0840</th>\n",
       "      <th>PD_0841</th>\n",
       "      <th>PD_0844</th>\n",
       "      <th>SAMN31952328</th>\n",
       "      <th>window_start</th>\n",
       "      <th>chrom</th>\n",
       "      <th>window_end</th>\n",
       "      <th>callable_frac</th>\n",
       "      <th>chr_type</th>\n",
       "      <th>species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>0.0</td>\n",
       "      <td>235.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0</td>\n",
       "      <td>HiC_scaffold_23</td>\n",
       "      <td>100000</td>\n",
       "      <td>0.93982</td>\n",
       "      <td>chrX</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.0</td>\n",
       "      <td>194.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>100000</td>\n",
       "      <td>HiC_scaffold_23</td>\n",
       "      <td>200000</td>\n",
       "      <td>0.94975</td>\n",
       "      <td>chrX</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>0.0</td>\n",
       "      <td>194.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>200000</td>\n",
       "      <td>HiC_scaffold_23</td>\n",
       "      <td>300000</td>\n",
       "      <td>0.96536</td>\n",
       "      <td>chrX</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>0.0</td>\n",
       "      <td>170.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>300000</td>\n",
       "      <td>HiC_scaffold_23</td>\n",
       "      <td>400000</td>\n",
       "      <td>0.95552</td>\n",
       "      <td>chrX</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>0.0</td>\n",
       "      <td>222.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>400000</td>\n",
       "      <td>HiC_scaffold_23</td>\n",
       "      <td>500000</td>\n",
       "      <td>0.94898</td>\n",
       "      <td>chrX</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3227</th>\n",
       "      <td>97.0</td>\n",
       "      <td>27.0</td>\n",
       "      <td>49.0</td>\n",
       "      <td>34.0</td>\n",
       "      <td>193600000</td>\n",
       "      <td>HiC_scaffold_1</td>\n",
       "      <td>193700000</td>\n",
       "      <td>0.96172</td>\n",
       "      <td>aut</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3228</th>\n",
       "      <td>92.0</td>\n",
       "      <td>21.0</td>\n",
       "      <td>129.0</td>\n",
       "      <td>24.0</td>\n",
       "      <td>193700000</td>\n",
       "      <td>HiC_scaffold_1</td>\n",
       "      <td>193800000</td>\n",
       "      <td>0.95889</td>\n",
       "      <td>aut</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3229</th>\n",
       "      <td>96.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>124.0</td>\n",
       "      <td>20.0</td>\n",
       "      <td>193800000</td>\n",
       "      <td>HiC_scaffold_1</td>\n",
       "      <td>193900000</td>\n",
       "      <td>0.96756</td>\n",
       "      <td>aut</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3230</th>\n",
       "      <td>111.0</td>\n",
       "      <td>15.0</td>\n",
       "      <td>93.0</td>\n",
       "      <td>102.0</td>\n",
       "      <td>193900000</td>\n",
       "      <td>HiC_scaffold_1</td>\n",
       "      <td>194000000</td>\n",
       "      <td>0.96973</td>\n",
       "      <td>aut</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3231</th>\n",
       "      <td>79.0</td>\n",
       "      <td>1.0</td>\n",
       "      <td>8.0</td>\n",
       "      <td>7.0</td>\n",
       "      <td>194000000</td>\n",
       "      <td>HiC_scaffold_1</td>\n",
       "      <td>194100000</td>\n",
       "      <td>0.93039</td>\n",
       "      <td>aut</td>\n",
       "      <td>Leontopithecus_chrysomelas_ssp</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3232 rows  10 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      PD_0840  PD_0841  PD_0844  SAMN31952328  window_start            chrom  \\\n",
       "0         0.0    235.0      0.0           0.0             0  HiC_scaffold_23   \n",
       "1         0.0    194.0      0.0           0.0        100000  HiC_scaffold_23   \n",
       "2         0.0    194.0      0.0           0.0        200000  HiC_scaffold_23   \n",
       "3         0.0    170.0      0.0           0.0        300000  HiC_scaffold_23   \n",
       "4         0.0    222.0      0.0           0.0        400000  HiC_scaffold_23   \n",
       "...       ...      ...      ...           ...           ...              ...   \n",
       "3227     97.0     27.0     49.0          34.0     193600000   HiC_scaffold_1   \n",
       "3228     92.0     21.0    129.0          24.0     193700000   HiC_scaffold_1   \n",
       "3229     96.0     15.0    124.0          20.0     193800000   HiC_scaffold_1   \n",
       "3230    111.0     15.0     93.0         102.0     193900000   HiC_scaffold_1   \n",
       "3231     79.0      1.0      8.0           7.0     194000000   HiC_scaffold_1   \n",
       "\n",
       "      window_end  callable_frac chr_type                         species  \n",
       "0         100000        0.93982     chrX  Leontopithecus_chrysomelas_ssp  \n",
       "1         200000        0.94975     chrX  Leontopithecus_chrysomelas_ssp  \n",
       "2         300000        0.96536     chrX  Leontopithecus_chrysomelas_ssp  \n",
       "3         400000        0.95552     chrX  Leontopithecus_chrysomelas_ssp  \n",
       "4         500000        0.94898     chrX  Leontopithecus_chrysomelas_ssp  \n",
       "...          ...            ...      ...                             ...  \n",
       "3227   193700000        0.96172      aut  Leontopithecus_chrysomelas_ssp  \n",
       "3228   193800000        0.95889      aut  Leontopithecus_chrysomelas_ssp  \n",
       "3229   193900000        0.96756      aut  Leontopithecus_chrysomelas_ssp  \n",
       "3230   194000000        0.96973      aut  Leontopithecus_chrysomelas_ssp  \n",
       "3231   194100000        0.93039      aut  Leontopithecus_chrysomelas_ssp  \n",
       "\n",
       "[3232 rows x 10 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "output_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "id": "c38d6fd6",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Leontopithecus_chrysomelas_ssp\n",
      "Leontopithecus_chrysomelas_ssp\n",
      "4 Leontopithecus_chrysomelas_ssp\n",
      "../zarr_data/Leontopithecus_chrysomelas_ssp/HiC_scaffold_23\n",
      "../zarr_data/Leontopithecus_chrysomelas_ssp/HiC_scaffold_1\n",
      "../zarr_data/Saguinus_labiatus_ssp\n",
      "Saguinus_labiatus_ssp\n",
      "2 Saguinus_labiatus_ssp\n",
      "../zarr_data/Saguinus_labiatus_ssp/CM063191.1\n",
      "../zarr_data/Saguinus_labiatus_ssp/JASSZA010000432.1\n",
      "../zarr_data/Saguinus_labiatus_ssp/CM063169.1\n",
      "../zarr_data/Plecturocebus_cupreus_ssp\n",
      "Plecturocebus_cupreus_ssp\n",
      "5 Plecturocebus_cupreus_ssp\n",
      "../zarr_data/Plecturocebus_cupreus_ssp/CM080837.1\n",
      "../zarr_data/Plecturocebus_cupreus_ssp/CM080815.1\n",
      "../zarr_data/Papio_hamadryas_ssp\n",
      "Papio_hamadryas_ssp\n",
      "47 Papio_hamadryas_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_fascicularis_ssp\n",
      "Macaca_fascicularis_ssp\n",
      "159 Macaca_fascicularis_ssp\n",
      "Skipping\n",
      "../zarr_data/Saimiri_macrodon_ssp\n",
      "Saimiri_macrodon_ssp\n",
      "3 Saimiri_macrodon_ssp\n",
      "../zarr_data/Saimiri_macrodon_ssp/NW_024100917.1\n",
      "../zarr_data/Saimiri_macrodon_ssp/NW_024100929.1\n",
      "../zarr_data/Saimiri_macrodon_ssp/NW_024100959.1\n",
      "../zarr_data/Cheirogaleus_crossleyi_ssp\n",
      "Cheirogaleus_crossleyi_ssp\n",
      "1 Cheirogaleus_crossleyi_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheirogaleus_medius_ssp\n",
      "Cheirogaleus_medius_ssp\n",
      "5 Cheirogaleus_medius_ssp\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01001483.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01002707.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01002350.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01000656.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01003151.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01000568.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01000937.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01001044.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01003018.1\n",
      "../zarr_data/Cheirogaleus_medius_ssp/VBSM01003185.1\n",
      "../zarr_data/Plecturocebus_caligatus_ssp\n",
      "Plecturocebus_caligatus_ssp\n",
      "3 Plecturocebus_caligatus_ssp\n",
      "../zarr_data/Plecturocebus_caligatus_ssp/CM080815.1\n",
      "../zarr_data/Plecturocebus_caligatus_ssp/CM080837.1\n",
      "../zarr_data/Callimico_goeldii_ssp\n",
      "Callimico_goeldii_ssp\n",
      "2 Callimico_goeldii_ssp\n",
      "../zarr_data/Callimico_goeldii_ssp/HiC_scaffold_1\n",
      "../zarr_data/Trachypithecus_laotum_ssp\n",
      "Trachypithecus_laotum_ssp\n",
      "1 Trachypithecus_laotum_ssp\n",
      "Skipping\n",
      "../zarr_data/Lepilemur_dorsalis_ssp\n",
      "Lepilemur_dorsalis_ssp\n",
      "2 Lepilemur_dorsalis_ssp\n",
      "../zarr_data/Lepilemur_dorsalis_ssp/NC_033692.1\n",
      "../zarr_data/Lepilemur_dorsalis_ssp/NC_033660.1\n",
      "../zarr_data/Miopithecus_ogouensis_ssp\n",
      "Miopithecus_ogouensis_ssp\n",
      "1 Miopithecus_ogouensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Tarsius_lariang_ssp\n",
      "Tarsius_lariang_ssp\n",
      "1 Tarsius_lariang_ssp\n",
      "Skipping\n",
      "../zarr_data/Mirza_coquereli_ssp\n",
      "Mirza_coquereli_ssp\n",
      "1 Mirza_coquereli_ssp\n",
      "Skipping\n",
      "../zarr_data/Ateles_belzebuth_ssp\n",
      "Ateles_belzebuth_ssp\n",
      "2 Ateles_belzebuth_ssp\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Ateles_belzebuth_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp\n",
      "Semnopithecus_schistaceus_ssp\n",
      "2 Semnopithecus_schistaceus_ssp\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022680471.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022680461.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022681471.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022681432.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022681445.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022681437.1\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp/NW_022680465.1\n",
      "../zarr_data/Saimiri_oerstedii_ssp\n",
      "Saimiri_oerstedii_ssp\n",
      "1 Saimiri_oerstedii_ssp\n",
      "Skipping\n",
      "../zarr_data/Piliocolobus_badius_ssp\n",
      "Piliocolobus_badius_ssp\n",
      "1 Piliocolobus_badius_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_brunneus_ssp\n",
      "Plecturocebus_brunneus_ssp\n",
      "2 Plecturocebus_brunneus_ssp\n",
      "../zarr_data/Plecturocebus_brunneus_ssp/CM080815.1\n",
      "../zarr_data/Plecturocebus_brunneus_ssp/CM080837.1\n",
      "../zarr_data/Cebus_imitator_ssp\n",
      "Cebus_imitator_ssp\n",
      "2 Cebus_imitator_ssp\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437082.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437065.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437044.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022436957.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437089.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437104.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437055.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437081.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437015.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437116.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437119.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437031.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437094.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437042.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437043.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022436941.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437085.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437013.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437131.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437038.1\n",
      "../zarr_data/Cebus_imitator_ssp/NW_022437033.1\n",
      "../zarr_data/Perodicticus_potto_ssp\n",
      "Perodicticus_potto_ssp\n",
      "3 Perodicticus_potto_ssp\n",
      "../zarr_data/Perodicticus_potto_ssp/NC_069780.1\n",
      "../zarr_data/Perodicticus_potto_ssp/NC_069804.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp\n",
      "Trachypithecus_poliocephalus_ssp\n",
      "12 Trachypithecus_poliocephalus_ssp\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022681471.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022681432.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022680471.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022680461.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022680465.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022681445.1\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp/NW_022681437.1\n",
      "../zarr_data/Symphalangus_syndactylus_ssp\n",
      "Symphalangus_syndactylus_ssp\n",
      "5 Symphalangus_syndactylus_ssp\n",
      "../zarr_data/Symphalangus_syndactylus_ssp/NC_072423.2\n",
      "../zarr_data/Symphalangus_syndactylus_ssp/NC_072447.2\n",
      "../zarr_data/Pongo_pygmaeus_ssp\n",
      "Pongo_pygmaeus_ssp\n",
      "26 Pongo_pygmaeus_ssp\n",
      "Skipping\n",
      "../zarr_data/Avahi_laniger_ssp\n",
      "Avahi_laniger_ssp\n",
      "2 Avahi_laniger_ssp\n",
      "../zarr_data/Avahi_laniger_ssp/HiC_scaffold_2\n",
      "../zarr_data/Avahi_laniger_ssp/HiC_scaffold_1\n",
      "../zarr_data/Lepilemur_sahamalazensis_ssp\n",
      "Lepilemur_sahamalazensis_ssp\n",
      "1 Lepilemur_sahamalazensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Allenopithecus_nigroviridis_ssp\n",
      "Allenopithecus_nigroviridis_ssp\n",
      "3 Allenopithecus_nigroviridis_ssp\n",
      "../zarr_data/Allenopithecus_nigroviridis_ssp/HiC_scaffold_1\n",
      "../zarr_data/Allenopithecus_nigroviridis_ssp/HiC_scaffold_2\n",
      "../zarr_data/Alouatta_belzebul_ssp\n",
      "Alouatta_belzebul_ssp\n",
      "3 Alouatta_belzebul_ssp\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_belzebul_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Cercopithecus_denti_ssp\n",
      "Cercopithecus_denti_ssp\n",
      "2 Cercopithecus_denti_ssp\n",
      "../zarr_data/Cercopithecus_denti_ssp/CM053398.1\n",
      "../zarr_data/Cercopithecus_denti_ssp/CM053363.1\n",
      "../zarr_data/Saimiri_ustus_ssp\n",
      "Saimiri_ustus_ssp\n",
      "10 Saimiri_ustus_ssp\n",
      "../zarr_data/Saimiri_ustus_ssp/NW_024100917.1\n",
      "../zarr_data/Saimiri_ustus_ssp/NW_024100929.1\n",
      "../zarr_data/Saimiri_ustus_ssp/NW_024100959.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp\n",
      "Trachypithecus_vetulus_ssp\n",
      "3 Trachypithecus_vetulus_ssp\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022680461.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022680471.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022681432.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022681471.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022681437.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022681445.1\n",
      "../zarr_data/Trachypithecus_vetulus_ssp/NW_022680465.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp\n",
      "Trachypithecus_francoisi_ssp\n",
      "9 Trachypithecus_francoisi_ssp\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681432.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681471.1\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022680461.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022680471.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022680465.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681437.1\n",
      "../zarr_data/Trachypithecus_francoisi_ssp/NW_022681445.1\n",
      "../zarr_data/Nomascus_gabriellae_ssp\n",
      "Nomascus_gabriellae_ssp\n",
      "1 Nomascus_gabriellae_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_hamlyni_ssp\n",
      "Cercopithecus_hamlyni_ssp\n",
      "2 Cercopithecus_hamlyni_ssp\n",
      "../zarr_data/Cercopithecus_hamlyni_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_hamlyni_ssp/CM053398.1\n",
      "../zarr_data/Leontocebus_nigricollis_ssp\n",
      "Leontocebus_nigricollis_ssp\n",
      "1 Leontocebus_nigricollis_ssp\n",
      "Skipping\n",
      "../zarr_data/Chiropotes_albinasus_ssp\n",
      "Chiropotes_albinasus_ssp\n",
      "4 Chiropotes_albinasus_ssp\n",
      "../zarr_data/Chiropotes_albinasus_ssp/CM052647.1\n",
      "../zarr_data/Chiropotes_albinasus_ssp/CM052624.1\n",
      "../zarr_data/Macaca_tonkeana_ssp\n",
      "Macaca_tonkeana_ssp\n",
      "8 Macaca_tonkeana_ssp\n",
      "../zarr_data/Macaca_tonkeana_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_tonkeana_ssp/NC_092125.1\n",
      "../zarr_data/Macaca_tonkeana_ssp/NW_027257673.1\n",
      "../zarr_data/Gorilla_beringei_ssp\n",
      "Gorilla_beringei_ssp\n",
      "28 Gorilla_beringei_ssp\n",
      "Skipping\n",
      "../zarr_data/Cebus_albifrons_ssp\n",
      "Cebus_albifrons_ssp\n",
      "5 Cebus_albifrons_ssp\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437031.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437119.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437042.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437094.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437043.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437085.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022436941.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437013.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437038.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437131.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437033.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437082.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437044.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437065.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437089.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022436957.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437055.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437104.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437015.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437081.1\n",
      "../zarr_data/Cebus_albifrons_ssp/NW_022437116.1\n",
      "../zarr_data/Mico_humeralifer_ssp\n",
      "Mico_humeralifer_ssp\n",
      "1 Mico_humeralifer_ssp\n",
      "Skipping\n",
      "../zarr_data/Cacajao_calvus_ssp\n",
      "Cacajao_calvus_ssp\n",
      "19 Cacajao_calvus_ssp\n",
      "../zarr_data/Cacajao_calvus_ssp/CM052624.1\n",
      "../zarr_data/Cacajao_calvus_ssp/CM052647.1\n",
      "../zarr_data/Macaca_nemestrina_ssp\n",
      "Macaca_nemestrina_ssp\n",
      "26 Macaca_nemestrina_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_sinica_ssp\n",
      "Macaca_sinica_ssp\n",
      "1 Macaca_sinica_ssp\n",
      "Skipping\n",
      "../zarr_data/Aotus_griseimembra_ssp\n",
      "Aotus_griseimembra_ssp\n",
      "4 Aotus_griseimembra_ssp\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988472.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988412.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988496.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988488.1\n",
      "../zarr_data/Aotus_griseimembra_ssp/NW_026988415.1\n",
      "../zarr_data/Cercopithecus_neglectus_ssp\n",
      "Cercopithecus_neglectus_ssp\n",
      "3 Cercopithecus_neglectus_ssp\n",
      "../zarr_data/Cercopithecus_neglectus_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_neglectus_ssp/CM053398.1\n",
      "../zarr_data/Pithecia_vanzolinii_ssp\n",
      "Pithecia_vanzolinii_ssp\n",
      "1 Pithecia_vanzolinii_ssp\n",
      "Skipping\n",
      "../zarr_data/Cebus_olivaceus_ssp\n",
      "Cebus_olivaceus_ssp\n",
      "2 Cebus_olivaceus_ssp\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437081.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437015.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437116.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022436957.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437089.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437104.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437055.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437082.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437065.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437044.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437033.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437013.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437131.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437038.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437043.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022436941.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437085.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437119.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437031.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437094.1\n",
      "../zarr_data/Cebus_olivaceus_ssp/NW_022437042.1\n",
      "../zarr_data/Semnopithecus_priam_ssp\n",
      "Semnopithecus_priam_ssp\n",
      "1 Semnopithecus_priam_ssp\n",
      "Skipping\n",
      "../zarr_data/Semnopithecus_entellus_ssp\n",
      "Semnopithecus_entellus_ssp\n",
      "9 Semnopithecus_entellus_ssp\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022680465.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681445.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681437.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681471.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022681432.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022680461.1\n",
      "../zarr_data/Semnopithecus_entellus_ssp/NW_022680471.1\n",
      "../zarr_data/Hylobates_pileatus_ssp\n",
      "Hylobates_pileatus_ssp\n",
      "3 Hylobates_pileatus_ssp\n",
      "../zarr_data/Hylobates_pileatus_ssp/CM038521.1\n",
      "../zarr_data/Hylobates_pileatus_ssp/CM038516.1\n",
      "../zarr_data/Galago_moholi_ssp\n",
      "Galago_moholi_ssp\n",
      "2 Galago_moholi_ssp\n",
      "../zarr_data/Galago_moholi_ssp/HiC_scaffold_31\n",
      "../zarr_data/Galago_moholi_ssp/HiC_scaffold_1\n",
      "../zarr_data/Rhinopithecus_strykeri_ssp\n",
      "Rhinopithecus_strykeri_ssp\n",
      "3 Rhinopithecus_strykeri_ssp\n",
      "../zarr_data/Rhinopithecus_strykeri_ssp/NC_044549.1\n",
      "../zarr_data/Rhinopithecus_strykeri_ssp/NC_044555.1\n",
      "../zarr_data/Otolemur_garnettii_ssp\n",
      "Otolemur_garnettii_ssp\n",
      "40 Otolemur_garnettii_ssp\n",
      "Skipping\n",
      "../zarr_data/Colobus_polykomos_ssp\n",
      "Colobus_polykomos_ssp\n",
      "1 Colobus_polykomos_ssp\n",
      "Skipping\n",
      "../zarr_data/Alouatta_discolor_ssp\n",
      "Alouatta_discolor_ssp\n",
      "2 Alouatta_discolor_ssp\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Alouatta_discolor_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Macaca_leucogenys_ssp\n",
      "Macaca_leucogenys_ssp\n",
      "4 Macaca_leucogenys_ssp\n",
      "../zarr_data/Macaca_leucogenys_ssp/NC_065578.1\n",
      "../zarr_data/Macaca_leucogenys_ssp/NC_065597.1\n",
      "../zarr_data/Macaca_leucogenys_ssp/NC_065598.1\n",
      "../zarr_data/Cercopithecus_mona_ssp\n",
      "Cercopithecus_mona_ssp\n",
      "15 Cercopithecus_mona_ssp\n",
      "../zarr_data/Cercopithecus_mona_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_mona_ssp/CM053398.1\n",
      "../zarr_data/Ateles_marginatus_ssp\n",
      "Ateles_marginatus_ssp\n",
      "1 Ateles_marginatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Callithrix_kuhlii_ssp\n",
      "Callithrix_kuhlii_ssp\n",
      "1 Callithrix_kuhlii_ssp\n",
      "Skipping\n",
      "../zarr_data/Propithecus_diadema_ssp\n",
      "Propithecus_diadema_ssp\n",
      "7 Propithecus_diadema_ssp\n",
      "../zarr_data/Propithecus_diadema_ssp/HiC_scaffold_2\n",
      "../zarr_data/Propithecus_diadema_ssp/HiC_scaffold_1\n",
      "../zarr_data/Rhinopithecus_brelichi_ssp\n",
      "Rhinopithecus_brelichi_ssp\n",
      "5 Rhinopithecus_brelichi_ssp\n",
      "../zarr_data/Rhinopithecus_brelichi_ssp/NC_044549.1\n",
      "../zarr_data/Rhinopithecus_brelichi_ssp/NC_044555.1\n",
      "../zarr_data/Eulemur_coronatus_ssp\n",
      "Eulemur_coronatus_ssp\n",
      "6 Eulemur_coronatus_ssp\n",
      "../zarr_data/Eulemur_coronatus_ssp/HiC_scaffold_1\n",
      "../zarr_data/Macaca_brunnescens_ssp\n",
      "Macaca_brunnescens_ssp\n",
      "1 Macaca_brunnescens_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_ascanius_ssp\n",
      "Cercopithecus_ascanius_ssp\n",
      "5 Cercopithecus_ascanius_ssp\n",
      "../zarr_data/Cercopithecus_ascanius_ssp/CM053363.1\n",
      "../zarr_data/Cercopithecus_ascanius_ssp/CM053398.1\n",
      "../zarr_data/Callithrix_jacchus_ssp\n",
      "Callithrix_jacchus_ssp\n",
      "12 Callithrix_jacchus_ssp\n",
      "../zarr_data/Callithrix_jacchus_ssp/NC_071442.1\n",
      "../zarr_data/Callithrix_jacchus_ssp/NC_071464.1\n",
      "../zarr_data/Avahi_peyrierasi_ssp\n",
      "Avahi_peyrierasi_ssp\n",
      "2 Avahi_peyrierasi_ssp\n",
      "../zarr_data/Avahi_peyrierasi_ssp/HiC_scaffold_2\n",
      "../zarr_data/Avahi_peyrierasi_ssp/HiC_scaffold_1\n",
      "../zarr_data/Tarsius_dentatus_ssp\n",
      "Tarsius_dentatus_ssp\n",
      "1 Tarsius_dentatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Indri_indri_ssp\n",
      "Indri_indri_ssp\n",
      "2 Indri_indri_ssp\n",
      "../zarr_data/Indri_indri_ssp/HiC_scaffold_2\n",
      "../zarr_data/Indri_indri_ssp/HiC_scaffold_1\n",
      "../zarr_data/Macaca_silenus_ssp\n",
      "Macaca_silenus_ssp\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "17 Macaca_silenus_ssp\n",
      "../zarr_data/Macaca_silenus_ssp/NW_027257673.1\n",
      "../zarr_data/Macaca_silenus_ssp/NC_092125.1\n",
      "../zarr_data/Macaca_silenus_ssp/NC_092145.1\n",
      "../zarr_data/Allochrocebus_preussi_ssp\n",
      "Allochrocebus_preussi_ssp\n",
      "1 Allochrocebus_preussi_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_petaurista_ssp\n",
      "Cercopithecus_petaurista_ssp\n",
      "1 Cercopithecus_petaurista_ssp\n",
      "Skipping\n",
      "../zarr_data/Mico_argentatus_ssp\n",
      "Mico_argentatus_ssp\n",
      "4 Mico_argentatus_ssp\n",
      "../zarr_data/Mico_argentatus_ssp/NC_071464.1\n",
      "../zarr_data/Mico_argentatus_ssp/NC_071442.1\n",
      "../zarr_data/Plecturocebus_hoffmannsi_ssp\n",
      "Plecturocebus_hoffmannsi_ssp\n",
      "1 Plecturocebus_hoffmannsi_ssp\n",
      "Skipping\n",
      "../zarr_data/Alouatta_juara_ssp\n",
      "Alouatta_juara_ssp\n",
      "2 Alouatta_juara_ssp\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_juara_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Hylobates_klossii_ssp\n",
      "Hylobates_klossii_ssp\n",
      "1 Hylobates_klossii_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheracebus_lucifer_ssp\n",
      "Cheracebus_lucifer_ssp\n",
      "4 Cheracebus_lucifer_ssp\n",
      "../zarr_data/Cheracebus_lucifer_ssp/CM080837.1\n",
      "../zarr_data/Cheracebus_lucifer_ssp/CM080815.1\n",
      "../zarr_data/Propithecus_deckenii_ssp\n",
      "Propithecus_deckenii_ssp\n",
      "1 Propithecus_deckenii_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheracebus_regulus_ssp\n",
      "Cheracebus_regulus_ssp\n",
      "2 Cheracebus_regulus_ssp\n",
      "../zarr_data/Cheracebus_regulus_ssp/CM080815.1\n",
      "../zarr_data/Cheracebus_regulus_ssp/CM080837.1\n",
      "../zarr_data/Alouatta_palliata_ssp\n",
      "Alouatta_palliata_ssp\n",
      "6 Alouatta_palliata_ssp\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_palliata_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Colobus_angolensis_ssp\n",
      "Colobus_angolensis_ssp\n",
      "1 Colobus_angolensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Chlorocebus_cynosuros_ssp\n",
      "Chlorocebus_cynosuros_ssp\n",
      "16 Chlorocebus_cynosuros_ssp\n"
     ]
    },
    {
     "ename": "ValueError",
     "evalue": "No objects to concatenate",
     "output_type": "error",
     "traceback": [
      "\u001b[31m---------------------------------------------------------------------------\u001b[39m",
      "\u001b[31mValueError\u001b[39m                                Traceback (most recent call last)",
      "\u001b[36mCell\u001b[39m\u001b[36m \u001b[39m\u001b[32mIn[11]\u001b[39m\u001b[32m, line 45\u001b[39m\n\u001b[32m     43\u001b[39m     df_sub[\u001b[33m\"\u001b[39m\u001b[33mvariants_used\u001b[39m\u001b[33m\"\u001b[39m] = \u001b[38;5;28mlen\u001b[39m(ds.variants)\n\u001b[32m     44\u001b[39m     df_l.append(df_sub)\n\u001b[32m---> \u001b[39m\u001b[32m45\u001b[39m output_df = \u001b[43mpd\u001b[49m\u001b[43m.\u001b[49m\u001b[43mconcat\u001b[49m\u001b[43m(\u001b[49m\u001b[43mdf_l\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m     46\u001b[39m output_df[\u001b[33m\"\u001b[39m\u001b[33mchr_type\u001b[39m\u001b[33m\"\u001b[39m] = output_df[\u001b[33m\"\u001b[39m\u001b[33mchrom\u001b[39m\u001b[33m\"\u001b[39m].map(\u001b[38;5;28mdict\u001b[39m(\u001b[38;5;28mzip\u001b[39m(regions_df.CONTIG_ID, regions_df.chr_type)))\n\u001b[32m     47\u001b[39m output_df[\u001b[33m\"\u001b[39m\u001b[33mspecies\u001b[39m\u001b[33m\"\u001b[39m] = long_form\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/primatediversity2/lib/python3.12/site-packages/pandas/core/reshape/concat.py:382\u001b[39m, in \u001b[36mconcat\u001b[39m\u001b[34m(objs, axis, join, ignore_index, keys, levels, names, verify_integrity, sort, copy)\u001b[39m\n\u001b[32m    379\u001b[39m \u001b[38;5;28;01melif\u001b[39;00m copy \u001b[38;5;129;01mand\u001b[39;00m using_copy_on_write():\n\u001b[32m    380\u001b[39m     copy = \u001b[38;5;28;01mFalse\u001b[39;00m\n\u001b[32m--> \u001b[39m\u001b[32m382\u001b[39m op = \u001b[43m_Concatenator\u001b[49m\u001b[43m(\u001b[49m\n\u001b[32m    383\u001b[39m \u001b[43m    \u001b[49m\u001b[43mobjs\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    384\u001b[39m \u001b[43m    \u001b[49m\u001b[43maxis\u001b[49m\u001b[43m=\u001b[49m\u001b[43maxis\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    385\u001b[39m \u001b[43m    \u001b[49m\u001b[43mignore_index\u001b[49m\u001b[43m=\u001b[49m\u001b[43mignore_index\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    386\u001b[39m \u001b[43m    \u001b[49m\u001b[43mjoin\u001b[49m\u001b[43m=\u001b[49m\u001b[43mjoin\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    387\u001b[39m \u001b[43m    \u001b[49m\u001b[43mkeys\u001b[49m\u001b[43m=\u001b[49m\u001b[43mkeys\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    388\u001b[39m \u001b[43m    \u001b[49m\u001b[43mlevels\u001b[49m\u001b[43m=\u001b[49m\u001b[43mlevels\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    389\u001b[39m \u001b[43m    \u001b[49m\u001b[43mnames\u001b[49m\u001b[43m=\u001b[49m\u001b[43mnames\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    390\u001b[39m \u001b[43m    \u001b[49m\u001b[43mverify_integrity\u001b[49m\u001b[43m=\u001b[49m\u001b[43mverify_integrity\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    391\u001b[39m \u001b[43m    \u001b[49m\u001b[43mcopy\u001b[49m\u001b[43m=\u001b[49m\u001b[43mcopy\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    392\u001b[39m \u001b[43m    \u001b[49m\u001b[43msort\u001b[49m\u001b[43m=\u001b[49m\u001b[43msort\u001b[49m\u001b[43m,\u001b[49m\n\u001b[32m    393\u001b[39m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    395\u001b[39m \u001b[38;5;28;01mreturn\u001b[39;00m op.get_result()\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/primatediversity2/lib/python3.12/site-packages/pandas/core/reshape/concat.py:445\u001b[39m, in \u001b[36m_Concatenator.__init__\u001b[39m\u001b[34m(self, objs, axis, join, keys, levels, names, ignore_index, verify_integrity, copy, sort)\u001b[39m\n\u001b[32m    442\u001b[39m \u001b[38;5;28mself\u001b[39m.verify_integrity = verify_integrity\n\u001b[32m    443\u001b[39m \u001b[38;5;28mself\u001b[39m.copy = copy\n\u001b[32m--> \u001b[39m\u001b[32m445\u001b[39m objs, keys = \u001b[38;5;28;43mself\u001b[39;49m\u001b[43m.\u001b[49m\u001b[43m_clean_keys_and_objs\u001b[49m\u001b[43m(\u001b[49m\u001b[43mobjs\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkeys\u001b[49m\u001b[43m)\u001b[49m\n\u001b[32m    447\u001b[39m \u001b[38;5;66;03m# figure out what our result ndim is going to be\u001b[39;00m\n\u001b[32m    448\u001b[39m ndims = \u001b[38;5;28mself\u001b[39m._get_ndims(objs)\n",
      "\u001b[36mFile \u001b[39m\u001b[32m~/miniconda3/envs/primatediversity2/lib/python3.12/site-packages/pandas/core/reshape/concat.py:507\u001b[39m, in \u001b[36m_Concatenator._clean_keys_and_objs\u001b[39m\u001b[34m(self, objs, keys)\u001b[39m\n\u001b[32m    504\u001b[39m     objs_list = \u001b[38;5;28mlist\u001b[39m(objs)\n\u001b[32m    506\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;28mlen\u001b[39m(objs_list) == \u001b[32m0\u001b[39m:\n\u001b[32m--> \u001b[39m\u001b[32m507\u001b[39m     \u001b[38;5;28;01mraise\u001b[39;00m \u001b[38;5;167;01mValueError\u001b[39;00m(\u001b[33m\"\u001b[39m\u001b[33mNo objects to concatenate\u001b[39m\u001b[33m\"\u001b[39m)\n\u001b[32m    509\u001b[39m \u001b[38;5;28;01mif\u001b[39;00m keys \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m:\n\u001b[32m    510\u001b[39m     objs_list = \u001b[38;5;28mlist\u001b[39m(com.not_none(*objs_list))\n",
      "\u001b[31mValueError\u001b[39m: No objects to concatenate"
     ]
    }
   ],
   "source": [
    "# Fst implementation.\n",
    "for x in glob.glob(zarr_path+\"*\"):\n",
    "    print(x)\n",
    "    # The name used to load all the files, short and long version\n",
    "    short_form = x.split(\"/\")[-1].split(\"_\")[0]\n",
    "    long_form = x.split(\"/\")[-1]\n",
    "    print(long_form)\n",
    "    # Loading the various metadata files. Metadata, contig information, callability bed.\n",
    "    metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "    metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "    metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "    regions_df[\"chr_type\"] = [\"chrX\" if x == 2 and y == 1 else \"aut\" for x, y in zip(regions_df.FEMALE_PLOIDY, regions_df.MALE_PLOIDY)]\n",
    "    large_contigs = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "    large_x = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID\n",
    "    # Skipping the large samples sizes and the singulars for the Fst calc\n",
    "    print(len(metadata_df.loc[metadata_df.GVCF_FOLDER == long_form]), long_form)\n",
    "    if ((len(metadata_df.loc[metadata_df.GVCF_FOLDER == long_form]) > 20) or\n",
    "    (len(metadata_df.loc[metadata_df.GVCF_FOLDER == long_form]) == 1)):\n",
    "        print(\"Skipping\")\n",
    "        continue\n",
    "    # Loading the genetic data.\n",
    "    df_l = []\n",
    "    for c in glob.glob(x+\"/*\"):\n",
    "        print(c)\n",
    "        ds = sg.load_dataset(c)\n",
    "        # This implementation is the pi implementation.\n",
    "        # Probably problematic in some cases with population structure, but it is easier to implement\n",
    "        ds[\"sample_cohort\"] = ds[\"samples\"]\n",
    "        # Subsetting and windowing the sgkit dataset. The rechunking handles what otherwise would cause an error.\n",
    "        #ds[\"call_genotype\"] = ds[\"call_genotype\"].clip(0)\n",
    "        ds = ds.sel(contigs=[ds.variant_contig[0].values])\n",
    "        if c.split(\"/\")[-1] in list(large_x):\n",
    "            ds = haploid_double(ds, \"call_genotype\", \"samples\")\n",
    "        missing_rate = ds.call_genotype_mask[:,:,0].sum(axis=1).values/ds.call_genotype_mask[:,:,0].count(axis=1).values\n",
    "        ds = ds.isel(variants=(missing_rate <= 0))\n",
    "        ds = sg.window_by_genome(ds)\n",
    "        ds = (sg.Fst(ds.chunk({\"variants\": 50000})))\n",
    "        df_sub = pd.DataFrame(ds.stat_Fst[0,:,], columns=ds.sample_id)\n",
    "        df_sub[\"chrom\"] = c.split(\"/\")[-1]\n",
    "        df_sub[\"variants_used\"] = len(ds.variants)\n",
    "        df_l.append(df_sub)\n",
    "    output_df = pd.concat(df_l)\n",
    "    output_df[\"chr_type\"] = output_df[\"chrom\"].map(dict(zip(regions_df.CONTIG_ID, regions_df.chr_type)))\n",
    "    output_df[\"species\"] = long_form\n",
    "    output_df.to_csv(\"../results/window_stats/{}_Fst.txt\".format(long_form), sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 46,
   "id": "59a431b6",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Leontopithecus_chrysomelas_ssp\n",
      "Leontopithecus_chrysomelas_ssp\n",
      "4 Leontopithecus_chrysomelas_ssp\n",
      "Skipping\n",
      "../zarr_data/Saguinus_labiatus_ssp\n",
      "Saguinus_labiatus_ssp\n",
      "2 Saguinus_labiatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_cupreus_ssp\n",
      "Plecturocebus_cupreus_ssp\n",
      "5 Plecturocebus_cupreus_ssp\n",
      "Skipping\n",
      "../zarr_data/Papio_hamadryas_ssp\n",
      "Papio_hamadryas_ssp\n",
      "47 Papio_hamadryas_ssp\n",
      "../zarr_data/Papio_hamadryas_ssp/NC_044976.1\n",
      "../zarr_data/Papio_hamadryas_ssp/NC_044996.1\n",
      "../zarr_data/Macaca_fascicularis_ssp\n",
      "Macaca_fascicularis_ssp\n",
      "159 Macaca_fascicularis_ssp\n",
      "Skipping\n",
      "../zarr_data/Saimiri_macrodon_ssp\n",
      "Saimiri_macrodon_ssp\n",
      "3 Saimiri_macrodon_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheirogaleus_crossleyi_ssp\n",
      "Cheirogaleus_crossleyi_ssp\n",
      "1 Cheirogaleus_crossleyi_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheirogaleus_medius_ssp\n",
      "Cheirogaleus_medius_ssp\n",
      "5 Cheirogaleus_medius_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_caligatus_ssp\n",
      "Plecturocebus_caligatus_ssp\n",
      "3 Plecturocebus_caligatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Callimico_goeldii_ssp\n",
      "Callimico_goeldii_ssp\n",
      "2 Callimico_goeldii_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_laotum_ssp\n",
      "Trachypithecus_laotum_ssp\n",
      "1 Trachypithecus_laotum_ssp\n",
      "Skipping\n",
      "../zarr_data/Lepilemur_dorsalis_ssp\n",
      "Lepilemur_dorsalis_ssp\n",
      "2 Lepilemur_dorsalis_ssp\n",
      "Skipping\n",
      "../zarr_data/Miopithecus_ogouensis_ssp\n",
      "Miopithecus_ogouensis_ssp\n",
      "1 Miopithecus_ogouensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Tarsius_lariang_ssp\n",
      "Tarsius_lariang_ssp\n",
      "1 Tarsius_lariang_ssp\n",
      "Skipping\n",
      "../zarr_data/Mirza_coquereli_ssp\n",
      "Mirza_coquereli_ssp\n",
      "1 Mirza_coquereli_ssp\n",
      "Skipping\n",
      "../zarr_data/Ateles_belzebuth_ssp\n",
      "Ateles_belzebuth_ssp\n",
      "2 Ateles_belzebuth_ssp\n",
      "Skipping\n",
      "../zarr_data/Semnopithecus_schistaceus_ssp\n",
      "Semnopithecus_schistaceus_ssp\n",
      "2 Semnopithecus_schistaceus_ssp\n",
      "Skipping\n",
      "../zarr_data/Saimiri_oerstedii_ssp\n",
      "Saimiri_oerstedii_ssp\n",
      "1 Saimiri_oerstedii_ssp\n",
      "Skipping\n",
      "../zarr_data/Piliocolobus_badius_ssp\n",
      "Piliocolobus_badius_ssp\n",
      "1 Piliocolobus_badius_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_brunneus_ssp\n",
      "Plecturocebus_brunneus_ssp\n",
      "2 Plecturocebus_brunneus_ssp\n",
      "Skipping\n",
      "../zarr_data/Cebus_imitator_ssp\n",
      "Cebus_imitator_ssp\n",
      "2 Cebus_imitator_ssp\n",
      "Skipping\n",
      "../zarr_data/Perodicticus_potto_ssp\n",
      "Perodicticus_potto_ssp\n",
      "3 Perodicticus_potto_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_poliocephalus_ssp\n",
      "Trachypithecus_poliocephalus_ssp\n",
      "12 Trachypithecus_poliocephalus_ssp\n",
      "Skipping\n",
      "../zarr_data/Symphalangus_syndactylus_ssp\n",
      "Symphalangus_syndactylus_ssp\n",
      "5 Symphalangus_syndactylus_ssp\n",
      "Skipping\n",
      "../zarr_data/Pongo_pygmaeus_ssp\n",
      "Pongo_pygmaeus_ssp\n",
      "26 Pongo_pygmaeus_ssp\n",
      "../zarr_data/Pongo_pygmaeus_ssp/NC_072396.2\n",
      "../zarr_data/Pongo_pygmaeus_ssp/NC_072373.2\n",
      "../zarr_data/Avahi_laniger_ssp\n",
      "Avahi_laniger_ssp\n",
      "2 Avahi_laniger_ssp\n",
      "Skipping\n",
      "../zarr_data/Lepilemur_sahamalazensis_ssp\n",
      "Lepilemur_sahamalazensis_ssp\n",
      "1 Lepilemur_sahamalazensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Allenopithecus_nigroviridis_ssp\n",
      "Allenopithecus_nigroviridis_ssp\n",
      "3 Allenopithecus_nigroviridis_ssp\n",
      "Skipping\n",
      "../zarr_data/Alouatta_belzebul_ssp\n",
      "Alouatta_belzebul_ssp\n",
      "3 Alouatta_belzebul_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_denti_ssp\n",
      "Cercopithecus_denti_ssp\n",
      "2 Cercopithecus_denti_ssp\n",
      "Skipping\n",
      "../zarr_data/Saimiri_ustus_ssp\n",
      "Saimiri_ustus_ssp\n",
      "10 Saimiri_ustus_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_vetulus_ssp\n",
      "Trachypithecus_vetulus_ssp\n",
      "3 Trachypithecus_vetulus_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_francoisi_ssp\n",
      "Trachypithecus_francoisi_ssp\n",
      "9 Trachypithecus_francoisi_ssp\n",
      "Skipping\n",
      "../zarr_data/Nomascus_gabriellae_ssp\n",
      "Nomascus_gabriellae_ssp\n",
      "1 Nomascus_gabriellae_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_hamlyni_ssp\n",
      "Cercopithecus_hamlyni_ssp\n",
      "2 Cercopithecus_hamlyni_ssp\n",
      "Skipping\n",
      "../zarr_data/Leontocebus_nigricollis_ssp\n",
      "Leontocebus_nigricollis_ssp\n",
      "1 Leontocebus_nigricollis_ssp\n",
      "Skipping\n",
      "../zarr_data/Chiropotes_albinasus_ssp\n",
      "Chiropotes_albinasus_ssp\n",
      "4 Chiropotes_albinasus_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_tonkeana_ssp\n",
      "Macaca_tonkeana_ssp\n",
      "8 Macaca_tonkeana_ssp\n",
      "Skipping\n",
      "../zarr_data/Gorilla_beringei_ssp\n",
      "Gorilla_beringei_ssp\n",
      "28 Gorilla_beringei_ssp\n",
      "../zarr_data/Gorilla_beringei_ssp/NC_073224.2\n",
      "../zarr_data/Gorilla_beringei_ssp/NC_073247.2\n",
      "../zarr_data/Cebus_albifrons_ssp\n",
      "Cebus_albifrons_ssp\n",
      "5 Cebus_albifrons_ssp\n",
      "Skipping\n",
      "../zarr_data/Mico_humeralifer_ssp\n",
      "Mico_humeralifer_ssp\n",
      "1 Mico_humeralifer_ssp\n",
      "Skipping\n",
      "../zarr_data/Cacajao_calvus_ssp\n",
      "Cacajao_calvus_ssp\n",
      "19 Cacajao_calvus_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_nemestrina_ssp\n",
      "Macaca_nemestrina_ssp\n",
      "26 Macaca_nemestrina_ssp\n",
      "../zarr_data/Macaca_nemestrina_ssp/NC_092125.1\n",
      "../zarr_data/Macaca_nemestrina_ssp/NC_092145.1\n",
      "../zarr_data/Macaca_nemestrina_ssp/NW_027257673.1\n",
      "../zarr_data/Macaca_sinica_ssp\n",
      "Macaca_sinica_ssp\n",
      "1 Macaca_sinica_ssp\n",
      "Skipping\n",
      "../zarr_data/Aotus_griseimembra_ssp\n",
      "Aotus_griseimembra_ssp\n",
      "4 Aotus_griseimembra_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_neglectus_ssp\n",
      "Cercopithecus_neglectus_ssp\n",
      "3 Cercopithecus_neglectus_ssp\n",
      "Skipping\n",
      "../zarr_data/Pithecia_vanzolinii_ssp\n",
      "Pithecia_vanzolinii_ssp\n",
      "1 Pithecia_vanzolinii_ssp\n",
      "Skipping\n",
      "../zarr_data/Cebus_olivaceus_ssp\n",
      "Cebus_olivaceus_ssp\n",
      "2 Cebus_olivaceus_ssp\n",
      "Skipping\n",
      "../zarr_data/Semnopithecus_priam_ssp\n",
      "Semnopithecus_priam_ssp\n",
      "1 Semnopithecus_priam_ssp\n",
      "Skipping\n",
      "../zarr_data/Semnopithecus_entellus_ssp\n",
      "Semnopithecus_entellus_ssp\n",
      "9 Semnopithecus_entellus_ssp\n",
      "Skipping\n",
      "../zarr_data/Hylobates_pileatus_ssp\n",
      "Hylobates_pileatus_ssp\n",
      "3 Hylobates_pileatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Galago_moholi_ssp\n",
      "Galago_moholi_ssp\n",
      "2 Galago_moholi_ssp\n",
      "Skipping\n",
      "../zarr_data/Rhinopithecus_strykeri_ssp\n",
      "Rhinopithecus_strykeri_ssp\n",
      "3 Rhinopithecus_strykeri_ssp\n",
      "Skipping\n",
      "../zarr_data/Otolemur_garnettii_ssp\n",
      "Otolemur_garnettii_ssp\n",
      "40 Otolemur_garnettii_ssp\n",
      "../zarr_data/Otolemur_garnettii_ssp/HiC_scaffold_1\n",
      "../zarr_data/Colobus_polykomos_ssp\n",
      "Colobus_polykomos_ssp\n",
      "1 Colobus_polykomos_ssp\n",
      "Skipping\n",
      "../zarr_data/Alouatta_discolor_ssp\n",
      "Alouatta_discolor_ssp\n",
      "2 Alouatta_discolor_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_leucogenys_ssp\n",
      "Macaca_leucogenys_ssp\n",
      "4 Macaca_leucogenys_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_mona_ssp\n",
      "Cercopithecus_mona_ssp\n",
      "15 Cercopithecus_mona_ssp\n",
      "Skipping\n",
      "../zarr_data/Ateles_marginatus_ssp\n",
      "Ateles_marginatus_ssp\n",
      "1 Ateles_marginatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Callithrix_kuhlii_ssp\n",
      "Callithrix_kuhlii_ssp\n",
      "1 Callithrix_kuhlii_ssp\n",
      "Skipping\n",
      "../zarr_data/Propithecus_diadema_ssp\n",
      "Propithecus_diadema_ssp\n",
      "7 Propithecus_diadema_ssp\n",
      "Skipping\n",
      "../zarr_data/Rhinopithecus_brelichi_ssp\n",
      "Rhinopithecus_brelichi_ssp\n",
      "5 Rhinopithecus_brelichi_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_coronatus_ssp\n",
      "Eulemur_coronatus_ssp\n",
      "6 Eulemur_coronatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_brunnescens_ssp\n",
      "Macaca_brunnescens_ssp\n",
      "1 Macaca_brunnescens_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_ascanius_ssp\n",
      "Cercopithecus_ascanius_ssp\n",
      "5 Cercopithecus_ascanius_ssp\n",
      "Skipping\n",
      "../zarr_data/Callithrix_jacchus_ssp\n",
      "Callithrix_jacchus_ssp\n",
      "12 Callithrix_jacchus_ssp\n",
      "Skipping\n",
      "../zarr_data/Avahi_peyrierasi_ssp\n",
      "Avahi_peyrierasi_ssp\n",
      "2 Avahi_peyrierasi_ssp\n",
      "Skipping\n",
      "../zarr_data/Tarsius_dentatus_ssp\n",
      "Tarsius_dentatus_ssp\n",
      "1 Tarsius_dentatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Indri_indri_ssp\n",
      "Indri_indri_ssp\n",
      "2 Indri_indri_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_silenus_ssp\n",
      "Macaca_silenus_ssp\n",
      "17 Macaca_silenus_ssp\n",
      "Skipping\n",
      "../zarr_data/Allochrocebus_preussi_ssp\n",
      "Allochrocebus_preussi_ssp\n",
      "1 Allochrocebus_preussi_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_petaurista_ssp\n",
      "Cercopithecus_petaurista_ssp\n",
      "1 Cercopithecus_petaurista_ssp\n",
      "Skipping\n",
      "../zarr_data/Mico_argentatus_ssp\n",
      "Mico_argentatus_ssp\n",
      "4 Mico_argentatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_hoffmannsi_ssp\n",
      "Plecturocebus_hoffmannsi_ssp\n",
      "1 Plecturocebus_hoffmannsi_ssp\n",
      "Skipping\n",
      "../zarr_data/Alouatta_juara_ssp\n",
      "Alouatta_juara_ssp\n",
      "2 Alouatta_juara_ssp\n",
      "Skipping\n",
      "../zarr_data/Hylobates_klossii_ssp\n",
      "Hylobates_klossii_ssp\n",
      "1 Hylobates_klossii_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheracebus_lucifer_ssp\n",
      "Cheracebus_lucifer_ssp\n",
      "4 Cheracebus_lucifer_ssp\n",
      "Skipping\n",
      "../zarr_data/Propithecus_deckenii_ssp\n",
      "Propithecus_deckenii_ssp\n",
      "1 Propithecus_deckenii_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheracebus_regulus_ssp\n",
      "Cheracebus_regulus_ssp\n",
      "2 Cheracebus_regulus_ssp\n",
      "Skipping\n",
      "../zarr_data/Alouatta_palliata_ssp\n",
      "Alouatta_palliata_ssp\n",
      "6 Alouatta_palliata_ssp\n",
      "Skipping\n",
      "../zarr_data/Colobus_angolensis_ssp\n",
      "Colobus_angolensis_ssp\n",
      "1 Colobus_angolensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Chlorocebus_cynosuros_ssp\n",
      "Chlorocebus_cynosuros_ssp\n",
      "16 Chlorocebus_cynosuros_ssp\n",
      "Skipping\n",
      "../zarr_data/Hylobates_moloch_ssp\n",
      "Hylobates_moloch_ssp\n",
      "4 Hylobates_moloch_ssp\n",
      "Skipping\n",
      "../zarr_data/Lemur_catta_ssp\n",
      "Lemur_catta_ssp\n",
      "18 Lemur_catta_ssp\n",
      "Skipping\n",
      "../zarr_data/Pongo_abelii_ssp\n",
      "Pongo_abelii_ssp\n",
      "27 Pongo_abelii_ssp\n",
      "../zarr_data/Pongo_abelii_ssp/NC_071985.2\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Pongo_abelii_ssp/NC_072008.2\n",
      "../zarr_data/Hapalemur_occidentalis_ssp\n",
      "Hapalemur_occidentalis_ssp\n",
      "1 Hapalemur_occidentalis_ssp\n",
      "Skipping\n",
      "../zarr_data/Allochrocebus_solatus_ssp\n",
      "Allochrocebus_solatus_ssp\n",
      "1 Allochrocebus_solatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Cebuella_niveiventris_ssp\n",
      "Cebuella_niveiventris_ssp\n",
      "2 Cebuella_niveiventris_ssp\n",
      "Skipping\n",
      "../zarr_data/Hoolock_hoolock_ssp\n",
      "Hoolock_hoolock_ssp\n",
      "11 Hoolock_hoolock_ssp\n",
      "Skipping\n",
      "../zarr_data/Chlorocebus_pygerythrus_ssp\n",
      "Chlorocebus_pygerythrus_ssp\n",
      "58 Chlorocebus_pygerythrus_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_roloway_ssp\n",
      "Cercopithecus_roloway_ssp\n",
      "1 Cercopithecus_roloway_ssp\n",
      "Skipping\n",
      "../zarr_data/Colobus_guereza_ssp\n",
      "Colobus_guereza_ssp\n",
      "9 Colobus_guereza_ssp\n",
      "Skipping\n",
      "../zarr_data/Callithrix_geoffroyi_ssp\n",
      "Callithrix_geoffroyi_ssp\n",
      "1 Callithrix_geoffroyi_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_obscurus_ssp\n",
      "Trachypithecus_obscurus_ssp\n",
      "3 Trachypithecus_obscurus_ssp\n",
      "Skipping\n",
      "../zarr_data/Saguinus_imperator_ssp\n",
      "Saguinus_imperator_ssp\n",
      "2 Saguinus_imperator_ssp\n",
      "Skipping\n",
      "../zarr_data/Gorilla_gorilla_ssp\n",
      "Gorilla_gorilla_ssp\n",
      "179 Gorilla_gorilla_ssp\n",
      "Skipping\n",
      "../zarr_data/Microcebus_tavaratra_ssp\n",
      "Microcebus_tavaratra_ssp\n",
      "1 Microcebus_tavaratra_ssp\n",
      "Skipping\n",
      "../zarr_data/Nycticebus_bengalensis_ssp\n",
      "Nycticebus_bengalensis_ssp\n",
      "9 Nycticebus_bengalensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_cristatus_ssp\n",
      "Trachypithecus_cristatus_ssp\n",
      "1 Trachypithecus_cristatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_rufifrons_ssp\n",
      "Eulemur_rufifrons_ssp\n",
      "6 Eulemur_rufifrons_ssp\n",
      "Skipping\n",
      "../zarr_data/Pithecia_albicans_ssp\n",
      "Pithecia_albicans_ssp\n",
      "1 Pithecia_albicans_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_mulatta_ssp\n",
      "Macaca_mulatta_ssp\n",
      "426 Macaca_mulatta_ssp\n",
      "Skipping\n",
      "../zarr_data/Cacajao_melanocephalus_ssp\n",
      "Cacajao_melanocephalus_ssp\n",
      "2 Cacajao_melanocephalus_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_hatinhensis_ssp\n",
      "Trachypithecus_hatinhensis_ssp\n",
      "1 Trachypithecus_hatinhensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Galagoides_demidoff_ssp\n",
      "Galagoides_demidoff_ssp\n",
      "1 Galagoides_demidoff_ssp\n",
      "Skipping\n",
      "../zarr_data/Aotus_trivirgatus_ssp\n",
      "Aotus_trivirgatus_ssp\n",
      "1 Aotus_trivirgatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Pithecia_mittermeieri_ssp\n",
      "Pithecia_mittermeieri_ssp\n",
      "2 Pithecia_mittermeieri_ssp\n",
      "Skipping\n",
      "../zarr_data/Hoolock_leuconedys_ssp\n",
      "Hoolock_leuconedys_ssp\n",
      "2 Hoolock_leuconedys_ssp\n",
      "Skipping\n",
      "../zarr_data/Rhinopithecus_avunculus_ssp\n",
      "Rhinopithecus_avunculus_ssp\n",
      "1 Rhinopithecus_avunculus_ssp\n",
      "Skipping\n",
      "../zarr_data/Pongo_tapanuliensis_ssp\n",
      "Pongo_tapanuliensis_ssp\n",
      "2 Pongo_tapanuliensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Nomascus_annamensis_ssp\n",
      "Nomascus_annamensis_ssp\n",
      "1 Nomascus_annamensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_nigra_ssp\n",
      "Macaca_nigra_ssp\n",
      "7 Macaca_nigra_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_diana_ssp\n",
      "Cercopithecus_diana_ssp\n",
      "1 Cercopithecus_diana_ssp\n",
      "Skipping\n",
      "../zarr_data/Nomascus_leucogenys_ssp\n",
      "Nomascus_leucogenys_ssp\n",
      "4 Nomascus_leucogenys_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_grovesi_ssp\n",
      "Plecturocebus_grovesi_ssp\n",
      "1 Plecturocebus_grovesi_ssp\n",
      "Skipping\n",
      "../zarr_data/Hylobates_lar_ssp\n",
      "Hylobates_lar_ssp\n",
      "4 Hylobates_lar_ssp\n",
      "Skipping\n",
      "../zarr_data/Papio_anubis_ssp\n",
      "Papio_anubis_ssp\n",
      "107 Papio_anubis_ssp\n",
      "Skipping\n",
      "../zarr_data/Leontocebus_fuscicollis_ssp\n",
      "Leontocebus_fuscicollis_ssp\n",
      "4 Leontocebus_fuscicollis_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_nigrescens_ssp\n",
      "Macaca_nigrescens_ssp\n",
      "2 Macaca_nigrescens_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_germaini_ssp\n",
      "Trachypithecus_germaini_ssp\n",
      "1 Trachypithecus_germaini_ssp\n",
      "Skipping\n",
      "../zarr_data/Ateles_geoffroyi_ssp\n",
      "Ateles_geoffroyi_ssp\n",
      "3 Ateles_geoffroyi_ssp\n",
      "Skipping\n",
      "../zarr_data/Cephalopachus_bancanus_ssp\n",
      "Cephalopachus_bancanus_ssp\n",
      "1 Cephalopachus_bancanus_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheracebus_lugens_ssp\n",
      "Cheracebus_lugens_ssp\n",
      "8 Cheracebus_lugens_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_mitis_ssp\n",
      "Cercopithecus_mitis_ssp\n",
      "5 Cercopithecus_mitis_ssp\n",
      "Skipping\n",
      "../zarr_data/Nomascus_concolor_ssp\n",
      "Nomascus_concolor_ssp\n",
      "1 Nomascus_concolor_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_cinerascens_ssp\n",
      "Plecturocebus_cinerascens_ssp\n",
      "2 Plecturocebus_cinerascens_ssp\n",
      "Skipping\n",
      "../zarr_data/Mandrillus_sphinx_ssp\n",
      "Mandrillus_sphinx_ssp\n",
      "3 Mandrillus_sphinx_ssp\n",
      "Skipping\n",
      "../zarr_data/Mico_humilis_ssp\n",
      "Mico_humilis_ssp\n",
      "1 Mico_humilis_ssp\n",
      "Skipping\n",
      "../zarr_data/Cebus_unicolor_ssp\n",
      "Cebus_unicolor_ssp\n",
      "4 Cebus_unicolor_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_mongoz_ssp\n",
      "Eulemur_mongoz_ssp\n",
      "7 Eulemur_mongoz_ssp\n",
      "Skipping\n",
      "../zarr_data/Mandrillus_leucophaeus_ssp\n",
      "Mandrillus_leucophaeus_ssp\n",
      "7 Mandrillus_leucophaeus_ssp\n",
      "Skipping\n",
      "../zarr_data/Pithecia_pithecia_ssp\n",
      "Pithecia_pithecia_ssp\n",
      "10 Pithecia_pithecia_ssp\n",
      "Skipping\n",
      "../zarr_data/Papio_kindae_ssp\n",
      "Papio_kindae_ssp\n",
      "28 Papio_kindae_ssp\n",
      "../zarr_data/Papio_kindae_ssp/NC_044976.1\n",
      "../zarr_data/Papio_kindae_ssp/NC_044996.1\n",
      "../zarr_data/Cheracebus_torquatus_ssp\n",
      "Cheracebus_torquatus_ssp\n",
      "1 Cheracebus_torquatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Lepilemur_septentrionalis_ssp\n",
      "Lepilemur_septentrionalis_ssp\n",
      "2 Lepilemur_septentrionalis_ssp\n",
      "Skipping\n",
      "../zarr_data/Saguinus_bicolor_ssp\n",
      "Saguinus_bicolor_ssp\n",
      "3 Saguinus_bicolor_ssp\n",
      "Skipping\n",
      "../zarr_data/Mirza_zaza_ssp\n",
      "Mirza_zaza_ssp\n",
      "6 Mirza_zaza_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_fuscata_ssp\n",
      "Macaca_fuscata_ssp\n",
      "67 Macaca_fuscata_ssp\n",
      "Skipping\n",
      "../zarr_data/Pan_troglodytes_ssp\n",
      "Pan_troglodytes_ssp\n",
      "199 Pan_troglodytes_ssp\n",
      "Skipping\n",
      "../zarr_data/Papio_papio_ssp\n",
      "Papio_papio_ssp\n",
      "14 Papio_papio_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_nemestrina_ssp_pca\n",
      "Macaca_nemestrina_ssp_pca\n",
      "0 Macaca_nemestrina_ssp_pca\n",
      "Skipping\n",
      "../zarr_data/Galago_senegalensis_ssp\n",
      "Galago_senegalensis_ssp\n",
      "3 Galago_senegalensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Ateles_paniscus_ssp\n",
      "Ateles_paniscus_ssp\n",
      "2 Ateles_paniscus_ssp\n",
      "Skipping\n",
      "../zarr_data/Varecia_variegata_ssp\n",
      "Varecia_variegata_ssp\n",
      "5 Varecia_variegata_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercocebus_lunulatus_ssp\n",
      "Cercocebus_lunulatus_ssp\n",
      "1 Cercocebus_lunulatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_siberu_ssp\n",
      "Macaca_siberu_ssp\n",
      "1 Macaca_siberu_ssp\n",
      "Skipping\n",
      "../zarr_data/Xanthonycticebus_pygmaeus_ssp\n",
      "Xanthonycticebus_pygmaeus_ssp\n",
      "3 Xanthonycticebus_pygmaeus_ssp\n",
      "Skipping\n",
      "../zarr_data/Avahi_occidentalis_ssp\n",
      "Avahi_occidentalis_ssp\n",
      "1 Avahi_occidentalis_ssp\n",
      "Skipping\n",
      "../zarr_data/Microcebus_mittermeieri_ssp\n",
      "Microcebus_mittermeieri_ssp\n",
      "1 Microcebus_mittermeieri_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_thibetana_thibetana_pca\n",
      "Macaca_thibetana_thibetana_pca\n",
      "0 Macaca_thibetana_thibetana_pca\n",
      "Skipping\n",
      "../zarr_data/Propithecus_edwardsi_ssp\n",
      "Propithecus_edwardsi_ssp\n",
      "1 Propithecus_edwardsi_ssp\n",
      "Skipping\n",
      "../zarr_data/Hapalemur_griseus_ssp\n",
      "Hapalemur_griseus_ssp\n",
      "8 Hapalemur_griseus_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_sylvanus_ssp\n",
      "Macaca_sylvanus_ssp\n",
      "2 Macaca_sylvanus_ssp\n",
      "Skipping\n",
      "../zarr_data/Pithecia_chrysocephala_ssp\n",
      "Pithecia_chrysocephala_ssp\n",
      "1 Pithecia_chrysocephala_ssp\n",
      "Skipping\n",
      "../zarr_data/Aotus_nancymaae_ssp\n",
      "Aotus_nancymaae_ssp\n",
      "31 Aotus_nancymaae_ssp\n",
      "../zarr_data/Aotus_nancymaae_ssp/NW_026988488.1\n",
      "../zarr_data/Aotus_nancymaae_ssp/NW_026988415.1\n",
      "../zarr_data/Aotus_nancymaae_ssp/NW_026988472.1\n",
      "../zarr_data/Aotus_nancymaae_ssp/NW_026988496.1\n",
      "../zarr_data/Aotus_nancymaae_ssp/NW_026988412.1\n",
      "../zarr_data/Macaca_assamensis_ssp\n",
      "Macaca_assamensis_ssp\n",
      "5 Macaca_assamensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_donacophilus_ssp\n",
      "Plecturocebus_donacophilus_ssp\n",
      "1 Plecturocebus_donacophilus_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_bernhardi_ssp\n",
      "Plecturocebus_bernhardi_ssp\n",
      "3 Plecturocebus_bernhardi_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_ebenus_ssp\n",
      "Trachypithecus_ebenus_ssp\n",
      "1 Trachypithecus_ebenus_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_maura_ssp\n",
      "Macaca_maura_ssp\n",
      "7 Macaca_maura_ssp\n",
      "Skipping\n",
      "../zarr_data/Alouatta_macconnelli_ssp\n",
      "Alouatta_macconnelli_ssp\n",
      "5 Alouatta_macconnelli_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_cinereiceps_ssp\n",
      "Eulemur_cinereiceps_ssp\n",
      "1 Eulemur_cinereiceps_ssp\n",
      "Skipping\n",
      "../zarr_data/Ateles_chamek_ssp\n",
      "Ateles_chamek_ssp\n",
      "11 Ateles_chamek_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_moloch_ssp\n",
      "Plecturocebus_moloch_ssp\n",
      "1 Plecturocebus_moloch_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_macaco_ssp\n",
      "Eulemur_macaco_ssp\n",
      "9 Eulemur_macaco_ssp\n",
      "Skipping\n",
      "../zarr_data/Pithecia_hirsuta_ssp\n",
      "Pithecia_hirsuta_ssp\n",
      "2 Pithecia_hirsuta_ssp\n",
      "Skipping\n",
      "../zarr_data/Hylobates_abbotti_ssp\n",
      "Hylobates_abbotti_ssp\n",
      "1 Hylobates_abbotti_ssp\n",
      "Skipping\n",
      "../zarr_data/Cacajao_hosomi_ssp\n",
      "Cacajao_hosomi_ssp\n",
      "2 Cacajao_hosomi_ssp\n",
      "Skipping\n",
      "../zarr_data/Saguinus_geoffroyi_ssp\n",
      "Saguinus_geoffroyi_ssp\n",
      "1 Saguinus_geoffroyi_ssp\n",
      "Skipping\n",
      "../zarr_data/Pithecia_pissinatti_ssp\n",
      "Pithecia_pissinatti_ssp\n",
      "3 Pithecia_pissinatti_ssp\n",
      "Skipping\n",
      "../zarr_data/Nasalis_larvatus_ssp\n",
      "Nasalis_larvatus_ssp\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "3 Nasalis_larvatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Propithecus_perrieri_ssp\n",
      "Propithecus_perrieri_ssp\n",
      "2 Propithecus_perrieri_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_radiata_ssp\n",
      "Macaca_radiata_ssp\n",
      "13 Macaca_radiata_ssp\n",
      "Skipping\n",
      "../zarr_data/Lepilemur_mustelinus_ssp\n",
      "Lepilemur_mustelinus_ssp\n",
      "1 Lepilemur_mustelinus_ssp\n",
      "Skipping\n",
      "../zarr_data/Prolemur_simus_ssp\n",
      "Prolemur_simus_ssp\n",
      "8 Prolemur_simus_ssp\n",
      "Skipping\n",
      "../zarr_data/Saguinus_oedipus_ssp\n",
      "Saguinus_oedipus_ssp\n",
      "1 Saguinus_oedipus_ssp\n",
      "Skipping\n",
      "../zarr_data/Piliocolobus_gordonorum_ssp\n",
      "Piliocolobus_gordonorum_ssp\n",
      "1 Piliocolobus_gordonorum_ssp\n",
      "Skipping\n",
      "../zarr_data/Alouatta_caraya_ssp\n",
      "Alouatta_caraya_ssp\n",
      "2 Alouatta_caraya_ssp\n",
      "Skipping\n",
      "../zarr_data/Lagothrix_lagotricha_ssp\n",
      "Lagothrix_lagotricha_ssp\n",
      "1 Lagothrix_lagotricha_ssp\n",
      "Skipping\n",
      "../zarr_data/Leontopithecus_rosalia_ssp\n",
      "Leontopithecus_rosalia_ssp\n",
      "3 Leontopithecus_rosalia_ssp\n",
      "Skipping\n",
      "../zarr_data/Daubentonia_madagascariensis_ssp\n",
      "Daubentonia_madagascariensis_ssp\n",
      "35 Daubentonia_madagascariensis_ssp\n",
      "../zarr_data/Daubentonia_madagascariensis_ssp/JBFSEQ010000009.1\n",
      "../zarr_data/Daubentonia_madagascariensis_ssp/JBFSEQ010000001.1\n",
      "../zarr_data/Propithecus_tattersalli_ssp\n",
      "Propithecus_tattersalli_ssp\n",
      "8 Propithecus_tattersalli_ssp\n",
      "Skipping\n",
      "../zarr_data/Hylobates_agilis_ssp\n",
      "Hylobates_agilis_ssp\n",
      "2 Hylobates_agilis_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_fascicularis_ssp_pca\n",
      "Macaca_fascicularis_ssp_pca\n",
      "0 Macaca_fascicularis_ssp_pca\n",
      "Skipping\n",
      "../zarr_data/Cacajao_ayresi_ssp\n",
      "Cacajao_ayresi_ssp\n",
      "3 Cacajao_ayresi_ssp\n",
      "Skipping\n",
      "../zarr_data/Lepilemur_ruficaudatus_ssp\n",
      "Lepilemur_ruficaudatus_ssp\n",
      "2 Lepilemur_ruficaudatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Saimiri_boliviensis_ssp\n",
      "Saimiri_boliviensis_ssp\n",
      "12 Saimiri_boliviensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_hecki_ssp\n",
      "Macaca_hecki_ssp\n",
      "5 Macaca_hecki_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_thibetana_ssp\n",
      "Macaca_thibetana_ssp\n",
      "3 Macaca_thibetana_ssp\n",
      "Skipping\n",
      "../zarr_data/Varecia_rubra_ssp\n",
      "Varecia_rubra_ssp\n",
      "4 Varecia_rubra_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_nictitans_ssp\n",
      "Cercopithecus_nictitans_ssp\n",
      "3 Cercopithecus_nictitans_ssp\n",
      "Skipping\n",
      "../zarr_data/Theropithecus_gelada_ssp\n",
      "Theropithecus_gelada_ssp\n",
      "73 Theropithecus_gelada_ssp\n",
      "Skipping\n",
      "../zarr_data/Microcebus_jonahi_ssp\n",
      "Microcebus_jonahi_ssp\n",
      "1 Microcebus_jonahi_ssp\n",
      "Skipping\n",
      "../zarr_data/Semnopithecus_hypoleucos_ssp\n",
      "Semnopithecus_hypoleucos_ssp\n",
      "4 Semnopithecus_hypoleucos_ssp\n",
      "Skipping\n",
      "../zarr_data/Brachyteles_hypoxanthus_ssp\n",
      "Brachyteles_hypoxanthus_ssp\n",
      "1 Brachyteles_hypoxanthus_ssp\n",
      "Skipping\n",
      "../zarr_data/Chiropotes_sagulatus_ssp\n",
      "Chiropotes_sagulatus_ssp\n",
      "4 Chiropotes_sagulatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Aotus_azarai_ssp\n",
      "Aotus_azarai_ssp\n",
      "14 Aotus_azarai_ssp\n",
      "Skipping\n",
      "../zarr_data/Papio_anubis_ssp_pca\n",
      "Papio_anubis_ssp_pca\n",
      "0 Papio_anubis_ssp_pca\n",
      "Skipping\n",
      "../zarr_data/Saguinus_inustus_ssp\n",
      "Saguinus_inustus_ssp\n",
      "1 Saguinus_inustus_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_fulvus_ssp\n",
      "Eulemur_fulvus_ssp\n",
      "2 Eulemur_fulvus_ssp\n",
      "Skipping\n",
      "../zarr_data/Hapalemur_meridionalis_ssp\n",
      "Hapalemur_meridionalis_ssp\n",
      "2 Hapalemur_meridionalis_ssp\n",
      "Skipping\n",
      "../zarr_data/Chlorocebus_tantalus_ssp\n",
      "Chlorocebus_tantalus_ssp\n",
      "11 Chlorocebus_tantalus_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercocebus_chrysogaster_ssp\n",
      "Cercocebus_chrysogaster_ssp\n",
      "1 Cercocebus_chrysogaster_ssp\n",
      "Skipping\n",
      "../zarr_data/Hylobates_muelleri_ssp\n",
      "Hylobates_muelleri_ssp\n",
      "1 Hylobates_muelleri_ssp\n",
      "Skipping\n",
      "../zarr_data/Allochrocebus_lhoesti_ssp\n",
      "Allochrocebus_lhoesti_ssp\n",
      "3 Allochrocebus_lhoesti_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_dubius_ssp\n",
      "Plecturocebus_dubius_ssp\n",
      "3 Plecturocebus_dubius_ssp\n",
      "Skipping\n",
      "../zarr_data/Miopithecus_talapoin_ssp\n",
      "Miopithecus_talapoin_ssp\n",
      "1 Miopithecus_talapoin_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercocebus_torquatus_ssp\n",
      "Cercocebus_torquatus_ssp\n",
      "2 Cercocebus_torquatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_leonina_ssp\n",
      "Macaca_leonina_ssp\n",
      "8 Macaca_leonina_ssp\n",
      "Skipping\n",
      "../zarr_data/Pygathrix_nemaeus_ssp\n",
      "Pygathrix_nemaeus_ssp\n",
      "7 Pygathrix_nemaeus_ssp\n",
      "Skipping\n",
      "../zarr_data/Otolemur_crassicaudatus_ssp\n",
      "Otolemur_crassicaudatus_ssp\n",
      "3 Otolemur_crassicaudatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_cephus_ssp\n",
      "Cercopithecus_cephus_ssp\n",
      "3 Cercopithecus_cephus_ssp\n",
      "Skipping\n",
      "../zarr_data/Loris_lydekkerianus_ssp\n",
      "Loris_lydekkerianus_ssp\n",
      "2 Loris_lydekkerianus_ssp\n",
      "Skipping\n",
      "../zarr_data/Papio_cynocephalus_ssp\n",
      "Papio_cynocephalus_ssp\n",
      "537 Papio_cynocephalus_ssp\n",
      "Skipping\n",
      "../zarr_data/Carlito_syrichta_ssp\n",
      "Carlito_syrichta_ssp\n",
      "1 Carlito_syrichta_ssp\n",
      "Skipping\n",
      "../zarr_data/Alouatta_seniculus_ssp\n",
      "Alouatta_seniculus_ssp\n",
      "21 Alouatta_seniculus_ssp\n",
      "../zarr_data/Alouatta_seniculus_ssp/CAJZLT010000040.1\n",
      "../zarr_data/Alouatta_seniculus_ssp/CAJZLT010000060.1\n",
      "../zarr_data/Alouatta_seniculus_ssp/CAJZLT010000072.1\n",
      "../zarr_data/Alouatta_seniculus_ssp/CAJZLT010000001.1\n",
      "../zarr_data/Alouatta_seniculus_ssp/CAJZLT010000080.1\n",
      "../zarr_data/Alouatta_seniculus_ssp/CAJZLT010000046.1\n",
      "../zarr_data/Alouatta_seniculus_ssp/CAJZLT010000034.1\n",
      "../zarr_data/Alouatta_seniculus_ssp/CAJZLT010000017.1\n",
      "../zarr_data/Cercocebus_atys_ssp\n",
      "Cercocebus_atys_ssp\n",
      "15 Cercocebus_atys_ssp\n",
      "Skipping\n",
      "../zarr_data/Pan_paniscus_ssp\n",
      "Pan_paniscus_ssp\n",
      "23 Pan_paniscus_ssp\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073267.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_085926.1\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073256.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073257.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_085927.1\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073266.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073264.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073255.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073254.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073265.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073261.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073271.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073270.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073260.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073259.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073262.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073272.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073253.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073268.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073249.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073269.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073252.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073263.2\n",
      "../zarr_data/Pan_paniscus_ssp/NC_073258.2\n",
      "../zarr_data/Presbytis_comata_ssp\n",
      "Presbytis_comata_ssp\n",
      "1 Presbytis_comata_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_sanfordi_ssp\n",
      "Eulemur_sanfordi_ssp\n",
      "5 Eulemur_sanfordi_ssp\n",
      "Skipping\n",
      "../zarr_data/Saimiri_sciureus_ssp\n",
      "Saimiri_sciureus_ssp\n",
      "2 Saimiri_sciureus_ssp\n",
      "Skipping\n",
      "../zarr_data/Saguinus_midas_ssp\n",
      "Saguinus_midas_ssp\n",
      "4 Saguinus_midas_ssp\n",
      "Skipping\n",
      "../zarr_data/Piliocolobus_kirkii_ssp\n",
      "Piliocolobus_kirkii_ssp\n",
      "1 Piliocolobus_kirkii_ssp\n",
      "Skipping\n",
      "../zarr_data/Propithecus_verreauxi_ssp\n",
      "Propithecus_verreauxi_ssp\n",
      "2 Propithecus_verreauxi_ssp\n",
      "Skipping\n",
      "../zarr_data/Nomascus_siki_ssp\n",
      "Nomascus_siki_ssp\n",
      "1 Nomascus_siki_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_phayrei_ssp\n",
      "Trachypithecus_phayrei_ssp\n",
      "5 Trachypithecus_phayrei_ssp\n",
      "Skipping\n",
      "../zarr_data/Propithecus_coquereli_ssp\n",
      "Propithecus_coquereli_ssp\n",
      "10 Propithecus_coquereli_ssp\n",
      "Skipping\n",
      "../zarr_data/Rhinopithecus_bieti_ssp\n",
      "Rhinopithecus_bieti_ssp\n",
      "36 Rhinopithecus_bieti_ssp\n",
      "../zarr_data/Rhinopithecus_bieti_ssp/NC_044549.1\n",
      "../zarr_data/Rhinopithecus_bieti_ssp/NC_044555.1\n",
      "../zarr_data/Eulemur_collaris_ssp\n",
      "Eulemur_collaris_ssp\n",
      "7 Eulemur_collaris_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheirogaleus_major_ssp\n",
      "Cheirogaleus_major_ssp\n",
      "3 Cheirogaleus_major_ssp\n",
      "Skipping\n",
      "../zarr_data/Cheirogaleus_sibreei_ssp\n",
      "Cheirogaleus_sibreei_ssp\n",
      "1 Cheirogaleus_sibreei_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_mulatta_ssp_pca\n",
      "Macaca_mulatta_ssp_pca\n",
      "0 Macaca_mulatta_ssp_pca\n",
      "Skipping\n",
      "../zarr_data/Lophocebus_aterrimus_ssp\n",
      "Lophocebus_aterrimus_ssp\n",
      "3 Lophocebus_aterrimus_ssp\n",
      "Skipping\n",
      "../zarr_data/Chlorocebus_dryas_ssp\n",
      "Chlorocebus_dryas_ssp\n",
      "1 Chlorocebus_dryas_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_arctoides_ssp\n",
      "Macaca_arctoides_ssp\n",
      "11 Macaca_arctoides_ssp\n",
      "Skipping\n",
      "../zarr_data/Chiropotes_israelita_ssp\n",
      "Chiropotes_israelita_ssp\n",
      "4 Chiropotes_israelita_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_johnii_ssp\n",
      "Trachypithecus_johnii_ssp\n",
      "2 Trachypithecus_johnii_ssp\n",
      "Skipping\n",
      "../zarr_data/Chlorocebus_sabaeus_ssp\n",
      "Chlorocebus_sabaeus_ssp\n",
      "53 Chlorocebus_sabaeus_ssp\n",
      "Skipping\n",
      "../zarr_data/Presbytis_melalophos_ssp\n",
      "Presbytis_melalophos_ssp\n",
      "1 Presbytis_melalophos_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_albifrons_ssp\n",
      "Eulemur_albifrons_ssp\n",
      "3 Eulemur_albifrons_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_rubriventer_ssp\n",
      "Eulemur_rubriventer_ssp\n",
      "7 Eulemur_rubriventer_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_pogonias_ssp\n",
      "Cercopithecus_pogonias_ssp\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "2 Cercopithecus_pogonias_ssp\n",
      "Skipping\n",
      "../zarr_data/Microcebus_griseorufus_ssp\n",
      "Microcebus_griseorufus_ssp\n",
      "1 Microcebus_griseorufus_ssp\n",
      "Skipping\n",
      "../zarr_data/Saguinus_mystax_ssp\n",
      "Saguinus_mystax_ssp\n",
      "1 Saguinus_mystax_ssp\n",
      "Skipping\n",
      "../zarr_data/Tarsius_wallacei_ssp\n",
      "Tarsius_wallacei_ssp\n",
      "1 Tarsius_wallacei_ssp\n",
      "Skipping\n",
      "../zarr_data/Arctocebus_calabarensis_ssp\n",
      "Arctocebus_calabarensis_ssp\n",
      "1 Arctocebus_calabarensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_auratus_ssp\n",
      "Trachypithecus_auratus_ssp\n",
      "2 Trachypithecus_auratus_ssp\n",
      "Skipping\n",
      "../zarr_data/Papio_ursinus_ssp\n",
      "Papio_ursinus_ssp\n",
      "8 Papio_ursinus_ssp\n",
      "Skipping\n",
      "../zarr_data/Plecturocebus_miltoni_ssp\n",
      "Plecturocebus_miltoni_ssp\n",
      "1 Plecturocebus_miltoni_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_flavifrons_ssp\n",
      "Eulemur_flavifrons_ssp\n",
      "10 Eulemur_flavifrons_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_wolfi_ssp\n",
      "Cercopithecus_wolfi_ssp\n",
      "3 Cercopithecus_wolfi_ssp\n",
      "Skipping\n",
      "../zarr_data/Aotus_vociferans_ssp\n",
      "Aotus_vociferans_ssp\n",
      "5 Aotus_vociferans_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_pileatus_ssp\n",
      "Trachypithecus_pileatus_ssp\n",
      "1 Trachypithecus_pileatus_ssp\n",
      "Skipping\n",
      "../zarr_data/Trachypithecus_geei_ssp\n",
      "Trachypithecus_geei_ssp\n",
      "6 Trachypithecus_geei_ssp\n",
      "Skipping\n",
      "../zarr_data/Microcebus_ravelobensis_ssp\n",
      "Microcebus_ravelobensis_ssp\n",
      "3 Microcebus_ravelobensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Erythrocebus_patas_ssp\n",
      "Erythrocebus_patas_ssp\n",
      "6 Erythrocebus_patas_ssp\n",
      "Skipping\n",
      "../zarr_data/Rhinopithecus_roxellana_ssp\n",
      "Rhinopithecus_roxellana_ssp\n",
      "57 Rhinopithecus_roxellana_ssp\n",
      "Skipping\n",
      "../zarr_data/Piliocolobus_tephrosceles_ssp\n",
      "Piliocolobus_tephrosceles_ssp\n",
      "1 Piliocolobus_tephrosceles_ssp\n",
      "Skipping\n",
      "../zarr_data/Eulemur_rufus_ssp\n",
      "Eulemur_rufus_ssp\n",
      "1 Eulemur_rufus_ssp\n",
      "Skipping\n",
      "../zarr_data/Lepilemur_ankaranensis_ssp\n",
      "Lepilemur_ankaranensis_ssp\n",
      "5 Lepilemur_ankaranensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Loris_tardigradus_ssp\n",
      "Loris_tardigradus_ssp\n",
      "2 Loris_tardigradus_ssp\n",
      "Skipping\n",
      "../zarr_data/Lepilemur_mittermeieri_ssp\n",
      "Lepilemur_mittermeieri_ssp\n",
      "1 Lepilemur_mittermeieri_ssp\n",
      "Skipping\n",
      "../zarr_data/Cebuella_pygmaea_ssp\n",
      "Cebuella_pygmaea_ssp\n",
      "2 Cebuella_pygmaea_ssp\n",
      "Skipping\n",
      "../zarr_data/Nycticebus_coucang_ssp\n",
      "Nycticebus_coucang_ssp\n",
      "3 Nycticebus_coucang_ssp\n",
      "Skipping\n",
      "../zarr_data/Sapajus_apella_ssp\n",
      "Sapajus_apella_ssp\n",
      "15 Sapajus_apella_ssp\n",
      "Skipping\n",
      "../zarr_data/Cercopithecus_campbelli_ssp\n",
      "Cercopithecus_campbelli_ssp\n",
      "2 Cercopithecus_campbelli_ssp\n",
      "Skipping\n",
      "../zarr_data/Microcebus_murinus_ssp\n",
      "Microcebus_murinus_ssp\n",
      "26 Microcebus_murinus_ssp\n",
      "../zarr_data/Microcebus_murinus_ssp/NC_033692.1\n",
      "../zarr_data/Microcebus_murinus_ssp/NC_033660.1\n",
      "../zarr_data/Pygathrix_cinerea_ssp\n",
      "Pygathrix_cinerea_ssp\n",
      "5 Pygathrix_cinerea_ssp\n",
      "Skipping\n",
      "../zarr_data/Microcebus_arnholdi_ssp\n",
      "Microcebus_arnholdi_ssp\n",
      "2 Microcebus_arnholdi_ssp\n",
      "Skipping\n",
      "../zarr_data/Pygathrix_nigripes_ssp\n",
      "Pygathrix_nigripes_ssp\n",
      "6 Pygathrix_nigripes_ssp\n",
      "Skipping\n",
      "../zarr_data/Macaca_cyclopis_ssp\n",
      "Macaca_cyclopis_ssp\n",
      "4 Macaca_cyclopis_ssp\n",
      "Skipping\n",
      "../zarr_data/Saimiri_cassiquiarensis_ssp\n",
      "Saimiri_cassiquiarensis_ssp\n",
      "9 Saimiri_cassiquiarensis_ssp\n",
      "Skipping\n",
      "../zarr_data/Hapalemur_aureus_ssp\n",
      "Hapalemur_aureus_ssp\n",
      "1 Hapalemur_aureus_ssp\n",
      "Skipping\n"
     ]
    }
   ],
   "source": [
    "# Fst implementation medium\n",
    "for x in glob.glob(zarr_path+\"*\"):\n",
    "    print(x)\n",
    "    # The name used to load all the files, short and long version\n",
    "    short_form = x.split(\"/\")[-1].split(\"_\")[0]\n",
    "    long_form = x.split(\"/\")[-1]\n",
    "    print(long_form)\n",
    "    # Loading the various metadata files. Metadata, contig information, callability bed.\n",
    "    metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "    metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "    metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "    regions_df[\"chr_type\"] = [\"chrX\" if x == 2 and y == 1 else \"aut\" for x, y in zip(regions_df.FEMALE_PLOIDY, regions_df.MALE_PLOIDY)]\n",
    "    large_contigs = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "    large_x = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID\n",
    "    # Skipping the large samples sizes and the singulars for the Fst calc\n",
    "    print(len(metadata_df.loc[metadata_df.GVCF_FOLDER == long_form]), long_form)\n",
    "    if ((len(metadata_df.loc[metadata_df.GVCF_FOLDER == long_form]) > 50) or\n",
    "    (len(metadata_df.loc[metadata_df.GVCF_FOLDER == long_form]) <= 20)):\n",
    "        print(\"Skipping\")\n",
    "        continue\n",
    "    # Loading the genetic data.\n",
    "    df_l = []\n",
    "    for c in glob.glob(x+\"/*\"):\n",
    "        print(c)\n",
    "        ds = sg.load_dataset(c)\n",
    "        # This implementation is the pi implementation.\n",
    "        # Probably problematic in some cases with population structure, but it is easier to implement\n",
    "        ds[\"sample_cohort\"] = ds[\"samples\"]\n",
    "        # Subsetting and windowing the sgkit dataset. The rechunking handles what otherwise would cause an error.\n",
    "        #ds[\"call_genotype\"] = ds[\"call_genotype\"].clip(0)\n",
    "        ds = ds.sel(contigs=[ds.variant_contig[0].values])\n",
    "        if c.split(\"/\")[-1] in list(large_x):\n",
    "            ds = haploid_double(ds, \"call_genotype\", \"samples\")\n",
    "        missing_rate = ds.call_genotype_mask[:,:,0].sum(axis=1).values/ds.call_genotype_mask[:,:,0].count(axis=1).values\n",
    "        ds = ds.isel(variants=(missing_rate <= 0))\n",
    "        ds = sg.window_by_genome(ds)\n",
    "        ds = (sg.Fst(ds.chunk({\"variants\": 50000})))\n",
    "        df_sub = pd.DataFrame(ds.stat_Fst[0,:,], columns=ds.sample_id)\n",
    "        df_sub[\"chrom\"] = c.split(\"/\")[-1]\n",
    "        df_sub[\"variants_used\"] = len(ds.variants)\n",
    "        df_l.append(df_sub)\n",
    "    output_df = pd.concat(df_l)\n",
    "    output_df[\"chr_type\"] = output_df[\"chrom\"].map(dict(zip(regions_df.CONTIG_ID, regions_df.chr_type)))\n",
    "    output_df[\"species\"] = long_form\n",
    "    output_df.to_csv(\"../results/window_stats/{}_Fst.txt\".format(long_form), sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "bcc90c66",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>PD_0120</th>\n",
       "      <th>PD_0348</th>\n",
       "      <th>chrom</th>\n",
       "      <th>variants_used</th>\n",
       "      <th>chr_type</th>\n",
       "      <th>species</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CM063191.1</td>\n",
       "      <td>1535859</td>\n",
       "      <td>chrX</td>\n",
       "      <td>Saguinus_labiatus_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CM063191.1</td>\n",
       "      <td>1535859</td>\n",
       "      <td>chrX</td>\n",
       "      <td>Saguinus_labiatus_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>-0.517100</td>\n",
       "      <td>JASSZA010000432.1</td>\n",
       "      <td>4587</td>\n",
       "      <td>aut</td>\n",
       "      <td>Saguinus_labiatus_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>-0.517100</td>\n",
       "      <td>NaN</td>\n",
       "      <td>JASSZA010000432.1</td>\n",
       "      <td>4587</td>\n",
       "      <td>aut</td>\n",
       "      <td>Saguinus_labiatus_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>0.184585</td>\n",
       "      <td>CM063169.1</td>\n",
       "      <td>3030007</td>\n",
       "      <td>aut</td>\n",
       "      <td>Saguinus_labiatus_ssp</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>0.184585</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CM063169.1</td>\n",
       "      <td>3030007</td>\n",
       "      <td>aut</td>\n",
       "      <td>Saguinus_labiatus_ssp</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    PD_0120   PD_0348              chrom  variants_used chr_type  \\\n",
       "0       NaN       NaN         CM063191.1        1535859     chrX   \n",
       "1       NaN       NaN         CM063191.1        1535859     chrX   \n",
       "0       NaN -0.517100  JASSZA010000432.1           4587      aut   \n",
       "1 -0.517100       NaN  JASSZA010000432.1           4587      aut   \n",
       "0       NaN  0.184585         CM063169.1        3030007      aut   \n",
       "1  0.184585       NaN         CM063169.1        3030007      aut   \n",
       "\n",
       "                 species  \n",
       "0  Saguinus_labiatus_ssp  \n",
       "1  Saguinus_labiatus_ssp  \n",
       "0  Saguinus_labiatus_ssp  \n",
       "1  Saguinus_labiatus_ssp  \n",
       "0  Saguinus_labiatus_ssp  \n",
       "1  Saguinus_labiatus_ssp  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "output_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "a8699942",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "../zarr_data/Macaca_fascicularis_ssp\n",
      "Macaca_fascicularis_ssp\n",
      "../zarr_data/Macaca_fascicularis_ssp/CP141342.1\n",
      "../zarr_data/Macaca_fascicularis_ssp/CP141341.1\n",
      "../zarr_data/Macaca_fascicularis_ssp/CP141361.1\n"
     ]
    }
   ],
   "source": [
    "for x in glob.glob(zarr_path+\"*\")[4:5]:\n",
    "    print(x)\n",
    "    # The name used to load all the files, short and long version\n",
    "    short_form = x.split(\"/\")[-1].split(\"_\")[0]\n",
    "    long_form = x.split(\"/\")[-1]\n",
    "    print(long_form)\n",
    "    # Loading the various metadata files. Metadata, contig information, callability bed.\n",
    "    metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "    metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "    metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "    regions_df[\"chr_type\"] = [\"chrX\" if x == 2 and y == 1 else \"aut\" for x, y in zip(regions_df.FEMALE_PLOIDY, regions_df.MALE_PLOIDY)]\n",
    "    large_contigs = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "    large_x = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID\n",
    "    # Loading the genetic data.\n",
    "    df_l = []\n",
    "    for c in glob.glob(x+\"/*\"):\n",
    "        print(c)\n",
    "        ds = sg.load_dataset(c)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f70a7cb",
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f85c614",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "zarr_path = \"../zarr_data/\"\n",
    "metadata_folders = glob.glob(metadata_path+\"*_individuals.txt\")\n",
    "\n",
    "for folder in metadata_folders[:0]:\n",
    "    metadata_df = pd.read_csv(folder, sep=\"\\t\")\n",
    "    short_form = folder.split(\"/\")[-1].split(\"_\")[0]\n",
    "    regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "    metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "    female_df = metadata_df[pd.to_numeric(metadata_df['AVG_COVERAGE_X'], errors='coerce').notnull()]\n",
    "    female_df = female_df.loc[(female_df.GENETIC_SEX == \"F\") & (female_df.AVG_COVERAGE_A >= 10)].sort_values(by=\"AVG_COVERAGE_A\", ascending=False)\n",
    "    for GVCF_FOLDER in female_df.GVCF_FOLDER.unique():\n",
    "        # Check that zarr data exists.\n",
    "        zarr_chroms = glob.glob(zarr_path+GVCF_FOLDER+\"/*\")\n",
    "        for c in zarr_chroms:\n",
    "            ds = sg.load_dataset(c, drop_variables=[\"variant_allele\"])\n",
    "            print(c)\n",
    "        reference = metadata_df.loc[metadata_df.GVCF_FOLDER == GVCF_FOLDER].REFERENCE_FOLDER.unique()[0]\n",
    "        regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "        regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "        large_contigs = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "        large_x = regions_df.loc[(regions_df.LENGTH >= size_cutoff) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID\n",
    "        bed_files = read_beds(GVCF_FOLDER)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "81f81bfd",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/Pongo_individuals.txt'"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "metadata_path+\"{}_individuals.txt\".format(short_form)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9462dfdb",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Loading the genetic data.\n",
    "    ds = sg.load_dataset(\"../data/{}\".format(long_form))\n",
    "    sex_map = dict(zip(metadata_df.GVCF_ID, metadata_df.SEX_I))\n",
    "    # If you want to split based on sex. I will make every individual unique\n",
    "    #ds[\"sample_cohort\"] = xr.DataArray(pd.Series(ds.sample_id.values).map(sex_map), dims=\"samples\")\n",
    "    ds[\"sample_cohort\"] = ds[\"samples\"]\n",
    "    # Only keeping contigs which are above the cutoff (1Mb or more)\n",
    "    kept_contigs =  [x for x in ds.contig_id.values if (x == large_contigs).any()]\n",
    "    contig_IDs = pd.Series(kept_contigs).map(dict(zip(ds.contig_id.values, range(len(ds.contig_id.values))))).values\n",
    "    # Subsetting and windowing the sgkit dataset. The rechunking handles what otherwise would cause an error.\n",
    "    ds_autx = ds.sel(variants=(ds.variant_contig.isin(contig_IDs).compute()), contigs=contig_IDs)\n",
    "    ds_autx = sg.window_by_position(ds_autx, size=window_size)\n",
    "    ds_autx[\"call_genotype\"] = ds_autx[\"call_genotype\"].clip(0)\n",
    "    ds_autx_diversity = sg.diversity(ds_autx.chunk({\"variants\": 50000}))\n",
    "    # Generating window callability\n",
    "    contig_l, window_start_l = [], []\n",
    "    v_c = pd.Series(ds_autx.window_contig).value_counts(sort=False)\n",
    "    for i in range(len(v_c)):\n",
    "        window_start_l.extend(list(range(0, v_c[i]*window_size, window_size)))\n",
    "        contig_l.extend([kept_contigs[i]]*v_c[i])\n",
    "    window_df = pd.DataFrame({\"chrom\": contig_l, \"window_start\": window_start_l})\n",
    "    intervals_callable = pos_windows(bed_files, window_size, kept_contigs)\n",
    "    window_df_call = window_df.merge(intervals_callable, on=[\"chrom\", \"window_start\"])\n",
    "    # Merge and save\n",
    "    output_df = pd.merge(window_df_call, pd.DataFrame(ds_autx_diversity[\"stat_diversity\"],\n",
    "                                      columns=ds_autx_diversity[\"sample_id\"]),\n",
    "                         left_index=True, right_index=True)\n",
    "    output_df.to_csv(\"../results/window_stats/{}_10kb_het.txt\".format(long_form), sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "f315c4e4",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_l = []\n",
    "for p in df_paths:\n",
    "    d = pd.read_csv(p, sep=\"\\t\", index_col=[0])\n",
    "    d[\"species\"] = p.split(\"/\")[-1].split(\"_1\")[0]\n",
    "    df_l.append(pd.melt(d, id_vars=[\"chrom\", \"window_start\", \"window_end\", \"callable_frac\", \"species\"], var_name=\"GVCF_ID\", value_name=\"het\"))\n",
    "het_df = pd.concat(df_l)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5de8c014",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Standardized parts.\n",
    "short_form = \"Papio\"\n",
    "long_form = \"Papio_papio_ssp\"\n",
    "size_cutoff = 1000000\n",
    "window_size = 100000 # Should maybe decrease to 25 or 10kb.\n",
    "\n",
    "metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "\n",
    "regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "large_contigs = regions_df.loc[(regions_df.LENGTH >= 1000000) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "large_x = regions_df.loc[(regions_df.LENGTH >= 1000000) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0fa50281",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "20    NC_044996.1\n",
       "Name: CONTIG_ID, dtype: object"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "large_x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "c3ea2789",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Standardized parts.\n",
    "short_form = \"Gorilla\"\n",
    "long_form = \"Gorilla_Gorilla_ssp\"\n",
    "size_cutoff = 1000000\n",
    "window_size = 100000 # Should maybe decrease to 25 or 10kb.\n",
    "\n",
    "metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "metadata_df = pd.read_csv(metadata_path+\"{}_individuals.txt\".format(short_form), sep=\"\\t\")\n",
    "metadata_df[\"SEX_I\"] = [0 if x == \"F\" else 1 for x in metadata_df.GENETIC_SEX]\n",
    "\n",
    "regions_df = pd.read_csv(metadata_path+\"{}_regions_and_batches.txt\".format(short_form), sep=\"\\t\")\n",
    "regions_df[\"LENGTH\"] = regions_df[\"END\"]-regions_df[\"START\"]\n",
    "large_contigs = regions_df.loc[(regions_df.LENGTH >= 1000000) & (regions_df.FEMALE_PLOIDY == 2)].CONTIG_ID.unique()\n",
    "large_x = regions_df.loc[(regions_df.LENGTH >= 1000000) & (regions_df.FEMALE_PLOIDY == 2) &\n",
    "                        (regions_df.MALE_PLOIDY == 1)].CONTIG_ID"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "id": "531a1b87",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "23    NC_073247.2\n",
       "25    NC_073247.2\n",
       "Name: CONTIG_ID, dtype: object"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "large_x"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "id": "4f5c4322",
   "metadata": {},
   "outputs": [],
   "source": [
    "het_df[\"chr_type\"] = [\"aut\" if x not in [\"NC_044996.1\", \"NC_073247.2\"] else \"chrX\" for x in het_df[\"chrom\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "id": "94d187a4",
   "metadata": {},
   "outputs": [],
   "source": [
    "het_df_high_call = het_df.loc[het_df.callable_frac >= 0.95]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 59,
   "id": "87ffdf3f",
   "metadata": {},
   "outputs": [],
   "source": [
    "metadata_path = \"/home/eriks/primatediversity/data/gVCFs_recalling_10_12_2024_metadata/\"\n",
    "metadata_df_gorilla = pd.read_csv(metadata_path+\"Gorilla_individuals.txt\".format(short_form), sep=\"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 47,
   "id": "ac38c294",
   "metadata": {},
   "outputs": [],
   "source": [
    "df_mean_het = het_df_high_call.groupby([\"GVCF_ID\", \"species\", \"chr_type\"])[\"het\"].mean().reset_index()"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.12.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
